--- a/Makefile.in
+++ b/Makefile.in
@@ -2230,8 +2230,7 @@
 	@echo "              *        Generating CLooG's documentation       *"
 	@echo "              *-----------------------------------------------*/"
 	doxygen ./autoconf/Doxyfile
-@HAVE_TEXI2DVI_TRUE@doc/cloog.pdf: doc/cloog.texi doc/gitversion.texi
-@HAVE_TEXI2DVI_TRUE@	$(TEXI2DVI) -I $(top_builddir)/doc --pdf $< -o $@
+@HAVE_TEXI2DVI_TRUE@doc/cloog.pdf:
 
 doc/gitversion.texi: @GIT_INDEX@
 	echo '@set VERSION '`$(top_builddir)/genversion.sh`'' > $@
