diff -Naur spider-0.2.1-orig/i18n.cpp spider-0.2.1/i18n.cpp
--- spider-0.2.1-orig/i18n.cpp	2008-03-29 03:01:00.000000000 +0100
+++ spider-0.2.1/i18n.cpp	2008-06-20 21:53:31.000000000 +0200
@@ -154,7 +154,7 @@
     "", // Norsk / Norwegian
     "", // suomi / Finnish
     "", // Polski / Polish
-    "", // Español / Spanish
+    "Seleccionar", // Español / Spanish
     "", // ÅëëçíéêÜ / Greek
     "", // Svenska / Swedish
     "", // Românã / Romanian
@@ -176,7 +176,7 @@
     "", // Norsk / Norwegian
     "", // suomi / Finnish
     "", // Polski / Polish
-    "", // Español / Spanish
+    " Contar baraja", // Español / Spanish
     "", // ÅëëçíéêÜ / Greek
     "", // Svenska / Swedish
     "", // Românã / Romanian
@@ -198,7 +198,7 @@
     "", // Norsk / Norwegian
     "", // suomi / Finnish
     "", // Polski / Polish
-    "", // Español / Spanish
+    " Contar montón", // Español / Spanish
     "", // ÅëëçíéêÜ / Greek
     "", // Svenska / Swedish
     "", // Românã / Romanian
@@ -220,7 +220,7 @@
     "", // Norsk / Norwegian
     "", // suomi / Finnish
     "", // Polski / Polish
-    "", // Español / Spanish
+    " Contar reparto", // Español / Spanish
     "", // ÅëëçíéêÜ / Greek
     "", // Svenska / Swedish
     "", // Românã / Romanian
@@ -242,7 +242,7 @@
     "", // Norsk / Norwegian
     "", // suomi / Finnish
     "", // Polski / Polish
-    "", // Español / Spanish
+    "Posición OSD izquierda", // Español / Spanish
     "", // ÅëëçíéêÜ / Greek
     "", // Svenska / Swedish
     "", // Românã / Romanian
@@ -264,7 +264,7 @@
     "", // Norsk / Norwegian
     "", // suomi / Finnish
     "", // Polski / Polish
-    "", // Español / Spanish
+    "Posición OSD superior", // Español / Spanish
     "", // ÅëëçíéêÜ / Greek
     "", // Svenska / Swedish
     "", // Românã / Romanian
@@ -286,7 +286,7 @@
     "", // Norsk / Norwegian
     "", // suomi / Finnish
     "", // Polski / Polish
-    "", // Español / Spanish
+    "Anchura OSD", // Español / Spanish
     "", // ÅëëçíéêÜ / Greek
     "", // Svenska / Swedish
     "", // Românã / Romanian
@@ -308,7 +308,7 @@
     "", // Norsk / Norwegian
     "", // suomi / Finnish
     "", // Polski / Polish
-    "", // Español / Spanish
+    "Altura OSD", // Español / Spanish
     "", // ÅëëçíéêÜ / Greek
     "", // Svenska / Swedish
     "", // Românã / Romanian
@@ -330,7 +330,7 @@
     "", // Norsk / Norwegian
     "", // suomi / Finnish
     "", // Polski / Polish
-    "", // Español / Spanish
+    "Error compensación OSD", // Español / Spanish
     "", // ÅëëçíéêÜ / Greek
     "", // Svenska / Swedish
     "", // Românã / Romanian
@@ -352,7 +352,7 @@
     "", // Norsk / Norwegian
     "", // suomi / Finnish
     "", // Polski / Polish
-    "", // Español / Spanish
+    "Encoger altura", // Español / Spanish
     "", // ÅëëçíéêÜ / Greek
     "", // Svenska / Swedish
     "", // Românã / Romanian
@@ -374,7 +374,7 @@
     "", // Norsk / Norwegian
     "", // suomi / Finnish
     "", // Polski / Polish
-    "", // Español / Spanish
+    "Encoger anchura", // Español / Spanish
     "", // ÅëëçíéêÜ / Greek
     "", // Svenska / Swedish
     "", // Românã / Romanian
@@ -396,7 +396,7 @@
     "", // Norsk / Norwegian
     "", // suomi / Finnish
     "", // Polski / Polish
-    "", // Español / Spanish
+    "Encoger anchura y altura", // Español / Spanish
     "", // ÅëëçíéêÜ / Greek
     "", // Svenska / Swedish
     "", // Românã / Romanian
@@ -418,7 +418,7 @@
     "", // Norsk / Norwegian
     "", // suomi / Finnish
     "", // Polski / Polish
-    "", // Español / Spanish
+    "Reducir colores", // Español / Spanish
     "", // ÅëëçíéêÜ / Greek
     "", // Svenska / Swedish
     "", // Românã / Romanian
@@ -440,7 +440,7 @@
     "", // Norsk / Norwegian
     "", // suomi / Finnish
     "", // Polski / Polish
-    "", // Español / Spanish
+    "Ocultar fila superior", // Español / Spanish
     "", // ÅëëçíéêÜ / Greek
     "", // Svenska / Swedish
     "", // Românã / Romanian
@@ -462,7 +462,7 @@
     "", // Norsk / Norwegian
     "", // suomi / Finnish
     "", // Polski / Polish
-    "", // Español / Spanish
+    "Reiniciar", // Español / Spanish
     "", // ÅëëçíéêÜ / Greek
     "", // Svenska / Swedish
     "", // Românã / Romanian
diff -Naur spider-0.2.1-orig/Makefile spider-0.2.1/Makefile
--- spider-0.2.1-orig/Makefile	2008-03-29 03:01:00.000000000 +0100
+++ spider-0.2.1/Makefile	2008-06-20 21:53:31.000000000 +0200
@@ -39,8 +39,8 @@
 
 ### The directory environment:
 
-VDRDIR = ../../..
-LIBDIR = ../../lib
+VDRDIR ?= ../../..
+LIBDIR ?= ../../lib
 TMPDIR = /tmp
 
 ### Allow user defined options to overwrite defaults:
diff -Naur spider-0.2.1-orig/po/es_ES.po spider-0.2.1/po/es_ES.po
--- spider-0.2.1-orig/po/es_ES.po	2008-03-29 03:01:00.000000000 +0100
+++ spider-0.2.1/po/es_ES.po	2008-06-22 11:38:31.000000000 +0200
@@ -1,7 +1,7 @@
 # Spanish translations for vdr-spider package.
 # Copyright (C) 2007-2008 Thomas Günther <tom@toms-cafe.de>
 # This file is distributed under the same license as the vdr-spider package.
-# bittor from open7x0.org, 2007.
+# bittor from open7x0.org <bittor7x0 _at_ gmail.com>, 2008.
 #
 # $Id: es_ES.po 129 2008-03-29 00:06:39Z tom $
 #
@@ -10,8 +10,8 @@
 "Project-Id-Version: vdr-spider 0.1.4\n"
 "Report-Msgid-Bugs-To: <tom@toms-cafe.de>\n"
 "POT-Creation-Date: 2008-03-26 00:41+0100\n"
-"PO-Revision-Date: 2007-09-23 14:56+0200\n"
-"Last-Translator: bittor from open7x0.org\n"
+"PO-Revision-Date: 2008-06-22 11:36+0200\n"
+"Last-Translator: bittor from open7x0.org <bittor7x0 _at_ gmail.com>\n"
 "Language-Team:  <vdr@linuxtv.org>\n"
 "MIME-Version: 1.0\n"
 "Content-Type: text/plain; charset=iso-8859-15\n"
@@ -40,55 +40,55 @@
 msgstr "Normal"
 
 msgid "Custom"
-msgstr ""
+msgstr "Seleccionar"
 
 msgid "Shrink height"
-msgstr ""
+msgstr "Encoger altura"
 
 msgid "Shrink width"
-msgstr ""
+msgstr "Encoger anchura"
 
 msgid "Shrink width and height"
-msgstr ""
+msgstr "Encoger anchura y altura"
 
 msgid "Reduce colors"
-msgstr ""
+msgstr "Reducir colores"
 
 msgid "Reset"
-msgstr ""
+msgstr "Reiniciar"
 
 msgid "Variation"
 msgstr "Variación"
 
 #. TRANSLATORS: note the leading blank!
 msgid " Deck count"
-msgstr ""
+msgstr " Contar baraja"
 
 #. TRANSLATORS: note the leading blank!
 msgid " Pile count"
-msgstr ""
+msgstr " Contar montón"
 
 #. TRANSLATORS: note the leading blank!
 msgid " Deal count"
-msgstr ""
+msgstr " Contar reparto"
 
 msgid "OSD position left"
-msgstr ""
+msgstr "Posición OSD izquierda"
 
 msgid "OSD position top"
-msgstr ""
+msgstr "Posición OSD superior"
 
 msgid "OSD width"
-msgstr ""
+msgstr "Anchura OSD"
 
 msgid "OSD height"
-msgstr ""
+msgstr "Altura OSD"
 
 msgid "OSD error compensation"
-msgstr ""
+msgstr "Error compensación OSD"
 
 msgid "Hide top row"
-msgstr ""
+msgstr "Ocultar fila superior"
 
 msgid "Spider Arachnid - the best patience game"
 msgstr "Spider Arachnid - mejor juego paciencia"
