diff -Naur uclibc_build_env_orig/make-incs/siemens-linux-kernel.mk uclibc_build_env/make-incs/siemens-linux-kernel.mk
--- uclibc_build_env_orig/make-incs/siemens-linux-kernel.mk	2009-09-20 19:45:47.000000000 +0200
+++ uclibc_build_env/make-incs/siemens-linux-kernel.mk	2009-09-19 10:19:58.000000000 +0200
@@ -208,7 +208,7 @@
 		$(TARGET_ROOT)/$(M7X0_KERNEL_DIR)/vmlinux \
 		$(TARGET_ROOT)/$(M7X0_KERNEL_DIR)/vmlinux.bin
 	-$(RM) -f $(TARGET_ROOT)/$(M7X0_KERNEL_DIR)/vmlinux.bin.lzma
-	$(LZMA_BIN) $(TARGET_ROOT)/$(M7X0_KERNEL_DIR)/vmlinux.bin
+	$(LZMA_BIN) e -lc1 -lp2 -pb2 -eos $(TARGET_ROOT)/$(M7X0_KERNEL_DIR)/vmlinux.bin $(TARGET_ROOT)/$(M7X0_KERNEL_DIR)/vmlinux.bin.lzma
 	PATH='$(PREFIX_BIN):$(PATH)' $(MAKE) CROSS_COMPILE=$(TARGET)- ARCH=mips \
 		CC=$(EGCS_BIN) -C $(SIEMENS-LINUX-KERNEL_DIR) \
 		INSTALL_MOD_PATH=$(TARGET_ROOT)/$(M7X0_KERNEL_DIR) modules_install
