diff -Naur klone-core-2.2.0/klone/http.h klone-core-2.2.0-atinar/klone/http.h
--- klone-core-2.2.0/klone/http.h	2007-11-09 23:06:26.000000000 +0100
+++ klone-core-2.2.0-atinar/klone/http.h	2009-03-24 02:23:21.000000000 +0100
@@ -31,6 +31,8 @@
     /**< request accepted but processing not completed */
     HTTP_STATUS_NO_CONTENT                = 204,
     /**< no body returned */
+    HTTP_PARTIAL_CONTENT                  = 206,
+    /**< partial content returned */
     HTTP_STATUS_MOVED_PERMANENTLY         = 301,
     /**< resource relocated permanently */
     HTTP_STATUS_MOVED_TEMPORARILY         = 302,
diff -Naur klone-core-2.2.0/src/libhttp/http.c klone-core-2.2.0-atinar/src/libhttp/http.c
--- klone-core-2.2.0/src/libhttp/http.c	2009-03-06 14:16:49.000000000 +0100
+++ klone-core-2.2.0-atinar/src/libhttp/http.c	2009-03-24 02:23:19.000000000 +0100
@@ -51,6 +51,7 @@
     { HTTP_STATUS_CREATED               , "Created"                 },
     { HTTP_STATUS_ACCEPTED              , "Accepted"                },
     { HTTP_STATUS_NO_CONTENT            , "No Content"              },
+    { HTTP_PARTIAL_CONTENT              , "Partial Content"         },
     { HTTP_STATUS_BAD_REQUEST           , "Bad Request"             },
     { HTTP_STATUS_UNAUTHORIZED          , "Unauthorized"            },
     { HTTP_STATUS_FORBIDDEN             , "Forbidden"               },
diff -Naur klone-core-2.2.0/src/libutils/utils.c klone-core-2.2.0-atinar/src/libutils/utils.c
--- klone-core-2.2.0/src/libutils/utils.c	2008-10-20 11:45:52.000000000 +0200
+++ klone-core-2.2.0-atinar/src/libutils/utils.c	2009-03-24 02:23:16.000000000 +0100
@@ -312,6 +312,7 @@
 static ssize_t u_urlncpy_encode(char *d, const char *s, size_t slen)
 {
     const char hexc[] = "0123456789ABCDEF";
+	const char reserved[] = "-_.~!*'();:@&=+$,/?%#[]";
     ssize_t wr = 0;
     unsigned char c;
 
@@ -324,7 +325,7 @@
         if(c == ' ') {
             *d++ = '+';
             wr++;
-        } else if(isalnum(c) || c == '_' || c == '-' || c == '.') {
+        } else if ((c>='A' && c<='Z') || (c>='a' && c<='z') || (c>='0' && c<='9') || strchr(reserved,c)) {
             *d++ = c;
             wr++;
         } else {
