diff -Naur busybox-1.20.2_pro-orig/networking/udhcp/dhcpc.c busybox-1.20.2_pro/networking/udhcp/dhcpc.c
--- busybox-1.20.2_pro-orig/networking/udhcp/dhcpc.c	2012-07-02 16:08:25.000000000 +0200
+++ busybox-1.20.2_pro/networking/udhcp/dhcpc.c	2012-06-24 15:48:41.000000000 +0200
@@ -24,7 +24,7 @@
 #include "common.h"
 #include "dhcpd.h"
 #include "dhcpc.h"
-
+#include <asm/types.h>
 #include <netinet/if_ether.h>
 #include <netpacket/packet.h>
 #include <linux/filter.h>
