diff -Naur launcher-0.0.3-orig/i18n.c launcher-0.0.3/i18n.c
--- launcher-0.0.3-orig/i18n.c	2007-09-01 21:46:22.000000000 +0200
+++ launcher-0.0.3/i18n.c	2007-12-23 12:13:58.000000000 +0100
@@ -20,7 +20,7 @@
     "",// TODO Norsk
     "Laajennokset",
     "",// TODO Polski
-    "",// TODO Español
+    "Plugins",// Español
     "",// TODO ÅëëçíéêÜ (Greek)
     "",// TODO Svenska
     "",// TODO Romaneste
@@ -39,7 +39,7 @@
     "",// TODO Norsk
     "Laajennos laajennoksille",
     "",// TODO Polski
-    "",// TODO Español
+    "Plugin que lanza otros plugins",// Español
     "",// TODO ÅëëçíéêÜ (Greek)
     "",// TODO Svenska
     "",// TODO Romaneste
@@ -58,7 +58,7 @@
     "",// TODO Norsk
     "Laajennoksella ei ole infotekstiä!",
     "",// TODO Polski
-    "",// TODO Español
+    "¡Este plugin no tiene información de texto!",// Español
     "",// TODO ÅëëçíéêÜ (Greek)
     "",// TODO Svenska
     "",// TODO Romaneste
diff -Naur launcher-0.0.3-orig/Makefile launcher-0.0.3/Makefile
--- launcher-0.0.3-orig/Makefile	2007-09-01 21:43:39.000000000 +0200
+++ launcher-0.0.3/Makefile	2007-12-23 12:11:51.000000000 +0100
@@ -16,13 +16,13 @@
 ### The C++ compiler and options:
 
 CXX      ?= g++
-CXXFLAGS ?= -O2 -Wall -Woverloaded-virtual
+CXXFLAGS ?= -fPIC -O2 -Wall -Woverloaded-virtual
 
 ### The directory environment:
 
-DVBDIR = ../../../../DVB
-VDRDIR = ../../..
-LIBDIR = ../../lib
+DVBDIR ?= ../../..
+VDRDIR ?= ../../..
+LIBDIR ?= ../../lib
 TMPDIR = /tmp
 
 ### Allow user defined options to overwrite defaults:
@@ -40,7 +40,7 @@
 
 ### Includes and Defines (add further entries here):
 
-INCLUDES += -I$(VDRDIR)/include -I$(DVBDIR)/include
+INCLUDES += -I$(VDRDIR)/include -I$(DVBDIR)
 
 DEFINES += -DPLUGIN_NAME_I18N='"$(PLUGIN)"'
 DEFINES += -D_GNU_SOURCE
@@ -56,7 +56,7 @@
 
 # Dependencies:
 
-MAKEDEP = g++ -MM -MG
+MAKEDEP = $(CXX) -MM -MG
 DEPFILE = .dependencies
 $(DEPFILE): Makefile
 	@$(MAKEDEP) $(DEFINES) $(INCLUDES) $(OBJS:%.o=%.c) > $@
@@ -94,7 +94,7 @@
 
 libvdr-$(PLUGIN).so: $(OBJS)
 	$(CXX) $(CXXFLAGS) -shared $(OBJS) -o $@
-	@cp $@ $(LIBDIR)/$@.$(APIVERSION)
+	@cp -f $@ $(LIBDIR)/$@.$(APIVERSION)
 
 dist: clean
 	@-rm -rf $(TMPDIR)/$(ARCHIVE)
diff -Naur launcher-0.0.3-orig/po/de_DE.po~ launcher-0.0.3/po/de_DE.po~
--- launcher-0.0.3-orig/po/de_DE.po~	2007-09-01 21:43:39.000000000 +0200
+++ launcher-0.0.3/po/de_DE.po~	1970-01-01 01:00:00.000000000 +0100
@@ -1,28 +0,0 @@
-# VDR plugin language source file.
-# Copyright (C) 2007 Klaus Schmidinger <kls@cadsoft.de>
-# This file is distributed under the same license as the VDR package.
-# Klaus Schmidinger <kls@cadsoft.de>, 2000
-#
-msgid ""
-msgstr ""
-"Project-Id-Version: VDR 1.5.7\n"
-"Report-Msgid-Bugs-To: cwieninger@gmx.de\n"
-"POT-Creation-Date: 2007-09-01 21:43+0200\n"
-"PO-Revision-Date: 2007-09-01 21:43+0200\n"
-"Last-Translator: Klaus Schmidinger <kls@cadsoft.de>\n"
-"Language-Team: <vdr@linuxtv.org>\n"
-"MIME-Version: 1.0\n"
-"Content-Type: text/plain; charset=ISO-8859-15\n"
-"Content-Transfer-Encoding: 8bit\n"
-
-#: launcher.c:16
-msgid "A plugin to launch other plugins"
-msgstr ""
-
-#: launcher.c:17 launcher.c:60
-msgid "Plugins"
-msgstr ""
-
-#: launcher.c:166
-msgid "This plugin has no info text!"
-msgstr "Dieses Plugin hat keinen Infotext!"
diff -Naur launcher-0.0.3-orig/po/es_ES.po launcher-0.0.3/po/es_ES.po
--- launcher-0.0.3-orig/po/es_ES.po	2007-09-01 21:47:37.000000000 +0200
+++ launcher-0.0.3/po/es_ES.po	2007-12-23 12:18:09.000000000 +0100
@@ -1,15 +1,15 @@
 # VDR plugin language source file.
 # Copyright (C) 2007 Klaus Schmidinger <kls@cadsoft.de>
 # This file is distributed under the same license as the VDR package.
-# Ruben Nunez Francisco <ruben.nunez@tang-it.com>, 2002
+# bittor from open7x0.org <bittor7x0 _at_ gmail.com>, 2007
 #
 msgid ""
 msgstr ""
 "Project-Id-Version: VDR 1.5.7\n"
 "Report-Msgid-Bugs-To: <cwieninger@gmx.de>\n"
 "POT-Creation-Date: 2007-09-01 21:47+0200\n"
-"PO-Revision-Date: 2007-09-01 21:43+0200\n"
-"Last-Translator: Ruben Nunez Francisco <ruben.nunez@tang-it.com>\n"
+"PO-Revision-Date: 2007-12-07 19:31+0200\n"
+"Last-Translator: bittor from open7x0.org <bittor7x0 _at_ gmail.com>\n"
 "Language-Team: <vdr@linuxtv.org>\n"
 "MIME-Version: 1.0\n"
 "Content-Type: text/plain; charset=ISO-8859-15\n"
@@ -17,12 +17,12 @@
 
 #: launcher.c:16
 msgid "A plugin to launch other plugins"
-msgstr ""
+msgstr "Plugin que lanza otros plugins"
 
 #: launcher.c:17 launcher.c:60
 msgid "Plugins"
-msgstr ""
+msgstr "Plugins"
 
 #: launcher.c:166
 msgid "This plugin has no info text!"
-msgstr ""
+msgstr "¡Este plugin no tiene información de texto!"
