diff -Naur epgsearch-0.9.24.beta27-orig/epgsearchtools.h epgsearch-0.9.24.beta27/epgsearchtools.h
--- epgsearch-0.9.24.beta27-orig/epgsearchtools.h	Wed Apr 16 19:32:02 2008
+++ epgsearch-0.9.24.beta27/epgsearchtools.h	Mon Apr 21 16:30:36 2008
@@ -77,7 +77,7 @@
 #define ISRADIO(x) ((x)->Vpid()==0||(x)->Vpid()==1||(x)->Vpid()==0x1fff)
 
 // Icons used in VDRSymbols-Font
-
+/*
 #define ICON_REC               0x8B
 #define ICON_RUNNING           0x92
 #define ICON_CLOCK             0x8C
@@ -87,19 +87,19 @@
 #define ICON_BAR_EMPTY         0x89
 #define ICON_BAR_CLOSE         0x8A
 #define ICON_VPS               0x93
+*/
 
-/*
 // previous WAREAGLE icons
-#define ICON_REC               0xF9
+#define ICON_REC               0xE4
 #define ICON_RUNNING           0xFB
 #define ICON_CLOCK             0xFD
 #define ICON_CLOCK_HALF        0x74
 #define ICON_BAR_OPEN          0x82
-#define ICON_BAR_FULL          0xFF
+#define ICON_BAR_FULL          0x7F
 #define ICON_BAR_EMPTY         0x83
 #define ICON_BAR_CLOSE         0x84
 #define ICON_VPS               0x56
-*/
+
 
 // UTF-8 Icons
 #define ICON_BAR_OPEN_UTF8     "\uE007"
diff -Naur epgsearch-0.9.24.beta27-orig/Makefile epgsearch-0.9.24.beta27/Makefile
--- epgsearch-0.9.24.beta27-orig/Makefile	Sun Apr 13 20:53:42 2008
+++ epgsearch-0.9.24.beta27/Makefile	Mon Apr 21 16:30:00 2008
@@ -34,9 +34,9 @@
 
 ### The directory environment:
 
-DVBDIR = ../../../../DVB
-VDRDIR = ../../..
-LIBDIR = ../../lib
+DVBDIR ?= ../../../../DVB
+VDRDIR ?= ../../..
+LIBDIR ?= ../../lib
 TMPDIR = /tmp
 
 ### Allow user defined options to overwrite defaults:
@@ -69,7 +69,7 @@
 
 ### Includes and Defines (add further entries here):
 
-INCLUDES += -I$(VDRDIR)/include -I$(DVBDIR)/include
+INCLUDES += -I$(VDRDIR)/include -I$(DVBDIR)
 #INCLUDES += -I$(VDRDIR)/include
 
 DEFINES += -D_GNU_SOURCE -DPLUGIN_NAME_I18N='"$(PLUGIN)"'
@@ -128,7 +128,7 @@
 
 # Dependencies:
 
-MAKEDEP = g++ -MM -MG
+MAKEDEP = $(CXX) -MM -MG
 DEPFILE = .dependencies
 $(DEPFILE): Makefile
 	@$(MAKEDEP) $(CXXFLAGS) $(DEFINES) $(INCLUDES) $(OBJS:%.o=%.c) $(OBJS2:%.o=%.c) $(OBJS3:%.o=%.c) $(OBJS4:%.o=%.c)> $@
