--- uClibc/libc/sysdeps/linux/mips/crti.S.old	2007-04-02 17:27:40.000000000 +0200
+++ uClibc/libc/sysdeps/linux/mips/crti.S	2007-04-02 17:29:16.000000000 +0200
@@ -4,10 +4,8 @@
 	.section .mdebug.abi32
 	.previous
 	.abicalls
-#APP
 	
 	.section .init
-#NO_APP
 	.align	2
 	.globl	_init
 	.ent	_init
@@ -23,13 +21,11 @@
 	.cprestore 16
 	sw	$31,28($sp)
 	sw	$28,24($sp)
-#APP
 	
 	.align 2
 	.end _init
 	
 	.section .fini
-#NO_APP
 	.align	2
 	.globl	_fini
 	.ent	_fini
@@ -45,7 +41,6 @@
 	.cprestore 16
 	sw	$31,28($sp)
 	sw	$28,24($sp)
-#APP
 	.align 2
 	.end _fini
 	
@@ -55,10 +50,8 @@
 	.section .mdebug.abiN32
 	.previous
 	.abicalls
-#APP
 	
 	.section .init
-#NO_APP
 	.align	2
 	.align	3
 	.globl	_init
@@ -74,13 +67,11 @@
 	addu	$28,$28,$25
 	addiu	$28,$28,%lo(%neg(%gp_rel(_init)))
 	sd	$31,8($sp)
-#APP
 	
 	.align 3
 	.end _init
 
 	.section .fini
-#NO_APP
 	.align	2
 	.align	3
 	.globl	_fini
@@ -96,7 +87,6 @@
 	addu	$28,$28,$25
 	addiu	$28,$28,%lo(%neg(%gp_rel(_fini)))
 	sd	$31,8($sp)
-#APP
 	.align 3
 	.end _fini
 	
@@ -106,11 +96,9 @@
 	.section .mdebug.abi64
 	.previous
 	.abicalls
-#APP
 	
 	
 	.section .init
-#NO_APP
 	.align	2
 	.globl	_init
 	.ent	_init
@@ -127,13 +115,11 @@
 	addiu	$1,$1,%lo(%neg(%gp_rel(_init)))
 	daddu	$gp,$1,$25
 	.set	at
-#APP
 	
 	.align 2
 	.end _init
 	
 	.section .fini
-#NO_APP
 	.align	2
 	.globl	_fini
 	.ent	_fini
@@ -150,7 +136,6 @@
 	addiu	$1,$1,%lo(%neg(%gp_rel(_fini)))
 	daddu	$gp,$1,$25
 	.set	at
-#APP
 	.align 2
 	.end _fini
 	
--- uClibc/libc/sysdeps/linux/mips/crtn.S.old	2007-04-02 17:29:54.000000000 +0200
+++ uClibc/libc/sysdeps/linux/mips/crtn.S	2007-04-02 17:31:51.000000000 +0200
@@ -4,17 +4,13 @@
 	.section .mdebug.abi32
 	.previous
 	.abicalls
-#APP
 	
 	.section .init
-#NO_APP
 	.align	2
 	.globl	_init
 	.ent	_init
 	.type	_init, @function
-#NO_APP
 	lw	$31,28($sp)
-	#nop
 	.set	noreorder
 	.set	nomacro
 	j	$31
@@ -23,17 +19,13 @@
 	.set	reorder
 
 	.end	_init
-#APP
 	
 	.section .fini
-#NO_APP
 	.align	2
 	.globl	_fini
 	.ent	_fini
 	.type	_fini, @function
-#NO_APP
 	lw	$31,28($sp)
-	#nop
 	.set	noreorder
 	.set	nomacro
 	j	$31
@@ -42,7 +34,6 @@
 	.set	reorder
 
 	.end	_fini
-#APP
 	
 	.ident	"GCC: (GNU) 3.3.2"
 #elif _MIPS_SIM == _MIPS_SIM_NABI32
@@ -50,16 +41,13 @@
 	.section .mdebug.abiN32
 	.previous
 	.abicalls
-#APP
 	
 	.section .init
-#NO_APP
 	.align	2
 	.align	3
 	.globl	_init
 	.ent	_init
 	.type	_init, @function
-#NO_APP
 	ld	$31,8($sp)
 	ld	$28,0($sp)
 	.set	noreorder
@@ -70,16 +58,13 @@
 	.set	reorder
 
 	.end	_init
-#APP
 	
 	.section .fini
-#NO_APP
 	.align	2
 	.align	3
 	.globl	_fini
 	.ent	_fini
 	.type	_fini, @function
-#NO_APP
 	ld	$31,8($sp)
 	ld	$28,0($sp)
 	.set	noreorder
@@ -90,7 +75,6 @@
 	.set	reorder
 
 	.end	_fini
-#APP
 	
 	.ident	"GCC: (GNU) 3.4.3"
 #else /* N64 */
@@ -98,19 +82,15 @@
 	.section .mdebug.abi64
 	.previous
 	.abicalls
-#APP
 	
 	
 	.section .init
-#NO_APP
 	.align	2
 	.globl	_init
 	.ent	_init
 	.type	_init, @function
-#NO_APP
 	ld	$31,24($sp)
 	ld	$28,16($sp)
-	#nop
 	.set	noreorder
 	.set	nomacro
 	j	$31
@@ -119,18 +99,14 @@
 	.set	reorder
 
 	.end	_init
-#APP
 	
 	.section .fini
-#NO_APP
 	.align	2
 	.globl	_fini
 	.ent	_fini
 	.type	_fini, @function
-#NO_APP
 	ld	$31,24($sp)
 	ld	$28,16($sp)
-	#nop
 	.set	noreorder
 	.set	nomacro
 	j	$31
@@ -139,7 +115,6 @@
 	.set	reorder
 
 	.end	_fini
-#APP
 	
 	.ident	"GCC: (GNU) 3.3.2"
 #endif	/* N64 */
