diff -Naur uClibc-0.9.30.3-orig/libc/sysdeps/linux/mips/crtn.S uClibc-0.9.30.3/libc/sysdeps/linux/mips/crtn.S
--- uClibc-0.9.30.3-orig/libc/sysdeps/linux/mips/crtn.S	2011-09-13 15:24:41.000000000 +0200
+++ uClibc-0.9.30.3/libc/sysdeps/linux/mips/crtn.S	2011-09-13 20:06:00.000000000 +0200
@@ -19,6 +19,7 @@
 	.set	reorder
 
 	.end	_init
+	.size	_init,.-.init
 	
 	.section .fini
 	.align	2
@@ -34,6 +35,7 @@
 	.set	reorder
 
 	.end	_fini
+	.size	_fini,.-.fini
 	
 	.ident	"GCC: (GNU) 3.3.2"
 #elif _MIPS_SIM == _MIPS_SIM_NABI32
@@ -58,6 +60,7 @@
 	.set	reorder
 
 	.end	_init
+	.size	_init,.-.init
 	
 	.section .fini
 	.align	2
@@ -75,6 +78,7 @@
 	.set	reorder
 
 	.end	_fini
+	.size	_fini,.-.fini
 	
 	.ident	"GCC: (GNU) 3.4.3"
 #else /* N64 */
@@ -99,6 +103,7 @@
 	.set	reorder
 
 	.end	_init
+	.size	_init,.-.init
 	
 	.section .fini
 	.align	2
@@ -115,6 +120,7 @@
 	.set	reorder
 
 	.end	_fini
+	.size	_fini,.-.fini
 	
 	.ident	"GCC: (GNU) 3.3.2"
 #endif	/* N64 */
