diff -Naur newsticker-0.0.4-orig/Makefile newsticker-0.0.4/Makefile
--- newsticker-0.0.4-orig/Makefile	2008-07-08 20:27:24.000000000 +0200
+++ newsticker-0.0.4/Makefile	2008-09-20 13:53:15.000000000 +0200
@@ -16,11 +16,10 @@
 ### The C++ compiler and options:
 
 CXX      ?= g++
-CXXFLAGS ?= -O2 -Wall -Woverloaded-virtual -Wno-deprecated
+CXXFLAGS ?= -fPIC -g -O2 -Wall -Woverloaded-virtual
 
 ### The directory environment:
 
-DVBDIR = ../../../../DVB
 VDRDIR = ../../..
 LIBDIR = ../../lib
 TMPDIR = /tmp
@@ -29,9 +28,9 @@
 
 -include $(VDRDIR)/Make.config
 
-### The version number of VDR (taken from VDR's "config.h"):
+### The version number of VDR's plugin API (taken from VDR's "config.h"):
 
-VDRVERSION = $(shell grep 'define VDRVERSION ' $(VDRDIR)/config.h | awk '{ print $$3 }' | sed -e 's/"//g')
+APIVERSION = $(shell sed -ne '/define APIVERSION/s/^.*"\(.*\)".*$$/\1/p' $(VDRDIR)/config.h)
 
 ### The name of the distribution archive:
 
@@ -69,7 +68,7 @@
 
 libvdr-$(PLUGIN).so: $(OBJS)
 	$(CXX) $(CXXFLAGS) -shared $(OBJS) -o $@
-	@cp $@ $(LIBDIR)/$@.$(VDRVERSION)
+	@cp --remove-destination $@ $(LIBDIR)/$@.$(APIVERSION)
 
 dist: clean
 	@-rm -rf $(TMPDIR)/$(ARCHIVE)
diff -Naur newsticker-0.0.4-orig/news.c newsticker-0.0.4/news.c
--- newsticker-0.0.4-orig/news.c	2008-07-08 20:27:24.000000000 +0200
+++ newsticker-0.0.4/news.c	2008-09-20 13:53:30.000000000 +0200
@@ -6,7 +6,9 @@
  * $Id: news.c 0.0.1 2003/11/22 ew $
  */
  
-#include "news.h"
+#include "news.h"
+
+using std::ifstream;
 
 int ifindchar(int start,char* str,char c)
 {
@@ -180,7 +182,7 @@
 			MyConfigDir = cPlugin::ConfigDirectory(plugin_Name);
 		if(!MyConfigDir)
 			dsyslog("Error getting the config directory!");
-		sprintf(tmpSystem, "wget -O %s/newsticker.rdf -o %s/newsticker.log -T 5 %s",MyConfigDir, MyConfigDir, url);
+		sprintf(tmpSystem, "wget -O %s/newsticker.rdf  %s",MyConfigDir, url);
 		//dsyslog("newsticker: %s" , tmpSystem);
 		 
 		int dow_i = system (tmpSystem);
diff -Naur newsticker-0.0.4-orig/news.h newsticker-0.0.4/news.h
--- newsticker-0.0.4-orig/news.h	2008-07-08 20:27:24.000000000 +0200
+++ newsticker-0.0.4/news.h	2008-09-20 13:53:30.000000000 +0200
@@ -20,8 +20,8 @@
 #include <netinet/in.h>
 #include <netdb.h>
 #include <unistd.h>
-#include <iostream.h>
-#include <fstream.h>
+#include <iostream>
+#include <fstream>
 
 #define INT_SIZEOFBUF 300 * 1024 // Buffer 200Kb 
 
diff -Naur newsticker-0.0.4-orig/newsticker.c newsticker-0.0.4/newsticker.c
--- newsticker-0.0.4-orig/newsticker.c	2008-07-08 20:27:24.000000000 +0200
+++ newsticker-0.0.4/newsticker.c	2008-09-20 13:53:30.000000000 +0200
@@ -98,7 +98,7 @@
   void Show(void) {Start(); }
   eOSState ProcessKey(eKeys Key);
   int scrollMessage(char* message, int speed, int position, cOsd *osd, tColor ColorFg, tColor ColorBg, eDvbFont font);
-  cOsd* cOSDNewsticker::createOSDLine(int row, eDvbFont font);
+  cOsd* createOSDLine(int row, eDvbFont font);
   };
 
 cOSDNewsticker::cOSDNewsticker(void)
