diff -Naur weatherng-0.0.7-pre2-orig/i18n.c weatherng-0.0.7-pre2/i18n.c
--- weatherng-0.0.7-pre2-orig/i18n.c	Wed Mar 25 13:32:58 2009
+++ weatherng-0.0.7-pre2/i18n.c	Mon Sep 28 08:47:19 2009
@@ -19,7 +19,7 @@
     "",// TODO
     "Sääennuste",
     "",// TODO
-    "",// TODO
+    "Previsión meteorológica",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -37,7 +37,7 @@
     "",// TODO
     "Weather.com:in sääennusteet",
     "",// TODO
-    "",// TODO
+    "Weather.com: Previsión meteorológica",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -55,7 +55,7 @@
     "",// TODO
     "Sää",
     "",// TODO
-    "",// TODO
+    "Previsión",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -73,7 +73,7 @@
     "",// TODO
     "Virhe: Säätietoja ei saatavilla!",
     "",// TODO
-    "",// TODO
+    "No hay datos",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -91,7 +91,7 @@
     "",// TODO
     "Paina 'OK' päivittääksesi sääennusteet ja tutkakartat",
     "",// TODO
-    "",// TODO
+    "Pulsa OK para descargar información",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -109,7 +109,7 @@
     "",// TODO
     "Olethan muokannut weatherng.sh:n toimivaksi?",
     "",// TODO
-    "",// TODO
+    "¿Has editado weather.sh?",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -127,7 +127,7 @@
     "",// TODO
     "Weatherng.sh:n sijainti:",
     "",// TODO
-    "",// TODO
+    "La ruta a weather.sh es: ",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -145,7 +145,7 @@
     "",// TODO
     "Säätietojen sijainti:",
     "",// TODO
-    "",// TODO
+    "La ruta a weatherdata es: ",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -163,7 +163,7 @@
     "",// TODO
     "Aurinko laskee: ",
     "",// TODO
-    "",// TODO
+    "Puesta de sol: ",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -181,7 +181,7 @@
     "",// TODO
     "Aurinko nousee:",
     "",// TODO
-    "",// TODO
+    "Amanecer: ",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -199,7 +199,7 @@
     "",// TODO
     "Säätila: ",
     "",// TODO
-    "",// TODO
+    "Previsión",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -217,7 +217,7 @@
     "",// TODO
     "Alin lämpötila: ",
     "",// TODO
-    "",// TODO
+    "Mínima",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -235,7 +235,7 @@
     "",// TODO
     "Ylin lämpötila: ",
     "",// TODO
-    "",// TODO
+    "Máxima",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -253,7 +253,7 @@
     "",// TODO
     "Sää: ",
     "",// TODO
-    "",// TODO
+    "Previsión para: ",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -271,7 +271,7 @@
     "",// TODO
     "Tuulen suunta: ",
     "",// TODO
-    "",// TODO
+    "Dirección del viento:",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -289,7 +289,7 @@
     "",// TODO
     "Tuulen nopeus: ",
     "",// TODO
-    "",// TODO
+    "Viento:",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -307,7 +307,7 @@
     "",// TODO
     " km/h",
     "",// TODO
-    "",// TODO
+    " km/h",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -325,7 +325,7 @@
     "",// TODO
     "Ilmankosteus: ",
     "",// TODO
-    "",// TODO
+    "Humedad: ",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -343,7 +343,7 @@
     "",// TODO
     "Sateen tod.näk.: ",
     "",// TODO
-    "",// TODO
+    "Precipitación: ",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -361,7 +361,7 @@
     "",// TODO
     " %",
     "",// TODO
-    "",// TODO
+    " %",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -379,7 +379,7 @@
     "",// TODO
     "°C",
     "",// TODO
-    "",// TODO
+    "°C",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -397,7 +397,7 @@
     "",// TODO
     "Päivitä säätiedot käynnistettäessä",
     "",// TODO
-    "",// TODO
+    "Actualizar datos al arranque de VDR",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -415,7 +415,7 @@
     "",// TODO
     "Tutkakuvan vaakakeskitys",
     "",// TODO
-    "",// TODO
+    "Posición imagen radar (izquierda)",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -433,7 +433,7 @@
     "",// TODO
     "Tutkakuvan pystykeskitys",
     "",// TODO
-    "",// TODO
+    "Posición imagen radar (arriba)",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -451,7 +451,7 @@
     "",// TODO
     "Tutkakuvan leveys",
     "",// TODO
-    "",// TODO
+    "Ancho imagen de radar",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -469,7 +469,7 @@
     "",// TODO
     "Tutkakuvan korkeus",
     "",// TODO
-    "",// TODO
+    "Alto imagen de radar",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -487,7 +487,7 @@
     "",// TODO
     "Näytä tutkakuvat 256 värillä",
     "",// TODO
-    "",// TODO
+    "Usar 256 colores para radarmaps",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -505,7 +505,7 @@
     "",// TODO
     "Käytä pieniä kirjasimia",
     "",// TODO
-    "",// TODO
+    "Usar fuentes pequeñas",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -523,7 +523,7 @@
     "",// TODO
     "Klassinen",
     "",// TODO
-    "",// TODO
+    "Clásico",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -541,7 +541,7 @@
     "",// TODO
     "enElchi",
     "",// TODO
-    "",// TODO
+    "enElchi",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -559,7 +559,7 @@
     "",// TODO
     "Moronimo",
     "",// TODO
-    "",// TODO
+    "Moronimo",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -577,7 +577,7 @@
     "",// TODO
     "DeepBlue",
     "",// TODO
-    "",// TODO
+    "DeepBlue",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -595,7 +595,7 @@
     "",// TODO
     "Enigma",
     "",// TODO
-    "",// TODO
+    "Enigma",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -613,7 +613,7 @@
     "",// TODO
     "SilverGreen",
     "",// TODO
-    "",// TODO
+    "SilverGreen",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -631,7 +631,7 @@
     "",// TODO
     "MoronimoMKII",
     "",// TODO
-    "",// TODO
+    "MoronimoMKII",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -649,7 +649,7 @@
     "",// TODO
     "MoronimoMKIII",
     "",// TODO
-    "",// TODO
+    "MoronimoMKIII",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -667,7 +667,7 @@
     "",// TODO
     "MoronimoMKIV",
     "",// TODO
-    "",// TODO
+    "MoronimoMKIV",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -685,7 +685,7 @@
     "",// TODO
     "Teema",
     "",// TODO
-    "",// TODO
+    "Tema",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -703,7 +703,7 @@
     "",// TODO
     "Korkeus",
     "",// TODO
-    "",// TODO
+    "Altura",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -721,7 +721,7 @@
     "",// TODO
     "Vaakakeskitys",
     "",// TODO
-    "",// TODO
+    "Desplazamiento horizontal",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -739,7 +739,7 @@
     "",// TODO
     "Pystykeskitys",
     "",// TODO
-    "",// TODO
+    "Desplazamiento vertical",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -757,7 +757,7 @@
     "",// TODO
     "Näytä symbolit invertoituna",
     "",// TODO
-    "",// TODO
+    "Invertir símbolos",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -775,7 +775,7 @@
     "",// TODO
     "Käytä pyöristettyjä kulmia",
     "",// TODO
-    "",// TODO
+    "Bordes redondeados",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -793,7 +793,7 @@
     "",// TODO
     "PÄIVÄ",
     "",// TODO
-    "",// TODO
+    "DIA",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -811,7 +811,7 @@
     "",// TODO
     "YÖ",
     "",// TODO
-    "",// TODO
+    "NOCHE",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -829,7 +829,7 @@
     "",// TODO
     "Maanantai",
     "",// TODO
-    "",// TODO
+    "Lunes",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -847,7 +847,7 @@
     "",// TODO
     "Tiistai",
     "",// TODO
-    "",// TODO
+    "Martes",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -865,7 +865,7 @@
     "",// TODO
     "Keskiviikko",
     "",// TODO
-    "",// TODO
+    "Miércoles",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -883,7 +883,7 @@
     "",// TODO
     "Torstai",
     "",// TODO
-    "",// TODO
+    "Jueves",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -901,7 +901,7 @@
     "",// TODO
     "Perjantai",
     "",// TODO
-    "",// TODO
+    "Viernes",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -919,7 +919,7 @@
     "",// TODO
     "Lauantai",
     "",// TODO
-    "",// TODO
+    "Sábado",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -937,7 +937,7 @@
     "",// TODO
     "Sunnuntai",
     "",// TODO
-    "",// TODO
+    "Domingo",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -955,7 +955,7 @@
     "",// TODO
     "aurinkoista",
     "",// TODO
-    "",// TODO
+    "Soleado",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -973,7 +973,7 @@
     "",// TODO
     "sateista",
     "",// TODO
-    "",// TODO
+    "Lluvia",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -991,7 +991,7 @@
     "",// TODO
     "ukkosta",
     "",// TODO
-    "",// TODO
+    "Truenos",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -1009,7 +1009,7 @@
     "",// TODO
     "pilvistä",
     "",// TODO
-    "",// TODO
+    "Nublado",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -1027,7 +1027,7 @@
     "",// TODO
     "enimmäkseen pilvistä",
     "",// TODO
-    "",// TODO
+    "Parcialmente nublado",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -1045,7 +1045,7 @@
     "",// TODO
     "iltapäivällä ukkoskuuroja",
     "",// TODO
-    "",// TODO
+    "Tormenta eléctrica nocturna",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -1063,7 +1063,7 @@
     "",// TODO
     "aamupäivällä ukkoskuuroja",
     "",// TODO
-    "",// TODO
+    "Tormenta eléctrica matinal",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -1081,7 +1081,7 @@
     "",// TODO
     "kevyttä sadetta",
     "",// TODO
-    "",// TODO
+    "Lluvia ligera",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -1099,7 +1099,7 @@
     "",// TODO
     "lumisadetta",
     "",// TODO
-    "",// TODO
+    "Nieve",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -1117,7 +1117,7 @@
     "",// TODO
     "sadekuuroja",
     "",// TODO
-    "",// TODO
+    "Chubascos",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -1135,16 +1135,16 @@
     "",// TODO
     "aamupäivällä sadekuuroja",
     "",// TODO
+    "Chubascos matinales",
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
-    "",// TODO
-    "¿àØÒÕâ!",                                                                                        
+    "¿àØÒÕâ!",
   },
-  { "PM Showers",                                                                                     
-    "nachmittags Schauer",                                                                        
+  { "PM Showers",
+    "nachmittags Schauer",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -1153,13 +1153,13 @@
     "",// TODO
     "iltapäivällä sadekuuroja",
     "",// TODO
+    "Chubascos nocturnos",
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
-    "",// TODO
-    "¿àØÒÕâ!",                                                                                        
+    "¿àØÒÕâ!",
   },
   { "Scattered Showers",
     "vereinzelt Schauer",
@@ -1171,7 +1171,7 @@
     "",// TODO
     "hajanaisia sadekuuroja",
     "",// TODO
-    "",// TODO
+    "Chubascos dispersos",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -1189,7 +1189,7 @@
     "",// TODO
     "sadetta / ukkosta",
     "",// TODO
-    "",// TODO
+    "Lluvia / Truenos",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -1207,7 +1207,7 @@
     "",// TODO
     "sadetta / tuulista",
     "",// TODO
-    "",// TODO
+    "Lluvia / Viento",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -1225,7 +1225,7 @@
     "",// TODO
     "enimmäkseen selkeää",
     "",// TODO
-    "",// TODO
+    "Parcialmente despejado",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -1243,7 +1243,7 @@
     "",// TODO
     "utua",
     "",// TODO
-    "",// TODO
+    "Bruma",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -1261,7 +1261,7 @@
     "",// TODO
     "sumua",
     "",// TODO
-    "",// TODO
+    "Niebla",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -1279,7 +1279,7 @@
     "",// TODO
     "tuulista",
     "",// TODO
-    "",// TODO
+    "Viento",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -1297,7 +1297,7 @@
     "",// TODO
     "poutaa",
     "",// TODO
-    "",// TODO
+    "Buen tiempo",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -1315,7 +1315,7 @@
     "",// TODO
     "selkeää",
     "",// TODO
-    "",// TODO
+    "Despejado",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -1333,7 +1333,7 @@
     "",// TODO
     "tuulista",
     "",// TODO
-    "",// TODO
+    "Ventoso",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -1351,7 +1351,7 @@
     "",// TODO
     "tihkua",
     "",// TODO
-    "",// TODO
+    "Llovizna",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -1369,7 +1369,7 @@
     "",// TODO
     "hallaa",
     "",// TODO
-    "",// TODO
+    "Heladas",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -1387,7 +1387,7 @@
     "",// TODO
     "enimmäkseen pilvistä",
     "",// TODO
-    "",// TODO
+    "Principalmente nublado",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -1405,13 +1405,13 @@
     "",// TODO
     "osittain pilvistä",
     "",// TODO
+    "Parcialmente nublado",
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
-    "",// TODO
-    "¿àØÒÕâ!",                                                                                        
+    "¿àØÒÕâ!",
   },
   { "M Clear",
     "meist klar",
@@ -1423,7 +1423,7 @@
     "",// TODO
     "enimmäkseen selkeää",
     "",// TODO
-    "",// TODO
+    "Principalmente despejado",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -1441,7 +1441,7 @@
     "",// TODO
     "ei tietoja",
     "",// TODO
-    "",// TODO
+    "N/D",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -1459,7 +1459,7 @@
     "",// TODO
     "enimmäkseen aurinkoista",
     "",// TODO
-    "",// TODO
+    "Principalmente soleado",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -1477,7 +1477,7 @@
     "",// TODO
     "osittain pilvistä",
     "",// TODO
-    "",// TODO
+    "Parcialmente nublado",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -1495,7 +1495,7 @@
     "",// TODO
     "sadetta myöhemmin",
     "",// TODO
-    "",// TODO
+    "Lluvia por la tarde",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -1513,7 +1513,7 @@
     "",// TODO
     "aamupäivällä pilvistä / iltapäivällä aurinkoista",
     "",// TODO
-    "",// TODO
+    "Mañana nubes / Tarde sol",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -1532,7 +1532,7 @@
     "",// TODO
     "hajanaisia ukkosmyrskyjä",
     "",// TODO
-    "",// TODO
+    "Tormentas eléctricas dispersas",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -1550,7 +1550,7 @@
     "",// TODO
     "pilvistä",
     "",// TODO
-    "",// TODO
+    "Encapotado",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -1568,7 +1568,7 @@
     "",// TODO
     "sumua",
     "",// TODO
-    "",// TODO
+    "Niebla",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -1586,7 +1586,7 @@
     "",// TODO
     "usvaa",
     "",// TODO
-    "",// TODO
+    "Niebla ligera",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -1604,7 +1604,7 @@
     "",// TODO
     "hernerokkasumua",
     "",// TODO
-    "",// TODO
+    "Niebla densa (maze)",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -1622,7 +1622,7 @@
     "",// TODO
     "myrskyä",
     "",// TODO
-    "",// TODO
+    "Tormenta",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -1640,7 +1640,7 @@
     "",// TODO
     "lumisadetta",
     "",// TODO
-    "",// TODO
+    "Nieve",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -1658,7 +1658,7 @@
     "",// TODO
     "kevyttä sadetta",
     "",// TODO
-    "",// TODO
+    "Lluvia ligera",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -1676,14 +1676,14 @@
     "",// TODO
     "kevyttä lumisadetta",
     "",// TODO
-    "",// TODO
+    "Nieve ligera",
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "¿àØÒÕâ!",
-  },  
+  },
     { "Few Snow",
     "wenig Schnee",
     "",// TODO
@@ -1694,14 +1694,14 @@
     "",// TODO
     "heikkoa lumisadetta",
     "",// TODO
-    "",// TODO
+    "Algo de nieve",
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "¿àØÒÕâ!",
-  }, 
+  },
     { "Clear",
     "Klar",
     "",// TODO
@@ -1712,14 +1712,14 @@
     "",// TODO
     "selkeää",
     "",// TODO
-    "",// TODO
+    "Despejado",
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "¿àØÒÕâ!",
-  },   
+  },
     { "Light Drizzle",
     "wenig Nieselregen",
     "",// TODO
@@ -1730,14 +1730,14 @@
     "",// TODO
     "kevyttä tihkua",
     "",// TODO
-    "",// TODO
+    "Llovizna",
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "¿àØÒÕâ!",
-  },  
+  },
     { "Scattered",
     "vereinzelt",
     "",// TODO
@@ -1748,14 +1748,14 @@
     "",// TODO
     "hajanaista",
     "",// TODO
-    "",// TODO
+    "Nubes dispersas",
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "¿àØÒÕâ!",
-  },  
+  },
     { "Smoke",
     "dichter Nebel",
     "",// TODO
@@ -1766,14 +1766,14 @@
     "",// TODO
     "paksua sumua",
     "",// TODO
-    "",// TODO
+    "Humo",
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "¿àØÒÕâ!",
-  },  
+  },
     { "Drifting Snow",
     "starkes Schneetreiben",
     "",// TODO
@@ -1784,14 +1784,14 @@
     "",// TODO
     "voimakasta lumisadetta",
     "",// TODO
-    "",// TODO
+    "Nieve levantada por el viento",
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "¿àØÒÕâ!",
-  },  
+  },
     { "steady",
     "",
     "",// TODO
@@ -1802,14 +1802,14 @@
     "",// TODO
     "tasaista",
     "",// TODO
-    "",// TODO
+    "constante",
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "¿àØÒÕâ!",
-  },  
+  },
     { "rising",
     "",
     "",// TODO
@@ -1820,14 +1820,14 @@
     "",// TODO
     "alkavaa",
     "",// TODO
-    "",// TODO
+    "el aumento de ",
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "¿àØÒÕâ!",
-  },  
+  },
     { "falling",
     "",
     "",// TODO
@@ -1838,14 +1838,14 @@
     "",// TODO
     "laantuvaa",
     "",// TODO
-    "",// TODO
+    "caída",
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "¿àØÒÕâ!",
-  },  
+  },
     { "Unlimited",
     "",
     "",// TODO
@@ -1856,14 +1856,14 @@
     "",// TODO
     "määrittelemätöntä",
     "",// TODO
-    "",// TODO
+    "Ilimitado",
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "¿àØÒÕâ!",
-  },  
+  },
     { "Showers in the Vicinity",
     "",
     "",// TODO
@@ -1874,14 +1874,14 @@
     "",// TODO
     "todennäköisesti sadekuuroja",
     "",// TODO
-    "",// TODO
+    "Chubascos en los alrededores",
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "¿àØÒÕâ!",
-  },  
+  },
     { "Waning Crescent",
     "",
     "",// TODO
@@ -1892,14 +1892,14 @@
     "",// TODO
     "väistyvä täysikuu",
     "",// TODO
-    "",// TODO
+    "Nueva visible",
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "¿àØÒÕâ!",
-  },  
+  },
     { "Waxing Crescent",
     "",
     "",// TODO
@@ -1910,14 +1910,14 @@
     "",// TODO
     "alkava täysikuu",
     "",// TODO
-    "",// TODO
+    "Media luna creciente",
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "¿àØÒÕâ!",
-  },  
+  },
     { "Waxing Gibbous",
     "",
     "",// TODO
@@ -1928,14 +1928,14 @@
     "",// TODO
     "väistyvä uusikuu",
     "",// TODO
-    "",// TODO
+    "Gibada creciente",
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "¿àØÒÕâ!",
-  },  
+  },
     { "Waning Gibbous",
     "",
     "",// TODO
@@ -1946,14 +1946,14 @@
     "",// TODO
     "väistyvä täysikuu",
     "",// TODO
-    "",// TODO
+    "Gibada menguante",
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "¿àØÒÕâ!",
-  },  
+  },
     { "T-Storms",
     "stürmisches Gewitter",
     "",// TODO
@@ -1964,14 +1964,14 @@
     "",// TODO
     "ukkosmyrskyjä",
     "",// TODO
-    "",// TODO
+    "Tormentas eléctricas",
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "¿àØÒÕâ!",
-  },  
+  },
     { "T-Storms Early",
     "zunächst schweres Gewitter",
     "",// TODO
@@ -1982,14 +1982,14 @@
     "",// TODO
     "ukkosmyrskyjä aikaisin",
     "",// TODO
-    "",// TODO
+    "Tormentas eléctricas tempranas",
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "¿àØÒÕâ!",
-  },  
+  },
     { "Light Rain Early",
     "zunächst leichter Regen",
     "",// TODO
@@ -2000,14 +2000,14 @@
     "",// TODO
     "kevyttä sadetta aikaisin",
     "",// TODO
-    "",// TODO
+    "Lluvia fina de mañana",
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "¿àØÒÕâ!",
-  },  
+  },
     { "Showers Early",
     "zunächst Regenschauer",
     "",// TODO
@@ -2018,7 +2018,7 @@
     "",// TODO
     "sadekuuroja aikaisin",
     "",// TODO
-    "",// TODO
+    "Chubascos de mañana",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -2036,7 +2036,7 @@
     "",// TODO
     "pilvistä aikaisin",
     "",// TODO
-    "",// TODO
+    "Nublado de mañana",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -2054,7 +2054,7 @@
     "",// TODO
     "iltapäivällä kevyttä sadetta",
     "",// TODO
-    "",// TODO
+    "Lluvia ligera por la tarde",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -2072,7 +2072,7 @@
     "",// TODO
     "aamupäivällä kevyttä sadetta",
     "",// TODO
-    "",// TODO
+    "Lluvia ligera por la mañana",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -2090,7 +2090,7 @@
     "",// TODO
     "kevyttä sadetta myöhään",
     "",// TODO
-    "",// TODO
+    "Lluvia ligera por la tarde",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -2108,13 +2108,67 @@
     "",// TODO
     "ajoittaisia sadekuuroja",
     "",// TODO
-    "",// TODO
+    "Algunos chubascos",
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "",// TODO
     "¿àØÒÕâ!",
+  },
+      { "Weather station 1",
+      "",
+      "",// TODO
+      "",// TODO
+      "",// TODO
+      "",// TODO
+      "",// TODO
+      "",// TODO
+      "",
+      "",// TODO
+      "Estación meteorológica 1",
+      "",// TODO
+      "",// TODO
+      "",// TODO
+      "",// TODO
+      "",// TODO
+      "¿àØÒÕâ!",
+  },
+      { "Weather station 2",
+      "",// TODO
+      "",// TODO
+      "",// TODO
+      "",// TODO
+      "",// TODO
+      "",// TODO
+      "",// TODO
+      "",// TODO
+      "",// TODO
+      "Estación meteorológica 2",
+      "",// TODO
+      "",// TODO
+      "",// TODO
+      "",// TODO
+      "",// TODO
+      "¿àØÒÕâ!",
+  },
+      { "Weather station 3",
+      "",// TODO
+      "",// TODO
+      "",// TODO
+      "",// TODO
+      "",// TODO
+      "",// TODO
+      "",// TODO
+      "",// TODO
+      "",// TODO
+      "Estación meteorológica 3",
+      "",// TODO
+      "",// TODO
+      "",// TODO
+      "",// TODO
+      "",// TODO
+      "¿àØÒÕâ!",
   },
   { NULL }
   };
diff -Naur weatherng-0.0.7-pre2-orig/OsdWeather.c weatherng-0.0.7-pre2/OsdWeather.c
--- weatherng-0.0.7-pre2-orig/OsdWeather.c	Wed Mar 25 13:32:58 2009
+++ weatherng-0.0.7-pre2/OsdWeather.c	Mon Sep 28 09:02:51 2009
@@ -271,24 +271,21 @@
 	char *url;
 
 	asprintf(&buffer, "%s%s",ScriptDir,"/data1.xml");
-//	asprintf(&url, "%s%s%s","http://xoap.weather.com/weather/local/",wetterSetup.st_data1,"?cc=*&unit=m&dayf=10&prod=xoap&par=1004124588&key=079f24145f208494");
-	asprintf(&url, "%s%s%s","http://xoap.weather.com/weather/local/",wetterSetup.st_data1,"?cc=*&unit=m&dayf=10");
+	asprintf(&url, "%s%s%s","http://xoap.weather.com/weather/local/",wetterSetup.st_data1,"?cc=*&unit=m&dayf=10&prod=xoap&par=1004124588&key=079f24145f208494");
 	dsyslog("DEBUG : weatherng: %s => %s",url, buffer);
 	httpGet(url,buffer);
 	free(buffer);
 	free(url);
 
 	asprintf(&buffer, "%s%s",ScriptDir,"/data2.xml");
-//	asprintf(&url, "%s%s%s","http://xoap.weather.com/weather/local/",wetterSetup.st_data2,"?cc=*&unit=m&dayf=10&prod=xoap&par=1004124588&key=079f24145f208494");
-	asprintf(&url, "%s%s%s","http://xoap.weather.com/weather/local/",wetterSetup.st_data2,"?cc=*&unit=m&dayf=10");
+	asprintf(&url, "%s%s%s","http://xoap.weather.com/weather/local/",wetterSetup.st_data2,"?cc=*&unit=m&dayf=10&prod=xoap&par=1004124588&key=079f24145f208494");
 	dsyslog("DEBUG : weatherng: %s => %s",url, buffer);
 	httpGet(url,buffer);
 	free(buffer);
 	free(url);
 
 	asprintf(&buffer, "%s%s",ScriptDir,"/data3.xml");
-//	asprintf(&url, "%s%s%s","http://xoap.weather.com/weather/local/",wetterSetup.st_data3,"?cc=*&unit=m&dayf=10&prod=xoap&par=1004124588&key=079f24145f208494");
-	asprintf(&url, "%s%s%s","http://xoap.weather.com/weather/local/",wetterSetup.st_data3,"?cc=*&unit=m&dayf=10");
+	asprintf(&url, "%s%s%s","http://xoap.weather.com/weather/local/",wetterSetup.st_data3,"?cc=*&unit=m&dayf=10&prod=xoap&par=1004124588&key=079f24145f208494");
 	dsyslog("DEBUG : weatherng: %s => %s",url, buffer);
 	httpGet(url,buffer);
 	free(buffer);
diff -Naur weatherng-0.0.7-pre2-orig/setup.c weatherng-0.0.7-pre2/setup.c
--- weatherng-0.0.7-pre2-orig/setup.c	Wed Mar 25 13:32:58 2009
+++ weatherng-0.0.7-pre2/setup.c	Mon Sep 28 08:47:19 2009
@@ -30,9 +30,9 @@
 	w_osdoffset_y = 0;
 	w_inverted = 1;
 	w_corner = 1;
-	strcpy(st_data1,"GMXX0040");
-	strcpy(st_data2,"GMXX0040");
-	strcpy(st_data3,"GMXX0040");
+	strcpy(st_data1,"SPXX0050");
+	strcpy(st_data2,"SPXX0015");
+	strcpy(st_data3,"SPXX0016");
 	st_pic0=NULL;
 	st_pic1=NULL;
 	st_pic2=NULL;
diff -Naur weatherng-0.0.7-pre2-orig/weatherng.c weatherng-0.0.7-pre2/weatherng.c
--- weatherng-0.0.7-pre2-orig/weatherng.c	Wed Mar 25 13:32:58 2009
+++ weatherng-0.0.7-pre2/weatherng.c	Mon Sep 28 09:03:04 2009
@@ -112,24 +112,21 @@
 		char *url;
 
 		asprintf(&buffer, "%s%s",ScriptDir,"/data1.xml");
-//		asprintf(&url, "%s%s%s","http://xoap.weather.com/weather/local/",wetterSetup.st_data1,"?cc=*&unit=m&dayf=10&prod=xoap&par=1004124588&key=079f24145f208494");
-		asprintf(&url, "%s%s%s","http://xoap.weather.com/weather/local/",wetterSetup.st_data1,"?cc=*&unit=m&dayf=10");
+		asprintf(&url, "%s%s%s","http://xoap.weather.com/weather/local/",wetterSetup.st_data1,"?cc=*&unit=m&dayf=10&prod=xoap&par=1004124588&key=079f24145f208494");
 		dsyslog("DEBUG : weatherng: Autoupdate %s => %s",url, buffer);
 		httpGet(url,buffer);
 		free(buffer);
 		free(url);
 
 		asprintf(&buffer, "%s%s",ScriptDir,"/data2.xml");
-//		asprintf(&url, "%s%s%s","http://xoap.weather.com/weather/local/",wetterSetup.st_data2,"?cc=*&unit=m&dayf=10&prod=xoap&par=1004124588&key=079f24145f208494");
-		asprintf(&url, "%s%s%s","http://xoap.weather.com/weather/local/",wetterSetup.st_data2,"?cc=*&unit=m&dayf=10");
+		asprintf(&url, "%s%s%s","http://xoap.weather.com/weather/local/",wetterSetup.st_data2,"?cc=*&unit=m&dayf=10&prod=xoap&par=1004124588&key=079f24145f208494");
 		dsyslog("DEBUG : weatherng: %s => %s",url, buffer);
 		httpGet(url,buffer);
 		free(buffer);
 		free(url);
 
 		asprintf(&buffer, "%s%s",ScriptDir,"/data3.xml");
-//		asprintf(&url, "%s%s%s","http://xoap.weather.com/weather/local/",wetterSetup.st_data3,"?cc=*&unit=m&dayf=10&prod=xoap&par=1004124588&key=079f24145f208494");
-		asprintf(&url, "%s%s%s","http://xoap.weather.com/weather/local/",wetterSetup.st_data3,"?cc=*&unit=m&dayf=10");
+		asprintf(&url, "%s%s%s","http://xoap.weather.com/weather/local/",wetterSetup.st_data3,"?cc=*&unit=m&dayf=10&prod=xoap&par=1004124588&key=079f24145f208494");
 		dsyslog("DEBUG : weatherng: %s => %s",url, buffer);
 		httpGet(url,buffer);
 		free(buffer);
