diff -Naur slin_m740_orig/fs/squashfs/Makefile slin_m740/fs/squashfs/Makefile
--- slin_m740_orig/fs/squashfs/Makefile	2009-07-22 21:35:02.000000000 +0200
+++ slin_m740/fs/squashfs/Makefile	2009-07-23 20:55:36.000000000 +0200
@@ -4,7 +4,7 @@
 
 O_TARGET := squashfs.o
 
-obj-y  := inode.o squashfs2_0.o
+obj-y  := inode.o
 
 obj-m := $(O_TARGET)
 
diff -Naur slin_m740_orig/fs/squashfs/squashfs.h slin_m740/fs/squashfs/squashfs.h
--- slin_m740_orig/fs/squashfs/squashfs.h	2009-07-22 21:35:02.000000000 +0200
+++ slin_m740/fs/squashfs/squashfs.h	2009-07-23 20:55:34.000000000 +0200
@@ -24,6 +24,9 @@
 #ifdef CONFIG_SQUASHFS_1_0_COMPATIBILITY
 #undef CONFIG_SQUASHFS_1_0_COMPATIBILITY
 #endif
+#ifdef CONFIG_SQUASHFS_2_0_COMPATIBILITY
+#undef CONFIG_SQUASHFS_2_0_COMPATIBILITY
+#endif
 #ifdef SQUASHFS_TRACE
 #define TRACE(s, args...)	printk(KERN_NOTICE "SQUASHFS: "s, ## args)
 #else
diff -Naur slin_m740_orig/include/linux/squashfs_fs.h slin_m740/include/linux/squashfs_fs.h
--- slin_m740_orig/include/linux/squashfs_fs.h	2009-07-22 21:35:02.000000000 +0200
+++ slin_m740/include/linux/squashfs_fs.h	2009-09-22 18:24:22.000000000 +0200
@@ -24,10 +24,6 @@
  * squashfs_fs.h
  */
 
-#ifndef CONFIG_SQUASHFS_2_0_COMPATIBILITY
-#define CONFIG_SQUASHFS_2_0_COMPATIBILITY
-#endif
-
 #ifdef CONFIG_SQUASHFS_VMALLOC
 #define SQUASHFS_ALLOC(a)		vmalloc(a)
 #define SQUASHFS_FREE(a)		vfree(a)
