diff -Naur a/extra/locale/Makefile.in b/extra/locale/Makefile.in
--- a/extra/locale/Makefile.in	2015-07-05 22:09:46.000000000 +0200
+++ b/extra/locale/Makefile.in	2015-12-28 22:37:24.421487891 +0100
@@ -104,6 +104,7 @@
 ifeq ($(UCLIBC_BUILD_MINIMAL_LOCALE),y)
 	$(Q)echo "$(CURDIR)/$(locale_DIR)/charmaps/ASCII.pairs" > $@
 	$(Q)echo "$(CURDIR)/$(locale_DIR)/charmaps/ISO-8859-1.pairs" >> $@
+	$(Q)echo "$(CURDIR)/$(locale_DIR)/charmaps/ISO-8859-15.pairs" >> $@
 else
 	$(Q)set -e; \
 		find $(CURDIR)/$(locale_DIR)/charmaps/ -name '*.pairs' | \
@@ -123,9 +124,13 @@
 	$(Q)echo "8-BIT yes" >> $@
 	$(Q)echo "#-" >> $@
 	$(Q)for locale in $(call qstrip,$(UCLIBC_BUILD_MINIMAL_LOCALES)); do \
-		echo "$$locale.UTF-8 UTF-8"; \
-		echo "$$locale ISO-8859-1"; \
-	done >> $@
+		echo "$$locale.UTF-8 UTF-8" >> $@; \
+		echo "$$locale ISO-8859-1" >> $@; \
+		if [ "$$locale" != "en_US" ]; then \
+			echo "$$locale.UTF-8@euro UTF-8" >> $@; \
+			echo "$$locale@euro ISO-8859-15" >> $@; \
+		fi; \
+	done
 else
 	$(Q)cat $< > $@
 endif
@@ -169,6 +174,7 @@
 	$(Q)for locale in $(call qstrip,$(UCLIBC_BUILD_MINIMAL_LOCALES)) C; do \
 		$< $(FLAG-locale-verbose) $$locale.UTF-8 > $@ || \
 		$< $(FLAG-locale-verbose) $$locale > $@ || \
+		$< $(FLAG-locale-verbose) $$locale.iso8859-15 > $@ || \
 		$< $(FLAG-locale-verbose) $$locale.iso8859-1 > $@ && break; \
 	done
 
