diff -Naur epgsearch-0.9.25.beta14-orig/epgsearchcfg.c epgsearch-0.9.25.beta14/epgsearchcfg.c
--- epgsearch-0.9.25.beta14-orig/epgsearchcfg.c	Mon Jul 27 10:51:10 2009
+++ epgsearch-0.9.25.beta14/epgsearchcfg.c	Fri Sep 04 09:47:46 2009
@@ -58,7 +58,16 @@
   DefSearchTemplateID = -1;
   addSubtitleToTimer = addSubtitleSmart;
   strcpy(mainmenuentry, "");
-  WarEagle = 0;
+#if APIVERSNUM >= 10503
+  if (strstr(Setup.FontOsd, "VDRSymbols") == Setup.FontOsd)
+    WarEagle = 1;
+  else
+    WarEagle = 0;
+#elif HAVE_ICONPATCH
+    WarEagle = Setup.WarEagleIcons;
+#else
+    WarEagle = 0;
+#endif
   showRadioChannels = 1;
   onePressTimerCreation = 1;
   conflictCheckIntervall = 30;
diff -Naur epgsearch-0.9.25.beta14-orig/epgsearchtools.c epgsearch-0.9.25.beta14/epgsearchtools.c
--- epgsearch-0.9.25.beta14-orig/epgsearchtools.c	2009-03-23 18:36:05.000000000 +0100
+++ epgsearch-0.9.25.beta14/epgsearchtools.c	2009-06-20 21:39:02.000000000 +0200
@@ -858,7 +858,10 @@
 
 void DelTimer(int index)
 {
-  cString cmdbuf = cString::sprintf("DELT %d", index);    
+  cString cmdbuf = cString::sprintf("MODT %d off", index);
+  LogFile.Log(2, "deactivate timer %d", index);
+  SendViaSVDRP(cmdbuf);
+  cmdbuf = cString::sprintf("DELT %d", index);
   LogFile.Log(2, "delete timer %d", index);
   SendViaSVDRP(cmdbuf);
   gl_timerStatusMonitor->SetConflictCheckAdvised(); 
diff -Naur epgsearch-0.9.25.beta14-orig/epgsearchtools.h epgsearch-0.9.25.beta14/epgsearchtools.h
--- epgsearch-0.9.25.beta14-orig/epgsearchtools.h	2009-03-23 18:30:36.000000000 +0100
+++ epgsearch-0.9.25.beta14/epgsearchtools.h	2009-06-20 20:44:44.000000000 +0200
@@ -92,7 +92,7 @@
 #define ICON_VPS               0x93
 #else
 // previous WAREAGLE icons
-#define ICON_REC               0xF9
+#define ICON_REC               0xE4
 #define ICON_RUNNING           0xFB
 #define ICON_CLOCK             0xFD
 #define ICON_CLOCK_HALF        0x74
diff -Naur epgsearch-0.9.25.beta14-orig/Makefile epgsearch-0.9.25.beta14/Makefile
--- epgsearch-0.9.25.beta14-orig/Makefile	2009-03-12 19:47:20.000000000 +0100
+++ epgsearch-0.9.25.beta14/Makefile	2009-06-20 20:49:09.000000000 +0200
@@ -18,7 +18,7 @@
 #WITHOUT_QUICKSEARCH=1
 
 ### comment this if you don't want epgsearch to auto config itself
-AUTOCONFIG=1
+#AUTOCONFIG=1
 
 ### if AUTOCONFIG is not active (i.e. commented) you can manually enable the 
 ### optional modules or patches for other plugins
@@ -35,7 +35,7 @@
 
 # uncomment this to enable support for the pin plugin.
 
-#USE_PINPLUGIN = 1
+USE_PINPLUGIN = 1
 
 # uncomment this to enable support for the graphtft plugin.
 #USE_GRAPHTFT = 1
@@ -64,9 +64,9 @@
 
 ### The directory environment:
 
-DVBDIR = ../../../../DVB
-VDRDIR = ../../..
-LIBDIR = ../../lib
+DVBDIR ?= ../../../../DVB
+VDRDIR ?= ../../..
+LIBDIR ?= ../../lib
 TMPDIR = /tmp
 
 ### auto configuring modules
@@ -115,7 +115,7 @@
 
 ### Includes and Defines (add further entries here):
 
-INCLUDES += -I$(VDRDIR)/include -I$(DVBDIR)/include
+INCLUDES += -I$(VDRDIR)/include -I$(DVBDIR)
 #INCLUDES += -I$(VDRDIR)/include
 
 DEFINES += -D_GNU_SOURCE -DPLUGIN_NAME_I18N='"$(PLUGIN)"'
@@ -127,6 +127,11 @@
 ### The object files (add further files here):
 
 OBJS = afuzzy.o blacklist.o changrp.o confdloader.o conflictcheck.o conflictcheck_thread.o distance.o $(PLUGIN).o epgsearchcats.o epgsearchcfg.o epgsearchext.o epgsearchsetup.o  epgsearchsvdrp.o epgsearchtools.o i18n.o mail.o md5.o menu_announcelist.o menu_blacklistedit.o menu_blacklists.o menu_commands.o menu_conflictcheck.o menu_deftimercheckmethod.o menu_dirselect.o menu_event.o menu_favorites.o menu_main.o menu_myedittimer.o menu_quicksearch.o menu_recsdone.o menu_search.o menu_searchactions.o menu_searchedit.o menu_searchresults.o menu_searchtemplate.o menu_switchtimers.o menu_templateedit.o menu_timersdone.o menu_whatson.o noannounce.o rcfile.o  recdone.o recstatus.o searchtimer_thread.o services.o switchtimer.o switchtimer_thread.o templatefile.o timer_thread.o timerdone.o timerstatus.o uservars.o varparser.o  
+
+# Check WarEagleIcons patch
+ifeq (exists, $(shell test -e $(VDRDIR)/iconpatch.h && echo exists))
+   DEFINES += -DHAVE_ICONPATCH
+endif
 
 ifdef HAVE_PCREPOSIX
 LIBS += $(shell pcre-config --libs-posix)
diff -Naur epgsearch-0.9.25.beta14-orig/menu_whatson.c epgsearch-0.9.25.beta14/menu_whatson.c
--- epgsearch-0.9.25.beta14-orig/menu_whatson.c	Mon Jul 27 10:51:09 2009
+++ epgsearch-0.9.25.beta14/menu_whatson.c	Thu Sep 03 19:00:29 2009
@@ -173,7 +173,7 @@
       else
       {
 	t[0] = event && hasMatch ? (timerMatch == tmFull) ? ((timer && timer->Recording())?'R':'T') : 't' : ' ';
-	v[0] = event && event->Vps() && (event->Vps() - event->StartTime()) ? 'V' : 'v';
+	v[0] = event && event->Vps() ? (event->Vps() - event->StartTime() ? 'V' : 'v') : ' ';
 	r[0] = event && event->IsRunning() ? '*' : ' ';
       }
 
diff -Naur epgsearch-0.9.25.beta14-orig/svdrpclient.h epgsearch-0.9.25.beta14/svdrpclient.h
--- epgsearch-0.9.25.beta14-orig/svdrpclient.h	2009-01-29 18:51:54.000000000 +0100
+++ epgsearch-0.9.25.beta14/svdrpclient.h	2009-06-20 21:39:27.000000000 +0200
@@ -152,6 +152,10 @@
 		{
 		    bCheckMultiLine = false;
 		    rc = atol(csResp);
+            if (rc != CMDSUCCESS )
+            {
+            LogFile.eSysLog("EPGSearch: SVDRP response '%s'", csResp);
+            }
 		}
 		free(csResp);
 		csResp = strdup("");
