--- configure	Wed Aug 15 14:16:57 2001
+++ configure-openbsd	Fri Nov 30 18:39:22 2007
@@ -3992,19 +3992,19 @@
   for any_driver in ${any_possible_drivers}; do
   case ${any_driver} in
 	  remotemaster)
-	  hw_module="${hw_module} hw_pixelview.o serial.o"
+	  hw_module="${hw_module} hw_pixelview.o"
 	  ;;
 	  pctv)
-	  hw_module="${hw_module} hw_pinsys.o serial.o"
+	  hw_module="${hw_module} hw_pinsys.o"
 	  ;;
 	  logitech)
-	  hw_module="${hw_module} hw_logitech.o serial.o"
+	  hw_module="${hw_module} hw_logitech.o"
 	  ;;
 	  slinke)
-	  hw_module="${hw_module} hw_slinke.o serial.o receive.o"
+	  hw_module="${hw_module} hw_slinke.o"
 	  ;;
 	  silitek)
-	  hw_module="${hw_module} hw_silitek.o serial.o"
+	  hw_module="${hw_module} hw_silitek.o"
 	  ;;
 	  mediafocusI)
 	  ;;
