diff -Naur a/PATCHES b/PATCHES
--- a/PATCHES	2011-06-30 01:17:40.000000000 +0200
+++ b/PATCHES	2019-11-04 15:40:39.051192330 +0100
@@ -0,0 +1 @@
+include-float
diff -Naur a/src/mpfr.h b/src/mpfr.h
--- a/src/mpfr.h	2019-01-31 21:29:48.000000000 +0100
+++ b/src/mpfr.h	2019-06-02 19:05:36.153226653 +0200
@@ -27,7 +27,7 @@
 #define MPFR_VERSION_MAJOR 4
 #define MPFR_VERSION_MINOR 0
 #define MPFR_VERSION_PATCHLEVEL 2
-#define MPFR_VERSION_STRING "4.0.2"
+#define MPFR_VERSION_STRING "4.0.2-p1"
 
 /* User macros:
    MPFR_USE_FILE:        Define it to make MPFR define functions dealing
diff -Naur a/src/mpfr-impl.h b/src/mpfr-impl.h
--- a/src/mpfr-impl.h	2019-01-27 19:30:16.000000000 +0100
+++ b/src/mpfr-impl.h	2019-06-02 19:05:36.145226719 +0200
@@ -57,6 +57,7 @@
 
 #include <stdlib.h>
 #include <limits.h>
+#include <float.h>  /* for FLT_RADIX, etc., tested below */
 
 
 /******************************************************
diff -Naur a/src/version.c b/src/version.c
--- a/src/version.c	2019-01-31 21:29:48.000000000 +0100
+++ b/src/version.c	2019-06-02 19:05:36.153226653 +0200
@@ -25,5 +25,5 @@
 const char *
 mpfr_get_version (void)
 {
-  return "4.0.2";
+  return "4.0.2-p1";
 }
diff -Naur a/VERSION b/VERSION
--- a/VERSION	2019-01-31 21:29:48.000000000 +0100
+++ b/VERSION	2019-06-02 19:05:36.157226621 +0200
@@ -1 +1 @@
-4.0.2
+4.0.2-p1
