diff -Naur pim-0.0.7-orig/calEvent.c pim-0.0.7/calEvent.c
--- pim-0.0.7-orig/calEvent.c	Tue Sep 02 08:57:43 2008
+++ pim-0.0.7/calEvent.c	Mon Sep 28 09:37:44 2009
@@ -22,6 +22,7 @@
 #include <string>
 #include <iostream>
 
+#include <stdlib.h>
 #include "i18n.h"
 #include "calEvent.h"
 
diff -Naur pim-0.0.7-orig/Makefile pim-0.0.7/Makefile
--- pim-0.0.7-orig/Makefile	Tue Sep 02 08:57:43 2008
+++ pim-0.0.7/Makefile	Mon Sep 28 09:37:44 2009
@@ -16,13 +16,13 @@
 ### The C++ compiler and options:
 
 CXX      ?= g++
-CXXFLAGS ?= -g -Wall -Woverloaded-virtual
+CXXFLAGS ?= -fPIC -g -Wall -Woverloaded-virtual
 
 ### The directory environment:
 
-DVBDIR = ../../../../DVB
-VDRDIR = ../../..
-LIBDIR = ../../lib
+DVBDIR ?= ../../..
+VDRDIR ?= ../../..
+LIBDIR ?= ../../lib
 TMPDIR = /tmp
 
 ### Allow user defined options to overwrite defaults:
@@ -58,7 +58,7 @@
 
 # Dependencies:
 
-MAKEDEP = g++ -MM -MG
+MAKEDEP = $(CXX) -MM -MG
 DEPFILE = .dependencies
 $(DEPFILE): Makefile
 	@$(MAKEDEP) $(DEFINES) $(INCLUDES) $(OBJS:%.o=%.c) > $@
@@ -71,7 +71,7 @@
 
 libvdr-$(PLUGIN).so: $(OBJS)
 	$(CXX) $(CXXFLAGS) -shared $(OBJS) -o $@
-	@cp $@ $(LIBDIR)/$@.$(APIVERSION)
+	@cp -f $@ $(LIBDIR)/$@.$(APIVERSION)
 
 dist: clean
 	@-rm -rf $(TMPDIR)/$(ARCHIVE)
