diff -Naurx .svn filebrowser-0.0.6b-orig/i18n.h filebrowser-0.0.6b/i18n.h
--- filebrowser-0.0.6b-orig/i18n.h	2009-02-04 04:39:22.000000000 +0100
+++ filebrowser-0.0.6b/i18n.h	2009-02-04 12:50:49.000000000 +0100
@@ -20,25 +20,774 @@
 
 const tI18nPhrase Phrases[] = {
   { "Filebrowser",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "",
+    "", // TODO /*Deutsch*/
+    "", // TODO /*Slovenski*/
+    "", // TODO /*Italiano*/
+    "", // TODO /*Nederlands*
+    "", // TODO /*Português*/
+    "", // TODO /*Français*/
+    "", // TODO /*Norsk*/
+    "", // TODO /*suomi*/
+    "", // TODO /*Polski*/
+    "Explorador de archivos", // /*Español*/
+    "", // TODO /*Ellinika*/
+    "", // TODO /*Svenska*/
+    "", // TODO /*Romaneste*/
+    "", // TODO /*Magyar*/
+    "", // TODO /*Català*/
+    "", // TODO /*Russian*/
+    "", // TODO /*Hrvatski*/
+    "", // TODO /*Eesti*/
+    "", // TODO /*Dansk*/
+    "", // TODO /*Eesky*/
+  },
+  { "Abort",
+    "", // TODO /*Deutsch*/
+    "", // TODO /*Slovenski*/
+    "", // TODO /*Italiano*/
+    "", // TODO /*Nederlands*
+    "", // TODO /*Português*/
+    "", // TODO /*Français*/
+    "", // TODO /*Norsk*/
+    "", // TODO /*suomi*/
+    "", // TODO /*Polski*/
+    "Cancelar", // /*Español*/
+    "", // TODO /*Ellinika*/
+    "", // TODO /*Svenska*/
+    "", // TODO /*Romaneste*/
+    "", // TODO /*Magyar*/
+    "", // TODO /*Català*/
+    "", // TODO /*Russian*/
+    "", // TODO /*Hrvatski*/
+    "", // TODO /*Eesti*/
+    "", // TODO /*Dansk*/
+    "", // TODO /*Eesky*/
+  },
+  { "Access code needed",
+    "", // TODO /*Deutsch*/
+    "", // TODO /*Slovenski*/
+    "", // TODO /*Italiano*/
+    "", // TODO /*Nederlands*
+    "", // TODO /*Português*/
+    "", // TODO /*Français*/
+    "", // TODO /*Norsk*/
+    "", // TODO /*suomi*/
+    "", // TODO /*Polski*/
+    "Código de acceso necesario", // /*Español*/
+    "", // TODO /*Ellinika*/
+    "", // TODO /*Svenska*/
+    "", // TODO /*Romaneste*/
+    "", // TODO /*Magyar*/
+    "", // TODO /*Català*/
+    "", // TODO /*Russian*/
+    "", // TODO /*Hrvatski*/
+    "", // TODO /*Eesti*/
+    "", // TODO /*Dansk*/
+    "", // TODO /*Eesky*/
+  },
+  { "Active",
+    "", // TODO /*Deutsch*/
+    "", // TODO /*Slovenski*/
+    "", // TODO /*Italiano*/
+    "", // TODO /*Nederlands*
+    "", // TODO /*Português*/
+    "", // TODO /*Français*/
+    "", // TODO /*Norsk*/
+    "", // TODO /*suomi*/
+    "", // TODO /*Polski*/
+    "Activo", // /*Español*/
+    "", // TODO /*Ellinika*/
+    "", // TODO /*Svenska*/
+    "", // TODO /*Romaneste*/
+    "", // TODO /*Magyar*/
+    "", // TODO /*Català*/
+    "", // TODO /*Russian*/
+    "", // TODO /*Hrvatski*/
+    "", // TODO /*Eesti*/
+    "", // TODO /*Dansk*/
+    "", // TODO /*Eesky*/
+  },
+  { "Code",
+    "", // TODO /*Deutsch*/
+    "", // TODO /*Slovenski*/
+    "", // TODO /*Italiano*/
+    "", // TODO /*Nederlands*
+    "", // TODO /*Português*/
+    "", // TODO /*Français*/
+    "", // TODO /*Norsk*/
+    "", // TODO /*suomi*/
+    "", // TODO /*Polski*/
+    "Código", // /*Español*/
+    "", // TODO /*Ellinika*/
+    "", // TODO /*Svenska*/
+    "", // TODO /*Romaneste*/
+    "", // TODO /*Magyar*/
+    "", // TODO /*Català*/
+    "", // TODO /*Russian*/
+    "", // TODO /*Hrvatski*/
+    "", // TODO /*Eesti*/
+    "", // TODO /*Dansk*/
+    "", // TODO /*Eesky*/
+  },
+  { "Failed to load directory",
+    "", // TODO /*Deutsch*/
+    "", // TODO /*Slovenski*/
+    "", // TODO /*Italiano*/
+    "", // TODO /*Nederlands*
+    "", // TODO /*Português*/
+    "", // TODO /*Français*/
+    "", // TODO /*Norsk*/
+    "", // TODO /*suomi*/
+    "", // TODO /*Polski*/
+    "Fallo al abrir el directorio", // /*Español*/
+    "", // TODO /*Ellinika*/
+    "", // TODO /*Svenska*/
+    "", // TODO /*Romaneste*/
+    "", // TODO /*Magyar*/
+    "", // TODO /*Català*/
+    "", // TODO /*Russian*/
+    "", // TODO /*Hrvatski*/
+    "", // TODO /*Eesti*/
+    "", // TODO /*Dansk*/
+    "", // TODO /*Eesky*/
+  },
+  { "Finished",
+    "", // TODO /*Deutsch*/
+    "", // TODO /*Slovenski*/
+    "", // TODO /*Italiano*/
+    "", // TODO /*Nederlands*
+    "", // TODO /*Português*/
+    "", // TODO /*Français*/
+    "", // TODO /*Norsk*/
+    "", // TODO /*suomi*/
+    "", // TODO /*Polski*/
+    "Finalizado", // /*Español*/
+    "", // TODO /*Ellinika*/
+    "", // TODO /*Svenska*/
+    "", // TODO /*Romaneste*/
+    "", // TODO /*Magyar*/
+    "", // TODO /*Català*/
+    "", // TODO /*Russian*/
+    "", // TODO /*Hrvatski*/
+    "", // TODO /*Eesti*/
+    "", // TODO /*Dansk*/
+    "", // TODO /*Eesky*/
+  },
+  { "Kill?",
+    "", // TODO /*Deutsch*/
+    "", // TODO /*Slovenski*/
+    "", // TODO /*Italiano*/
+    "", // TODO /*Nederlands*
+    "", // TODO /*Português*/
+    "", // TODO /*Français*/
+    "", // TODO /*Norsk*/
+    "", // TODO /*suomi*/
+    "", // TODO /*Polski*/
+    "¿Terminar?", // /*Español*/
+    "", // TODO /*Ellinika*/
+    "", // TODO /*Svenska*/
+    "", // TODO /*Romaneste*/
+    "", // TODO /*Magyar*/
+    "", // TODO /*Català*/
+    "", // TODO /*Russian*/
+    "", // TODO /*Hrvatski*/
+    "", // TODO /*Eesti*/
+    "", // TODO /*Dansk*/
+    "", // TODO /*Eesky*/
+  },
+  { "Kill",
+    "", // TODO /*Deutsch*/
+    "", // TODO /*Slovenski*/
+    "", // TODO /*Italiano*/
+    "", // TODO /*Nederlands*
+    "", // TODO /*Português*/
+    "", // TODO /*Français*/
+    "", // TODO /*Norsk*/
+    "", // TODO /*suomi*/
+    "", // TODO /*Polski*/
+    "Terminar", // /*Español*/
+    "", // TODO /*Ellinika*/
+    "", // TODO /*Svenska*/
+    "", // TODO /*Romaneste*/
+    "", // TODO /*Magyar*/
+    "", // TODO /*Català*/
+    "", // TODO /*Russian*/
+    "", // TODO /*Hrvatski*/
+    "", // TODO /*Eesti*/
+    "", // TODO /*Dansk*/
+    "", // TODO /*Eesky*/
+  },
+  { "Mark",
+    "", // TODO /*Deutsch*/
+    "", // TODO /*Slovenski*/
+    "", // TODO /*Italiano*/
+    "", // TODO /*Nederlands*
+    "", // TODO /*Português*/
+    "", // TODO /*Français*/
+    "", // TODO /*Norsk*/
+    "", // TODO /*suomi*/
+    "", // TODO /*Polski*/
+    "Marcar", // /*Español*/
+    "", // TODO /*Ellinika*/
+    "", // TODO /*Svenska*/
+    "", // TODO /*Romaneste*/
+    "", // TODO /*Magyar*/
+    "", // TODO /*Català*/
+    "", // TODO /*Russian*/
+    "", // TODO /*Hrvatski*/
+    "", // TODO /*Eesti*/
+    "", // TODO /*Dansk*/
+    "", // TODO /*Eesky*/
+  },
+  { "Mark all",
+    "", // TODO /*Deutsch*/
+    "", // TODO /*Slovenski*/
+    "", // TODO /*Italiano*/
+    "", // TODO /*Nederlands*
+    "", // TODO /*Português*/
+    "", // TODO /*Français*/
+    "", // TODO /*Norsk*/
+    "", // TODO /*suomi*/
+    "", // TODO /*Polski*/
+    "Marcar todo", // /*Español*/
+    "", // TODO /*Ellinika*/
+    "", // TODO /*Svenska*/
+    "", // TODO /*Romaneste*/
+    "", // TODO /*Magyar*/
+    "", // TODO /*Català*/
+    "", // TODO /*Russian*/
+    "", // TODO /*Hrvatski*/
+    "", // TODO /*Eesti*/
+    "", // TODO /*Dansk*/
+    "", // TODO /*Eesky*/
+  },
+  { "New",
+    "", // TODO /*Deutsch*/
+    "", // TODO /*Slovenski*/
+    "", // TODO /*Italiano*/
+    "", // TODO /*Nederlands*
+    "", // TODO /*Português*/
+    "", // TODO /*Français*/
+    "", // TODO /*Norsk*/
+    "", // TODO /*suomi*/
+    "", // TODO /*Polski*/
+    "Nuevo", // /*Español*/
+    "", // TODO /*Ellinika*/
+    "", // TODO /*Svenska*/
+    "", // TODO /*Romaneste*/
+    "", // TODO /*Magyar*/
+    "", // TODO /*Català*/
+    "", // TODO /*Russian*/
+    "", // TODO /*Hrvatski*/
+    "", // TODO /*Eesti*/
+    "", // TODO /*Dansk*/
+    "", // TODO /*Eesky*/
+  },
+  { "New file",
+    "", // TODO /*Deutsch*/
+    "", // TODO /*Slovenski*/
+    "", // TODO /*Italiano*/
+    "", // TODO /*Nederlands*
+    "", // TODO /*Português*/
+    "", // TODO /*Français*/
+    "", // TODO /*Norsk*/
+    "", // TODO /*suomi*/
+    "", // TODO /*Polski*/
+    "Nuevo archivo", // /*Español*/
+    "", // TODO /*Ellinika*/
+    "", // TODO /*Svenska*/
+    "", // TODO /*Romaneste*/
+    "", // TODO /*Magyar*/
+    "", // TODO /*Català*/
+    "", // TODO /*Russian*/
+    "", // TODO /*Hrvatski*/
+    "", // TODO /*Eesti*/
+    "", // TODO /*Dansk*/
+    "", // TODO /*Eesky*/
+  },
+  { "No Updates",
+    "", // TODO /*Deutsch*/
+    "", // TODO /*Slovenski*/
+    "", // TODO /*Italiano*/
+    "", // TODO /*Nederlands*
+    "", // TODO /*Português*/
+    "", // TODO /*Français*/
+    "", // TODO /*Norsk*/
+    "", // TODO /*suomi*/
+    "", // TODO /*Polski*/
+    "No actualizar", // /*Español*/
+    "", // TODO /*Ellinika*/
+    "", // TODO /*Svenska*/
+    "", // TODO /*Romaneste*/
+    "", // TODO /*Magyar*/
+    "", // TODO /*Català*/
+    "", // TODO /*Russian*/
+    "", // TODO /*Hrvatski*/
+    "", // TODO /*Eesti*/
+    "", // TODO /*Dansk*/
+    "", // TODO /*Eesky*/
+  },
+  { "OK",
+    "", // TODO /*Deutsch*/
+    "", // TODO /*Slovenski*/
+    "", // TODO /*Italiano*/
+    "", // TODO /*Nederlands*
+    "", // TODO /*Português*/
+    "", // TODO /*Français*/
+    "", // TODO /*Norsk*/
+    "", // TODO /*suomi*/
+    "", // TODO /*Polski*/
+    "Aceptar", // /*Español*/
+    "", // TODO /*Ellinika*/
+    "", // TODO /*Svenska*/
+    "", // TODO /*Romaneste*/
+    "", // TODO /*Magyar*/
+    "", // TODO /*Català*/
+    "", // TODO /*Russian*/
+    "", // TODO /*Hrvatski*/
+    "", // TODO /*Eesti*/
+    "", // TODO /*Dansk*/
+    "", // TODO /*Eesky*/
+  },
+  { "Other",
+    "", // TODO /*Deutsch*/
+    "", // TODO /*Slovenski*/
+    "", // TODO /*Italiano*/
+    "", // TODO /*Nederlands*
+    "", // TODO /*Português*/
+    "", // TODO /*Français*/
+    "", // TODO /*Norsk*/
+    "", // TODO /*suomi*/
+    "", // TODO /*Polski*/
+    "Otros", // /*Español*/
+    "", // TODO /*Ellinika*/
+    "", // TODO /*Svenska*/
+    "", // TODO /*Romaneste*/
+    "", // TODO /*Magyar*/
+    "", // TODO /*Català*/
+    "", // TODO /*Russian*/
+    "", // TODO /*Hrvatski*/
+    "", // TODO /*Eesti*/
+    "", // TODO /*Dansk*/
+    "", // TODO /*Eesky*/
+  },
+  { "Other Commands",
+    "", // TODO /*Deutsch*/
+    "", // TODO /*Slovenski*/
+    "", // TODO /*Italiano*/
+    "", // TODO /*Nederlands*
+    "", // TODO /*Português*/
+    "", // TODO /*Français*/
+    "", // TODO /*Norsk*/
+    "", // TODO /*suomi*/
+    "", // TODO /*Polski*/
+    "Otros comandos", // /*Español*/
+    "", // TODO /*Ellinika*/
+    "", // TODO /*Svenska*/
+    "", // TODO /*Romaneste*/
+    "", // TODO /*Magyar*/
+    "", // TODO /*Català*/
+    "", // TODO /*Russian*/
+    "", // TODO /*Hrvatski*/
+    "", // TODO /*Eesti*/
+    "", // TODO /*Dansk*/
+    "", // TODO /*Eesky*/
+  },
+  { "Pause",
+    "", // TODO /*Deutsch*/
+    "", // TODO /*Slovenski*/
+    "", // TODO /*Italiano*/
+    "", // TODO /*Nederlands*
+    "", // TODO /*Português*/
+    "", // TODO /*Français*/
+    "", // TODO /*Norsk*/
+    "", // TODO /*suomi*/
+    "", // TODO /*Polski*/
+    "Pausa", // /*Español*/
+    "", // TODO /*Ellinika*/
+    "", // TODO /*Svenska*/
+    "", // TODO /*Romaneste*/
+    "", // TODO /*Magyar*/
+    "", // TODO /*Català*/
+    "", // TODO /*Russian*/
+    "", // TODO /*Hrvatski*/
+    "", // TODO /*Eesti*/
+    "", // TODO /*Dansk*/
+    "", // TODO /*Eesky*/
+  },
+  { "Paused",
+    "", // TODO /*Deutsch*/
+    "", // TODO /*Slovenski*/
+    "", // TODO /*Italiano*/
+    "", // TODO /*Nederlands*
+    "", // TODO /*Português*/
+    "", // TODO /*Français*/
+    "", // TODO /*Norsk*/
+    "", // TODO /*suomi*/
+    "", // TODO /*Polski*/
+    "Pausado", // /*Español*/
+    "", // TODO /*Ellinika*/
+    "", // TODO /*Svenska*/
+    "", // TODO /*Romaneste*/
+    "", // TODO /*Magyar*/
+    "", // TODO /*Català*/
+    "", // TODO /*Russian*/
+    "", // TODO /*Hrvatski*/
+    "", // TODO /*Eesti*/
+    "", // TODO /*Dansk*/
+    "", // TODO /*Eesky*/
+  },
+  { "Play",
+    "", // TODO /*Deutsch*/
+    "", // TODO /*Slovenski*/
+    "", // TODO /*Italiano*/
+    "", // TODO /*Nederlands*
+    "", // TODO /*Português*/
+    "", // TODO /*Français*/
+    "", // TODO /*Norsk*/
+    "", // TODO /*suomi*/
+    "", // TODO /*Polski*/
+    "Ejecutar", // /*Español*/
+    "", // TODO /*Ellinika*/
+    "", // TODO /*Svenska*/
+    "", // TODO /*Romaneste*/
+    "", // TODO /*Magyar*/
+    "", // TODO /*Català*/
+    "", // TODO /*Russian*/
+    "", // TODO /*Hrvatski*/
+    "", // TODO /*Eesti*/
+    "", // TODO /*Dansk*/
+    "", // TODO /*Eesky*/
+  },
+  { "Remove",
+    "", // TODO /*Deutsch*/
+    "", // TODO /*Slovenski*/
+    "", // TODO /*Italiano*/
+    "", // TODO /*Nederlands*
+    "", // TODO /*Português*/
+    "", // TODO /*Français*/
+    "", // TODO /*Norsk*/
+    "", // TODO /*suomi*/
+    "", // TODO /*Polski*/
+    "Borrar", // /*Español*/
+    "", // TODO /*Ellinika*/
+    "", // TODO /*Svenska*/
+    "", // TODO /*Romaneste*/
+    "", // TODO /*Magyar*/
+    "", // TODO /*Català*/
+    "", // TODO /*Russian*/
+    "", // TODO /*Hrvatski*/
+    "", // TODO /*Eesti*/
+    "", // TODO /*Dansk*/
+    "", // TODO /*Eesky*/
+  },
+  { "Resume",
+    "", // TODO /*Deutsch*/
+    "", // TODO /*Slovenski*/
+    "", // TODO /*Italiano*/
+    "", // TODO /*Nederlands*
+    "", // TODO /*Português*/
+    "", // TODO /*Français*/
+    "", // TODO /*Norsk*/
+    "", // TODO /*suomi*/
+    "", // TODO /*Polski*/
+    "Reanudar", // /*Español*/
+    "", // TODO /*Ellinika*/
+    "", // TODO /*Svenska*/
+    "", // TODO /*Romaneste*/
+    "", // TODO /*Magyar*/
+    "", // TODO /*Català*/
+    "", // TODO /*Russian*/
+    "", // TODO /*Hrvatski*/
+    "", // TODO /*Eesti*/
+    "", // TODO /*Dansk*/
+    "", // TODO /*Eesky*/
+  },
+  { "Running",
+    "", // TODO /*Deutsch*/
+    "", // TODO /*Slovenski*/
+    "", // TODO /*Italiano*/
+    "", // TODO /*Nederlands*
+    "", // TODO /*Português*/
+    "", // TODO /*Français*/
+    "", // TODO /*Norsk*/
+    "", // TODO /*suomi*/
+    "", // TODO /*Polski*/
+    "Ejecutandose", // /*Español*/
+    "", // TODO /*Ellinika*/
+    "", // TODO /*Svenska*/
+    "", // TODO /*Romaneste*/
+    "", // TODO /*Magyar*/
+    "", // TODO /*Català*/
+    "", // TODO /*Russian*/
+    "", // TODO /*Hrvatski*/
+    "", // TODO /*Eesti*/
+    "", // TODO /*Dansk*/
+    "", // TODO /*Eesky*/
+  },
+  { "Running threads",
+    "", // TODO /*Deutsch*/
+    "", // TODO /*Slovenski*/
+    "", // TODO /*Italiano*/
+    "", // TODO /*Nederlands*
+    "", // TODO /*Português*/
+    "", // TODO /*Français*/
+    "", // TODO /*Norsk*/
+    "", // TODO /*suomi*/
+    "", // TODO /*Polski*/
+    "Procesos ejecutandose", // /*Español*/
+    "", // TODO /*Ellinika*/
+    "", // TODO /*Svenska*/
+    "", // TODO /*Romaneste*/
+    "", // TODO /*Magyar*/
+    "", // TODO /*Català*/
+    "", // TODO /*Russian*/
+    "", // TODO /*Hrvatski*/
+    "", // TODO /*Eesti*/
+    "", // TODO /*Dansk*/
+    "", // TODO /*Eesky*/
+  },
+  { "Select Destination",
+    "", // TODO /*Deutsch*/
+    "", // TODO /*Slovenski*/
+    "", // TODO /*Italiano*/
+    "", // TODO /*Nederlands*
+    "", // TODO /*Português*/
+    "", // TODO /*Français*/
+    "", // TODO /*Norsk*/
+    "", // TODO /*suomi*/
+    "", // TODO /*Polski*/
+    "Seleccionar destino", // /*Español*/
+    "", // TODO /*Ellinika*/
+    "", // TODO /*Svenska*/
+    "", // TODO /*Romaneste*/
+    "", // TODO /*Magyar*/
+    "", // TODO /*Català*/
+    "", // TODO /*Russian*/
+    "", // TODO /*Hrvatski*/
+    "", // TODO /*Eesti*/
+    "", // TODO /*Dansk*/
+    "", // TODO /*Eesky*/
+  },
+  { "Show full path in title",
+    "", // TODO /*Deutsch*/
+    "", // TODO /*Slovenski*/
+    "", // TODO /*Italiano*/
+    "", // TODO /*Nederlands*
+    "", // TODO /*Português*/
+    "", // TODO /*Français*/
+    "", // TODO /*Norsk*/
+    "", // TODO /*suomi*/
+    "", // TODO /*Polski*/
+    "Mostrar la ruta completa en el título", // /*Español*/
+    "", // TODO /*Ellinika*/
+    "", // TODO /*Svenska*/
+    "", // TODO /*Romaneste*/
+    "", // TODO /*Magyar*/
+    "", // TODO /*Català*/
+    "", // TODO /*Russian*/
+    "", // TODO /*Hrvatski*/
+    "", // TODO /*Eesti*/
+    "", // TODO /*Dansk*/
+    "", // TODO /*Eesky*/
+  },
+  { "Show hidden files",
+    "", // TODO /*Deutsch*/
+    "", // TODO /*Slovenski*/
+    "", // TODO /*Italiano*/
+    "", // TODO /*Nederlands*
+    "", // TODO /*Português*/
+    "", // TODO /*Français*/
+    "", // TODO /*Norsk*/
+    "", // TODO /*suomi*/
+    "", // TODO /*Polski*/
+    "Mostrar los archivos ocultos", // /*Español*/
+    "", // TODO /*Ellinika*/
+    "", // TODO /*Svenska*/
+    "", // TODO /*Romaneste*/
+    "", // TODO /*Magyar*/
+    "", // TODO /*Català*/
+    "", // TODO /*Russian*/
+    "", // TODO /*Hrvatski*/
+    "", // TODO /*Eesti*/
+    "", // TODO /*Dansk*/
+    "", // TODO /*Eesky*/
+  },
+  { "Sources",
+    "", // TODO /*Deutsch*/
+    "", // TODO /*Slovenski*/
+    "", // TODO /*Italiano*/
+    "", // TODO /*Nederlands*
+    "", // TODO /*Português*/
+    "", // TODO /*Français*/
+    "", // TODO /*Norsk*/
+    "", // TODO /*suomi*/
+    "", // TODO /*Polski*/
+    "Directorios", // /*Español*/
+    "", // TODO /*Ellinika*/
+    "", // TODO /*Svenska*/
+    "", // TODO /*Romaneste*/
+    "", // TODO /*Magyar*/
+    "", // TODO /*Català*/
+    "", // TODO /*Russian*/
+    "", // TODO /*Hrvatski*/
+    "", // TODO /*Eesti*/
+    "", // TODO /*Dansk*/
+    "", // TODO /*Eesky*/
+  },
+  { "Threads",
+    "", // TODO /*Deutsch*/
+    "", // TODO /*Slovenski*/
+    "", // TODO /*Italiano*/
+    "", // TODO /*Nederlands*
+    "", // TODO /*Português*/
+    "", // TODO /*Français*/
+    "", // TODO /*Norsk*/
+    "", // TODO /*suomi*/
+    "", // TODO /*Polski*/
+    "Procesos", // /*Español*/
+    "", // TODO /*Ellinika*/
+    "", // TODO /*Svenska*/
+    "", // TODO /*Romaneste*/
+    "", // TODO /*Magyar*/
+    "", // TODO /*Català*/
+    "", // TODO /*Russian*/
+    "", // TODO /*Hrvatski*/
+    "", // TODO /*Eesti*/
+    "", // TODO /*Dansk*/
+    "", // TODO /*Eesky*/
+  },
+  { " to ",
+    "", // TODO /*Deutsch*/
+    "", // TODO /*Slovenski*/
+    "", // TODO /*Italiano*/
+    "", // TODO /*Nederlands*
+    "", // TODO /*Português*/
+    "", // TODO /*Français*/
+    "", // TODO /*Norsk*/
+    "", // TODO /*suomi*/
+    "", // TODO /*Polski*/
+    " a ", // /*Español*/
+    "", // TODO /*Ellinika*/
+    "", // TODO /*Svenska*/
+    "", // TODO /*Romaneste*/
+    "", // TODO /*Magyar*/
+    "", // TODO /*Català*/
+    "", // TODO /*Russian*/
+    "", // TODO /*Hrvatski*/
+    "", // TODO /*Eesti*/
+    "", // TODO /*Dansk*/
+    "", // TODO /*Eesky*/
+  },
+  { "Unm. all",
+    "", // TODO /*Deutsch*/
+    "", // TODO /*Slovenski*/
+    "", // TODO /*Italiano*/
+    "", // TODO /*Nederlands*
+    "", // TODO /*Português*/
+    "", // TODO /*Français*/
+    "", // TODO /*Norsk*/
+    "", // TODO /*suomi*/
+    "", // TODO /*Polski*/
+    "Desm. todo", // /*Español*/
+    "", // TODO /*Ellinika*/
+    "", // TODO /*Svenska*/
+    "", // TODO /*Romaneste*/
+    "", // TODO /*Magyar*/
+    "", // TODO /*Català*/
+    "", // TODO /*Russian*/
+    "", // TODO /*Hrvatski*/
+    "", // TODO /*Eesti*/
+    "", // TODO /*Dansk*/
+    "", // TODO /*Eesky*/
+  },
+  { "Unmark",
+    "", // TODO /*Deutsch*/
+    "", // TODO /*Slovenski*/
+    "", // TODO /*Italiano*/
+    "", // TODO /*Nederlands*
+    "", // TODO /*Português*/
+    "", // TODO /*Français*/
+    "", // TODO /*Norsk*/
+    "", // TODO /*suomi*/
+    "", // TODO /*Polski*/
+    "Desmarcar", // /*Español*/
+    "", // TODO /*Ellinika*/
+    "", // TODO /*Svenska*/
+    "", // TODO /*Romaneste*/
+    "", // TODO /*Magyar*/
+    "", // TODO /*Català*/
+    "", // TODO /*Russian*/
+    "", // TODO /*Hrvatski*/
+    "", // TODO /*Eesti*/
+    "", // TODO /*Dansk*/
+    "", // TODO /*Eesky*/
+  },
+  { "Updates",
+    "", // TODO /*Deutsch*/
+    "", // TODO /*Slovenski*/
+    "", // TODO /*Italiano*/
+    "", // TODO /*Nederlands*
+    "", // TODO /*Português*/
+    "", // TODO /*Français*/
+    "", // TODO /*Norsk*/
+    "", // TODO /*suomi*/
+    "", // TODO /*Polski*/
+    "Actualizar", // /*Español*/
+    "", // TODO /*Ellinika*/
+    "", // TODO /*Svenska*/
+    "", // TODO /*Romaneste*/
+    "", // TODO /*Magyar*/
+    "", // TODO /*Català*/
+    "", // TODO /*Russian*/
+    "", // TODO /*Hrvatski*/
+    "", // TODO /*Eesti*/
+    "", // TODO /*Dansk*/
+    "", // TODO /*Eesky*/
+  },
+  { "View",
+    "", // TODO /*Deutsch*/
+    "", // TODO /*Slovenski*/
+    "", // TODO /*Italiano*/
+    "", // TODO /*Nederlands*
+    "", // TODO /*Português*/
+    "", // TODO /*Français*/
+    "", // TODO /*Norsk*/
+    "", // TODO /*suomi*/
+    "", // TODO /*Polski*/
+    "Ver", // /*Español*/
+    "", // TODO /*Ellinika*/
+    "", // TODO /*Svenska*/
+    "", // TODO /*Romaneste*/
+    "", // TODO /*Magyar*/
+    "", // TODO /*Català*/
+    "", // TODO /*Russian*/
+    "", // TODO /*Hrvatski*/
+    "", // TODO /*Eesti*/
+    "", // TODO /*Dansk*/
+    "", // TODO /*Eesky*/
+  },
+  { "Wrong code!",
+    "", // TODO /*Deutsch*/
+    "", // TODO /*Slovenski*/
+    "", // TODO /*Italiano*/
+    "", // TODO /*Nederlands*
+    "", // TODO /*Português*/
+    "", // TODO /*Français*/
+    "", // TODO /*Norsk*/
+    "", // TODO /*suomi*/
+    "", // TODO /*Polski*/
+    "¡Código erróneo!", // /*Español*/
+    "", // TODO /*Ellinika*/
+    "", // TODO /*Svenska*/
+    "", // TODO /*Romaneste*/
+    "", // TODO /*Magyar*/
+    "", // TODO /*Català*/
+    "", // TODO /*Russian*/
+    "", // TODO /*Hrvatski*/
+    "", // TODO /*Eesti*/
+    "", // TODO /*Dansk*/
+    "", // TODO /*Eesky*/
   },
   {
     NULL
