diff -Naur klone-core-2.3.0/klone/http.h klone-core-2.3.0-atinar/klone/http.h
--- klone-core-2.3.0/klone/http.h	2010-12-09 11:41:37.000000000 +0100
+++ klone-core-2.3.0-atinar/klone/http.h	2010-12-11 02:17:18.000000000 +0100
@@ -31,6 +31,8 @@
     /**< request accepted but processing not completed */
     HTTP_STATUS_NO_CONTENT                = 204,
     /**< no body returned */
+    HTTP_PARTIAL_CONTENT                  = 206,
+    /**< partial content returned */
     HTTP_STATUS_MOVED_PERMANENTLY         = 301,
     /**< resource relocated permanently */
     HTTP_STATUS_MOVED_TEMPORARILY         = 302,
diff -Naur klone-core-2.3.0/klone/utils.h klone-core-2.3.0-atinar/klone/utils.h
--- klone-core-2.3.0/klone/utils.h	2010-12-09 11:41:37.000000000 +0100
+++ klone-core-2.3.0-atinar/klone/utils.h	2010-12-11 02:17:18.000000000 +0100
@@ -54,6 +54,7 @@
 enum { U_COPY_VERBATIM, U_COPY_ENCODE, U_COPY_DECODE };
 
 enum { URLCPY_VERBATIM, URLCPY_ENCODE, URLCPY_DECODE };
+ssize_t u_urlncpy_encode(char *d, const char *s, size_t slen, const char *keep);
 ssize_t u_urlncpy(char *dst, const char *src, size_t slen, int flags);
 
 enum { HEXCPY_VERBATIM, HEXCPY_ENCODE, HEXCPY_DECODE };
diff -Naur klone-core-2.3.0/src/libhttp/http.c klone-core-2.3.0-atinar/src/libhttp/http.c
--- klone-core-2.3.0/src/libhttp/http.c	2010-12-09 11:41:37.000000000 +0100
+++ klone-core-2.3.0-atinar/src/libhttp/http.c	2010-12-11 02:17:18.000000000 +0100
@@ -51,6 +51,7 @@
     { HTTP_STATUS_CREATED               , "Created"                 },
     { HTTP_STATUS_ACCEPTED              , "Accepted"                },
     { HTTP_STATUS_NO_CONTENT            , "No Content"              },
+    { HTTP_PARTIAL_CONTENT              , "Partial Content"         },
     { HTTP_STATUS_BAD_REQUEST           , "Bad Request"             },
     { HTTP_STATUS_UNAUTHORIZED          , "Unauthorized"            },
     { HTTP_STATUS_FORBIDDEN             , "Forbidden"               },
diff -Naur klone-core-2.3.0/src/libutils/utils.c klone-core-2.3.0-atinar/src/libutils/utils.c
--- klone-core-2.3.0/src/libutils/utils.c	2010-12-09 11:41:37.000000000 +0100
+++ klone-core-2.3.0-atinar/src/libutils/utils.c	2010-12-11 02:17:18.000000000 +0100
@@ -307,11 +307,13 @@
     return -1;
 }
 
-static ssize_t u_urlncpy_encode(char *d, const char *s, size_t slen)
+ssize_t u_urlncpy_encode(char *d, const char *s, size_t slen, const char *keep)
 {
     const char hexc[] = "0123456789ABCDEF";
     ssize_t wr = 0;
     unsigned char c;
+    const char default_keep[] = "-_."; //~/:?=&"; //!*'();@+$,[]#%"; 
+    if (keep==NULL) keep=default_keep;
 
     dbg_return_if (d == NULL, -1);
     dbg_return_if (s == NULL, -1);
@@ -322,7 +324,7 @@
         if(c == ' ') {
             *d++ = '+';
             wr++;
-        } else if(isalnum(c) || c == '_' || c == '-' || c == '.') {
+        } else if ((c>='A' && c<='Z') || (c>='a' && c<='z') || (c>='0' && c<='9') || strchr(keep,c)) {
             *d++ = c;
             wr++;
         } else {
@@ -394,7 +396,7 @@
     switch(flags)
     {
     case URLCPY_ENCODE:
-        return u_urlncpy_encode(d, s, slen);
+        return u_urlncpy_encode(d, s, slen, NULL);
     case URLCPY_DECODE:
         return u_urlncpy_decode(d, s, slen);
     default:
