diff -Naur setup-0.0.1-orig/i18n.cpp setup-0.0.1/i18n.cpp
--- setup-0.0.1-orig/i18n.cpp	2007-03-06 08:21:04.000000000 +0100
+++ setup-0.0.1/i18n.cpp	2007-07-15 19:44:15.000000000 +0200
@@ -12,6 +12,46 @@
 
 const tI18nPhrase Phrases[] = {
   {
+    "Gigaset 7x0 Firmware Setup",
+    "Gigaset 7x0 Firmware Setup",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "Configurar Firmware Gigaset 7x0",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+  },
+  {
+    "open7x0-Setup (based on Minivdr-Setup)",
+    "open7x0-Setup (based on Minivdr-Setup)",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "Configuración-open7x0 (basado en Minivdr-Setup)",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+  },
+  {
     "Gigaset Firmware setup - open7x0.org (based on setup plugin by minivdr.de)",
     "Gigaset Firmware setup - open7x0.org (based on setup plugin by minivdr.de)",
     "",
@@ -22,7 +62,7 @@
     "",
     "",
     "",
-    "Configuración - diseñado para minivdr.de (c)",
+    "Configurar Firmware Gigaset - open7x0.org (basado en plugin setup por minivdr.de)",
     "",
     "",
     "",
@@ -1094,7 +1134,7 @@
     "",
     "",
     "",
-    "",
+    "Cargando configuración firmware ...",
     "",
     "",
     "",
@@ -1115,7 +1155,7 @@
     "",
     "",
     "",
-    "",
+    "Guardando configuración firmware ...",
     "",
     "",
     "",
diff -Naur setup-0.0.1-orig/Makefile setup-0.0.1/Makefile
--- setup-0.0.1-orig/Makefile	2007-05-15 08:20:51.000000000 +0200
+++ setup-0.0.1/Makefile	2007-07-30 11:02:00.000000000 +0200
@@ -23,11 +23,11 @@
 ### The C++ compiler and options:
 
 CXX      ?= g++
-CXXFLAGS ?= -g -O2 -Wall -Woverloaded-virtual
+CXXFLAGS ?= -fPIC -g -O2 -Wall -Woverloaded-virtual
 
 ### The directory environment:
 
-DVBDIR = ../../../../DVB
+DVBDIR ?= ../../..
 VDRDIR ?= ../../..
 LIBDIR ?= ../../lib
 TMPDIR = /tmp
@@ -50,7 +50,7 @@
 
 ### Includes and Defines (add further entries here):
 
-INCLUDES += -I$(VDRDIR)/include -I$(DVBDIR)/include
+INCLUDES += -I$(VDRDIR)/include -I$(DVBDIR)
 
 
 #just for TESTING VDR SUBmenu
diff -Naur setup-0.0.1-orig/setup.cpp setup-0.0.1/setup.cpp
--- setup-0.0.1-orig/setup.cpp	2007-03-06 08:21:04.000000000 +0100
+++ setup-0.0.1/setup.cpp	2007-07-07 15:08:44.000000000 +0200
@@ -30,13 +30,13 @@
   cPluginSetup(void);
   virtual ~cPluginSetup();
   virtual const char *Version(void) { return VERSION; }
-  virtual const char *Description(void) { return DESCRIPTION; }
+  virtual const char *Description(void) { return tr(DESCRIPTION); }
   virtual const char *CommandLineHelp(void);
   virtual bool ProcessArgs(int argc, char *argv[]);
   virtual bool Initialize(void);
   virtual bool Start(void);
   virtual void Housekeeping(void);
-  virtual const char *MainMenuEntry(void) { return MAINMENUENTRY; }
+  virtual const char *MainMenuEntry(void) { return tr(MAINMENUENTRY); }
   virtual cOsdObject *MainMenuAction(void);
   virtual cMenuSetupPage *SetupMenu(void);
   virtual bool SetupParse(const char *Name, const char *Value);
