--- slin_m740/include/linux/ip.h.orig	2007-12-23 10:09:42.000000000 +0100
+++ slin_m740/include/linux/ip.h	2007-12-23 10:12:09.000000000 +0100
@@ -108,7 +108,7 @@
   unsigned char __pad1;
   unsigned char __pad2;
   unsigned char __data[0];
-};
+} __attribute__((packed));
 
 #define optlength(opt) (sizeof(struct ip_options) + opt->optlen)
 #endif
@@ -133,6 +133,6 @@
 	__u32	saddr;
 	__u32	daddr;
 	/*The options start here. */
-};
+} __attribute__((packed));
 
 #endif	/* _LINUX_IP_H */
--- slin_m740/include/linux/tcp.h.orig	2007-12-23 10:16:59.000000000 +0100
+++ slin_m740/include/linux/tcp.h	2007-12-23 10:22:12.000000000 +0100
@@ -53,7 +53,7 @@
 	__u16	window;
 	__u16	check;
 	__u16	urg_ptr;
-};
+} __attribute__((packed));
 
 
 enum {
@@ -97,7 +97,7 @@
 union tcp_word_hdr { 
 	struct tcphdr hdr;
 	__u32 		  words[5];
-}; 
+} __attribute__((packed)); 
 
 #define tcp_flag_word(tp) ( ((union tcp_word_hdr *)(tp))->words [3]) 
 
--- slin_m740/include/net/sock.h.orig	2007-12-23 16:22:01.000000000 +0100
+++ slin_m740/include/net/sock.h	2007-12-23 16:22:34.000000000 +0100
@@ -251,7 +251,7 @@
 struct tcp_sack_block {
 	__u32	start_seq;
 	__u32	end_seq;
-};
+} __attribute__((packed));
 
 struct tcp_opt {
 	int	tcp_header_len;	/* Bytes of tcp header to send		*/
