diff -Naur channellists-0.0.2-orig/i18n.c channellists-0.0.2/i18n.c
--- channellists-0.0.2-orig/i18n.c	Wed Jan 16 01:22:10 2008
+++ channellists-0.0.2/i18n.c	Fri Jan 18 10:19:25 2008
@@ -18,7 +18,7 @@
     "", // Norsk
     "", // suomi
     "", // Polski
-    "", // Español
+    "Listas de canales", // Español
     "", // ÅëëçíéêÜ (Greek)
     "", // Svenska
     "", // Romaneste
@@ -40,7 +40,7 @@
     "", // Norsk
     "", // suomi
     "", // Polski
-    "", // Español
+    "Administra tus listas de canales", // Español
     "", // ÅëëçíéêÜ (Greek)
     "", // Svenska
     "", // Romaneste
@@ -62,7 +62,7 @@
     "", // Norsk
     "", // suomi
     "", // Polski
-    "", // Español
+    "Ocultar entrada del menú principal", // Español
     "", // ÅëëçíéêÜ (Greek)
     "", // Svenska
     "", // Romaneste
@@ -84,7 +84,7 @@
     "", // Norsk
     "", // suomi
     "", // Polski
-    "", // Español
+    "¿Cambiar la lista de canales actual?", // Español
     "", // ÅëëçíéêÜ (Greek)
     "", // Svenska
     "", // Romaneste
@@ -106,7 +106,7 @@
     "", // Norsk
     "", // suomi
     "", // Polski
-    "", // Español
+    "Renombrar", // Español
     "", // ÅëëçíéêÜ (Greek)
     "", // Svenska
     "", // Romaneste
@@ -128,7 +128,7 @@
     "", // Norsk
     "", // suomi
     "", // Polski
-    "", // Español
+    "Copiar", // Español
     "", // ÅëëçíéêÜ (Greek)
     "", // Svenska
     "", // Romaneste
@@ -150,7 +150,7 @@
     "", // Norsk
     "", // suomi
     "", // Polski
-    "", // Español
+    "Guardar", // Español
     "", // ÅëëçíéêÜ (Greek)
     "", // Svenska
     "", // Romaneste
@@ -172,7 +172,7 @@
     "", // Norsk
     "", // suomi
     "", // Polski
-    "", // Español
+    "Nombre", // Español
     "", // ÅëëçíéêÜ (Greek)
     "", // Svenska
     "", // Romaneste
@@ -194,7 +194,7 @@
     "", // Norsk
     "", // suomi
     "", // Polski
-    "", // Español
+    "¿Borrar la lista de canales?", // Español
     "", // ÅëëçíéêÜ (Greek)
     "", // Svenska
     "", // Romaneste
@@ -216,7 +216,7 @@
     "", // Norsk
     "", // suomi
     "", // Polski
-    "", // Español
+    "No es posible crear el directorio", // Español
     "", // ÅëëçíéêÜ (Greek)
     "", // Svenska
     "", // Romaneste
@@ -238,7 +238,7 @@
     "", // Norsk
     "", // suomi
     "", // Polski
-    "", // Español
+    "Se produjo un error, ver syslog...", // Español
     "", // ÅëëçíéêÜ (Greek)
     "", // Svenska
     "", // Romaneste
@@ -260,7 +260,7 @@
     "", // Norsk
     "", // suomi
     "", // Polski
-    "", // Español
+    "Renombrar", // Español
     "", // ÅëëçíéêÜ (Greek)
     "", // Svenska
     "", // Romaneste
@@ -282,7 +282,7 @@
     "", // Norsk
     "", // suomi
     "", // Polski
-    "", // Español
+    "Copiar", // Español
     "", // ÅëëçíéêÜ (Greek)
     "", // Svenska
     "", // Romaneste
@@ -304,7 +304,7 @@
     "", // Norsk
     "", // suomi
     "", // Polski
-    "", // Español
+    "Borrar", // Español
     "", // ÅëëçíéêÜ (Greek)
     "", // Svenska
     "", // Romaneste
@@ -326,7 +326,7 @@
     "", // Norsk
     "", // suomi
     "", // Polski
-    "", // Español
+    "Guardar", // Español
     "", // ÅëëçíéêÜ (Greek)
     "", // Svenska
     "", // Romaneste
diff -Naur channellists-0.0.2-orig/Makefile channellists-0.0.2/Makefile
--- channellists-0.0.2-orig/Makefile	Sat Jan 12 19:38:52 2008
+++ channellists-0.0.2/Makefile	Fri Jan 18 09:22:34 2008
@@ -22,8 +22,8 @@
 
 ### The directory environment:
 
-VDRDIR = ../../..
-LIBDIR = ../../lib
+VDRDIR ?= ../../..
+LIBDIR ?= ../../lib
 TMPDIR = /tmp
 
 ### Allow user defined options to overwrite defaults:
diff -Naur channellists-0.0.2-orig/po/es_ES.po channellists-0.0.2/po/es_ES.po
--- channellists-0.0.2-orig/po/es_ES.po	Wed Jan 16 08:50:18 2008
+++ channellists-0.0.2/po/es_ES.po	Fri Jan 18 10:00:19 2008
@@ -1,61 +1,61 @@
 # VDR plugin language source file.
 # Copyright (C) 2007 Klaus Schmidinger <kls@cadsoft.de>
 # This file is distributed under the same license as the VDR package.
-# Ruben Nunez Francisco <ruben.nunez@tang-it.com>, 2002
+# bittor from open7x0.org <bittor7x0 _at_ gmail.com>, 2008
 #
 msgid ""
 msgstr ""
 "Project-Id-Version: VDR 1.5.7\n"
 "Report-Msgid-Bugs-To: <see README>\n"
 "POT-Creation-Date: 2008-01-16 01:23+0100\n"
-"PO-Revision-Date: 2008-01-16 01:22+0100\n"
-"Last-Translator: Ruben Nunez Francisco <ruben.nunez@tang-it.com>\n"
+"PO-Revision-Date: 2008-01-18 20:25+0100\n"
+"Last-Translator: bittor from open7x0.org <bittor7x0 _at_ gmail.com>\n"
 "Language-Team: <vdr@linuxtv.org>\n"
 "MIME-Version: 1.0\n"
 "Content-Type: text/plain; charset=ISO-8859-15\n"
 "Content-Transfer-Encoding: 8bit\n"
 
 msgid "Manage your channellists"
-msgstr ""
+msgstr "Administra tus listas de canales"
 
 msgid "Channellists"
-msgstr ""
+msgstr "Listas de canales"
 
 msgid "Hide main menu entry"
-msgstr ""
+msgstr "Ocultar entrada del menú principal"
 
 msgid "Name"
-msgstr ""
+msgstr "Nombre"
 
 msgid "An error occurs, see syslog..."
-msgstr ""
+msgstr "Se produjo un error, ver syslog..."
 
 msgid "Could not create dir"
-msgstr ""
+msgstr "No es posible crear el directorio"
 
 msgid "Switch current channellist?"
-msgstr ""
+msgstr "¿Cambiar la lista de canales actual?"
 
 msgid "Rename"
-msgstr ""
+msgstr "Renombrar"
 
 msgid "Copy"
-msgstr ""
+msgstr "Copiar"
 
 msgid "Delete channellist?"
-msgstr ""
+msgstr "¿Borrar la lista de canales?"
 
 msgid "Save"
-msgstr ""
+msgstr "Guardar"
 
 msgid "Button$Rename"
-msgstr ""
+msgstr "Renombrar"
 
 msgid "Button$Copy"
-msgstr ""
+msgstr "Copiar"
 
 msgid "Button$Delete"
-msgstr ""
+msgstr "Borrar"
 
 msgid "Button$Save"
-msgstr ""
+msgstr "Guardar"
