diff -Naur klone-2.4.0/makl/helpers/mkdep.gcc.sh klone-2.4.0-fixed/makl/helpers/mkdep.gcc.sh
--- klone-2.4.0/makl/helpers/mkdep.gcc.sh	2010-12-09 11:41:25.000000000 +0100
+++ klone-2.4.0-fixed/makl/helpers/mkdep.gcc.sh	2011-04-08 21:42:46.000000000 +0200
@@ -79,10 +79,11 @@
 CC=${CC-"cc"}
 MKDEP_CPP=${MKDEP_CPP-"${CC} -E"}
 MKDEP_CPP_OPTS=${MKDEP_CPP_OPTS-"-M"};
+MKDEP_CPP_FILTERED=$(echo "$@" | sed s/-a\ // | sed s/-O2\ //)
 
 echo "# $@" > $TMP	# store arguments for debugging
 
-if $MKDEP_CPP $MKDEP_CPP_OPTS "$@" >> $TMP; then :
+if $MKDEP_CPP $MKDEP_CPP_OPTS $MKDEP_CPP_FILTERED >> $TMP; then :
 else
 	echo 'mkdep: compile failed' >&2
 	exit 1
diff -Naur klone-2.4.0-orig/configure klone-2.4.0/configure
--- klone-2.4.0-orig/configure	2011-05-02 16:52:18.000000000 +0200
+++ klone-2.4.0/configure	2012-04-20 19:44:15.550995982 +0200
@@ -190,11 +190,6 @@
 # makl_add_var_mk "CFLAGS" "\$(LIBOPENSSL_CFLAGS)"
 makl_add_var_mk "CFLAGS" "\$(LIBZ_CFLAGS)"

-# if debug is not enabled add optimization flags
-if [ -z "`makl_get_var_mk "KLONE_DEBUG"`" ]; then
-    makl_append_var_mk "CFLAGS"   "-O2"
-fi
-
 # adds -fstack-protector-all to the CFLAGS when supported by the compiler
 if [ -z "`makl_get_var_mk "DISABLE_STACK_PROTECTOR"`" ]; then
     if [ "`makl_get_var_mk "HAVE_STACK_PROTECTOR"`" ]; then
