diff -Naur uclibc_build_env_orig/make-incs/e2fsprogs.mk uclibc_build_env/make-incs/e2fsprogs.mk
--- uclibc_build_env_orig/make-incs/e2fsprogs.mk	2010-02-10 15:52:03.000000000 +0100
+++ uclibc_build_env/make-incs/e2fsprogs.mk	2010-02-09 17:48:39.000000000 +0100
@@ -29,7 +29,7 @@
 
 E2FSPROGS_FILE := e2fsprogs-$(E2FSPROGS_VERSION).tar.gz
 E2FSPROGS_DLFILE := $(DOWNLOAD_DIR)/$(E2FSPROGS_FILE)
-E2FSPROGS_URL := http://dfn.dl.sourceforge.net/sourceforge/e2fsprogs/$(E2FSPROGS_FILE)
+E2FSPROGS_URL := http://sourceforge.net/projects/e2fsprogs/files/e2fsprogs/$(E2FSPROGS_VERSION)/$(E2FSPROGS_FILE)/download
 E2FSPROGS_DIR := $(BUILD_DIR)/e2fsprogs-$(E2FSPROGS_VERSION)
 E2FSPROGS_BUILD_DIR := $(BUILD_DIR)/e2fsprogs.build
 
