diff -Naur uclibc_build_env_orig/make-incs/lirc.mk uclibc_build_env/make-incs/lirc.mk
--- uclibc_build_env_orig/make-incs/lirc.mk	2010-02-10 15:56:11.000000000 +0100
+++ uclibc_build_env/make-incs/lirc.mk	2010-02-09 17:48:39.000000000 +0100
@@ -29,7 +29,7 @@
 LIRC_PATCHES_DIR := $(PATCHES_DIR)/lirc/$(LIRC_VERSION)
 LIRC_FILE := lirc-$(LIRC_VERSION).tar.bz2
 LIRC_DLFILE := $(DOWNLOAD_DIR)/$(LIRC_FILE)
-LIRC_URL := http://sourceforge.net/projects/lirc/files/LIRC/$(LIRC_VERSION)/$(LIRC_FILE)/download
+LIRC_URL := http://dfn.dl.sourceforge.net/sourceforge/lirc/$(LIRC_FILE)
 LIRC_DIR := $(BUILD_DIR)/lirc-$(LIRC_VERSION)
 
 LIRC_INSTALLED = $(STAGEFILES_DIR)/.lirc_installed
