diff -Naur bootmenu-ori/make-incs/bootmenu.mk bootmenu/make-incs/bootmenu.mk
--- bootmenu-ori/make-incs/bootmenu.mk	Fri Jan 25 12:37:11 2008
+++ bootmenu/make-incs/bootmenu.mk	Fri Feb 22 18:55:23 2008
@@ -27,6 +27,7 @@
 BOOTMENU_SVN := svn://open7x0.org/boxamp
 BOOTMENU_DIR := $(BUILD_DIR)/bootmenu
 BOOTMENU_TC_FILE := $(BOOTMENU_DIR)/linux-mips-uclibc.tc
+BOOTMENU_PATCHES_DIR := $(PATCHES_DIR)/bootmenu
 
 LASTDIR := `pwd`
 
@@ -67,6 +68,7 @@
 # compile ptypes and bootmenu
 #
 $(STAGEFILES_DIR)/.bootmenu_compiled: $(STAGEFILES_DIR)/.bootmenu_checkedout
+	$(PATCHES_DIR)/patch-kernel.sh $(BOOTMENU_DIR)/src $(BOOTMENU_PATCHES_DIR)
 	$(CD) $(BOOTMENU_DIR)/src && env $(UCLIBC_ENV) $(MAKE) -f Makefile.tc $(UCLIBC_ENV)
 	$(TOUCH) $(STAGEFILES_DIR)/.bootmenu_compiled
 
@@ -83,10 +85,8 @@
 .PHONY: clean-bootmenu distclean-bootmenu
 
 clean-bootmenu:
-	(if [ -f $(BOOTMENU_DIR)/src ] ; then \
-		$(CD) $(BOOTMENU_DIR)/src && $(MAKE) -f Makefile.tc clean ; \
-	fi );
-	
+	-$(RM) -rf $(BOOTMENU_DIR)/src
+
 distclean-bootmenu:
 	-$(RM) -rf $(BOOTMENU_DIR)
 	-$(RM) -f $(STAGEFILES_DIR)/.bootmenu_checkedout
