diff -Naur uclibc_build_env_orig/make-incs/vdr-m7x0.mk uclibc_build_env/make-incs/vdr-m7x0.mk
--- uclibc_build_env_orig/make-incs/vdr-m7x0.mk	2008-02-21 20:54:34.000000000 +0100
+++ uclibc_build_env/make-incs/vdr-m7x0.mk	2008-05-10 15:27:30.000000000 +0200
@@ -26,6 +26,7 @@
 VDR_DIR := $(BUILD_DIR)/vdr-m7x0
 VDR_CONF_COMMON_DIR := $(PRG_CONFIGS_DIR)/vdr-m7x0/common
 VDR_CONF_SYSTYPE_DIR := $(PRG_CONFIGS_DIR)/vdr-m7x0/$(CONFIG_M7X0_TYPE)
+VDR_PATCHES_DIR := $(PATCHES_DIR)/vdr-m7x0
 
 VDR_BRANCH_DIR = $(VDR_DIR)/$(or $(strip $(filter trunk,$(CONFIG_VDR_TREE))), \
                                  branches/$(CONFIG_VDR_TREE))
@@ -76,6 +77,14 @@
 	$(TOUCH) $(STAGEFILES_DIR)/.vdr_local_downloaded
 
 #
+# patch vdr
+#
+
+	(if [ ! -f $(VDR_BRANCH_DIR)/HISTORY-liemikuutio ] ; then \
+	$(PATCHES_DIR)/patch-kernel.sh $(VDR_BRANCH_DIR) $(VDR_PATCHES_DIR) ; \
+	fi);
+
+#
 # compile vdr
 #
 
