--- uclibc_build_env_orig/make-incs/vdr-m7x0.mk	2008-09-15 21:14:17.000000000 +0200
+++ uclibc_build_env/make-incs/vdr-m7x0.mk	2008-09-15 21:54:51.000000000 +0200
@@ -26,6 +26,7 @@
 VDR_DIR := $(BUILD_DIR)/vdr-m7x0
 VDR_CONF_COMMON_DIR := $(PRG_CONFIGS_DIR)/vdr-m7x0/common
 VDR_CONF_SYSTYPE_DIR := $(PRG_CONFIGS_DIR)/vdr-m7x0/$(CONFIG_M7X0_TYPE)
+VDR_PATCHES_DIR := $(PATCHES_DIR)/vdr-m7x0
 
 VDR_BRANCH_DIR = $(VDR_DIR)/$(or $(strip $(filter trunk,$(CONFIG_VDR_TREE))), \
                                  branches/$(CONFIG_VDR_TREE))
@@ -64,7 +65,7 @@
 #
 
 $(STAGEFILES_DIR)/.vdr_downloaded: $(TC_INIT_RULE)
-	$(SVN) co $(VDR_SVN_URL) $(VDR_DIR)
+	$(SVN) co -r 348 $(VDR_SVN_URL) $(VDR_DIR)
 	$(TOUCH) $(STAGEFILES_DIR)/.vdr_downloaded
 
 $(STAGEFILES_DIR)/.vdr_local_downloaded: $(CONFIG_VDR_LOCAL_PATH)/*.c \
@@ -76,11 +77,14 @@
 	$(TOUCH) $(STAGEFILES_DIR)/.vdr_local_downloaded
 
 #
-# compile vdr
+# patch and compile vdr
 #
 
 $(STAGEFILES_DIR)/.vdr_$(CONFIG_VDR_TREE)_compiled: $$(VDR_DEPS) \
         $(STAGEFILES_DIR)/.vdr_$(filter local_,$(CONFIG_VDR_TREE)_)downloaded
+	(if [ ! -f $(VDR_BRANCH_DIR)/HISTORY-liemikuutio ] ; then \
+	$(call patch_package, $(VDR_BRANCH_DIR), $(VDR_PATCHES_DIR)) ; \
+	fi);
 	$(UCLIBC_ENV) $(MAKE) -C $(VDR_BRANCH_DIR) clean
 	$(UCLIBC_ENV) $(MAKE) -C $(VDR_BRANCH_DIR) all
 	$(TOUCH) $(STAGEFILES_DIR)/.vdr_$(CONFIG_VDR_TREE)_compiled
