diff -Naur uclibc_build_env_orig/make-incs/vdr-m7x0-plugins.mk uclibc_build_env/make-incs/vdr-m7x0-plugins.mk
--- uclibc_build_env_orig/make-incs/vdr-m7x0-plugins.mk	2008-10-07 20:20:25.000000000 +0200
+++ uclibc_build_env/make-incs/vdr-m7x0-plugins.mk	2008-10-13 15:05:51.000000000 +0200
@@ -24,8 +24,10 @@
 
 VDR-PLUGINS_SVN_URL := svn://open7x0.org/vdr-m7x0-PLUGINS
 VDR-PLUGINS_DIR := $(BUILD_DIR)/vdr-m7x0-PLUGINS
+VDR-PLUGINS_EXTRA_DIR := $(BUILD_DIR)/vdr-m7x0-PLUGINS-extra
 VDR-PLUGINS_CONF_COMMON_DIR := $(PRG_CONFIGS_DIR)/vdr-m7x0-plugins/common
 VDR-PLUGINS_CONF_SYSTYPE_DIR := $(PRG_CONFIGS_DIR)/vdr-m7x0-plugins/$(CONFIG_M7X0_TYPE)
+VDR-PLUGINS_BRANCH := branches/`$(ECHO) $(CONFIG_VDR-PLUGINS) | cut -d'/' -f1`
 
 vdr-plugins_rules = \
    $(foreach plugin,$(CONFIG_VDR-PLUGINS), \
@@ -81,6 +83,22 @@
 #
 $(STAGEFILES_DIR)/.vdr-plugins_downloaded: $(TC_INIT_RULE)
 	$(SVN) co $(VDR-PLUGINS_SVN_URL) $(VDR-PLUGINS_DIR)
+	$(RM) -rf $(VDR-PLUGINS_DIR)/$(VDR-PLUGINS_BRANCH)/.svn
+	$(CP) -Trf ../VDR-Plugins/Source+patch $(VDR-PLUGINS_EXTRA_DIR)
+	(for dp in `ls $(VDR-PLUGINS_EXTRA_DIR)`; do \
+		if [ -d $(VDR-PLUGINS_EXTRA_DIR)/$$dp ] ; then \
+			for dps in `ls $(VDR-PLUGINS_EXTRA_DIR)/$$dp/`; do \
+				if [ -d $(VDR-PLUGINS_EXTRA_DIR)/$$dp/$$dps ] ; then \
+					dptarget=`$(ECHO) $$dps | $(CUT) -d'-' -f1` ; \
+					$(RM) -rf $(VDR-PLUGINS_DIR)/$(VDR-PLUGINS_BRANCH)/$$dptarget/ ; \
+					$(MKDIR) -p $(VDR-PLUGINS_DIR)/$(VDR-PLUGINS_BRANCH)/$$dptarget/ ; \
+					$(CP) -Trf $(VDR-PLUGINS_EXTRA_DIR)/$$dp/$$dps $(VDR-PLUGINS_DIR)/$(VDR-PLUGINS_BRANCH)/$$dptarget ; \
+					$(FIND) $(VDR-PLUGINS_DIR)/$(VDR-PLUGINS_BRANCH)/$$dptarget/ '(' -name '.svn'  ')' -exec $(RM) -rf {} \; 2> /dev/null ; \
+					$(PATCHES_DIR)/patch-kernel.sh $(VDR-PLUGINS_DIR)/$(VDR-PLUGINS_BRANCH)/$$dptarget/ $(VDR-PLUGINS_EXTRA_DIR)/$$dp ; \
+				fi ; \
+			done ; \
+		fi ; \
+	done)
 	$(TOUCH) $(STAGEFILES_DIR)/.vdr-plugins_downloaded
 
 #
@@ -119,6 +137,7 @@
 
 $(FILELIST_DIR)/vdr-plugins.lst: $(STAGEFILES_DIR)/.vdr-plugins_installed
 	($(ECHO)   "usr/lib/vdr       -                 d 755 0 0 - - - - -"; \
+	 $(ECHO)   "etc/plugins-lib   -                 d 755 0 0 - - - - -"; \
 	for lib in $(VDR-PLUGINS_LIBNAMES) ; do \
 		$(ECHO) "usr/lib/vdr/$$lib usr/lib/vdr/$$lib f 755 0 0 - - - - -"; \
 		$(ECHO) "usr/lib/vdr/$$lib -                 s   - - - - - - - -"; \
@@ -136,6 +155,7 @@
 
 clean-vdr-plugins:
 	-$(RM) -rf $(VDR-PLUGINS_DIR)
+	-$(RM) -rf $(VDR-PLUGINS_EXTRA_DIR)
 
 distclean-vdr-plugins:
 	-$(RM) -f $(STAGEFILES_DIR)/.vdr-plugins_*
