diff -Naur uclibc_build_env_orig/make-incs/bash.mk uclibc_build_env/make-incs/bash.mk
--- uclibc_build_env_orig/make-incs/bash.mk	2008-01-20 18:11:33.000000000 +0100
+++ uclibc_build_env/make-incs/bash.mk	2008-01-19 20:38:37.000000000 +0100
@@ -144,6 +144,8 @@
 
 $(STAGEFILES_DIR)/.bash_installed: $(STAGEFILES_DIR)/.bash_compiled
 	$(UCLIBC_ENV) $(MAKE) -C $(BASH_BUILD_DIR) install
+	$(CP) $(BASH_DIR)/profile $(TARGET_ROOT)/etc/profile
+	chmod 644 $(TARGET_ROOT)/etc/profile
 	$(TOUCH) $(STAGEFILES_DIR)/.bash_installed
 
 $(FILELIST_DIR)/bash.lst: $(STAGEFILES_DIR)/.bash_installed
diff -Naur uclibc_build_env_orig/make-incs/busybox.mk uclibc_build_env/make-incs/busybox.mk
--- uclibc_build_env_orig/make-incs/busybox.mk	2008-01-20 18:11:33.000000000 +0100
+++ uclibc_build_env/make-incs/busybox.mk	2008-01-21 20:04:11.000000000 +0100
@@ -77,6 +77,12 @@
 $(STAGEFILES_DIR)/.busybox_configured: $(STAGEFILES_DIR)/.busybox_patched
 	$(SED) -ne 's,^CONFIG_PREFIX=.*,CONFIG_PREFIX=\"$(TARGET_ROOT)\",g' \
 		-e 'w$(BUSYBOX_DIR)/.config' $(BUSYBOX_CONFIG)
+  ifeq ($(CONFIG_BASH),y)
+	$(SED) -i -e 's,^CONFIG_FEATURE_SH_IS_ASH=.*,# CONFIG_FEATURE_SH_IS_ASH is not set,g' \
+		$(BUSYBOX_DIR)/.config
+	$(SED) -i -e 's,^# CONFIG_FEATURE_SH_IS_NONE.*,CONFIG_FEATURE_SH_IS_NONE=y,g' \
+		$(BUSYBOX_DIR)/.config
+  endif
 	$(UCLIBC_ENV) $(MAKE) -C $(BUSYBOX_DIR) CROSS_COMPILE=mips-linux-uclibc- \
 		ARCH=mips oldconfig
 	$(TOUCH) $(STAGEFILES_DIR)/.busybox_configured
diff -Naur uclibc_build_env_orig/rootfs-file-lists/bash.lst uclibc_build_env/rootfs-file-lists/bash.lst
--- uclibc_build_env_orig/rootfs-file-lists/bash.lst	2008-01-20 18:11:29.000000000 +0100
+++ uclibc_build_env/rootfs-file-lists/bash.lst	2008-01-21 20:07:57.000000000 +0100
@@ -1,3 +1,5 @@
 bin/bash    bin/bash    f 755 0 0 - - - - -
 bin/bash    -           s   - - - - - - - -
+bin/sh      bash        l 777 0 0 - - - - -
 bin/bashbug bin/bashbug f 755 0 0 - - - - -
+etc/profile etc/profile f 644 0 0 - - - - -
