diff -Naur uclibc_build_env_orig/rootfs-file-lists/ncurses.lst uclibc_build_env/rootfs-file-lists/ncurses.lst
--- uclibc_build_env_orig/rootfs-file-lists/ncurses.lst	2008-02-14 22:04:01.000000000 +0100
+++ uclibc_build_env/rootfs-file-lists/ncurses.lst	2008-11-13 20:07:15.000000000 +0100
@@ -33,6 +33,7 @@
 usr/share/terminfo/l/linux         usr/share/terminfo/l/linux         f 644 0 0 - - - - -
 usr/share/terminfo/v               -                                  d 755 0 0 - - - - -
 usr/share/terminfo/v/vt100         usr/share/terminfo/v/vt100         f 644 0 0 - - - - -
+usr/share/terminfo/v/vt102         vt100                              l 777 0 0 - - - - -
 usr/share/terminfo/v/vt320         usr/share/terminfo/v/vt320         f 644 0 0 - - - - -
 usr/share/terminfo/v/vt300         vt320                              l 777 0 0 - - - - -
 usr/share/terminfo/v/vt400         usr/share/terminfo/v/vt400         f 644 0 0 - - - - -
