diff -Naur uclibc_build_env_orig/make-incs/webif.mk uclibc_build_env/make-incs/webif.mk
--- uclibc_build_env_orig/make-incs/webif.mk	2008-10-03 21:52:13.000000000 +0200
+++ uclibc_build_env/make-incs/webif.mk	2008-10-13 10:39:24.000000000 +0200
@@ -40,18 +40,6 @@
 CLEAN_RULES += clean-webif
 DISTCLEAN_RULES += distclean-webif
 
-ifneq ($(and $(filter y,$(CONFIG_WEBIF)), $(wildcard $(WEBIF_DIR))),)
-$(info Updating webif download ...)
-webif_svn_changed := $(call svn_changed, $(WEBIF_DIR))
-
-ifeq ($(webif_svn_changed),0)
-$(info webif is up to date)
-else
-$(info $(webif_svn_changed) file(s) updated)
-webif_tmp := $(shell $(TOUCH) $(STAGEFILES_DIR)/.webapp_downloaded)
-endif
-endif
-
 #
 # download webif
 #
