diff -Naur uclibc_build_env_orig/configs/siemens-linux-kernel/m740.config uclibc_build_env/configs/siemens-linux-kernel/m740.config
--- uclibc_build_env_orig/configs/siemens-linux-kernel/m740.config	2009-04-08 18:21:47.000000000 +0200
+++ uclibc_build_env/configs/siemens-linux-kernel/m740.config	2009-04-08 19:05:23.000000000 +0200
@@ -466,6 +466,14 @@
 # CONFIG_HIPPI is not set
 # CONFIG_PLIP is not set
 # CONFIG_PPP is not set
+# CONFIG_PPP_MULTILINK is not set
+# CONFIG_PPP_FILTER is not set
+# CONFIG_PPP_ASYNC is not set
+# CONFIG_PPP_SYNC_TTY is not set
+# CONFIG_PPP_DEFLATE is not set
+# CONFIG_PPP_BSDCOMP is not set
+# CONFIG_PPP_MPPE is not set
+# CONFIG_PPPOE is not set
 # CONFIG_SLIP is not set
 
 #
@@ -914,6 +922,33 @@
 # CONFIG_MIPS_UNCACHED is not set
 
 #
+# Cryptographic options
+#
+# CONFIG_CRYPTO is not set
+# CONFIG_CRYPTO_HMAC is not set
+# CONFIG_CRYPTO_NULL is not set
+# CONFIG_CRYPTO_MD4 is not set
+# CONFIG_CRYPTO_MD5 is not set
+# CONFIG_CRYPTO_SHA1 is not set
+# CONFIG_CRYPTO_SHA256 is not set
+# CONFIG_CRYPTO_SHA512 is not set
+# CONFIG_CRYPTO_WP512 is not set
+# CONFIG_CRYPTO_DES is not set
+# CONFIG_CRYPTO_BLOWFISH is not set
+# CONFIG_CRYPTO_TWOFISH is not set
+# CONFIG_CRYPTO_SERPENT is not set
+# CONFIG_CRYPTO_AES is not set
+# CONFIG_CRYPTO_CAST5 is not set
+# CONFIG_CRYPTO_CAST6 is not set
+# CONFIG_CRYPTO_TEA is not set
+# CONFIG_CRYPTO_KHAZAD is not set
+# CONFIG_CRYPTO_ANUBIS is not set
+# CONFIG_CRYPTO_ARC4 is not set
+# CONFIG_CRYPTO_DEFLATE is not set
+# CONFIG_CRYPTO_MICHAEL_MIC is not set
+# CONFIG_CRYPTO_TEST is not set
+
+#
 # Library routines
 #
 CONFIG_ZLIB_INFLATE=y
diff -Naur uclibc_build_env_orig/configs/siemens-linux-kernel/m750s.config uclibc_build_env/configs/siemens-linux-kernel/m750s.config
--- uclibc_build_env_orig/configs/siemens-linux-kernel/m750s.config	2009-04-08 18:21:47.000000000 +0200
+++ uclibc_build_env/configs/siemens-linux-kernel/m750s.config	2009-04-08 19:04:55.000000000 +0200
@@ -466,6 +466,14 @@
 # CONFIG_HIPPI is not set
 # CONFIG_PLIP is not set
 # CONFIG_PPP is not set
+# CONFIG_PPP_MULTILINK is not set
+# CONFIG_PPP_FILTER is not set
+# CONFIG_PPP_ASYNC is not set
+# CONFIG_PPP_SYNC_TTY is not set
+# CONFIG_PPP_DEFLATE is not set
+# CONFIG_PPP_BSDCOMP is not set
+# CONFIG_PPP_MPPE is not set
+# CONFIG_PPPOE is not set
 # CONFIG_SLIP is not set
 
 #
@@ -914,6 +922,33 @@
 # CONFIG_MIPS_UNCACHED is not set
 
 #
+# Cryptographic options
+#
+# CONFIG_CRYPTO is not set
+# CONFIG_CRYPTO_HMAC is not set
+# CONFIG_CRYPTO_NULL is not set
+# CONFIG_CRYPTO_MD4 is not set
+# CONFIG_CRYPTO_MD5 is not set
+# CONFIG_CRYPTO_SHA1 is not set
+# CONFIG_CRYPTO_SHA256 is not set
+# CONFIG_CRYPTO_SHA512 is not set
+# CONFIG_CRYPTO_WP512 is not set
+# CONFIG_CRYPTO_DES is not set
+# CONFIG_CRYPTO_BLOWFISH is not set
+# CONFIG_CRYPTO_TWOFISH is not set
+# CONFIG_CRYPTO_SERPENT is not set
+# CONFIG_CRYPTO_AES is not set
+# CONFIG_CRYPTO_CAST5 is not set
+# CONFIG_CRYPTO_CAST6 is not set
+# CONFIG_CRYPTO_TEA is not set
+# CONFIG_CRYPTO_KHAZAD is not set
+# CONFIG_CRYPTO_ANUBIS is not set
+# CONFIG_CRYPTO_ARC4 is not set
+# CONFIG_CRYPTO_DEFLATE is not set
+# CONFIG_CRYPTO_MICHAEL_MIC is not set
+# CONFIG_CRYPTO_TEST is not set
+
+#
 # Library routines
 #
 CONFIG_ZLIB_INFLATE=y
diff -Naur uclibc_build_env_orig/make-incs/siemens-linux-kernel.mk uclibc_build_env/make-incs/siemens-linux-kernel.mk
--- uclibc_build_env_orig/make-incs/siemens-linux-kernel.mk	2009-04-08 18:21:37.000000000 +0200
+++ uclibc_build_env/make-incs/siemens-linux-kernel.mk	2009-04-08 19:16:46.000000000 +0200
@@ -60,6 +60,17 @@
    lib/modules/2.4.21-xfs/kernel/fs/cifs/cifs.o \
    lib/modules/2.4.21-xfs/kernel/drivers/usb/host/ehci-hcd.o \
    lib/modules/2.4.21-xfs/kernel/drivers/usb/host/usb-ohci.o 
+
+ifeq ($(CONFIG_PPTPD),y)
+   SIEMENS-LINUX-KERNEL_MODLST += \
+	lib/modules/2.4.21-xfs/kernel/crypto/sha1.o \
+	lib/modules/2.4.21-xfs/kernel/crypto/arc4.o \
+	lib/modules/2.4.21-xfs/kernel/drivers/net/slhc.o \
+	lib/modules/2.4.21-xfs/kernel/drivers/net/ppp_generic.o \
+	lib/modules/2.4.21-xfs/kernel/drivers/net/ppp_async.o \
+	lib/modules/2.4.21-xfs/kernel/drivers/net/ppp_mppe.o
+endif
+
 SIEMENS-LINUX-KERNEL_DIRLST := \
    lib/modules \
    lib/modules/2.4.21-xfs \
@@ -69,6 +80,13 @@
    lib/modules/2.4.21-xfs/kernel/drivers \
    lib/modules/2.4.21-xfs/kernel/drivers/usb \
    lib/modules/2.4.21-xfs/kernel/drivers/usb/host     
+
+ifeq ($(CONFIG_PPTPD),y)
+   SIEMENS-LINUX-KERNEL_DIRLST += \
+	lib/modules/2.4.21-xfs/kernel/crypto \
+	lib/modules/2.4.21-xfs/kernel/drivers/net
+endif
+
 #
 # unpack siemens linux kernel
 #
@@ -104,6 +122,12 @@
 # There are some hard coded defines in tree, which are system type depending
 	$(call patch_package, $(SIEMENS-LINUX-KERNEL_DIR), \
 		$(SIEMENS-LINUX-KERNEL_PATCHES_DIR)/$(CONFIG_M7X0_TYPE))
+	if [ X"$(CONFIG_PPTPD)" = X"y" ] ; then \
+		$(call patch_package, $(SIEMENS-LINUX-KERNEL_DIR), \
+			$(SIEMENS-LINUX-KERNEL_PATCHES_DIR)/others/cryptoapi) ; \
+		$(call patch_package, $(SIEMENS-LINUX-KERNEL_DIR), \
+			$(SIEMENS-LINUX-KERNEL_PATCHES_DIR)/others/mppe) ; \
+	fi
 	$(TOUCH) $(STAGEFILES_DIR)/.siemens-linux-kernel_$(CONFIG_M7X0_TYPE)_patched
 
 #
@@ -167,6 +191,26 @@
 	PATH='$(PREFIX_BIN):$(PATH)' $(MAKE) CROSS_COMPILE=$(TARGET)- ARCH=mips \
 		CC=$(EGCS_BIN) -C $(SIEMENS-LINUX-KERNEL_DIR) \
 		INSTALL_MOD_PATH=$(TARGET_ROOT)/$(M7X0_KERNEL_DIR) modules_install
+ifeq ($(CONFIG_PPTPD),y)
+	(if [ X"`$(GREP) sha1\.o $(BUILDIN_DIR)/$(CONFIG_SCRIPT_BASE)/common/etc/rc.local.net`" = X"" ] ; then \
+		$(SED) -i 's/\/usr\/sbin\/inetd\ \/etc\/inetd.conf\ \&/\/sbin\/insmod\ \/lib\/modules\/2.4.21-xfs\/kernel\/crypto\/sha1.o\n\ \ \ \ \/usr\/sbin\/inetd\ \/etc\/inetd.conf\ \&/g' $(BUILDIN_DIR)/$(CONFIG_SCRIPT_BASE)/common/etc/rc.local.net ; \
+	fi);
+	(if [ X"`$(GREP) arc4\.o $(BUILDIN_DIR)/$(CONFIG_SCRIPT_BASE)/common/etc/rc.local.net`" = X"" ] ; then \
+		$(SED) -i 's/\/usr\/sbin\/inetd\ \/etc\/inetd.conf\ \&/\/sbin\/insmod\ \/lib\/modules\/2.4.21-xfs\/kernel\/crypto\/arc4.o\n\ \ \ \ \/usr\/sbin\/inetd\ \/etc\/inetd.conf\ \&/g' $(BUILDIN_DIR)/$(CONFIG_SCRIPT_BASE)/common/etc/rc.local.net ; \
+	fi);
+	(if [ X"`$(GREP) slhc\.o $(BUILDIN_DIR)/$(CONFIG_SCRIPT_BASE)/common/etc/rc.local.net`" = X"" ] ; then \
+		$(SED) -i 's/\/usr\/sbin\/inetd\ \/etc\/inetd.conf\ \&/\/sbin\/insmod\ \/lib\/modules\/2.4.21-xfs\/kernel\/drivers\/net\/slhc.o\n\ \ \ \ \/usr\/sbin\/inetd\ \/etc\/inetd.conf\ \&/g' $(BUILDIN_DIR)/$(CONFIG_SCRIPT_BASE)/common/etc/rc.local.net ; \
+	fi);
+	(if [ X"`$(GREP) ppp_generic\.o $(BUILDIN_DIR)/$(CONFIG_SCRIPT_BASE)/common/etc/rc.local.net`" = X"" ] ; then \
+		$(SED) -i 's/\/usr\/sbin\/inetd\ \/etc\/inetd.conf\ \&/\/sbin\/insmod\ \/lib\/modules\/2.4.21-xfs\/kernel\/drivers\/net\/ppp_generic.o\n\ \ \ \ \/usr\/sbin\/inetd\ \/etc\/inetd.conf\ \&/g' $(BUILDIN_DIR)/$(CONFIG_SCRIPT_BASE)/common/etc/rc.local.net ; \
+	fi);
+	(if [ X"`$(GREP) ppp_async\.o $(BUILDIN_DIR)/$(CONFIG_SCRIPT_BASE)/common/etc/rc.local.net`" = X"" ] ; then \
+		$(SED) -i 's/\/usr\/sbin\/inetd\ \/etc\/inetd.conf\ \&/\/sbin\/insmod\ \/lib\/modules\/2.4.21-xfs\/kernel\/drivers\/net\/ppp_async.o\n\ \ \ \ \/usr\/sbin\/inetd\ \/etc\/inetd.conf\ \&/g' $(BUILDIN_DIR)/$(CONFIG_SCRIPT_BASE)/common/etc/rc.local.net ; \
+	fi);
+	(if [ X"`$(GREP) ppp_mppe\.o $(BUILDIN_DIR)/$(CONFIG_SCRIPT_BASE)/common/etc/rc.local.net`" = X"" ] ; then \
+		$(SED) -i 's/\/usr\/sbin\/inetd\ \/etc\/inetd.conf\ \&/\/sbin\/insmod\ \/lib\/modules\/2.4.21-xfs\/kernel\/drivers\/net\/ppp_mppe.o\n\ \ \ \ \/usr\/sbin\/inetd\ \/etc\/inetd.conf\ \&/g' $(BUILDIN_DIR)/$(CONFIG_SCRIPT_BASE)/common/etc/rc.local.net ; \
+	fi);
+endif
 	$(TOUCH) $(STAGEFILES_DIR)/.siemens-linux-kernel_$(CONFIG_M7X0_TYPE)_installed
 
 
