diff -Naur uclibc_build_env_orig/make-incs/pure-ftpd.mk uclibc_build_env/make-incs/pure-ftpd.mk
--- uclibc_build_env_orig/make-incs/pure-ftpd.mk	2009-03-28 10:12:21.000000000 +0100
+++ uclibc_build_env/make-incs/pure-ftpd.mk	2009-06-01 20:14:26.000000000 +0200
@@ -140,6 +140,14 @@
 
 $(STAGEFILES_DIR)/.pure-ftpd_installed: $(STAGEFILES_DIR)/.pure-ftpd_compiled
 	$(UCLIBC_ENV) $(MAKE) -C $(PURE-FTPD_DIR) install
+	-$(RM) -rf $(TARGET_ROOT)/etc/pure-ftpd
+	$(MKDIR) -p $(TARGET_ROOT)/etc/pure-ftpd
+	$(TOUCH) $(TARGET_ROOT)/etc/pure-ftpd/pureftpd.passwd
+	$(SVN) co svn://open7x0.org/uclibc_build_env/branches/experimental/buildin/base/common/etc/pure-ftpd/pureftpd.pdb $(TARGET_ROOT)/etc/pure-ftpd/pureftpd.pdb
+	chmod 644 $(TARGET_ROOT)/etc/pure-ftpd/pureftpd.pdb
+	(if [ X"`$(GREP) pure-ftpd $(BUILDIN_DIR)/$(CONFIG_SCRIPT_BASE)/common/etc/inetd.conf`" = X"" ] ; then \
+		$(SED) -i '$$aftp\tstream tcp nowait root /usr/sbin/pure-ftpd -lpuredb' $(BUILDIN_DIR)/$(CONFIG_SCRIPT_BASE)/common/etc/inetd.conf ; \
+	fi);
 	$(TOUCH) $(STAGEFILES_DIR)/.pure-ftpd_installed
 
 $(FILELIST_DIR)/pure-ftpd.lst: $(STAGEFILES_DIR)/.pure-ftpd_installed
diff -Naur uclibc_build_env_orig/rootfs-file-lists/pure-ftpd.lst uclibc_build_env/rootfs-file-lists/pure-ftpd.lst
--- uclibc_build_env_orig/rootfs-file-lists/pure-ftpd.lst	2009-01-30 19:50:32.000000000 +0100
+++ uclibc_build_env/rootfs-file-lists/pure-ftpd.lst	2009-06-02 20:08:57.000000000 +0200
@@ -1,2 +1,5 @@
-usr/sbin/pure-ftpd    usr/sbin/pure-ftpd    f 755 0 0 - - - - -
-usr/sbin/pure-ftpd    -                     s   - - - - - - - -
+usr/sbin/pure-ftpd            usr/sbin/pure-ftpd            f 755 0 0 - - - - -
+usr/sbin/pure-ftpd            -                             s   - - - - - - - -
+etc/pure-ftpd                 -                             d 755 0 0 - - - - -
+etc/pure-ftpd/pureftpd.passwd etc/pure-ftpd/pureftpd.passwd f 644 0 0 - - - - -
+etc/pure-ftpd/pureftpd.pdb    etc/pure-ftpd/pureftpd.pdb    f 644 0 0 - - - - -
