diff -Naur uclibc_build_env_orig/make-incs/m7x0-modules.mk uclibc_build_env/make-incs/m7x0-modules.mk
--- uclibc_build_env_orig/make-incs/m7x0-modules.mk	2009-12-11 12:03:35.000000000 +0100
+++ uclibc_build_env/make-incs/m7x0-modules.mk	2010-03-12 09:32:05.818115034 +0100
@@ -68,12 +68,12 @@
 M7X0-MODULES-FW-m750s-VERSION := 2.4
 M7X0-MODULES_FILE := fw-siemens-$(CONFIG_M7X0_TYPE)-$(M7X0-MODULES-FW-$(CONFIG_M7X0_TYPE)-VERSION).zip
 M7X0-MODULES_DLFILE := $(DOWNLOAD_DIR)/$(M7X0-MODULES_FILE)
-M7X0-REFERER_URL := http://gigaset.siemens.com/
+M7X0-REFERER_URL := http://gigaset.com/
 
 ifeq ($(CONFIG_M7X0_TYPE),m740)
-M7X0-MODULES_URL := http://shc-download.siemens.com/repository/1374/137469/Gigaset_M740_AV_V_2.4.exe
+M7X0-MODULES_URL := http://gigaset.com/medias/sys_master/Gigaset_M740_AV_V_2.4.exe
 else ifeq ($(CONFIG_M7X0_TYPE),m750s)
-M7X0-MODULES_URL := http://shc-download.siemens.com/repository/1374/137472/Gigaset_M750_S_V_2.4.exe
+M7X0-MODULES_URL := http://gigaset.com/medias/sys_master/Gigaset_M750_S_V_2.4.exe
 else
 $(error unknown system type given)
 endif
diff -Naur uclibc_build_env_orig/make-incs/siemens-gpl-src.mk uclibc_build_env/make-incs/siemens-gpl-src.mk
--- uclibc_build_env_orig/make-incs/siemens-gpl-src.mk	2009-12-11 12:03:35.000000000 +0100
+++ uclibc_build_env/make-incs/siemens-gpl-src.mk	2010-03-12 09:33:46.309230523 +0100
@@ -33,8 +33,8 @@
 SIEMENS-BUILD-ENV_FILE := siemens-build_env.tar.gz
 SIEMENS-BUILD-ENV_DLFILE = $(DOWNLOAD_DIR)/$(SIEMENS-BUILD-ENV_FILE)
 
-SIEMENS-GPL-SRC_URL := http://gigaset.siemens.com/repository/1222/122244/gigaset_m750_gplsw_1_4_3_tar.bz2
-SIEMENS-BUILD-ENV_URL := http://gigaset.siemens.com/repository/1237/123736/wavebox_build_env.tar.gz
+SIEMENS-GPL-SRC_URL := http://zenvoid.org/software/m7x0/siemens-m750-gpl-src.tar.bz2
+SIEMENS-BUILD-ENV_URL := http://zenvoid.org/software/m7x0/wavebox_build_env.tar.gz
 # As the extracted archive is very big (around 1GB) and
 # we need only a very small part of it we don't extracted fully
 # BTW: This thing has wrong naming it should called siemens-gpl-trash ...
diff -Naur uclibc_build_env_orig/hostutils-make-incs/cramfs.mk uclibc_build_env/hostutils-make-incs/cramfs.mk
--- uclibc_build_env_orig/hostutils-make-incs/cramfs.mk	2009-12-11 12:03:31.000000000 +0100
+++ uclibc_build_env/hostutils-make-incs/cramfs.mk	2010-02-15 12:19:20.957516987 +0100
@@ -32,7 +32,7 @@
 CRAMFS_HOSTDLFILE := $(DOWNLOAD_DIR)/$(CRAMFS_HOSTFILE)
 
 CRAMFS_HOSTPATCHES_DIR := $(PATCHES_DIR)/cramfs/$(CRAMFS_HOSTVERSION)
-CRAMFS_HOSTURL := http://dfn.dl.sourceforge.net/sourceforge/cramfs/$(CRAMFS_HOSTFILE)
+CRAMFS_HOSTURL := http://sourceforge.net/projects/cramfs/files/cramfs/$(CRAMFS_HOSTVERSION)/$(CRAMFS_HOSTFILE)/download
 CRAMFS_HOSTDIR := $(HOSTUTILS_BUILD_DIR)/cramfs-$(CRAMFS_HOSTVERSION)
 
 CLEAN_RULES += clean-cramfs-host
diff -Naur uclibc_build_env_orig/hostutils-make-incs/squashfs.mk uclibc_build_env/hostutils-make-incs/squashfs.mk
--- uclibc_build_env_orig/hostutils-make-incs/squashfs.mk	2009-12-11 12:03:31.000000000 +0100
+++ uclibc_build_env/hostutils-make-incs/squashfs.mk	2010-02-15 12:17:32.729290653 +0100
@@ -35,7 +35,7 @@
 SQUASHFS_HOSTDLFILE := $(DOWNLOAD_DIR)/$(SQUASHFS_HOSTFILE)
 
 SQUASHFS_HOSTPATCHES_DIR := $(PATCHES_DIR)/squashfs/$(SQUASHFS_HOSTVERSION)
-SQUASHFS_HOSTURL := http://surfnet.dl.sourceforge.net/sourceforge/squashfs/$(SQUASHFS_HOSTFILE)
+SQUASHFS_HOSTURL := http://mesh.dl.sourceforge.net/sourceforge/squashfs/$(SQUASHFS_HOSTFILE)
 SQUASHFS_HOSTDIR := $(HOSTUTILS_BUILD_DIR)/squashfs$(SQUASHFS_HOSTVERSION)
 
 CLEAN_RULES += clean-squashfs-host
diff -Naur uclibc_build_env_orig/make-incs/e2fsprogs.mk uclibc_build_env/make-incs/e2fsprogs.mk
--- uclibc_build_env_orig/make-incs/e2fsprogs.mk	2009-12-11 12:03:35.000000000 +0100
+++ uclibc_build_env/make-incs/e2fsprogs.mk	2010-02-15 17:30:40.780024398 +0100
@@ -29,7 +29,7 @@
 
 E2FSPROGS_FILE := e2fsprogs-$(E2FSPROGS_VERSION).tar.gz
 E2FSPROGS_DLFILE := $(DOWNLOAD_DIR)/$(E2FSPROGS_FILE)
-E2FSPROGS_URL := http://dfn.dl.sourceforge.net/sourceforge/e2fsprogs/$(E2FSPROGS_FILE)
+E2FSPROGS_URL := http://sourceforge.net/projects/e2fsprogs/files/e2fsprogs/$(E2FSPROGS_VERSION)/$(E2FSPROGS_FILE)/download
 E2FSPROGS_DIR := $(BUILD_DIR)/e2fsprogs-$(E2FSPROGS_VERSION)
 E2FSPROGS_BUILD_DIR := $(BUILD_DIR)/e2fsprogs.build
 
diff -Naur uclibc_build_env_orig/make-incs/lirc.mk uclibc_build_env/make-incs/lirc.mk
--- uclibc_build_env_orig/make-incs/lirc.mk	2009-12-11 12:03:35.000000000 +0100
+++ uclibc_build_env/make-incs/lirc.mk	2010-02-15 17:30:56.648752439 +0100
@@ -29,7 +29,7 @@
 LIRC_PATCHES_DIR := $(PATCHES_DIR)/lirc/$(LIRC_VERSION)
 LIRC_FILE := lirc-$(LIRC_VERSION).tar.bz2
 LIRC_DLFILE := $(DOWNLOAD_DIR)/$(LIRC_FILE)
-LIRC_URL := http://dfn.dl.sourceforge.net/sourceforge/lirc/$(LIRC_FILE)
+LIRC_URL := http://sourceforge.net/projects/lirc/files/LIRC/$(LIRC_VERSION)/$(LIRC_FILE)/download
 LIRC_DIR := $(BUILD_DIR)/lirc-$(LIRC_VERSION)
 
 LIRC_INSTALLED = $(STAGEFILES_DIR)/.lirc_installed
diff -Naur uclibc_build_env_orig/patches/ssmtp/2.64/300_ssmtp_conf.patch uclibc_build_env/patches/ssmtp/2.64/300_ssmtp_conf.patch
--- uclibc_build_env_orig/patches/ssmtp/2.64/300_ssmtp_conf.patch	1970-01-01 01:00:00.000000000 +0100
+++ uclibc_build_env/patches/ssmtp/2.64/300_ssmtp_conf.patch	2009-10-02 12:58:56.000000000 +0200
@@ -0,0 +1,64 @@
+--- ssmtp-2.61-13/ssmtp.conf	2007-11-18 19:34:23.000000000 +0100
++++ conf-m740av/ssmtp.conf	2007-11-18 21:27:01.000000000 +0100
+@@ -1,15 +1,18 @@
+ #
+-# /etc/ssmtp.conf -- a config file for sSMTP sendmail.
++# Config file for sSMTP sendmail
+ #
++# Get enhanced (*really* enhanced) debugging information in the logs
++# If you want to have debugging of the config file parsing, move this option
++# to the top of the config file and uncomment
++#Debug=YES
+ 
+ # The person who gets all mail for userids < 1000
+ # Make this empty to disable rewriting.
+ root=postmaster
+ 
+-# The place where the mail goes. The actual machine name is required
+-# no MX records are consulted. Commonly mailhosts are named mail.domain.com
+-# The example will fit if you are in domain.com and your mailhub is so named.
+-mailhub=mail
++# The place where the mail goes. The actual machine name is required no
++# MX records are consulted. Commonly mailhosts are named mail.domain.com
++mailhub=mail.your_provider.de
+ 
+ # Example for SMTP port number 2525
+ # mailhub=mail.your.domain:2525
+@@ -18,26 +21,18 @@
+ # Example for SSL encrypted connection
+ # mailhub=mail.your.domain:465
+ 
+-# Where will the mail seem to come from?
+-rewriteDomain=
+-
+-# The full hostname
+-hostname=_HOSTNAME_
+-
+-# Set this to never rewrite the "From:" line (unless not given) and to
+-# use that address in the "from line" of the envelope.
+-#FromLineOverride=YES
+ 
+-# Use SSL/TLS to send secure messages to server.
+-#UseTLS=YES
++# Falls SMTP-AUTH verwendet :
++AuthUser=KlausMaier
++AuthPass=geheim
+ 
+-# Use SSL/TLS certificate to authenticate against smtp host.
+-#UseTLSCert=YES
++# Where will the mail seem to come from?
++#rewriteDomain=
+ 
+-# Use this RSA certificate.
+-#TLSCert=/etc/ssl/certs/ssmtp.pem
++# The full hostname
++hostname=m740av
+ 
+-# Get enhanced (*really* enhanced) debugging information in the logs
+-# If you want to have debugging of the config file parsing, move this option
+-# to the top of the config file and uncomment
+-#Debug=YES
++# Are users allowed to set their own From: address?
++# YES - Allow the user to specify their own From: address
++# NO - Use the system generated From: address
++# FromLineOverride=YES
diff -Naur uclibc_build_env_orig/patches/ssmtp/2.64/400_revaliases.patch uclibc_build_env/patches/ssmtp/2.64/400_revaliases.patch
--- uclibc_build_env_orig/patches/ssmtp/2.64/400_revaliases.patch	1970-01-01 01:00:00.000000000 +0100
+++ uclibc_build_env/patches/ssmtp/2.64/400_revaliases.patch	2009-10-02 12:58:56.000000000 +0200
@@ -0,0 +1,13 @@
+--- ssmtp-2.61-13/revaliases	2001-04-16 13:46:22.000000000 +0200
++++ conf-m740av/revaliases	2007-11-18 00:12:00.000000000 +0100
+@@ -1,6 +1,7 @@
+ # sSMTP aliases
+-# 
+-# Format:	local_account:outgoing_address:mailhub
+ #
+-# Example: root:your_login@your.domain:mailhub.your.domain[:port]
++# Format:       local_account:outgoing_address:mailhub
++#
++# Example:      root:your_login@your.domain:mailhub.your.domain[:port]
+ # where [:port] is an optional port number that defaults to 25.
++root:KlausMaier@your_provider.de:mail.your_provider.de
diff -Naur uclibc_build_env_orig/make-incs/ssmtp.mk uclibc_build_env/make-incs/ssmtp.mk
--- uclibc_build_env_orig/make-incs/ssmtp.mk	2009-10-02 12:58:54.000000000 +0200
+++ uclibc_build_env/make-incs/ssmtp.mk	2009-12-18 16:50:38.000000000 +0100
@@ -25,11 +25,11 @@
 # Put dependencies here all pack should depend on $$(BASE_BUILD_STAGEFILE)
 SSMTP_DEPS = $(BASE_BUILD_STAGEFILE)
 
-SSMTP_VERSION := 2.63
+SSMTP_VERSION := 2.64
 SSMTP_PATCHES_DIR := $(PATCHES_DIR)/ssmtp/$(SSMTP_VERSION)
 
 #http://ftp.de.debian.org/debian/pool/main/s/ssmtp/ssmtp_2.61.orig.tar.gz
-SSMTP_FILE := ssmtp_$(SSMTP_VERSION).orig.tar.gz
+SSMTP_FILE := ssmtp_$(SSMTP_VERSION).orig.tar.bz2
 SSMTP_DLFILE := $(DOWNLOAD_DIR)/$(SSMTP_FILE)
 SSMTP_URL := http://ftp.de.debian.org/debian/pool/main/s/ssmtp/$(SSMTP_FILE)
 SSMTP_DIR := $(BUILD_DIR)/ssmtp-$(SSMTP_VERSION)
@@ -60,7 +60,7 @@
                                            $(wildcard $(SSMTP_PATCHES_DIR)/*.patch) \
                                            $$(SSMTP_DEPS)
 	-$(RM) -rf $(SSMTP_DIR)
-	$(TAR) -C $(BUILD_DIR) -zf $(SSMTP_DLFILE)
+	$(TAR) -C $(BUILD_DIR) -jf $(SSMTP_DLFILE)
 	$(TOUCH) $(STAGEFILES_DIR)/.ssmtp_unpacked
 
 #
