diff -Naur vdr-m7x0-orig/menu.c vdr-m7x0/menu.c
--- vdr-m7x0-orig/menu.c	2007-06-23 18:09:43.000000000 +0200
+++ vdr-m7x0/menu.c	2007-06-23 18:12:41.000000000 +0200
@@ -2411,6 +2411,19 @@
          }
       }
   free(LastItemText);
+
+//kw
+  char *buffer = NULL;
+  int FreeMB;
+  int Percent = VideoDiskSpace(&FreeMB);
+  int Minutes = int(double(FreeMB) / MB_PER_MINUTE);
+  int Hours = Minutes / 60;
+  Minutes %= 60;
+  asprintf(&buffer, "%s   %d%%  %2d:%02d %s  (%.1f GB)", tr("Disk"), Percent, Hours, Minutes, tr("free"), FreeMB / 1024.0 );
+  SetTitle(buffer);
+  free(buffer);
+//kw
+
   if (Refresh)
      Display();
 }
