diff -Naur VDR-NG-orig/remote.c VDR-NG/remote.c
--- VDR-NG-orig/remote.c	2008-05-09 21:08:54.000000000 +0200
+++ VDR-NG/remote.c	2008-05-09 22:19:35.000000000 +0200
@@ -22,6 +22,7 @@
 #define REPEATTIMEOUT  1000 // ms
 
 eKeys cRemote::keys[MaxKeys];
+bool cRemote::disabled = false;
 int cRemote::in = 0;
 int cRemote::out = 0;
 cTimeMs cRemote::repeatTimeout;
@@ -185,7 +186,7 @@
          if ((k & k_Repeat) != 0)
             repeatTimeout.Set(REPEATTIMEOUT);
          lastActivity = time(NULL);
-         return k;
+         return disabled ? kNone : k;
          }
       else if (!WaitMs || !keyPressed.TimedWait(mutex, WaitMs) && repeatTimeout.TimedOut())
          return kNone;
diff -Naur VDR-NG-orig/remote.h VDR-NG/remote.h
--- VDR-NG-orig/remote.h	2008-05-09 21:08:54.000000000 +0200
+++ VDR-NG/remote.h	2008-05-09 22:17:46.000000000 +0200
@@ -32,6 +32,7 @@
   static const char *keyMacroPlugin;
   static const char *callPlugin;
   char *name;
+  static bool disabled;
 protected:
   cRemote(const char *Name);
   const char *GetSetup(void);
@@ -45,6 +46,8 @@
   const char *Name(void) { return name; }
   static void SetLearning(cRemote *Learning) { learning = Learning; }
   static bool IsLearning() { return learning != NULL; }
+  static void Disable() { disabled = true; }
+  static void Enable() { disabled = false; }
   static void Clear(void);
   static bool Put(eKeys Key, bool AtFront = false);
   static bool PutMacro(eKeys Key);
diff -Naur VDR-NG-orig/svdrp.c VDR-NG/svdrp.c
--- VDR-NG-orig/svdrp.c	2008-05-09 21:28:54.000000000 +0200
+++ VDR-NG/svdrp.c	2008-05-09 22:17:46.000000000 +0200
@@ -187,6 +187,8 @@
   "    it returns the current channel number and name.",
   "CLRE\n"
   "    Clear the entire EPG list.",
+  "CTRL <on|off>\n"
+  "    Enable/Disable Input devices.",
   "DELC <number>\n"
   "    Delete channel.",
   "DELR <number>\n"
@@ -539,6 +541,26 @@
   Reply(250, "EPG data cleared");
 }
 
+void cSVDRP::CmdCTRL(const char *Option)
+{
+  if (*Option) {
+    if (!strcasecmp( Option, "on" )) {
+      cRemote::Enable();
+      Reply(250, "Input Control enabled");
+      }
+    else {
+      if (!strcasecmp( Option, "off" )) {
+        cRemote::Disable();
+        Reply(250, "Input Control disabled");
+        }
+      else
+        Reply(250, "Illegal Command %s", Option);
+      }
+    }
+  else
+    Reply(250, "Parameter missing");
+}
+
 void cSVDRP::CmdDELC(const char *Option)
 {
   if (*Option) {
@@ -1538,6 +1560,7 @@
   s = skipspace(s);
   if      (CMD("CHAN"))  CmdCHAN(s);
   else if (CMD("CLRE"))  CmdCLRE(s);
+  else if (CMD("CTRL"))  CmdCTRL(s);
   else if (CMD("DELC"))  CmdDELC(s);
   else if (CMD("DELR"))  CmdDELR(s);
   else if (CMD("DELT"))  CmdDELT(s);
diff -Naur VDR-NG-orig/svdrp.h VDR-NG/svdrp.h
--- VDR-NG-orig/svdrp.h	2008-05-09 21:28:54.000000000 +0200
+++ VDR-NG/svdrp.h	2008-05-09 22:17:46.000000000 +0200
@@ -56,6 +56,7 @@
   void PrintHelpTopics(const char **hp);
   void CmdCHAN(const char *Option);
   void CmdCLRE(const char *Option);
+  void CmdCTRL(const char *Option);
   void CmdDELC(const char *Option);
   void CmdDELR(const char *Option);
   void CmdDELT(const char *Option);
