diff -Naur VDR-NG-orig/i18n.c VDR-NG/i18n.c
--- VDR-NG-orig/i18n.c	Fri Jun 20 09:41:14 2008
+++ VDR-NG/i18n.c	Fri Jun 20 09:51:38 2008
@@ -1372,7 +1372,7 @@
     "",//TODO
     "",//TODO
     "",//TODO
-    "",//TODO
+    "Editando - ¿reiniciar igualmente?",
     "",//TODO
     "",//TODO
     "",//TODO
@@ -1460,7 +1460,7 @@
     "",//TODO
     "",//TODO
     "",//TODO
-    "",//TODO
+    "Pulse una tecla para interrumpir reinicio",
     "",//TODO
     "",//TODO
     "",//TODO
@@ -1482,7 +1482,7 @@
     "",//TODO
     "VDR sammuu myöhemmin - pakota virtakytkimellä",
     "",//TODO
-    "",//TODO
+    "VDR se apagará más tarde - pulse Apagar para forzar",
     "",//TODO
     "",//TODO
     "",//TODO
@@ -1504,7 +1504,7 @@
     "",//TODO
     "VDR sammuu %s minuutin kuluttua",
     "",//TODO
-    "",//TODO
+    "VDR se apagará en %s minutos",
     "",//TODO
     "",//TODO
     "",//TODO
@@ -1526,7 +1526,7 @@
     "",//TODO
     "Leikkaus kesken - sammutetaanko?",
     "",//TODO
-    "",//TODO
+    "Editando - ¿apagar igualmente?",
     "",//TODO
     "",//TODO
     "",//TODO
@@ -1548,7 +1548,7 @@
     "",//TODO
     "Laajennos %s herää %ld minuutin kuluttua - sammutetaanko?",
     "",//TODO
-    "",//TODO
+    "El plugin %s iniciará en %ld min, ¿continuar?",
     "",//TODO
     "",//TODO
     "",//TODO
