diff -Naur vdr-m7x0-orig/Makefile vdr-m7x0/Makefile
--- vdr-m7x0-orig/Makefile	2007-07-28 11:31:07.000000000 +0200
+++ vdr-m7x0/Makefile	2007-07-28 11:33:21.000000000 +0200
@@ -195,9 +195,12 @@
 	for i in `ls $(PLUGINDIR)/src | grep -v '[^a-z0-9]'`; do\
 	    echo "Plugin $$i:";\
 	    if ! grep -q "\$$(LIBDIR)/.*\$$(APIVERSION)" "$(PLUGINDIR)/src/$$i/Makefile" ; then\
-	       echo "ERROR: plugin $$i doesn't honor APIVERSION - not compiled!";\
-	       noapiv="$$noapiv $$i";\
-	       continue;\
+	       sed -i -e s/VDRVERSION/APIVERSION/g $(PLUGINDIR)/src/$$i/Makefile;\
+	       if ! grep -q "\$$(LIBDIR)/.*\$$(APIVERSION)" "$(PLUGINDIR)/src/$$i/Makefile" ; then\
+	         echo "ERROR: plugin $$i doesn't honor APIVERSION - not compiled!";\
+	         noapiv="$$noapiv $$i";\
+	         continue;\
+	         fi;\
 	       fi;\
 	    DVBDIR=$(abspath .) VDRDIR=$(abspath .) LIBDIR=$(abspath $(PLUGINLIBDIR)) $(MAKE) -C "$(PLUGINDIR)/src/$$i" all || failed="$$failed $$i";\
 	    done;\
