diff -Naur VDR-NG-orig/config.c VDR-NG/config.c
--- VDR-NG-orig/config.c	2009-07-05 11:58:28.000000000 +0200
+++ VDR-NG/config.c	2009-07-05 12:05:09.000000000 +0200
@@ -342,6 +342,7 @@
 #ifdef USE_DIRECT_IO
   ReplayUseDirectIO = 0;
 #endif
+  ShutdownMessage = 0;
   AbortWhenPluginFails = 0;
   EmergencyExit = 0;
   LircRepeatDelay = 350;
@@ -539,6 +540,7 @@
 #ifdef USE_DIRECT_IO
   else if (!strcasecmp(Name, "ReplayUseDirectIO"))   ReplayUseDirectIO  = atoi(Value);
 #endif
+  else if (!strcasecmp(Name, "ShutdownMessage"))     ShutdownMessage    = atoi(Value);
   else if (!strcasecmp(Name, "AbortWhenPluginFails")) AbortWhenPluginFails = atoi(Value);
   else if (!strcasecmp(Name, "EmergencyExit"))       EmergencyExit      = atoi(Value);
   else if (!strcasecmp(Name, "LircRepeatDelay"))     LircRepeatDelay    = atoi(Value);
@@ -643,6 +645,7 @@
 #ifdef USE_DIRECT_IO
   Store("ReplayUseDirectIO",  ReplayUseDirectIO);
 #endif
+  Store("ShutdownMessage",    ShutdownMessage);
   Store("AbortWhenPluginFails", AbortWhenPluginFails);
   Store("EmergencyExit",      EmergencyExit);
   Store("LircRepeatDelay",    LircRepeatDelay);
diff -Naur VDR-NG-orig/config.h VDR-NG/config.h
--- VDR-NG-orig/config.h	2009-07-05 11:58:28.000000000 +0200
+++ VDR-NG/config.h	2009-07-05 12:05:45.000000000 +0200
@@ -299,6 +299,7 @@
 #ifdef USE_DIRECT_IO
   int ReplayUseDirectIO;
 #endif
+  int ShutdownMessage;
   int AbortWhenPluginFails;
   int EmergencyExit;
   int LircRepeatDelay;
diff -Naur VDR-NG-orig/i18n.c VDR-NG/i18n.c
--- VDR-NG-orig/i18n.c	2009-07-05 11:58:28.000000000 +0200
+++ VDR-NG/i18n.c	2009-07-05 12:08:53.000000000 +0200
@@ -5246,6 +5246,48 @@
     "som før",
     "jako naposledy",
   },
+  { "Setup.Miscellaneous$Show cancel message at shutdown",
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "Mensaje de cancelación al apagar",
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+  },
+  { "System shutdown",
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "Apagando el sistema",
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+  },
   { "Setup.Miscellaneous$Abort when Plugin fails to load",
     "Abbrechen wenn Plugin nicht lädt",
     "",// TODO
diff -Naur VDR-NG-orig/menu.c VDR-NG/menu.c
--- VDR-NG-orig/menu.c	2009-07-05 11:58:28.000000000 +0200
+++ VDR-NG/menu.c	2009-07-05 12:10:54.000000000 +0200
@@ -3278,6 +3278,7 @@
   Add(new cMenuEditStraItem(tr("Setup.Miscellaneous$Filter channels"),            &data.ChannelFilter, 3, channelFilterTexts));;    
   Add(new cMenuEditChanItem(tr("Setup.Miscellaneous$Initial channel"),            &data.InitialChannel, tr("Setup.Miscellaneous$as before")));
   Add(new cMenuEditIntItem( tr("Setup.Miscellaneous$Initial volume"),             &data.InitialVolume, -1, 255, tr("Setup.Miscellaneous$as before")));
+  Add(new cMenuEditBoolItem(tr("Setup.Miscellaneous$Show cancel message at shutdown"), &data.ShutdownMessage, tr("no"), tr("yes")));
   Add(new cMenuEditBoolItem(tr("Setup.Miscellaneous$Abort when Plugin fails to load"), &data.AbortWhenPluginFails));
   Add(new cMenuEditBoolItem(tr("Setup.Miscellaneous$Emergency exit"),             &data.EmergencyExit));
   Add(new cMenuEditIntItem( tr("Setup.Miscellaneous$LIRC repeat delay"),          &data.LircRepeatDelay, 0, 1000));
diff -Naur VDR-NG-orig/vdr.c VDR-NG/vdr.c
--- VDR-NG-orig/vdr.c	2009-07-05 11:58:28.000000000 +0200
+++ VDR-NG/vdr.c	2009-07-05 12:13:02.000000000 +0200
@@ -1163,10 +1163,14 @@
                    ShutdownHandler.SetUserInactive();
                    break;
                    }
+		   if(Setup.ShutdownMessage){
                 // Ask the final question:
                 if (!Interface->Confirm(tr("Press any key to cancel shutdown"), SHUTDOWNCANCELROMPT, true))
                    // If final question was canceled, continue to be active:
                    break;
+		   } else
+                Skins.Message(mtInfo, tr("System shutdown"));
+
                 // Ok, now call the shutdown script:
                 ShutdownHandler.DoShutdown(true);
                 // Set VDR to be non-interactive and power down again later:
