diff -Naur VDR-NG-orig/epg.c VDR-NG/epg.c
--- VDR-NG-orig/epg.c	2008-09-27 12:40:29.000000000 +0200
+++ VDR-NG/epg.c	2008-09-27 12:43:12.000000000 +0200
@@ -116,6 +116,9 @@
   duration = 0;
   vps = 0;
   SetSeen();
+//M7X0 BEGIN AK
+  cachedDateStringLang = -1;
+//M7X0 END AK
 }
 
 cEvent::~cEvent()
@@ -194,6 +197,11 @@
      startTime = StartTime;
      if (schedule)
         schedule->HashEvent(this);
+//M7X0 BEGIN AK
+     cachedDateString = DateString(startTime);
+     cachedTimeString = TimeString(startTime);
+     cachedDateStringLang = Setup.OSDLanguage;
+//M7X0 END AK
      }
 }
 
@@ -236,12 +244,19 @@
 
 cString cEvent::GetDateString(void) const
 {
-  return DateString(startTime);
+//M7X0 BEGIN AK
+  if (cachedDateStringLang != Setup.OSDLanguage) {
+     cachedDateString = DateString(startTime);
+     }
+  return cachedDateString;
+//M7X0 END AK
 }
 
 cString cEvent::GetTimeString(void) const
 {
-  return TimeString(startTime);
+//M7X0 BEGIN AK
+  return cachedTimeString;
+//M7X0 END AK
 }
 
 cString cEvent::GetEndTimeString(void) const
diff -Naur VDR-NG-orig/epg.h VDR-NG/epg.h
--- VDR-NG-orig/epg.h	2008-09-27 12:40:29.000000000 +0200
+++ VDR-NG/epg.h	2008-09-27 12:43:17.000000000 +0200
@@ -66,6 +66,11 @@
   int duration;            // Duration of this event in seconds
   time_t vps;              // Video Programming Service timestamp (VPS, aka "Programme Identification Label", PIL)
   time_t seen;             // When this event was last seen in the data stream
+//M7X0 BEGIN AK
+  mutable int cachedDateStringLang;
+  mutable cString cachedTimeString;
+  mutable cString cachedDateString;
+//M7X0 END AK
 public:
   cEvent(tEventID EventID);
   ~cEvent();
diff -Naur VDR-NG-orig/tools.c VDR-NG/tools.c
--- VDR-NG-orig/tools.c	2008-09-27 12:40:29.000000000 +0200
+++ VDR-NG/tools.c	2008-09-27 12:43:21.000000000 +0200
@@ -672,25 +672,21 @@
      }
   return "???";
 }
-
+//M7X0 BEGIN AK
 cString DateString(time_t t)
 {
-  char buf[32];
   struct tm tm_r;
-  tm *tm = localtime_r(&t, &tm_r);
-  char *p = stpcpy(buf, WeekDayName(tm->tm_wday));
-  *p++ = ' ';
-  strftime(p, sizeof(buf) - (p - buf), "%d.%m.%Y", tm);
-  return buf;
+  localtime_r(&t, &tm_r);
+  return cString::sprintf("%s %02d.%02d.%04d", *WeekDayName(tm_r.tm_wday),tm_r.tm_mday,tm_r.tm_mon+1,tm_r.tm_year+1900);;
 }
 
 cString TimeString(time_t t)
 {
-  char buf[25];
   struct tm tm_r;
-  strftime(buf, sizeof(buf), "%R", localtime_r(&t, &tm_r));
-  return buf;
+  localtime_r(&t, &tm_r);
+  return cString::sprintf("%02d:%02d",tm_r.tm_hour,tm_r.tm_min);
 }
+//M7X0 END AK
 // --- RgbToJpeg -------------------------------------------------------------
 //M7X0 BEGIN AK
 #ifdef WITH_LIBJPEG
