diff -Naur VDR-NG-EM-orig/vdr.c VDR-NG-EM/vdr.c
--- VDR-NG-EM-orig/vdr.c	2008-03-16 23:25:59.000000000 +0100
+++ VDR-NG-EM/vdr.c	2008-03-16 23:26:18.000000000 +0100
@@ -678,10 +678,6 @@
         }
      }
 
-  // Check for interactive mode
-
-  ShutdownHandler.CheckManualStart();
-
   // User interface:
 
   Interface = new cInterface(SVDRPport);
@@ -758,6 +754,10 @@
      alarm(WatchdogTimeout); // Initial watchdog timer start
      }
 
+  // Check for interactive mode
+
+  ShutdownHandler.CheckManualStart();
+
   // Main program loop:
 
 #define DELETE_MENU ((IsInfoMenu &= (Menu == NULL)), delete Menu, Menu = NULL)
diff -Naur VDR-NG-EM-orig/eit.c VDR-NG-EM/eit.c
--- VDR-NG-EM-orig/eit.c        2008-03-19 18:09:36.000000000 +0100
+++ VDR-NG-EM/eit.c    2008-03-19 18:07:25.000000000 +0100
@@ -15,6 +15,7 @@
 #include "epg.h"
 //M7X0 BEGIN AK
 #include "epgmode.h"
+#include "shutdown.h"
 //M7X0 END AK
 #include "i18n.h"
 #include "libsi/section.h"
@@ -457,6 +458,7 @@
      if (abs(diff - lastDiff) < 3) {
         isyslog("System Time = %s (%ld)", *TimeToString(loctim), loctim);
         isyslog("Local Time  = %s (%ld)", *TimeToString(sattim), sattim);
+        ShutdownHandler.CheckManualStart();
         if (stime(&sattim) < 0){
                                char __errorstr[256];
                                strerror_r(errno,__errorstr,256); 
