diff -Naur VDR-NG-EM-orig/dvbdevice.c VDR-NG-EM/dvbdevice.c
--- VDR-NG-EM-orig/dvbdevice.c	2008-03-16 13:06:47.000000000 +0100
+++ VDR-NG-EM/dvbdevice.c	2008-05-17 19:16:17.000000000 +0100
@@ -3617,7 +3617,7 @@
   while (payl < Length && Data[payl] == 0xFF)
         payl++;
   if (payl < Length && (Data[payl] & 0xC0) == 0x40)
-     payl++;
+     payl += 2;
 
   if (payl >= Length) {
      esyslog("ERROR: Invalid MPEG1 PES Header!");
@@ -3646,6 +3646,8 @@
   playBufferFill = 0;
   data_start = Data;
   ts_start = Data + ts_offset;
+  if ((Pid & 0xF0) == 0x80)
+     payload_offset += 4;
   Data += payload_offset;
   Length -= payload_offset;
   if (Length <= 0)
@@ -3712,6 +3714,9 @@
 
 int cDvbDevice::PlayVideo(const uchar *Data, int Length)
 {
+  if (Data[3] == 0xBE) // skip padding stream
+     return Length;
+
   if (!playTrickSpeed) {
      PESPacket2TS(Data, Length, 0xef, playTsCcounter[0]);
      int r = safe_write(fd_playDvr, playBuffer, playBufferFill);
