diff -Naur VDR-NG-orig/menu.c VDR-NG/menu.c
--- VDR-NG-orig/menu.c	2009-02-28 00:11:15.000000000 +0100
+++ VDR-NG/menu.c	2009-02-28 00:14:23.000000000 +0100
@@ -4654,6 +4654,7 @@
 
 cRecordControl *cRecordControls::GetRecordControl(const char *FileName)
 {
+ if(FileName)
   for (int i = 0; i < MAXRECORDCONTROLS; i++) {
       if (RecordControls[i] && strcmp(RecordControls[i]->FileName(), FileName) == 0)
          return RecordControls[i];
