diff -Naur VDR-NG-orig/tools.c VDR-NG/tools.c
--- VDR-NG-orig/tools.c	2009-02-21 10:21:05.000000000 +0100
+++ VDR-NG/tools.c	2009-03-22 16:49:28.000000000 +0100
@@ -559,11 +559,47 @@
      LOG_ERROR_STR(FileName);
 }
 
+extern const char *VideoDirectory;
+static time_t FirstUpdateFileTime=0;
+static time_t LastUpdateFileTime=0;
 time_t LastModifiedTime(const char *FileName)
 {
   struct stat fs;
   if (stat(FileName, &fs) == 0)
+  {
+     if(!strncmp(FileName,VideoDirectory,strlen(VideoDirectory)))
+     {
+        if(!FirstUpdateFileTime) 
+        {
+           LastUpdateFileTime=0;
+           FirstUpdateFileTime=time(NULL);
+           isyslog("VideoDirectory detected %s %ld file(%ld)",FileName,FirstUpdateFileTime,fs.st_mtime);
+        }
+        if(FirstUpdateFileTime>fs.st_mtime) 
+           return FirstUpdateFileTime; 
+     }
      return fs.st_mtime;
+  } 
+  if(!strncmp(FileName,VideoDirectory,strlen(VideoDirectory)))
+  { 
+     if(DirectoryOk(VideoDirectory,true))
+     {
+        if(!FirstUpdateFileTime) 
+        {
+           LastUpdateFileTime=0;
+           FirstUpdateFileTime=time(NULL);
+           isyslog("VideoDirectory detected %s %ld",FileName,FirstUpdateFileTime);
+        }
+        return FirstUpdateFileTime;     
+     }
+     if(!LastUpdateFileTime)
+     {
+        FirstUpdateFileTime=0;
+        LastUpdateFileTime=time(NULL);
+        isyslog("VideoDirectory lost %s %ld",FileName,LastUpdateFileTime);
+     }
+     return LastUpdateFileTime; 
+  }
   return 0;
 }
 
diff -Naur VDR-NG-orig/vdr.c VDR-NG/vdr.c
--- VDR-NG-orig/vdr.c	2009-02-21 10:23:05.000000000 +0100
+++ VDR-NG/vdr.c	2009-03-20 20:48:50.000000000 +0100
@@ -525,7 +525,6 @@
 
   if (!DirectoryOk(VideoDirectory, true)) {
      fprintf(stderr, "vdr: can't access video directory %s\n", VideoDirectory);
-     return 2;
      }
 
   // Daemon mode:
