diff -Naur VDR-NG-orig/cutter.c VDR-NG/cutter.c
--- VDR-NG-orig/cutter.c	Tue Jun 23 11:06:21 2009
+++ VDR-NG/cutter.c	Thu Jun 25 11:46:15 2009
@@ -27,7 +27,7 @@
   cUnbufferedFile *fromFile, *toFile;
   cFileName *fromFileName, *toFileName;
   cIndexFile *fromIndex, *toIndex;
-  cMarks fromMarks, toMarks;
+  cMarks fromMarks;
 protected:
   virtual void Action(void);
 public:
@@ -48,7 +48,6 @@
      toFileName = new cFileName(ToFileName, true, true);
      fromIndex = new cIndexFile(FromFileName, false);
      toIndex = new cIndexFile(ToFileName, true);
-     toMarks.Load(ToFileName); // doesn't actually load marks, just sets the file name
      Start();
      }
   else
@@ -92,9 +91,6 @@
      int FileSize = 0;
      int CurrentFileNumber = 0;
      bool SkipThisSourceFile = false;
-     int LastIFrame = 0;
-     toMarks.Add(0);
-     toMarks.Save();
      uchar buffer[MAXFRAMESIZE];
      bool LastMark = false;
      bool cutIn = true;
@@ -224,7 +220,6 @@
                     }
                  FileSize = 0;
                  }
-              LastIFrame = 0;
 
               if (!SkipThisSourceFile && cutIn) {
                  cRemux::SetBrokenLink(buffer, Length);
@@ -240,17 +235,11 @@
               break;
               }
            FileSize += Length;
-           if (!LastIFrame)
-              LastIFrame = toIndex->Last();
 
            // Check editing marks:
 
            if (Mark && Index >= Mark->position) {
               Mark = fromMarks.Next(Mark);
-              toMarks.Add(LastIFrame);
-              if (Mark)
-                 toMarks.Add(toIndex->Last() + 1);
-              toMarks.Save();
               if (Mark) {
                  Index = Mark->position;
                  Mark = fromMarks.Next(Mark);
