diff -Naur VDR-NG-orig/pat.c VDR-NG/pat.c
--- VDR-NG-orig/pat.c	2010-04-21 18:29:06.000000000 +0200
+++ VDR-NG/pat.c	2010-04-29 18:26:45.144368859 +0200
@@ -341,6 +341,7 @@
             switch (stream.getStreamType()) {
               case 1: // STREAMTYPE_11172_VIDEO
               case 2: // STREAMTYPE_13818_VIDEO
+              case 0x1B: // MPEG4
                       Vpid = stream.getPid();
                       ProcessCaDescriptors = true;
                       break;
diff -Naur VDR-NG-orig/sdt.c VDR-NG/sdt.c
--- VDR-NG-orig/sdt.c	2010-04-21 18:29:06.000000000 +0200
+++ VDR-NG/sdt.c	2010-04-27 18:07:11.697192957 +0200
@@ -55,6 +55,7 @@
                    case 0x02: // digital radio sound service
                    case 0x04: // NVOD reference service
                    case 0x05: // NVOD time-shifted service
+                   case 0x19: // digital HD television service
                         {
                         char NameBuf[1024];
                         char ShortNameBuf[1024];
