diff -Naur remoteosd-0.0.2-orig/Makefile remoteosd-0.0.2/Makefile
--- remoteosd-0.0.2-orig/Makefile	2006-06-04 17:16:54.000000000 +0200
+++ remoteosd-0.0.2/Makefile	2007-06-16 20:10:53.000000000 +0200
@@ -20,8 +20,8 @@
 
 ### The directory environment:
 
-VDRDIR = ../../..
-LIBDIR = ../../lib
+VDRDIR ?= ../../..
+LIBDIR ?= ../../lib
 TMPDIR = /tmp
 
 ### Allow user defined options to overwrite defaults:
@@ -69,7 +69,7 @@
 libvdr-$(PLUGIN).so: $(OBJS)
 	$(CXX) $(CXXFLAGS) -shared $(OBJS) -o $@
 #	@cp $@ $(LIBDIR)/$@.$(VDRVERSION)
-	@cp $@ $(LIBDIR)/$@.$(APIVERSION)
+	@cp -f $@ $(LIBDIR)/$@.$(APIVERSION)
 
 dist: clean
 	@-rm -rf $(TMPDIR)/$(ARCHIVE)
diff -Naur remoteosd-0.0.2-orig/remoteosd.c remoteosd-0.0.2/remoteosd.c
--- remoteosd-0.0.2-orig/remoteosd.c	2007-02-23 23:07:41.000000000 +0100
+++ remoteosd-0.0.2/remoteosd.c	2007-06-16 13:34:06.000000000 +0200
@@ -108,6 +108,10 @@
 {
   if (!strcasecmp(Name, "HideMainMenuEntry"))
 	  RemoteOsdSetup.hideMainMenuEntry = atoi(Value);
+  else if (!strcasecmp(Name, "ReplaceSchedule"))
+	  RemoteOsdSetup.replaceSchedule = atoi(Value);
+  else if (!strcasecmp(Name, "ReplaceTimers"))
+	  RemoteOsdSetup.replaceTimers = atoi(Value);
   else if (!strcasecmp(Name, "ServerIp"))
 	  strn0cpy(RemoteOsdSetup.serverIp, Value, sizeof(RemoteOsdSetup.serverIp));
   else if (!strcasecmp(Name, "ServerPort"))
