diff -Naur epgsearch-0.9.24.beta17-orig/i18n-generated.h epgsearch-0.9.24.beta17/i18n-generated.h
--- epgsearch-0.9.24.beta17-orig/i18n-generated.h	2008-01-17 20:16:09.000000000 +0100
+++ epgsearch-0.9.24.beta17/i18n-generated.h	2008-01-20 12:20:05.000000000 +0100
@@ -1164,7 +1164,7 @@
     "",
     "",
     "",
-    "",
+    "antes de 'después'",
     "",
     "",
     "",
diff -Naur epgsearch-0.9.24.beta17-orig/Makefile epgsearch-0.9.24.beta17/Makefile
--- epgsearch-0.9.24.beta17-orig/Makefile	2008-01-13 20:19:49.000000000 +0100
+++ epgsearch-0.9.24.beta17/Makefile	2008-01-20 12:17:36.000000000 +0100
@@ -34,9 +34,9 @@
 
 ### The directory environment:
 
-DVBDIR = ../../../../DVB
-VDRDIR = ../../..
-LIBDIR = ../../lib
+DVBDIR ?= ../../../../DVB
+VDRDIR ?= ../../..
+LIBDIR ?= ../../lib
 TMPDIR = /tmp
 
 ### Allow user defined options to overwrite defaults:
@@ -69,7 +69,7 @@
 
 ### Includes and Defines (add further entries here):
 
-INCLUDES += -I$(VDRDIR)/include -I$(DVBDIR)/include
+INCLUDES += -I$(VDRDIR)/include -I$(DVBDIR)
 #INCLUDES += -I$(VDRDIR)/include
 
 DEFINES += -D_GNU_SOURCE -DPLUGIN_NAME_I18N='"$(PLUGIN)"'
@@ -128,7 +128,7 @@
 
 # Dependencies:
 
-MAKEDEP = g++ -MM -MG
+MAKEDEP = $(CXX) -MM -MG
 DEPFILE = .dependencies
 $(DEPFILE): Makefile
 	@$(MAKEDEP) $(CXXFLAGS) $(DEFINES) $(INCLUDES) $(OBJS:%.o=%.c) $(OBJS2:%.o=%.c) $(OBJS3:%.o=%.c) $(OBJS4:%.o=%.c)> $@
diff -Naur epgsearch-0.9.24.beta17-orig/po/es_ES.po epgsearch-0.9.24.beta17/po/es_ES.po
--- epgsearch-0.9.24.beta17-orig/po/es_ES.po	2008-01-17 20:02:59.000000000 +0100
+++ epgsearch-0.9.24.beta17/po/es_ES.po	2008-01-20 12:19:43.000000000 +0100
@@ -113,7 +113,7 @@
 msgstr "después hora-def."
 
 msgid "before 'next'"
-msgstr ""
+msgstr "antes de 'después'"
 
 msgid "General"
 msgstr "General"
