diff -Naur launcher-0.0.2a-orig/i18n.c launcher-0.0.2a/i18n.c
--- launcher-0.0.2a-orig/i18n.c	2004-06-25 19:18:23.000000000 +0200
+++ launcher-0.0.2a/i18n.c	2007-08-05 18:26:42.000000000 +0200
@@ -19,7 +19,7 @@
     "",// TODO Norsk
     "Laajennokset",
     "",// TODO Polski
-    "",// TODO Español
+    "Plugins",// Español
     "",// TODO ÅëëçíéêÜ (Greek)
     "",// TODO Svenska
     "",// TODO Romaneste
@@ -38,7 +38,7 @@
     "",// TODO Norsk
     "Laajennos laajennoksille",
     "",// TODO Polski
-    "",// TODO Español
+    "Plugin para lanzar otros plugins",// Español
     "",// TODO ÅëëçíéêÜ (Greek)
     "",// TODO Svenska
     "",// TODO Romaneste
@@ -57,7 +57,7 @@
     "",// TODO Norsk
     "Laajennoksella ei ole infotekstiä!",
     "",// TODO Polski
-    "",// TODO Español
+    "¡Este plugin no tiene información de texto!",// Español
     "",// TODO ÅëëçíéêÜ (Greek)
     "",// TODO Svenska
     "",// TODO Romaneste
diff -Naur launcher-0.0.2a-orig/Makefile launcher-0.0.2a/Makefile
--- launcher-0.0.2a-orig/Makefile	2004-06-25 19:11:40.000000000 +0200
+++ launcher-0.0.2a/Makefile	2007-07-31 17:55:13.000000000 +0200
@@ -16,13 +16,13 @@
 ### The C++ compiler and options:
 
 CXX      ?= g++
-CXXFLAGS ?= -O2 -Wall -Woverloaded-virtual
+CXXFLAGS ?= -fPIC -O2 -Wall -Woverloaded-virtual
 
 ### The directory environment:
 
-DVBDIR = ../../../../DVB
-VDRDIR = ../../..
-LIBDIR = ../../lib
+DVBDIR ?= ../../..
+VDRDIR ?= ../../..
+LIBDIR ?= ../../lib
 TMPDIR = /tmp
 
 ### Allow user defined options to overwrite defaults:
@@ -31,7 +31,7 @@
 
 ### The version number of VDR (taken from VDR's "config.h"):
 
-VDRVERSION = $(shell grep 'define VDRVERSION ' $(VDRDIR)/config.h | awk '{ print $$3 }' | sed -e 's/"//g')
+APIVERSION = $(shell grep 'define APIVERSION ' $(VDRDIR)/config.h | awk '{ print $$3 }' | sed -e 's/"//g')
 
 ### The name of the distribution archive:
 
@@ -40,7 +40,7 @@
 
 ### Includes and Defines (add further entries here):
 
-INCLUDES += -I$(VDRDIR)/include -I$(DVBDIR)/include
+INCLUDES += -I$(VDRDIR)/include -I$(DVBDIR)
 
 DEFINES += -DPLUGIN_NAME_I18N='"$(PLUGIN)"'
 DEFINES += -D_GNU_SOURCE
@@ -56,7 +56,7 @@
 
 # Dependencies:
 
-MAKEDEP = g++ -MM -MG
+MAKEDEP = $(CXX) -MM -MG
 DEPFILE = .dependencies
 $(DEPFILE): Makefile
 	@$(MAKEDEP) $(DEFINES) $(INCLUDES) $(OBJS:%.o=%.c) > $@
@@ -69,7 +69,7 @@
 
 libvdr-$(PLUGIN).so: $(OBJS)
 	$(CXX) $(CXXFLAGS) -shared $(OBJS) -o $@
-	@cp $@ $(LIBDIR)/$@.$(VDRVERSION)
+	@cp -f $@ $(LIBDIR)/$@.$(APIVERSION)
 
 dist: clean
 	@-rm -rf $(TMPDIR)/$(ARCHIVE)
