diff -Naur a/extra/Configs/Config.in b/extra/Configs/Config.in
--- a/extra/Configs/Config.in	Mon Feb 01 16:41:03 2016
+++ b/extra/Configs/Config.in	Fri Feb 26 09:57:01 2016
@@ -497,7 +497,6 @@
 		   !TARGET_arm && \
 		   !TARGET_i386 && \
 		   !TARGET_metag && \
-		   !TARGET_mips && \
 		   !TARGET_powerpc && \
 		   !TARGET_sparc && \
 		   !TARGET_x86_64 && \
@@ -516,7 +515,6 @@
 		   !TARGET_arm && \
 		   !TARGET_i386 && \
 		   !TARGET_metag && \
-		   !TARGET_mips && \
 		   !TARGET_powerpc && \
 		   !TARGET_sh && \
 		   !TARGET_sparc && \
@@ -546,6 +544,7 @@
 		   !TARGET_ia64 && \
 		   !TARGET_m68k && \
 		   !TARGET_microblaze && \
+		   !TARGET_mips && \
 		   !TARGET_nios2 && \
 		   !TARGET_or1k && \
 		   ARCH_USE_MMU
