diff -Naur cinebars-0.0.5-orig/i18n.c cinebars-0.0.5/i18n.c
--- cinebars-0.0.5-orig/i18n.c	2008-01-17 15:01:13.000000000 +0100
+++ cinebars-0.0.5/i18n.c	2008-01-20 13:34:07.000000000 +0100
@@ -19,7 +19,7 @@
     "",// Norsk
     "",// suomi
     "",// Polski
-    "",// Español
+    "Superposición \"bandas de cine\"",	// Español
     "",// ÅëëçíéêÜ
     "",// Svenska
     "",// Romaneste
@@ -39,7 +39,7 @@
     "",// Norsk
     "",// suomi
     "",// Polski
-    "",// Español
+    "Bandas de cine",	// Español
     "",// ÅëëçíéêÜ
     "",// Svenska
     "",// Romaneste
@@ -59,7 +59,7 @@
     "", // Norsk
     "Piilota valinta päävalikosta", 		// suomi
     "", // Polski
-    "", // Español
+    "Ocultar entrada en menú principal",	// Español
     "", // ÅëëçíéêÜ
     "", // Svenska
     "", // Romaneste
@@ -79,7 +79,7 @@
     "",// TODO
     "",
     "",// TODO
-    "",// TODO
+    "Banda superior",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -99,7 +99,7 @@
     "",// TODO
     "",
     "",// TODO
-    "",// TODO
+    "Altura banda superior",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -119,7 +119,7 @@
     "",// TODO
     "",
     "",// TODO
-    "",// TODO
+    "Banda inferior",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -139,7 +139,7 @@
     "",// TODO
     "",
     "",// TODO
-    "",// TODO
+    "Altura banda inferior",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -159,7 +159,7 @@
     "",// TODO
     "",
     "",// TODO
-    "",// TODO
+    "Incremento de altura",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -179,7 +179,7 @@
     "",// TODO
     "",
     "",// TODO
-    "",// TODO
+    "Modo OSD",
     "",// TODO
     "",// TODO
     "",// TODO
diff -Naur cinebars-0.0.5-orig/Makefile cinebars-0.0.5/Makefile
--- cinebars-0.0.5-orig/Makefile	2008-01-17 15:12:22.000000000 +0100
+++ cinebars-0.0.5/Makefile	2008-01-20 13:25:54.000000000 +0100
@@ -22,8 +22,8 @@
 
 ### The directory environment:
 #DVBDIR = ../../../../DVB
-VDRDIR = ../../..
-LIBDIR = ../../lib
+VDRDIR ?= ../../..
+LIBDIR ?= ../../lib
 TMPDIR = /tmp
 
 ### Allow user defined options to overwrite defaults:
@@ -44,7 +44,7 @@
 PACKAGE = vdr-$(ARCHIVE)
 
 ### Includes and Defines (add further entries here):
-INCLUDES += -I$(VDRDIR)/include -I$(DVBDIR)/include
+INCLUDES += -I$(VDRDIR)/include -I$(DVBDIR)
 DEFINES += -D_GNU_SOURCE -DPLUGIN_NAME_I18N='"$(PLUGIN)"'
 
 ### The object files (add further files here):
@@ -96,7 +96,7 @@
 ### Targets:
 libvdr-$(PLUGIN).so: $(OBJS)
 	$(CXX) $(CXXFLAGS) -shared $(OBJS) -o $@
-	@cp $@ $(LIBDIR)/$@.$(APIVERSION)
+	@cp -f $@ $(LIBDIR)/$@.$(APIVERSION)
 
 dist: clean
 	@-rm -rf $(TMPDIR)/$(ARCHIVE)
diff -Naur cinebars-0.0.5-orig/po/es_ES.po cinebars-0.0.5/po/es_ES.po
--- cinebars-0.0.5-orig/po/es_ES.po	1970-01-01 01:00:00.000000000 +0100
+++ cinebars-0.0.5/po/es_ES.po	2008-01-20 13:27:51.000000000 +0100
@@ -0,0 +1,52 @@
+# VDR plugin language source file.
+# Copyright (C) 2007 Klaus Schmidinger <kls@cadsoft.de>
+# This file is distributed under the same license as the VDR package.
+# Klaus Schmidinger <kls@cadsoft.de>, 2000
+#
+msgid ""
+msgstr ""
+"Project-Id-Version: VDR 1.5.7\n"
+"Report-Msgid-Bugs-To: <egal@egal-vdr.de>\n"
+"POT-Creation-Date: 2007-08-24 18:54+0200\n"
+"PO-Revision-Date: 2007-12-07 19:31+0200\n"
+"Last-Translator: bittor from open7x0.org <bittor7x0 _at_ gmail.com>\n"
+"Language-Team: <vdr@linuxtv.org>\n"
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=ISO-8859-15\n"
+"Content-Transfer-Encoding: 8bit\n"
+
+#: cinebars.c:19
+msgid "Overlays \"cinebars\""
+msgstr "Superposición \"bandas de cine\""
+
+#: cinebars.c:20
+msgid "Cinebars"
+msgstr "Bandas de cine"
+
+#: cinebars.c:248
+msgid "Hide mainmenu entry"
+msgstr "Ocultar entrada en menú principal"
+
+#: cinebars.c:249
+msgid "Top Bar"
+msgstr "Banda superior"
+
+#: cinebars.c:250
+msgid "Top BarHeight"
+msgstr "Altura banda superior"
+
+#: cinebars.c:251
+msgid "Bottom Bar"
+msgstr "Banda inferior"
+
+#: cinebars.c:252
+msgid "Bottom BarHeight"
+msgstr "Altura banda inferior"
+
+#: cinebars.c:253
+msgid "StepHeight"
+msgstr "Incremento de altura"
+
+#: cinebars.c:255
+msgid "OSD-Mode"
+msgstr "Modo OSD"
