diff -Naur suspendoutput-0.2.5-orig/Makefile suspendoutput-0.2.5/Makefile
--- suspendoutput-0.2.5-orig/Makefile	2006-08-07 01:47:32.000000000 +0200
+++ suspendoutput-0.2.5/Makefile	2007-06-16 20:50:27.000000000 +0200
@@ -16,12 +16,12 @@
 ### The C++ compiler and options:
 
 CXX      ?= g++
-CXXFLAGS ?= -O2 -Wall -Woverloaded-virtual
+CXXFLAGS ?= -fPIC -O2 -Wall -Woverloaded-virtual
 
 ### The directory environment:
 
-VDRDIR = ../../..
-LIBDIR = ../../lib
+VDRDIR ?= ../../..
+LIBDIR ?= ../../lib
 TMPDIR = /tmp
 
 ### Allow user defined options to overwrite defaults:
@@ -64,7 +64,7 @@
 
 # Dependencies:
 
-MAKEDEP = g++ -MM -MG
+MAKEDEP = $(CXX) -MM -MG
 DEPFILE = .dependencies
 $(DEPFILE): Makefile
 	@$(MAKEDEP) $(DEFINES) $(INCLUDES) $(OBJS:%.o=%.c) > $@
@@ -76,7 +76,7 @@
 all: libvdr-$(PLUGIN).so
 
 mpg2c: mpg2c.c
-	$(CC) mpg2c.c -o $@
+	gcc mpg2c.c -o $@
 
 vdrlogo_720x576.c: mpg2c vdrlogo_720x576.mpg
 	@./mpg2c vdrlogo vdrlogo_720x576.mpg vdrlogo_720x576.c
@@ -85,7 +85,7 @@
 
 libvdr-$(PLUGIN).so: $(OBJS) vdrlogo_720x576.o black_720x576.o
 	$(CXX) $(CXXFLAGS) -shared $(OBJS) vdrlogo_720x576.o black_720x576.o -o $@
-	@cp $@ $(LIBDIR)/$@.$(APIVERSION)
+	@cp -f $@ $(LIBDIR)/$@.$(APIVERSION)
 
 dist: clean
 	@-rm -rf $(TMPDIR)/$(ARCHIVE)
@@ -97,4 +97,4 @@
 
 clean:
 	@-rm -f $(OBJS) $(DEPFILE) *.so *.tgz core* *~ mpg2c black_720x576.c nosignal_720x576.c vdrlogo_720x576.c vdrlogo_720x576.o
- 
\ No hay ningún carácter de nueva línea al final del fichero
+ 
