--- slin_m740/include/linux/ip.h.orig	2007-12-23 10:09:42.000000000 +0100
+++ slin_m740/include/linux/ip.h	2007-12-23 10:12:09.000000000 +0100
@@ -108,7 +108,7 @@
   unsigned char __pad1;
   unsigned char __pad2;
   unsigned char __data[0];
-};
+} __attribute__((packed, aligned(2)));
 
 #define optlength(opt) (sizeof(struct ip_options) + opt->optlen)
 #endif
@@ -133,6 +133,6 @@
 	__u32	saddr;
 	__u32	daddr;
 	/*The options start here. */
-};
+} __attribute__((packed, aligned(2)));
 
 #endif	/* _LINUX_IP_H */
--- slin_m740/include/linux/tcp.h.orig	2007-12-23 10:16:59.000000000 +0100
+++ slin_m740/include/linux/tcp.h	2007-12-23 10:22:12.000000000 +0100
@@ -53,7 +53,7 @@
 	__u16	window;
 	__u16	check;
 	__u16	urg_ptr;
-};
+} __attribute__((packed, aligned(2)));
 
 
 enum {
@@ -97,7 +97,7 @@
 union tcp_word_hdr { 
 	struct tcphdr hdr;
 	__u32 		  words[5];
-}; 
+} __attribute__((packed, aligned(2))); 
 
 #define tcp_flag_word(tp) ( ((union tcp_word_hdr *)(tp))->words [3]) 
 
--- slin_m740/include/net/sock.h.orig	2007-12-23 16:22:01.000000000 +0100
+++ slin_m740/include/net/sock.h	2007-12-23 16:22:34.000000000 +0100
@@ -251,7 +251,7 @@
 struct tcp_sack_block {
 	__u32	start_seq;
 	__u32	end_seq;
-};
+} __attribute__((packed, aligned(2)));
 
 struct tcp_opt {
 	int	tcp_header_len;	/* Bytes of tcp header to send		*/
--- slin_m740/include/linux/icmp.h.orig	2007-12-26 12:59:20.000000000 +0100
+++ slin_m740/include/linux/icmp.h	2007-12-26 13:00:00.000000000 +0100
@@ -78,7 +78,7 @@
 		__u16	mtu;
 	} frag;
   } un;
-};
+} __attribute__((packed, aligned(2)));
 
 /*
  *	constants for (set|get)sockopt
--- a/include/linux/igmp.h	Sat May 29 14:46:20 2004
+++ b/include/linux/igmp.h	Tue Mar 11 23:09:56 2014
@@ -30,7 +30,7 @@
 	__u8 code;		/* For newer IGMP */
 	__u16 csum;
 	__u32 group;
-};
+} __attribute__((packed, aligned(2)));
 
 #define IGMP_HOST_MEMBERSHIP_QUERY	0x11	/* From RFC1112 */
 #define IGMP_HOST_MEMBERSHIP_REPORT	0x12	/* Ditto */
--- a/include/linux/in.h	Mon Nov 08 11:55:33 2004
+++ b/include/linux/in.h	Tue Mar 11 23:10:27 2014
@@ -50,7 +50,7 @@
 /* Internet address. */
 struct in_addr {
 	__u32	s_addr;
-};
+} __attribute__((packed, aligned(2)));
 
 #define IP_TOS		1
 #define IP_TTL		2
--- a/include/linux/udp.h	Sat May 29 14:46:21 2004
+++ b/include/linux/udp.h	Tue Mar 11 23:09:08 2014
@@ -23,7 +23,7 @@
 	__u16	dest;
 	__u16	len;
 	__u16	check;
-};
+} __attribute__((packed, aligned(2)));
 
 
 #endif	/* _LINUX_UDP_H */
