diff -Naur klone-core-2.1.1rc9/klone/http.h klone-core-2.1.1rc9-atinar/klone/http.h
--- klone-core-2.1.1rc9/klone/http.h	2007-11-09 23:06:26.000000000 +0100
+++ klone-core-2.1.1rc9-atinar/klone/http.h	2008-12-04 01:30:43.000000000 +0100
@@ -31,6 +31,8 @@
     /**< request accepted but processing not completed */
     HTTP_STATUS_NO_CONTENT                = 204,
     /**< no body returned */
+    HTTP_PARTIAL_CONTENT                  = 206,
+    /**< partial content returned */
     HTTP_STATUS_MOVED_PERMANENTLY         = 301,
     /**< resource relocated permanently */
     HTTP_STATUS_MOVED_TEMPORARILY         = 302,
diff -Naur klone-core-2.1.1rc9/src/libhttp/http.c klone-core-2.1.1rc9-atinar/src/libhttp/http.c
--- klone-core-2.1.1rc9/src/libhttp/http.c	2008-06-04 19:48:02.000000000 +0200
+++ klone-core-2.1.1rc9-atinar/src/libhttp/http.c	2008-12-04 01:30:41.000000000 +0100
@@ -50,6 +50,7 @@
     { HTTP_STATUS_CREATED               , "Created"                 },
     { HTTP_STATUS_ACCEPTED              , "Accepted"                },
     { HTTP_STATUS_NO_CONTENT            , "No Content"              },
+    { HTTP_PARTIAL_CONTENT              , "Partial Content"         },
     { HTTP_STATUS_BAD_REQUEST           , "Bad Request"             },
     { HTTP_STATUS_UNAUTHORIZED          , "Unauthorized"            },
     { HTTP_STATUS_FORBIDDEN             , "Forbidden"               },
