diff -Naur uClibc++-0.2.2-orig/Rules.mak uClibc++-0.2.2/Rules.mak
--- uClibc++-0.2.2-orig/Rules.mak	Mon Jun 04 00:51:12 2007
+++ uClibc++-0.2.2/Rules.mak	Thu Jul 21 17:22:23 2011
@@ -92,7 +92,7 @@
 PICFLAG:=-fPIC
 
 # use '-Os' optimization if available, else use -O2, allow Config to override
-OPTIMIZATION+=$(call check_gcc,-Os,-O2)
+# OPTIMIZATION+=$(call check_gcc,-Os,-O2)
 
 # Add a bunch of extra pedantic annoyingly strict checks
 XWARNINGS=$(subst ",, $(strip $(WARNINGS))) -Wno-trigraphs -pedantic
