--- samba-2.2.12-orig/source/include/debug.h	2011-03-04 17:03:40.000000000 +0100
+++ samba-2.2.12/source/include/debug.h	2011-03-04 17:07:23.569922464 +0100
@@ -143,52 +143,38 @@
   (0)
 
 #define ZERODEBUGLVL( level ) \
-  ( ((DEBUGLEVEL_CLASS[ DBGC_CLASS ] >= (level))||  \
-     (DEBUGLEVEL_CLASS[ DBGC_ALL   ] >= (level))  ) \
-   && dbghdr( level, FILE_MACRO, FUNCTION_MACRO, (__LINE__) ) )
+  (0)
 
 
 #define DEBUGLVLC( dbgc_class, level ) \
   (0)
 
 #define ZERODEBUGLVLC( dbgc_class, level ) \
-  ( ((DEBUGLEVEL_CLASS[ dbgc_class ] >= (level))||  \
-     (DEBUGLEVEL_CLASS[ DBGC_ALL   ] >= (level))  ) \
-   && dbghdr( level, FILE_MACRO, FUNCTION_MACRO, (__LINE__) ) )
+  (0)
 
 
 #define DEBUG( level, body ) \
   (void)(0)
 
 #define ZERODEBUG( level, body ) \
-  (void)( ((DEBUGLEVEL_CLASS[ DBGC_CLASS ] >= (level))||  \
-           (DEBUGLEVEL_CLASS[ DBGC_ALL   ] >= (level))  ) \
-       && (dbghdr( level, FILE_MACRO, FUNCTION_MACRO, (__LINE__) )) \
-       && (dbgtext body) )
+  (void)(0)
 
 #define DEBUGC( dbgc_class, level, body ) \
   (void)(0)
 
 #define ZERODEBUGC( dbgc_class, level, body ) \
-  (void)( ((DEBUGLEVEL_CLASS[ dbgc_class ] >= (level))||  \
-           (DEBUGLEVEL_CLASS[ DBGC_ALL   ] >= (level))  ) \
-       && (dbghdr( level, FILE_MACRO, FUNCTION_MACRO, (__LINE__) )) \
-       && (dbgtext body) )
+  (void)(0)
 
 #define DEBUGADD( level, body ) \
   (void)(0)
 
 #define ZERODEBUGADD( level, body ) \
-  (void)( ((DEBUGLEVEL_CLASS[ DBGC_CLASS ] >= (level))||  \
-           (DEBUGLEVEL_CLASS[ DBGC_ALL   ] >= (level))  ) \
-       && (dbgtext body) )
+  (void)(0)
 
 #define DEBUGADDC( dbgc_class, level, body ) \
   (void)(0)
 
 #define ZERODEBUGADDC( dbgc_class, level, body ) \
-  (void)( ((DEBUGLEVEL_CLASS[ dbgc_class ] >= (level))||  \
-           (DEBUGLEVEL_CLASS[ DBGC_ALL   ] >= (level))  ) \
-       && (dbgtext body) )
+  (void)(0)
 
 #endif
