diff -Naur uclibc_build_env_orig/configs/siemens-linux-kernel/m740.config uclibc_build_env/configs/siemens-linux-kernel/m740.config
--- uclibc_build_env_orig/configs/siemens-linux-kernel/m740.config	Fri Jan 25 12:37:18 2008
+++ uclibc_build_env/configs/siemens-linux-kernel/m740.config	Wed Mar 19 12:12:45 2008
@@ -646,9 +646,9 @@
 #
 # CONFIG_CODA_FS is not set
 # CONFIG_INTERMEZZO_FS is not set
-CONFIG_NFS_FS=y
-CONFIG_NFS_V3=y
-CONFIG_NFS_DIRECTIO=y
+# CONFIG_NFS_FS is not set
+# CONFIG_NFS_V3 is not set
+# CONFIG_NFS_DIRECTIO is not set
 # CONFIG_ROOT_NFS is not set
 # CONFIG_NFSD is not set
 # CONFIG_NFSD_V3 is not set
@@ -656,7 +656,7 @@
 CONFIG_SUNRPC=y
 CONFIG_LOCKD=y
 CONFIG_LOCKD_V4=y
-CONFIG_CIFS=m
+# CONFIG_CIFS is not set
 # CONFIG_CIFS_STATS is not set
 # CONFIG_CIFS_POSIX is not set
 # CONFIG_SMB_FS is not set
diff -Naur uclibc_build_env_orig/configs/siemens-linux-kernel/m750s.config uclibc_build_env/configs/siemens-linux-kernel/m750s.config
--- uclibc_build_env_orig/configs/siemens-linux-kernel/m750s.config	Fri Jan 25 12:37:18 2008
+++ uclibc_build_env/configs/siemens-linux-kernel/m750s.config	Wed Mar 19 12:13:41 2008
@@ -646,9 +646,9 @@
 #
 # CONFIG_CODA_FS is not set
 # CONFIG_INTERMEZZO_FS is not set
-CONFIG_NFS_FS=y
-CONFIG_NFS_V3=y
-CONFIG_NFS_DIRECTIO=y
+# CONFIG_NFS_FS is not set
+# CONFIG_NFS_V3 is not set
+# CONFIG_NFS_DIRECTIO is not set
 # CONFIG_ROOT_NFS is not set
 # CONFIG_NFSD is not set
 # CONFIG_NFSD_V3 is not set
@@ -656,7 +656,7 @@
 CONFIG_SUNRPC=y
 CONFIG_LOCKD=y
 CONFIG_LOCKD_V4=y
-CONFIG_CIFS=m
+# CONFIG_CIFS is not set
 # CONFIG_CIFS_STATS is not set
 # CONFIG_CIFS_POSIX is not set
 # CONFIG_SMB_FS is not set
diff -Naur uclibc_build_env_orig/make-incs/siemens-linux-kernel.mk uclibc_build_env/make-incs/siemens-linux-kernel.mk
--- uclibc_build_env_orig/make-incs/siemens-linux-kernel.mk	Thu Jan 08 10:52:31 2009
+++ uclibc_build_env/make-incs/siemens-linux-kernel.mk	Mon Mar 09 18:27:41 2009
@@ -57,7 +57,6 @@
 DISTCLEAN_RULES += distclean-siemens-linux-kernel
 
 SIEMENS-LINUX-KERNEL_MODLST := \
-   lib/modules/2.4.21-xfs/kernel/fs/cifs/cifs.o \
    lib/modules/2.4.21-xfs/kernel/drivers/usb/host/ehci-hcd.o \
    lib/modules/2.4.21-xfs/kernel/drivers/usb/host/usb-ohci.o 
 
@@ -76,8 +75,6 @@
    lib/modules \
    lib/modules/2.4.21-xfs \
    lib/modules/2.4.21-xfs/kernel \
-   lib/modules/2.4.21-xfs/kernel/fs \
-   lib/modules/2.4.21-xfs/kernel/fs/cifs \
    lib/modules/2.4.21-xfs/kernel/drivers \
    lib/modules/2.4.21-xfs/kernel/drivers/usb \
    lib/modules/2.4.21-xfs/kernel/drivers/usb/host     
@@ -95,7 +92,6 @@
 $(STAGEFILES_DIR)/.siemens-linux-kernel_$(CONFIG_M7X0_TYPE)_unpacked: \
       $(wildcard $(SIEMENS-LINUX-KERNEL_PATCHES_DIR)/common/*.patch) \
       $(wildcard $(SIEMENS-LINUX-KERNEL_PATCHES_DIR)/$(CONFIG_M7X0_TYPE)/*.patch) \
-      $(SIEMENS-LINUX-KERNEL_PATCHES_DIR)/common/cifs-fs.tar.bz2 \
       $$(SIEMENS-LINUX-KERNEL_DEPS) $(SIEMENS-LINUX-KERNEL_CONFIG)
 	-$(RM) -rf $(SIEMENS-LINUX-KERNEL_DIR)
 	$(BZCAT) $(SIEMENS-GPL-SRC_DLFILE) | $(TAR) -C $(BUILD_DIR) \
@@ -111,9 +107,6 @@
 
 $(STAGEFILES_DIR)/.siemens-linux-kernel_$(CONFIG_M7X0_TYPE)_patched: \
       $(STAGEFILES_DIR)/.siemens-linux-kernel_$(CONFIG_M7X0_TYPE)_unpacked
-# CIFS for 2.4er kernel (it's much easier to handle as smbfs and faster)
-	$(BZCAT) $(SIEMENS-LINUX-KERNEL_PATCHES_DIR)/common/cifs-fs.tar.bz2 | \
-		 $(TAR) -C $(SIEMENS-LINUX-KERNEL_DIR) -f -
 # Needed patches to get kernel compiling
 	$(call patch_package, $(SIEMENS-LINUX-KERNEL_DIR), \
 		$(SIEMENS-LINUX-KERNEL_PATCHES_DIR)/common)
