diff -Naur uclibc_build_env_orig/configs/siemens-linux-kernel/m740.config uclibc_build_env/configs/siemens-linux-kernel/m740.config
--- uclibc_build_env_orig/configs/siemens-linux-kernel/m740.config	2009-12-13 18:10:34.000000000 +0100
+++ uclibc_build_env/configs/siemens-linux-kernel/m740.config	2009-12-13 18:14:40.000000000 +0100
@@ -658,17 +658,17 @@
 #
 # CONFIG_CODA_FS is not set
 # CONFIG_INTERMEZZO_FS is not set
-CONFIG_NFS_FS=m
-CONFIG_NFS_V3=y
-CONFIG_NFS_DIRECTIO=y
+# CONFIG_NFS_FS is not set
+# CONFIG_NFS_V3 is not set
+# CONFIG_NFS_DIRECTIO is not set
 # CONFIG_ROOT_NFS is not set
 # CONFIG_NFSD is not set
 # CONFIG_NFSD_V3 is not set
 # CONFIG_NFSD_TCP is not set
-CONFIG_SUNRPC=m
-CONFIG_LOCKD=m
-CONFIG_LOCKD_V4=y
-CONFIG_CIFS=m
+# CONFIG_SUNRPC is not set
+# CONFIG_LOCKD is not set
+# CONFIG_LOCKD_V4 is not set
+# CONFIG_CIFS is not set
 # CONFIG_CIFS_STATS is not set
 # CONFIG_CIFS_POSIX is not set
 # CONFIG_SMB_FS is not set
diff -Naur uclibc_build_env_orig/configs/siemens-linux-kernel/m750s.config uclibc_build_env/configs/siemens-linux-kernel/m750s.config
--- uclibc_build_env_orig/configs/siemens-linux-kernel/m750s.config	2009-12-13 18:10:34.000000000 +0100
+++ uclibc_build_env/configs/siemens-linux-kernel/m750s.config	2009-12-13 18:14:40.000000000 +0100
@@ -658,17 +658,17 @@
 #
 # CONFIG_CODA_FS is not set
 # CONFIG_INTERMEZZO_FS is not set
-CONFIG_NFS_FS=m
-CONFIG_NFS_V3=y
-CONFIG_NFS_DIRECTIO=y
+# CONFIG_NFS_FS is not set
+# CONFIG_NFS_V3 is not set
+# CONFIG_NFS_DIRECTIO is not set
 # CONFIG_ROOT_NFS is not set
 # CONFIG_NFSD is not set
 # CONFIG_NFSD_V3 is not set
 # CONFIG_NFSD_TCP is not set
-CONFIG_SUNRPC=m
-CONFIG_LOCKD=m
-CONFIG_LOCKD_V4=y
-CONFIG_CIFS=m
+# CONFIG_SUNRPC is not set
+# CONFIG_LOCKD is not set
+# CONFIG_LOCKD_V4 is not set
+# CONFIG_CIFS is not set
 # CONFIG_CIFS_STATS is not set
 # CONFIG_CIFS_POSIX is not set
 # CONFIG_SMB_FS is not set
diff -Naur uclibc_build_env_orig/make-incs/siemens-linux-kernel.mk uclibc_build_env/make-incs/siemens-linux-kernel.mk
--- uclibc_build_env_orig/make-incs/siemens-linux-kernel.mk	2009-12-13 18:10:39.000000000 +0100
+++ uclibc_build_env/make-incs/siemens-linux-kernel.mk	2009-12-13 18:16:20.000000000 +0100
@@ -60,17 +60,12 @@
 DISTCLEAN_RULES += distclean-siemens-linux-kernel
 
 SIEMENS-LINUX-KERNEL_MODLST := \
-   lib/modules/2.4.21-xfs/kernel/fs/cifs/cifs.o \
    lib/modules/2.4.21-xfs/kernel/fs/ext2/ext2.o \
    lib/modules/2.4.21-xfs/kernel/fs/ext3/ext3.o \
    lib/modules/2.4.21-xfs/kernel/fs/fat/fat.o \
    lib/modules/2.4.21-xfs/kernel/fs/jbd/jbd.o \
-   lib/modules/2.4.21-xfs/kernel/fs/lockd/lockd.o \
-   lib/modules/2.4.21-xfs/kernel/fs/nfs/nfs.o \
    lib/modules/2.4.21-xfs/kernel/fs/nls/nls_iso8859-15.o \
    lib/modules/2.4.21-xfs/kernel/fs/vfat/vfat.o \
-   lib/modules/2.4.21-xfs/kernel/net/packet/af_packet.o \
-   lib/modules/2.4.21-xfs/kernel/net/sunrpc/sunrpc.o \
    lib/modules/2.4.21-xfs/kernel/drivers/scsi/scsi_mod.o \
    lib/modules/2.4.21-xfs/kernel/drivers/scsi/sd_mod.o \
    lib/modules/2.4.21-xfs/kernel/drivers/usb/usbcore.o \
@@ -94,18 +89,12 @@
    lib/modules/2.4.21-xfs \
    lib/modules/2.4.21-xfs/kernel \
    lib/modules/2.4.21-xfs/kernel/fs \
-   lib/modules/2.4.21-xfs/kernel/fs/cifs \
    lib/modules/2.4.21-xfs/kernel/fs/ext2 \
    lib/modules/2.4.21-xfs/kernel/fs/ext3 \
    lib/modules/2.4.21-xfs/kernel/fs/fat \
    lib/modules/2.4.21-xfs/kernel/fs/jbd \
-   lib/modules/2.4.21-xfs/kernel/fs/lockd \
-   lib/modules/2.4.21-xfs/kernel/fs/nfs \
    lib/modules/2.4.21-xfs/kernel/fs/nls \
    lib/modules/2.4.21-xfs/kernel/fs/vfat \
-   lib/modules/2.4.21-xfs/kernel/net \
-   lib/modules/2.4.21-xfs/kernel/net/packet \
-   lib/modules/2.4.21-xfs/kernel/net/sunrpc \
    lib/modules/2.4.21-xfs/kernel/drivers \
    lib/modules/2.4.21-xfs/kernel/drivers/scsi \
    lib/modules/2.4.21-xfs/kernel/drivers/usb \
@@ -125,7 +114,6 @@
 $(STAGEFILES_DIR)/.siemens-linux-kernel_$(CONFIG_M7X0_TYPE)_unpacked: \
       $(wildcard $(SIEMENS-LINUX-KERNEL_PATCHES_DIR)/common/*.patch) \
       $(wildcard $(SIEMENS-LINUX-KERNEL_PATCHES_DIR)/$(CONFIG_M7X0_TYPE)/*.patch) \
-      $(SIEMENS-LINUX-KERNEL_PATCHES_DIR)/common/cifs-fs.tar.bz2 \
       $$(SIEMENS-LINUX-KERNEL_DEPS) $(SIEMENS-LINUX-KERNEL_CONFIG)
 	-$(RM) -rf $(SIEMENS-LINUX-KERNEL_DIR)
 	$(BZCAT) $(SIEMENS-GPL-SRC_DLFILE) | $(TAR) -C $(BUILD_DIR) \
@@ -141,9 +129,6 @@
 
 $(STAGEFILES_DIR)/.siemens-linux-kernel_$(CONFIG_M7X0_TYPE)_patched: \
       $(STAGEFILES_DIR)/.siemens-linux-kernel_$(CONFIG_M7X0_TYPE)_unpacked
-# CIFS for 2.4er kernel (it's much easier to handle as smbfs and faster)
-	$(BZCAT) $(SIEMENS-LINUX-KERNEL_PATCHES_DIR)/common/cifs-fs.tar.bz2 | \
-		 $(TAR) -C $(SIEMENS-LINUX-KERNEL_DIR) -f -
 # Needed patches to get kernel compiling
 	$(call patch_package, $(SIEMENS-LINUX-KERNEL_DIR), \
 		$(SIEMENS-LINUX-KERNEL_PATCHES_DIR)/common)
