diff -Naur noepgmenu-0.0.4-orig/Makefile noepgmenu-0.0.4/Makefile
--- noepgmenu-0.0.4-orig/Makefile	2007-09-01 15:47:43.000000000 +0200
+++ noepgmenu-0.0.4/Makefile	2007-12-23 11:57:56.000000000 +0100
@@ -20,9 +20,9 @@
 
 ### The directory environment:
 
-DVBDIR = ../../../../DVB
-VDRDIR = ../../..
-LIBDIR = ../../lib
+DVBDIR ?= ../../..
+VDRDIR ?= ../../..
+LIBDIR ?= ../../lib
 TMPDIR = /tmp
 
 ### Allow user defined options to overwrite defaults:
@@ -43,7 +43,7 @@
 
 ### Includes and Defines (add further entries here):
 
-INCLUDES += -I$(VDRDIR)/include -I$(DVBDIR)/include
+INCLUDES += -I$(VDRDIR)/include -I$(DVBDIR)
 
 DEFINES += -D_GNU_SOURCE -DPLUGIN_NAME_I18N='"$(PLUGIN)"'
 
@@ -100,7 +100,7 @@
 
 libvdr-$(PLUGIN).so: $(OBJS)
 	$(CXX) $(CXXFLAGS) -shared $(OBJS) -o $@
-	@cp $@ $(LIBDIR)/$@.$(APIVERSION)
+	@cp -f $@ $(LIBDIR)/$@.$(APIVERSION)
 
 dist: clean
 	@-rm -rf $(TMPDIR)/$(ARCHIVE)
diff -Naur noepgmenu-0.0.4-orig/noepgmenu.c noepgmenu-0.0.4/noepgmenu.c
--- noepgmenu-0.0.4-orig/noepgmenu.c	2007-09-01 15:55:44.000000000 +0200
+++ noepgmenu-0.0.4/noepgmenu.c	2007-12-23 11:57:56.000000000 +0100
@@ -36,7 +36,7 @@
     "",// TODO Norsk
     "",
     "",// TODO Polski
-    "",// TODO Español
+    "Menú para el parche noEPG",// Español
     "",// TODO ÅëëçíéêÜ (Greek)
     "",// TODO Svenska
     "",// TODO Romaneste
@@ -61,7 +61,7 @@
     "",// TODO Norsk
     "",
     "",// TODO Polski
-    "",// TODO Español
+    "Canales sin EPG",// Español
     "",// TODO ÅëëçíéêÜ (Greek)
     "",// TODO Svenska
     "",// TODO Romaneste
@@ -86,7 +86,7 @@
     "",// TODO
     "Piilota valinta päävalikosta",
     "",// TODO
-    "",// TODO
+    "Ocultar en el menú pricipal",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -111,7 +111,7 @@
     "",// TODO
     "",
     "",// TODO
-    "",// TODO
+    "¡demasiados canales!",
     "",// TODO
     "",// TODO
     "",// TODO
diff -Naur noepgmenu-0.0.4-orig/po/es_ES.po noepgmenu-0.0.4/po/es_ES.po
--- noepgmenu-0.0.4-orig/po/es_ES.po	2007-09-01 15:46:27.000000000 +0200
+++ noepgmenu-0.0.4/po/es_ES.po	2007-12-23 12:02:21.000000000 +0100
@@ -1,15 +1,15 @@
 # VDR plugin language source file.
 # Copyright (C) 2007 Klaus Schmidinger <kls@cadsoft.de>
 # This file is distributed under the same license as the VDR package.
-# Ruben Nunez Francisco <ruben.nunez@tang-it.com>, 2002
+# bittor from open7x0.org <bittor7x0 _at_ gmail.com>, 2007
 #
 msgid ""
 msgstr ""
 "Project-Id-Version: VDR 1.5.7\n"
 "Report-Msgid-Bugs-To: <cwieninger@gmx.de>\n"
 "POT-Creation-Date: 2007-09-01 15:46+0200\n"
-"PO-Revision-Date: 2007-09-01 15:40+0200\n"
-"Last-Translator: Ruben Nunez Francisco <ruben.nunez@tang-it.com>\n"
+"PO-Revision-Date: 2007-12-07 19:31+0200\n"
+"Last-Translator: bittor from open7x0.org <bittor7x0 _at_ gmail.com>\n"
 "Language-Team: <vdr@linuxtv.org>\n"
 "MIME-Version: 1.0\n"
 "Content-Type: text/plain; charset=ISO-8859-15\n"
@@ -17,16 +17,16 @@
 
 #: noepgmenu.c:22
 msgid "a menu for noEPG patch"
-msgstr ""
+msgstr "Menú para el parche noEPG"
 
 #: noepgmenu.c:23 noepgmenu.c:188
 msgid "noEPG channels"
-msgstr ""
+msgstr "Canales sin EPG"
 
 #: noepgmenu.c:233
 msgid "too many channels!"
-msgstr ""
+msgstr "¡demasiados canales!"
 
 #: noepgmenu.c:345
 msgid "Hide main menu entry"
-msgstr ""
+msgstr "Ocultar en el menú pricipal"
