diff -Naur skinenigmang-0.0.5-orig/doc-M7x0/README.M7x0 skinenigmang-0.0.5/doc-M7x0/README.M7x0
--- skinenigmang-0.0.5-orig/doc-M7x0/README.M7x0	1970-01-01 01:00:00.000000000 +0100
+++ skinenigmang-0.0.5/doc-M7x0/README.M7x0	2007-06-15 20:54:49.000000000 +0200
@@ -0,0 +1,39 @@
+EnigmaNG Kompiliert f¸r den VDR auf der M7x0
+--------------------------------------------
+
+Die Datei "libvdr-skinenigmang.so.1.4.4" muﬂ in den Ordner .../PLUGIN/lib kopiert werden.
+Die Dateien und Ordner unter "conf" m¸ssen in den Konfigurationsordner kopiert werde. 
+Dieser befindet sich normalerweise dort wo auch das vdr Programm liegt und heiﬂt conf-Name der Box-MAC Adresse der Box.
+
+Bei mir sieht das dann so aus:
++---PLUGINS
+|   \---lib
+|           libvdr-setup.so.1.4.4
+|           libvdr-skinenigmang.so.1.4.4
+|
++---conf-M740AV2-0XA163F48FD3
+    +---themes
+    |       EnigmaNG-AppleGreen.theme
+    |       EnigmaNG-DarkBlue.theme
+    |       EnigmaNG-WineRed.theme
+    |       EnigmaNG-WomenLike.theme
+    |       EnigmaNG-default.theme
+    |       EnigmaNG-YellowSun.theme
+    |
+    \---plugins
+        +---setup
+        |   \---help
+        \---skinenigmang
+            +---flags
+            +---hqlogos
+            +---icons
+            |   +---message
+            |   +---replay
+            |   +---audio
+            |   +---volume
+            |   \---menu
+            \---logos
+(Die Dateien unterhalb von conf-M740AV2-0001E3FA8FD3/plugins/skinenigmang habe ich der ‹bersichtlichkeit weggelassen.)
+
+Viel Spaﬂ 
+    Hanspeter
\ No hay ning√∫n car√°cter de nueva l√≠nea al final del fichero
diff -Naur skinenigmang-0.0.5-orig/doc-M7x0/README.M7x0.EN skinenigmang-0.0.5/doc-M7x0/README.M7x0.EN
--- skinenigmang-0.0.5-orig/doc-M7x0/README.M7x0.EN	1970-01-01 01:00:00.000000000 +0100
+++ skinenigmang-0.0.5/doc-M7x0/README.M7x0.EN	2007-06-15 20:54:49.000000000 +0200
@@ -0,0 +1,39 @@
+EnigmaNG compiled for the vdr on the M7x0
+--------------------------------------------
+
+The file "libvdr-skinenigmang.so.1.4.4" needs to be copied into the .../PLUGIN/lib folder.
+All files and folder in the "conf" folder needs to be copied into your configuration folder. 
+This folder is normally there were you have your vdr and is named "conf-DNS name-MAC address".
+
+On my disk it looks like this:
++---PLUGINS
+|   \---lib
+|           libvdr-setup.so.1.4.4
+|           libvdr-skinenigmang.so.1.4.4
+|
++---conf-M740AV2-0XA163F48FD3
+    +---themes
+    |       EnigmaNG-AppleGreen.theme
+    |       EnigmaNG-DarkBlue.theme
+    |       EnigmaNG-WineRed.theme
+    |       EnigmaNG-WomenLike.theme
+    |       EnigmaNG-default.theme
+    |       EnigmaNG-YellowSun.theme
+    |
+    \---plugins
+        +---setup
+        |   \---help
+        \---skinenigmang
+            +---flags
+            +---hqlogos
+            +---icons
+            |   +---message
+            |   +---replay
+            |   +---audio
+            |   +---volume
+            |   \---menu
+            \---logos
+(I have removed the files below conf-M740AV2-0001E3FA8FD3/plugins/skinenigmang )
+
+Have fun
+    Hanspeter
\ No hay ning√∫n car√°cter de nueva l√≠nea al final del fichero
diff -Naur skinenigmang-0.0.5-orig/i18n.c skinenigmang-0.0.5/i18n.c
--- skinenigmang-0.0.5-orig/i18n.c	2007-06-28 11:26:00.000000000 +0200
+++ skinenigmang-0.0.5/i18n.c	2007-08-01 13:59:47.000000000 +0200
@@ -12,7 +12,7 @@
  *  - FranÁais:   Patrice Staudt <pat @vdr-portal.de>
  *  - Finnish:    Rolf Ahrenberg <rofafor @vdr-portal.de>
  *  - Polski:     ???
- *  - EspaÒol:    ???
+ *  - EspaÒol:    bittor from open7x0.org <bittor7x0 _@_ gmail.com>
  *  - Svenska:    ???
  *  - Russian:    SergArb @vdr-portal.de / neptunvasja @vdr-portal.de
  *  - Eesti:      ???
@@ -277,7 +277,7 @@
    "",                          // Norsk
    "Stereo",                    // suomi (Finnish)
    "Stereo",                    // Polski
-   "Estereo",                   // EspaÒol
+   "EstÈreo",                   // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "Stereo",                    // Svenska
    "",                          // Rom‚n„
@@ -409,7 +409,7 @@
    "",                          // Norsk
    "  Tunnista kanava",           // suomi (Finnish)
    "  Rozpoznaj kana≥ po",        // Polski
-   "  Identificar el canal por...",       // EspaÒol
+   "  Identificar el canal por",       // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "  Identifiera kanal med",     // Svenska
    "",                          // Rom‚n„
@@ -431,7 +431,7 @@
    "",                          // Norsk
    "nimest‰",                   // suomi (Finnish)
    "nazwa",                     // Polski
-   "Nombre",                    // EspaÒol
+   "nombre",                    // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "namn",                      // Svenska
    "",                          // Rom‚n„
@@ -453,7 +453,7 @@
    "",                          // Norsk
    "tiedoista",                 // suomi (Finnish)
    "parametry",                 // Polski
-   "Datos",                     // EspaÒol
+   "datos",                     // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "data",                      // Svenska
    "",                          // Rom‚n„
@@ -475,7 +475,7 @@
    "",                          // Norsk
    "V‰limuistin koko kanavalogoille",   // suomi (Finnish)
    "Rozmiar cache dla logo kana≥u",     // Polski
-   "TamaÒo de la cache de los logos",   // EspaÒol
+   "TamaÒo de cachÈ para los logos",   // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "Cachestorlek fˆr logotyper",        // Svenska
    "",                          // Rom‚n„
@@ -497,7 +497,7 @@
    "",                          // Norsk
    "Tyhjenn‰",                  // suomi (Finnish)
    "OprÛønij cache",            // Polski
-   "Limpiar la cache",          // EspaÒol
+   "Limpiar cachÈ",             // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "Tˆm cachen",                // Svenska
    "",                          // Rom‚n„
@@ -519,7 +519,7 @@
    "",                          // Norsk
    "Tyhjennet‰‰n v‰limuistia...",       // suomi (Finnish)
    "OprÛøniam cache logo kana≥u...",    // Polski
-   "Limpiando la cache...",     // EspaÒol
+   "Limpiando la cachÈ...",     // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "Tˆmmer cachen...",          // Svenska
    "",                          // Rom‚n„
@@ -541,7 +541,7 @@
    "",                          // Norsk
    "Kielet",                    // suomi (Finnish)
    "JÍzyki",                    // Polski
-   "Lenguajes",                 // EspaÒol
+   "Idiomas",                   // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "SprÂk",                     // Svenska
    "",                          // Rom‚n„
@@ -563,7 +563,7 @@
    "",                          // Norsk
    "Lis‰tiedot",                // suomi (Finnish)
    "Informacje pomocnicze",     // Polski
-   "InformaciÛn auxiliar",      // EspaÒol
+   "InformaciÛn adicional",     // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "Extrainformation",          // Svenska
    "",                          // Rom‚n„
@@ -585,7 +585,7 @@
    "",                          // Norsk
    "N‰yt‰ lis‰tiedot",          // suomi (Finnish)
    "Pokazuj informacje pomocnicze",     // Polski
-   "Mostrar informaciÛn auxiliar",      // EspaÒol
+   "Mostrar informaciÛn adicional",     // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "Visa extrainformation",     // Svenska
    "",                          // Rom‚n„
@@ -607,7 +607,7 @@
    "",                          // Norsk
    "N‰yt‰ tapahtuman aika",     // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Mostrar tiempo restante/transcurrido",// EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -629,7 +629,7 @@
    "",                          // Norsk
    "j‰ljell‰ oleva",            // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "restante",                  // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -651,7 +651,7 @@
    "",                          // Norsk
    "kulunut",                   // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "transcurrido",              // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -673,7 +673,7 @@
    "",                          // Norsk
    "N‰yt‰ infoalue p‰‰valikossa", // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Mostrar info en men˙ principal",// EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -695,7 +695,7 @@
    "",                          // Norsk
    "N‰yt‰ valintasymboli",      // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Mostrar marcador en listas",// EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -761,7 +761,7 @@
    "",                          // Norsk
    "N‰yt‰ VPS-tieto",           // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Mostrar VPS",               // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -783,7 +783,7 @@
    "",                          // Norsk
    "VAROITUS",                  // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "AVISO",                     // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -805,7 +805,7 @@
    "",                          // Norsk
    "P‰‰llekk‰inen ajastin",     // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Conflicto de programaciÛn", // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -827,7 +827,7 @@
    "",                          // Norsk
    "P‰‰llekk‰isi‰ ajastimia",   // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Conflictos de programaciÛn",// EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -849,7 +849,7 @@
    "",                          // Norsk
    "AJASTIMET",                 // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "PROGRAMACIONES",            // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -871,7 +871,7 @@
    "",                          // Norsk
    "TOISTUVAT TAPAHTUMAT",      // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "REPETICIONES",              // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -893,7 +893,7 @@
    "",                          // Norsk
    "*** Virheellinen kanava ***", // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "*** Canal no v·lido ***",   // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -915,7 +915,7 @@
    "",                          // Norsk
    "Koko",                      // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "TamaÒo",                    // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -937,7 +937,7 @@
    "",                          // Norsk
    "Mykistetty",                // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Silenciar",                 // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -959,7 +959,7 @@
    "",                          // Norsk
    "ƒ‰nenvoimakkuus",           // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Volumen",                   // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -981,7 +981,7 @@
    "",                          // Norsk
    "N‰yt‰ tapahtumien symbolit",// suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Mostrar sÌmbolos en listas",// EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -993,6 +993,7 @@
    "",                          // Dansk
    "",                          // »esky (Czech)
    },
+#ifndef VDRO7OVERSION
   {"Try 8bpp single area",      // English
    "Einen 8bpp Bereich versuchen", // Deutsch
    "",                          // Slovenski
@@ -1003,7 +1004,7 @@
    "",                          // Norsk
    "Suosi yht‰ 8bpp kuva-aluetta", // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Probar un ·rea de 8bpp",    // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1015,6 +1016,7 @@
    "",                          // Dansk
    "",                          // »esky (Czech)
    },
+#endif
   {"Show flags",                // English
    "Fahnen anzeigen",           // Deutsch
    "",                          // Slovenski
@@ -1025,7 +1027,7 @@
    "",                          // Norsk
    "N‰yt‰ liput",               // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Mostrar banderas",          // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1047,7 +1049,7 @@
    "",                          // Norsk
    "N‰yt‰ valikkosymbolit",     // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Mostrar sÌmbolos en men˙",  // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1069,7 +1071,7 @@
    "",                          // Norsk
    "ei koskaan",                // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "nunca",                     // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1091,7 +1093,7 @@
    "",                          // Norsk
    "aina",                      // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "siempre",                   // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1113,7 +1115,7 @@
    "",                          // Norsk
    "prosentti",                 // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "porcentaje",                // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1135,7 +1137,7 @@
    "",                          // Norsk
    "jos olemassa",              // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "si existe",                 // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1157,7 +1159,7 @@
    "",                          // Norsk
    "Uusintojen lukum‰‰r‰",      // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "N˙mero de repeticiones",    // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1179,7 +1181,7 @@
    "",                          // Norsk
    "K‰yt‰ lyhytt‰ kuvausta uusinnoille", // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Usar subtÌtulo para repeticiones",// EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1201,7 +1203,7 @@
    "",                          // Norsk
    "N‰yt‰ p‰‰llekk‰iset ajastimet", // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Mostrar conflictos de programaciÛn",// EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1223,7 +1225,7 @@
    "",                          // Norsk
    "Ajastimien valvonta",       // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Comprobar programaciÛn",    // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1245,7 +1247,7 @@
    "",                          // Norsk
    "EPGSearch",                 // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "EPGSearch",                 // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1267,7 +1269,7 @@
    "",                          // Norsk
    "Logot ja symbolit",         // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Logos y sÌmbolos",          // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1289,7 +1291,7 @@
    "",                          // Norsk
    "vain jos size.vdr",         // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "sÛlo usar size.vdr",        // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1311,7 +1313,7 @@
    "",                          // Norsk
    "N‰yt‰ tallenteen koko",     // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Mostrar tamaÒo de grabaciÛn",// EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1333,7 +1335,7 @@
    "",                          // Norsk
    "n‰ytteistys",               // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Algoritmo pÌxel",           // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1355,7 +1357,7 @@
    "",                          // Norsk
    "skaalaus",                  // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Algoritmo ratio",           // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1377,7 +1379,7 @@
    "",                          // Norsk
    "zoomaus",                   // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Aumentar imagen",           // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1399,7 +1401,7 @@
    "",                                // Norsk
    "N‰yt‰ kuvat lis‰tietovalikossa",  // suomi (Finnish)
    "",                                // Polski
-   "",                                // EspaÒol
+   "Mostrar im·genes grabaciÛn/suceso",// EspaÒol
    "",                                // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                                // Svenska
    "",                                // Rom‚n„
@@ -1421,7 +1423,7 @@
    "",                          // Norsk
    "  Muokkaa kuvien kokoa",    // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "  Redimensionar im·genes",  // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1443,7 +1445,7 @@
    "",                          // Norsk
    "  Kuvien leveys",           // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "  Anchura de imagen",       // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1465,7 +1467,7 @@
    "",                          // Norsk
    "  Kuvien korkeus",          // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "  Altura de imagen",        // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1487,7 +1489,7 @@
    "",                          // Norsk
    "  Kuvaformaatti",           // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "  Formato de imagen",       // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1509,7 +1511,7 @@
    "",                          // Norsk
    "Levit‰ otsikkopalkki",      // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Ancho completo para tÌtulo",// EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1532,7 +1534,7 @@
    "",                          // Norsk
    "N‰yt‰ s‰hkˆpostikuvake",    // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Mostrar icono correo",      // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1554,7 +1556,7 @@
    "",                          // Norsk
    "jos uutta postia",          // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "sÛlo si hay nuevo correo",  // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1578,7 +1580,7 @@
    "",                          // Norsk
    "Tekstien el‰vˆinti",        // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Texto animado",             // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1600,7 +1602,7 @@
    "",                          // Norsk
    "K‰yt‰ el‰vˆinti‰",          // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Activar",                   // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1622,7 +1624,7 @@
    "",                          // Norsk
    "  Vieritystapa",            // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "  Comportamiento del scroll",// EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1644,7 +1646,7 @@
    "",                          // Norsk
    "vasemmalle",                // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "a la izquierda",            // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1666,7 +1668,7 @@
    "",                          // Norsk
    "edestakaisin",              // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "izquierda y derecha",       // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1688,7 +1690,7 @@
    "",                          // Norsk
    "  Vierityksen viive (ms)",  // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "  Retardo del scroll (ms)", // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1710,7 +1712,7 @@
    "",                          // Norsk
    "  Vierityksen tauko (ms)",  // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "  Pausa del scroll (ms)",   // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1732,7 +1734,7 @@
    "",                          // Norsk
    "  Vilkutuksen tauko (ms)",  // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "  Pausa del parpadeo (ms)", // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1754,7 +1756,7 @@
    "",                          // Norsk
    "  Vierit‰ valikon otsikkoa",// suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "  TÌtulo OSD del scroll",   // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1776,7 +1778,7 @@
    "",                          // Norsk
    "  Vierit‰ infoaluetta",     // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "  Scroll para el ·rea info",// EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1798,7 +1800,7 @@
    "",                          // Norsk
    "  Vierit‰ aktiivista valintaa", // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "  Scroll para la lista activa",// EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1820,7 +1822,7 @@
    "",                          // Norsk
    "  Vierit‰ muita alueita",   // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "  Scroll para otros datos", // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1843,7 +1845,7 @@
    "",                          // Norsk
    "  N‰yt‰ symbolit valikossa", // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "  Mostrar sÌmbolos en men˙",// EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1865,7 +1867,7 @@
    "",                          // Norsk
    "  N‰yt‰ symbolit toistettaessa", // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "  Mostrar sÌmbolos en reproducciÛn",// EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1887,7 +1889,7 @@
    "",                          // Norsk
    "  N‰yt‰ symbolit viesteiss‰", // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "  Mostrar sÌmbolos en mensajes",// EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1909,7 +1911,7 @@
    "",                          // Norsk
    "  N‰yt‰ symbolit ‰‰nivalikossa", // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "  Mostrar sÌmbolos en audio",// EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1931,7 +1933,7 @@
    "",                          // Norsk
    "Kirjasintyypit",            // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Fuentes",                   // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1953,7 +1955,7 @@
    "",                          // Norsk
    "Valikon otsikko",           // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "TÌtulo OSD",                // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1975,7 +1977,7 @@
    "",                          // Norsk
    "P‰iv‰ys",                   // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Fecha",                     // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -1997,7 +1999,7 @@
    "",                          // Norsk
    "Viestit",                   // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Mensajes",                  // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -2019,7 +2021,7 @@
    "",                          // Norsk
    "V‰rin‰pp‰imet",             // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Botones de ayuda",          // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -2041,7 +2043,7 @@
    "",                          // Norsk
    "Kanavatieto: ohjelman nimi",// suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Info canal: tÌtulo",        // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -2063,7 +2065,7 @@
    "",                          // Norsk
    "Kanavatieto: ohjelman kuvaus", // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Info canal: subtÌtulo",     // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -2085,7 +2087,7 @@
    "",                          // Norsk
    "Kanavatieto: kieli",        // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Info canal: idioma",        // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -2107,7 +2109,7 @@
    "",                          // Norsk
    "Listat",                    // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Listado",                   // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -2129,7 +2131,7 @@
    "",                          // Norsk
    "Infoalue: ajastimen otsikko", // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "¡rea info: tÌtulo programaciÛn",// EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -2151,7 +2153,7 @@
    "",                          // Norsk
    "Infoalue: ajastimen leip‰teksti", // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "¡rea info: texto programaciÛn",// EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -2173,7 +2175,7 @@
    "",                          // Norsk
    "Infoalue: varoituksen otsikko",  // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "¡rea info: tÌtulo aviso",   // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -2195,7 +2197,7 @@
    "",                          // Norsk
    "Infoalue: varoituksen leip‰teksti", // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "¡rea info: texto aviso",    // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -2217,7 +2219,7 @@
    "",                          // Norsk
    "Lis‰tiedot: ohjelman nimi", // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Detalles: tÌtulo",          // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -2239,7 +2241,7 @@
    "",                          // Norsk
    "Lis‰tiedot: ohjelman kuvaus", // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Detalles: subtÌtulo",       // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -2261,7 +2263,7 @@
    "",                          // Norsk
    "Lis‰tiedot: p‰iv‰m‰‰r‰",    // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Detalles: fecha",           // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -2283,7 +2285,7 @@
    "",                          // Norsk
    "Lis‰tiedot: leip‰teksti",   // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Detalles: texto",           // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -2305,7 +2307,7 @@
    "",                          // Norsk
    "Toisto: kellonajat",        // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Reproducir: veces",         // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -2327,7 +2329,7 @@
    "",                          // Norsk
    "oletuskirjasintyyppi",      // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Fuente OSD por defecto",    // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -2349,7 +2351,7 @@
    "",                          // Norsk
    "tasav‰linen kirjasintyyppi", // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Fuente tamaÒo fijo por defecto",// EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -2371,7 +2373,7 @@
    "",                          // Norsk
    "pieni kirjasintyyppi",      // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Fuente pequeÒa por defecto",// EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -2393,7 +2395,7 @@
    "",                          // Norsk
    "Kanava:",                   // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Canal:",                    // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -2415,7 +2417,7 @@
    "",                          // Norsk
    "Hakutapa:",                 // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Buscar patrÛn:",            // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -2437,7 +2439,7 @@
    "",                          // Norsk
    "Ei valvontaa",              // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "No comprobar programaciÛn", // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -2459,7 +2461,7 @@
    "",                          // Norsk
    "Yleiset",                   // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "General",                   // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -2481,7 +2483,7 @@
    "",                          // Norsk
    "Nimi",                      // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Nombre",                    // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -2503,7 +2505,7 @@
    "",                          // Norsk
    "Leveys",                    // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Anchura",                   // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -2525,7 +2527,7 @@
    "",                          // Norsk
    "Aseta",                     // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Cambiar",                   // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -2547,7 +2549,7 @@
    "",                          // Norsk
    "TrueType-kirjasin",         // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Fuente TrueType",           // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -2569,7 +2571,7 @@
    "",                          // Norsk
    "Ei TrueType-kirjasintyyppi tukea!", // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "°Sin soporte TTF!",         // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -2591,7 +2593,7 @@
    "",                          // Norsk
    "Tasav‰linen kirjasintyyppi", // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "Fuente Fija",               // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -2614,7 +2616,7 @@
    "",                          // Norsk
    "Vaihtuva kuvaruutun‰ytˆn koko", // suomi (Finnish)
    "",                          // Polski
-   "",                          // EspaÒol
+   "TamaÒo OSD din·mico",       // EspaÒol
    "",                          // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                          // Svenska
    "",                          // Rom‚n„
@@ -2637,7 +2639,7 @@
    "",                                      // Norsk
    "TrueType-kirjasintyyppej‰ ei lˆydy!",   // suomi (Finnish)
    "",                                      // Polski
-   "",                                      // EspaÒol
+   "°Ninguna fuente TrueType instalada!",   // EspaÒol
    "",                                      // ≈ÎÎÁÌÈÍ‹ (Greek)
    "",                                      // Svenska
    "",                                      // Rom‚n„
diff -Naur skinenigmang-0.0.5-orig/Makefile skinenigmang-0.0.5/Makefile
--- skinenigmang-0.0.5-orig/Makefile	2007-06-25 12:44:11.000000000 +0200
+++ skinenigmang-0.0.5/Makefile	2007-08-01 13:58:38.000000000 +0200
@@ -5,7 +5,7 @@
 
 # If you are using the epgsearch plugin and want to see the number of
 # timer conflicts in the main menu's info area.
-#SKINENIGMA_HAVE_EPGSEARCH = 1
+SKINENIGMA_HAVE_EPGSEARCH = 1
 
 # This turns usage of logos in the main menu complete. This might also
 # improve the performance of the menus. EXPERIMENTAL!!!
@@ -21,7 +21,7 @@
 # If you use the mailbox plugin this will include support for it.
 # NOTE: this can also be defined if you don't know if the mailbox
 # plugin will be used because it has no compile time requirements.
-SKINENIGMA_USE_PLUGIN_MAILBOX = 1
+#SKINENIGMA_USE_PLUGIN_MAILBOX = 1
 
 # EXPERIMENTAL!!! NOT YET OFFICIALY SUPPORTED!!!
 #
@@ -61,8 +61,8 @@
 
 ### The directory environment:
 
-VDRDIR = ../../..
-LIBDIR = ../../lib
+VDRDIR ?= ../../..
+LIBDIR ?= ../../lib
 TMPDIR = /tmp
 
 ### Allow user defined options to overwrite defaults:
@@ -81,10 +81,14 @@
 
 ### Includes and Defines (add further entries here):
 
-INCLUDES += -I$(VDRDIR)/include
+INCLUDES += -I$(VDRDIR)/include -I$(VDRDIR)
 
 DEFINES += -D_GNU_SOURCE -DPLUGIN_NAME_I18N='"$(PLUGIN)"'
 
+# Compiled for VDR-M7x0 Siemens Gigaset from open7x0.org
+# Comment following line in other cases
+DEFINES += -DVDRO7OVERSION
+
 ifdef SKINENIGMA_HAVE_EPGSEARCH
 DEFINES += -DSKINENIGMA_HAVE_EPGSEARCH
 endif
diff -Naur skinenigmang-0.0.5-orig/setup.c skinenigmang-0.0.5/setup.c
--- skinenigmang-0.0.5-orig/setup.c	2007-06-25 10:00:04.000000000 +0200
+++ skinenigmang-0.0.5/setup.c	2007-08-01 16:40:19.000000000 +0200
@@ -211,7 +211,9 @@
   int current = Current();
   Clear();
 
+#ifndef VDRO7OVERSION
   Add(new cMenuEditBoolItem(tr("Try 8bpp single area"), &data->singleArea8Bpp));
+#endif
   Add(new cMenuEditBoolItem(tr("Show info area in main menu"), &data->showInfo));
   Add(new cMenuEditBoolItem(tr("Show auxiliary information"), &data->showAuxInfo, tr("top"), tr("bottom")));
   Add(new cMenuEditStraItem(tr("Show recording's size"), &data->showRecSize, 3, showRecSizeTexts));
