diff -Naur vsftpd-2.1.0-orig/Makefile vsftpd-2.1.0/Makefile
--- vsftpd-2.1.0-orig/Makefile	2009-02-18 23:28:05.000000000 +0100
+++ vsftpd-2.1.0/Makefile	2009-02-28 18:26:03.000000000 +0100
@@ -1,11 +1,11 @@
 # Makefile for systems with GNU tools
-CC 	=	gcc
+#CC 	=	gcc
 INSTALL	=	install
 IFLAGS  = -idirafter dummyinc
 #CFLAGS = -g
 CFLAGS	=	-O2 -Wall -W -Wshadow #-pedantic -Werror -Wconversion
 
-LIBS	=	`./vsf_findlibs.sh`
+LIBS	=	-lcrypt
 LINK	=	-Wl,-s
 
 OBJS	=	main.o utility.o prelogin.o ftpcmdio.o postlogin.o privsock.o \
