diff -Naur usbautomounter-orig/usbautomounter.c usbautomounter/usbautomounter.c
--- usbautomounter-orig/usbautomounter.c	2009-05-13 15:03:36.000000000 +0200
+++ usbautomounter/usbautomounter.c	2009-05-13 15:03:22.000000000 +0200
@@ -823,6 +823,15 @@
 		return 1;
 	}
 
+	if(!strcasecmp(argv[1],"usb"))
+	{
+		char *Interface=getenv("INTERFACE");
+		if(!Interface)
+			return 1;
+		if(strncmp(Interface,"8/6/",4))
+			return 1;
+	}
+
 	int lazy=1;
 	int force_fsck = 0;
 	int f;
