diff -Naur spider-0.1.4-orig/Makefile spider-0.1.4/Makefile
--- spider-0.1.4-orig/Makefile	2006-04-25 03:02:00.000000000 +0200
+++ spider-0.1.4/Makefile	2007-06-25 20:55:01.000000000 +0200
@@ -21,8 +21,8 @@
 
 ### The directory environment:
 
-VDRDIR = ../../..
-LIBDIR = ../../lib
+VDRDIR ?= ../../..
+LIBDIR ?= ../../lib
 TMPDIR = /tmp
 
 ### Allow user defined options to overwrite defaults:
@@ -70,7 +70,7 @@
 
 libvdr-$(PLUGIN).so: $(OBJS)
 	$(CXX) $(CXXFLAGS) -shared $(OBJS) -o $@
-	@cp $@ $(LIBDIR)/$@.$(APIVERSION)
+	@cp -f $@ $(LIBDIR)/$@.$(APIVERSION)
 
 dist: clean
 	@-rm -rf $(TMPDIR)/$(ARCHIVE)
