diff -Naur busybox-1.18.5-orig/shell/ash.c busybox-1.18.5/shell/ash.c
--- busybox-1.18.5-orig/shell/ash.c	2011-06-12 17:41:52.000000000 +0200
+++ busybox-1.18.5/shell/ash.c	2011-07-26 12:37:25.770306000 +0200
@@ -8433,7 +8433,16 @@
 #if !defined(__alpha__) || (defined(__GNUC__) && __GNUC__ >= 3)
 static
 #endif
+// Dirty hack for fix compilation with Link Time Optimization
+#ifndef LINK_TIME_OPTIMIZATION
 void evaltreenr(union node *, int) __attribute__ ((alias("evaltree"),__noreturn__));
+#else
+void evaltreenr(union node *n, int flags)
+{
+	evaltree(n, flags);
+	abort();
+}
+#endif
 
 static void
 evalloop(union node *n, int flags)
