# Patch by Matthias Schwarzott <zzam@gentoo.org>

diff -Naur vdr-m7x0-orig/config.c vdr-m7x0/config.c
--- vdr-m7x0-orig/config.c	2007-04-30 13:41:45.000000000 +0200
+++ vdr-m7x0/config.c	2007-04-30 13:46:25.000000000 +0200
@@ -284,6 +284,7 @@
   JumpSeconds = 60;
   JumpSecondsRepeat = 180;
   JumpFramesRepeat = 15;
+  AbortWhenPluginFails = 0;
 }
 
 cSetup& cSetup::operator= (const cSetup &s)
@@ -454,6 +455,7 @@
   else if (!strcasecmp(Name, "JumpSeconds"))         JumpSeconds        = atoi(Value);
   else if (!strcasecmp(Name, "JumpSecondsRepeat"))   JumpSecondsRepeat  = atoi(Value);
   else if (!strcasecmp(Name, "JumpFramesRepeat"))    JumpFramesRepeat   = atoi(Value);
+  else if (!strcasecmp(Name, "AbortWhenPluginFails")) AbortWhenPluginFails = atoi(Value);
   else
      return false;
   return true;
@@ -531,6 +533,7 @@
   Store("JumpSeconds",        JumpSeconds);
   Store("JumpSecondsRepeat",  JumpSecondsRepeat);
   Store("JumpFramesRepeat",   JumpFramesRepeat);
+  Store("AbortWhenPluginFails", AbortWhenPluginFails);
 
   Sort();
 
diff -Naur vdr-m7x0-orig/config.h vdr-m7x0/config.h
--- vdr-m7x0-orig/config.h	2007-04-30 13:41:45.000000000 +0200
+++ vdr-m7x0/config.h	2007-04-30 13:47:49.000000000 +0200
@@ -258,6 +258,7 @@
   int InitialVolume;
   int ShowRecDate, ShowRecTime, ShowRecLength, ShowProgressBar, MenuCmdPosition;
   int JumpSeconds, JumpSecondsRepeat, JumpFramesRepeat;
+  int AbortWhenPluginFails;
   int __EndData__;
   cSetup(void);
   cSetup& operator= (const cSetup &s);
diff -Naur vdr-m7x0-orig/i18n.c vdr-m7x0/i18n.c
--- vdr-m7x0-orig/i18n.c	2007-04-30 13:41:45.000000000 +0200
+++ vdr-m7x0/i18n.c	2007-04-30 13:44:19.000000000 +0200
@@ -4778,6 +4778,27 @@
     "som før",
     "jako naposledy",
   },
+  { "Setup.Miscellaneous$Abort when Plugin fails to load",
+    "Abbrechen wenn Plugin nicht lädt",
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "Detener si un Plugin falla al cargar",
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+  },
   // The days of the week:
   { "MTWTFSS",
     "MDMDFSS",
diff -Naur vdr-m7x0-orig/menu.c vdr-m7x0/menu.c
--- vdr-m7x0-orig/menu.c	2007-04-30 13:42:40.000000000 +0200
+++ vdr-m7x0/menu.c	2007-04-30 13:44:19.000000000 +0200
@@ -3061,6 +3061,7 @@
   Add(new cMenuEditIntItem( tr("Setup.Miscellaneous$Zap timeout (s)"),            &data.ZapTimeout));
   Add(new cMenuEditChanItem(tr("Setup.Miscellaneous$Initial channel"),            &data.InitialChannel, tr("Setup.Miscellaneous$as before")));
   Add(new cMenuEditIntItem( tr("Setup.Miscellaneous$Initial volume"),             &data.InitialVolume, -1, 255, tr("Setup.Miscellaneous$as before")));
+  Add(new cMenuEditBoolItem( tr("Setup.Miscellaneous$Abort when Plugin fails to load"),    &data.AbortWhenPluginFails));
 }
 
 // --- cMenuSetupPluginItem --------------------------------------------------
diff -Naur vdr-m7x0-orig/plugin.c vdr-m7x0/plugin.c
--- vdr-m7x0-orig/plugin.c	2007-04-30 13:41:45.000000000 +0200
+++ vdr-m7x0/plugin.c	2007-04-30 13:44:19.000000000 +0200
@@ -344,7 +344,8 @@
 {
   for (cDll *dll = dlls.First(); dll; dll = dlls.Next(dll)) {
       if (!dll->Load(Log))
-         return false;
+         if (Setup.AbortWhenPluginFails)
+            return false;
       }
   return true;
 }
