diff -Naur VDR-NG-orig/config.c VDR-NG/config.c
--- VDR-NG-orig/config.c	2008-04-17 11:03:02.000000000 +0200
+++ VDR-NG/config.c	2008-04-17 11:10:27.000000000 +0200
@@ -341,6 +341,9 @@
   JumpFramesRepeat = 15;
   AbortWhenPluginFails = 0;
   ExitOnFailure = 1;
+  LircRepeatDelay = 350;
+  LircRepeatFreq = 100;
+  LircRepeatTimeout = 500;
 }
 
 cSetup& cSetup::operator= (const cSetup &s)
@@ -532,6 +535,9 @@
   else if (!strcasecmp(Name, "JumpFramesRepeat"))    JumpFramesRepeat   = atoi(Value);
   else if (!strcasecmp(Name, "AbortWhenPluginFails")) AbortWhenPluginFails = atoi(Value);
   else if (!strcasecmp(Name, "ExitOnFailure"))       ExitOnFailure      = atoi(Value);
+  else if (!strcasecmp(Name, "LircRepeatDelay"))     LircRepeatDelay    = atoi(Value);
+  else if (!strcasecmp(Name, "LircRepeatFreq"))      LircRepeatFreq     = atoi(Value);
+  else if (!strcasecmp(Name, "LircRepeatTimeout"))   LircRepeatTimeout  = atoi(Value);
   else
      return false;
   return true;
@@ -630,6 +636,9 @@
   Store("JumpFramesRepeat",   JumpFramesRepeat);
   Store("AbortWhenPluginFails", AbortWhenPluginFails);
   Store("ExitOnFailure",      ExitOnFailure);
+  Store("LircRepeatDelay",    LircRepeatDelay);
+  Store("LircRepeatFreq",     LircRepeatFreq);
+  Store("LircRepeatTimeout",  LircRepeatTimeout);
 
   Sort();
 
diff -Naur VDR-NG-orig/config.h VDR-NG/config.h
--- VDR-NG-orig/config.h	2008-04-17 10:44:33.000000000 +0200
+++ VDR-NG/config.h	2008-04-17 11:13:04.000000000 +0200
@@ -298,6 +298,9 @@
   int JumpSeconds, JumpSecondsRepeat, JumpFramesRepeat;
   int AbortWhenPluginFails;
   int ExitOnFailure;
+  int LircRepeatDelay;
+  int LircRepeatFreq;
+  int LircRepeatTimeout;
   int __EndData__;
   cSetup(void);
   cSetup& operator= (const cSetup &s);
diff -Naur VDR-NG-orig/i18n.c VDR-NG/i18n.c
--- VDR-NG-orig/i18n.c	2008-04-17 10:53:59.000000000 +0200
+++ VDR-NG/i18n.c	2008-04-17 11:09:14.000000000 +0200
@@ -5266,6 +5266,69 @@
     "",// TODO
     "",// TODO
   },
+  { "Setup.Miscellaneous$Lirc repeat delay",
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "Lirc repetir retraso",
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+  },
+  { "Setup.Miscellaneous$Lirc repeat freq",
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "Lirc repetir frecuencia",
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+  },
+  { "Setup.Miscellaneous$Lirc repeat timeout",
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "Lirc repetir tiempo de espera",
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+  },
   { "Setup.Miscellaneous$Filter channels",
     "",// TODO
     "",// TODO
diff -Naur VDR-NG-orig/lirc.c VDR-NG/lirc.c
--- VDR-NG-orig/lirc.c	2008-05-09 21:08:54.000000000 +0200
+++ VDR-NG/lirc.c	2008-04-17 11:09:14.000000000 +0200
@@ -12,6 +12,7 @@
 #include "lirc.h"
 #include <netinet/in.h>
 #include <sys/socket.h>
+#include "config.h"
 
 #define REPEATDELAY 350 // ms
 #define REPEATFREQ 100 // ms
@@ -127,7 +128,7 @@
               continue;
               }
            if (count == 0) {
-              if (strcmp(KeyName, LastKeyName) == 0 && FirstTime.Elapsed() < REPEATDELAY)
+              if (strcmp(KeyName, LastKeyName) == 0 && FirstTime.Elapsed() < (unsigned int)Setup.LircRepeatDelay)
                  continue; // skip keys coming in too fast
               if (repeat)
                  Put(LastKeyName, false, true);
@@ -139,18 +140,18 @@
 //M7X0 END AK
               }
            else {
-              if (LastTime.Elapsed() < REPEATFREQ)
+              if (LastTime.Elapsed() < (unsigned int)Setup.LircRepeatFreq)
                  continue; // repeat function kicks in after a short delay (after last key instead of first key)
-              if (FirstTime.Elapsed() < REPEATDELAY)
+              if (FirstTime.Elapsed() < (unsigned int)Setup.LircRepeatDelay)
                  continue; // skip keys coming in too fast (for count != 0 as well)
               repeat = true;
-              timeout = REPEATDELAY;
+              timeout = Setup.LircRepeatDelay;
               }
            LastTime.Set();
            Put(KeyName, repeat);
            }
         else if (repeat) { // the last one was a repeat, so let's generate a release
-           if (LastTime.Elapsed() >= REPEATTIMEOUT) {
+           if (LastTime.Elapsed() >= (unsigned int)Setup.LircRepeatTimeout) {
               Put(LastKeyName, false, true);
               repeat = false;
               *LastKeyName = 0;
diff -Naur VDR-NG-orig/menu.c VDR-NG/menu.c
--- VDR-NG-orig/menu.c	2008-04-17 11:03:09.000000000 +0200
+++ VDR-NG/menu.c	2008-04-17 11:11:36.000000000 +0200
@@ -3282,6 +3282,9 @@
   Add(new cMenuEditIntItem( tr("Setup.Miscellaneous$Initial volume"),             &data.InitialVolume, -1, 255, tr("Setup.Miscellaneous$as before")));
   Add(new cMenuEditBoolItem(tr("Setup.Miscellaneous$Abort when Plugin fails to load"), &data.AbortWhenPluginFails));
   Add(new cMenuEditBoolItem(tr("Setup.Miscellaneous$Exit on video stream failure"), &data.ExitOnFailure));
+  Add(new cMenuEditIntItem( tr("Setup.Miscellaneous$Lirc repeat delay"),          &data.LircRepeatDelay, 0, 1000));
+  Add(new cMenuEditIntItem( tr("Setup.Miscellaneous$Lirc repeat freq"),           &data.LircRepeatFreq, 0, 1000));
+  Add(new cMenuEditIntItem( tr("Setup.Miscellaneous$Lirc repeat timeout"),        &data.LircRepeatTimeout, 0, 5000));
 }
 
 // --- cMenuSetupPluginItem --------------------------------------------------
