diff -Naur skinenigmang-0.0.6-orig/Makefile skinenigmang-0.0.6/Makefile
--- skinenigmang-0.0.6-orig/Makefile	2008-03-09 09:31:30.000000000 +0100
+++ skinenigmang-0.0.6/Makefile	2008-03-09 19:20:34.000000000 +0100
@@ -5,7 +5,7 @@
 
 # This turns usage of logos in the main menu complete. This might also
 # improve the performance of the menus. EXPERIMENTAL!!!
-#SKINENIGMA_NO_MENULOGO = 1
+SKINENIGMA_NO_MENULOGO = 1
 
 # Debugging on/off 
 #SKINENIGMA_DEBUG = 1
@@ -16,15 +16,15 @@
 
 # If you are using the epgsearch plugin and want to see the number of
 # timer conflicts in the main menu's info area.
-SKINENIGMA_USE_PLUGIN_EPGSEARCH = 1
+#SKINENIGMA_USE_PLUGIN_EPGSEARCH = 1
 
 # If you use the mailbox plugin this will include support for it.
-SKINENIGMA_USE_PLUGIN_MAILBOX = 1
+#SKINENIGMA_USE_PLUGIN_MAILBOX = 1
 
 # Include code to support the Avards plugin:
 #  - Dynamic OSD size depending on current WSS mode.
 #  - Display current WSS mode in channel info and replay OSDs.
-SKINENIGMA_USE_PLUGIN_AVARDS = 1
+#SKINENIGMA_USE_PLUGIN_AVARDS = 1
 
 # Disable any code that is used for scrolling or blinking text.
 # NOTE: this is only useful if you want to save some bytes because you
@@ -60,8 +60,8 @@
 
 ### The directory environment:
 
-VDRDIR = ../../..
-LIBDIR = ../../lib
+VDRDIR ?= ../../..
+LIBDIR ?= ../../lib
 TMPDIR = /tmp
 
 ### Allow user defined options to overwrite defaults:
@@ -84,10 +84,14 @@
 
 ### Includes and Defines (add further entries here):
 
-INCLUDES += -I$(VDRDIR)/include
+INCLUDES += -I$(VDRDIR)/include -I$(VDRDIR)
 
 DEFINES += -D_GNU_SOURCE -DPLUGIN_NAME_I18N='"$(PLUGIN)"'
 
+# Compiled for VDR-M7x0 Siemens Gigaset from open7x0.org
+# Comment following line in other cases
+DEFINES += -DVDRO7OVERSION
+
 ifdef SKINENIGMA_USE_PLUGIN_EPGSEARCH 
 DEFINES += -DUSE_PLUGIN_EPGSEARCH
 else
diff -Naur skinenigmang-0.0.6-orig/setup.c skinenigmang-0.0.6/setup.c
--- skinenigmang-0.0.6-orig/setup.c	2008-01-07 10:15:43.000000000 +0100
+++ skinenigmang-0.0.6/setup.c	2008-03-09 19:21:31.000000000 +0100
@@ -232,7 +232,9 @@
   int current = Current();
   Clear();
 
+#ifndef VDRO7OVERSION
   Add(new cMenuEditBoolItem(tr("Try 8bpp single area"), &data->singleArea8Bpp));
+#endif
   Add(new cMenuEditBoolItem(tr("Show info area in main menu"), &data->showInfo));
   Add(new cMenuEditBoolItem(tr("Show auxiliary information"), &data->showAuxInfo, trVDR("top"), trVDR("bottom")));
   Add(new cMenuEditStraItem(tr("Show recording's size"), &data->showRecSize, 3, showRecSizeTexts));
