diff -Naur noepgmenu-0.0.3a-orig/Makefile noepgmenu-0.0.3a/Makefile
--- noepgmenu-0.0.3a-orig/Makefile	2006-04-25 18:34:09.000000000 +0200
+++ noepgmenu-0.0.3a/Makefile	2007-07-29 18:07:53.000000000 +0200
@@ -20,9 +20,9 @@
 
 ### The directory environment:
 
-DVBDIR = ../../../../DVB
-VDRDIR = ../../..
-LIBDIR = ../../lib
+DVBDIR ?= ../../..
+VDRDIR ?= ../../..
+LIBDIR ?= ../../lib
 TMPDIR = /tmp
 
 ### Allow user defined options to overwrite defaults:
@@ -43,7 +43,7 @@
 
 ### Includes and Defines (add further entries here):
 
-INCLUDES += -I$(VDRDIR)/include -I$(DVBDIR)/include
+INCLUDES += -I$(VDRDIR)/include -I$(DVBDIR)
 
 DEFINES += -D_GNU_SOURCE -DPLUGIN_NAME_I18N='"$(PLUGIN)"'
 
@@ -71,7 +71,7 @@
 
 libvdr-$(PLUGIN).so: $(OBJS)
 	$(CXX) $(CXXFLAGS) -shared $(OBJS) -o $@
-	@cp $@ $(LIBDIR)/$@.$(APIVERSION)
+	@cp -f $@ $(LIBDIR)/$@.$(APIVERSION)
 
 dist: clean
 	@-rm -rf $(TMPDIR)/$(ARCHIVE)
diff -Naur noepgmenu-0.0.3a-orig/noepgmenu.c noepgmenu-0.0.3a/noepgmenu.c
--- noepgmenu-0.0.3a-orig/noepgmenu.c	2006-04-27 10:12:46.000000000 +0200
+++ noepgmenu-0.0.3a/noepgmenu.c	2007-06-23 11:37:47.000000000 +0200
@@ -28,7 +28,7 @@
     "",// TODO Norsk
     "",
     "",// TODO Polski
-    "",// TODO Español
+    "Menú para el parche noEPG",// Español
     "",// TODO ÅëëçíéêÜ (Greek)
     "",// TODO Svenska
     "",// TODO Romaneste
@@ -53,7 +53,7 @@
     "",// TODO Norsk
     "",
     "",// TODO Polski
-    "",// TODO Español
+    "Canales sin EPG",// Español
     "",// TODO ÅëëçíéêÜ (Greek)
     "",// TODO Svenska
     "",// TODO Romaneste
@@ -78,7 +78,7 @@
     "",// TODO
     "Piilota valinta päävalikosta",
     "",// TODO
-    "",// TODO
+    "Ocultar en el menú pricipal",
     "",// TODO
     "",// TODO
     "",// TODO
@@ -103,7 +103,7 @@
     "",// TODO
     "",
     "",// TODO
-    "",// TODO
+    "¡demasiados canales!",
     "",// TODO
     "",// TODO
     "",// TODO
