From 445f926088b7bd09b3da4e6f1bf8ef7c2a0239fb Mon Sep 17 00:00:00 2001
From: Mike Frysinger <vapier@gentoo.org>
Date: Tue, 11 Jan 2011 14:09:02 -0500
Subject: [PATCH] README: fix typo in tcp wrapper doc

Signed-off-by: Mike Frysinger <vapier@gentoo.org>
Signed-off-by: Neil Brown <neilb@suse.de>
---
 README | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/README b/README
index 0da54cc..916de7e 100644
--- a/README
+++ b/README
@@ -18,7 +18,7 @@ There is no "./configure", just use "make".
 
 Some make variable can be used to control compilation.
 
- NO_TCP_WRAPPER=  if non-empty, doen't use tcp_wrappers
+ NO_TCP_WRAPPER=  if non-empty, don't use tcp_wrappers
  USE_DNS=	  if set, tcp_wrappers can check peers based on hostname
 		  as well as IP address.  This should only be used if you
 		  are certain that gethostbyname will never trigger a
