diff -Naur sudoku-0.1.3-orig/Makefile sudoku-0.1.3/Makefile
--- sudoku-0.1.3-orig/Makefile	2006-04-25 03:02:00.000000000 +0200
+++ sudoku-0.1.3/Makefile	2007-06-16 20:14:24.000000000 +0200
@@ -23,8 +23,8 @@
 
 ### The directory environment:
 
-VDRDIR = ../../..
-LIBDIR = ../../lib
+VDRDIR ?= ../../..
+LIBDIR ?= ../../lib
 TMPDIR = /tmp
 
 ### Allow user defined options to overwrite defaults:
@@ -73,7 +73,7 @@
 
 libvdr-$(PLUGIN).so: $(OBJS)
 	$(CXX) $(CXXFLAGS) -shared $(OBJS) -o $@
-	@cp $@ $(LIBDIR)/$@.$(APIVERSION)
+	@cp -f $@ $(LIBDIR)/$@.$(APIVERSION)
 
 dist: clean
 	@-rm -rf $(TMPDIR)/$(ARCHIVE)
diff -Naur sudoku-0.1.3-orig/tools/Makefile sudoku-0.1.3/tools/Makefile
--- sudoku-0.1.3-orig/tools/Makefile	2006-04-25 03:02:00.000000000 +0200
+++ sudoku-0.1.3/tools/Makefile	2007-06-23 12:26:44.000000000 +0200
@@ -19,7 +19,7 @@
                       ../sudoku.cpp)
 
 CXX      ?= g++
-CXXFLAGS ?= -g -O2 -Wall -Woverloaded-virtual
+CXXFLAGS ?= -fPIC -g -O2 -Wall -Woverloaded-virtual
 DEFINES  += -D_GNU_SOURCE -DVERSION=\"$(VERSION)\"
 
 ifdef STATIC_LINK
@@ -30,6 +30,10 @@
     DEFINES += -DWITH_TEST
 endif
 
+### Allow user defined options to overwrite defaults:
+
+-include $(VDRDIR)/Make.config
+
 ### Targets:
 
 all: $(PROGRAM)
