diff -Naur busybox-1.19.3-orig/networking/udhcp/dhcpc.c busybox-1.19.3/networking/udhcp/dhcpc.c
--- busybox-1.19.3-orig/networking/udhcp/dhcpc.c	2011-11-01 14:42:48.000000000 +0100
+++ busybox-1.19.3/networking/udhcp/dhcpc.c	2011-11-01 14:34:12.000000000 +0100
@@ -24,7 +24,7 @@
 #include "common.h"
 #include "dhcpd.h"
 #include "dhcpc.h"
-
+#include <asm/types.h>
 #include <netinet/if_ether.h>
 #include <netpacket/packet.h>
 #include <linux/filter.h>
