--- dropbear-0.52/options.h.orig	2009-08-12 12:06:17.000000000 +0200
+++ dropbear-0.52/options.h	2009-08-12 12:08:24.000000000 +0200
@@ -48,7 +48,7 @@
 /* Set this if you want to use the DROPBEAR_SMALL_CODE option. This can save
 several kB in binary size, however will make the symmetrical ciphers (AES, DES
 etc) slower (perhaps by 50%). Recommended for most small systems. */
-#define DROPBEAR_SMALL_CODE
+/* #define DROPBEAR_SMALL_CODE */
 
 /* Enable X11 Forwarding - server only */
 #define ENABLE_X11FWD
--- dropbear-0.52/options.h.orig	2009-08-12 13:20:00.000000000 +0200
+++ dropbear-0.52/options.h	2009-08-12 13:21:16.000000000 +0200
@@ -263,7 +263,7 @@
 #define DEFAULT_IDLE_TIMEOUT 0
 
 /* The default path. This will often get replaced by the shell */
-#define DEFAULT_PATH "/usr/bin:/bin"
+#define DEFAULT_PATH "/usr/sbin:/sbin:/usr/bin:/bin"
 
 /* Some other defines (that mostly should be left alone) are defined
  * in sysoptions.h */
