Bug #2374 - 2nd pass: error reading index
https://projects.vdr-developer.org/issues/2374

--- a/command/markad-standalone.cpp
+++ b/command/markad-standalone.cpp
@@ -965,8 +965,10 @@
         off_t offset;
         int number,frame,iframes;
         int frange=macontext.Video.Info.FramesPerSecond*120; // 40s + 80s
+        int frange_begin=p1->position-frange; // 120 seconds before first mark
+        if (frange_begin<0) frange_begin=0; // but not before beginning of broadcast
 
-        if (marks.ReadIndex(directory,isTS,p1->position-frange,frange,&number,&offset,&frame,&iframes))
+        if (marks.ReadIndex(directory,isTS,frange_begin,frange,&number,&offset,&frame,&iframes))
         {
             if (!ProcessFile2ndPass(&p1,NULL,number,offset,frame,iframes)) break;
 
