diff -Naur a/libbb/loop.c b/libbb/loop.c
--- a/libbb/loop.c
+++ b/libbb/loop.c
@@ -145,7 +145,8 @@
 #endif
 		safe_strncpy((char *)loopinfo.lo_file_name, file, LO_NAME_SIZE);
 		loopinfo.lo_offset = offset;
-		loopinfo.lo_sizelimit = sizelimit;
+		/* Our kernel does not support lo_sizelimit */
+		//loopinfo.lo_sizelimit = sizelimit;
 		/*
 		 * LO_FLAGS_READ_ONLY is not set because RO is controlled
 		 * by open type of the lfd.
