diff -Naur scheduler-0.1.1-orig/Makefile scheduler-0.1.1/Makefile
--- scheduler-0.1.1-orig/Makefile	2008-01-13 20:21:19.000000000 +0100
+++ scheduler-0.1.1/Makefile	2008-01-20 12:40:23.000000000 +0100
@@ -22,8 +22,8 @@
 
 ### The directory environment:
 
-VDRDIR = ../../..
-LIBDIR = ../../lib
+VDRDIR ?= ../../..
+LIBDIR ?= ../../lib
 TMPDIR = /tmp
 
 ### Allow user defined options to overwrite defaults:
