diff -Naur autotimer-0.2.0-orig/Makefile autotimer-0.2.0/Makefile
--- autotimer-0.2.0-orig/Makefile	2006-06-02 05:41:44.000000000 +0200
+++ autotimer-0.2.0/Makefile	2007-06-16 20:02:00.000000000 +0200
@@ -20,8 +20,8 @@
 
 ### The directory environment:
 
-VDRDIR = ../../..
-LIBDIR = ../../lib
+VDRDIR ?= ../../..
+LIBDIR ?= ../../lib
 TMPDIR = /tmp
 
 ### Allow user defined options to overwrite defaults:
@@ -67,7 +67,7 @@
 
 libvdr-$(PLUGIN).so: $(OBJS)
 	$(CXX) $(CXXFLAGS) -shared $(OBJS) -o $@
-	@cp $@ $(LIBDIR)/$@.$(APIVERSION)
+	@cp -f $@ $(LIBDIR)/$@.$(APIVERSION)
 
 dist: clean
 	@-rm -rf $(TMPDIR)/$(ARCHIVE)
