diff -Naur filebrowser-0.0.6b-orig/Makefile filebrowser-0.0.6b/Makefile
--- filebrowser-0.0.6b-orig/Makefile	2006-12-07 20:31:05.000000000 +0100
+++ filebrowser-0.0.6b/Makefile	2007-07-29 18:03:38.000000000 +0200
@@ -8,6 +8,7 @@
 # By default the main source file also carries this name.
 #
 PLUGIN = filebrowser
+DEBUG = 1
 
 ### The version number of this plugin (taken from the main source file):
 
@@ -16,13 +17,13 @@
 ### The C++ compiler and options:
 
 CXX      ?= g++
-CXXFLAGS ?= -O2 -Wall -Woverloaded-virtual
+CXXFLAGS ?= -fPIC -O2 -Wall -Woverloaded-virtual
 
 ### The directory environment:
 
-DVBDIR = ../../../../DVB
-VDRDIR = ../../..
-LIBDIR = ../../lib
+DVBDIR ?= ../../..
+VDRDIR ?= ../../..
+LIBDIR ?= ../../lib
 TMPDIR = /tmp
 
 ### Add Debug macro
@@ -50,7 +51,7 @@
 
 ### Includes and Defines (add further entries here):
 
-INCLUDES += -I$(VDRDIR)/include -I$(DVBDIR)/include
+INCLUDES += -I$(VDRDIR)/include -I$(DVBDIR)
 
 DEFINES += -D_GNU_SOURCE -DPLUGIN_NAME_I18N='"$(PLUGIN)"'
 
@@ -78,7 +79,7 @@
 
 libvdr-$(PLUGIN).so: $(OBJS)
 	$(CXX) $(CXXFLAGS) -shared $(OBJS) -o $@
-	@cp $@ $(LIBDIR)/$@.$(APIVERSION)
+	@cp -f $@ $(LIBDIR)/$@.$(APIVERSION)
 
 dist: clean
 	@$(MAKE) clean -C developers/filebrowserdemo > /dev/null
