diff -Naur -x svn setup/common.h setup-orig/common.h
--- setup/common.h	2008-09-18 17:53:38.000000000 +0200
+++ setup-orig/common.h	2008-09-18 17:39:34.000000000 +0200
@@ -10,23 +10,13 @@
 #define SETUP_COMMON_H
 
 #include <vdr/tools.h>
-#include "vdr/plugin.h"
 
-#define STR_YES "YES"
-#define STR_NO "NO"
-
-
-
-#ifdef DEBUG_H
+#ifdef DEBUG
 #define debug(x...) dsyslog("open 7x0 Setup: " x);
-#define info(x...) isyslog("open 7x0 Setup: " x);
 #define error(x...) esyslog("open 7x0 Setup: " x);
 #else
 #define debug(x...) ;
-#define info(x...) isyslog("open 7x0 Setup: " x);
 #define error(x...) esyslog("open 7x0 Setup: " x);
 #endif
 
-
-
 #endif // __SKINSOPPALUSIKKA_COMMON_H
diff -Naur -x svn setup/config.cpp setup-orig/config.cpp
--- setup/config.cpp	2008-09-18 17:44:47.000000000 +0200
+++ setup-orig/config.cpp	2008-09-18 17:39:34.000000000 +0200
@@ -18,6 +18,7 @@
 #include "config.h"
 #include "sysconfig.h"
 #include "plugins.h"
+#include "debug.h"
 #include "util.h"
 
 /*
@@ -35,7 +36,7 @@
 	debug("Config.constructor");
 	//Init sysconfig values
   _filename  	= Util::Strdupnew(fname);
-  debug("Load file: [%s]",fname);
+  debug("Load: %s",fname);
   _xmlDoc    	= TiXmlDocument( fname );
   debug("xml document created")
 
@@ -78,13 +79,13 @@
         // Set some default Values
         const char *tmp=NULL;
         if( (tmp =root->Attribute("bootLinux"))==NULL)
-          _bootLinux=Util::Strdupnew("pic_tool reboot 3 10");
+          _bootLinux=Util::Strdupnew("reboot");
         else
           _bootLinux=Util::Strdupnew(tmp);
 
 		//Read command to get plugin path from XML
-        if( (tmp =root->Attribute("VDRlibDirCMD"))==NULL)
-          _libDirCMD = Util::Strdupnew("/var/vdr/lib");
+        if( (tmp =root->Attribute("VDRlibDirCmd"))==NULL)
+          _libDirCMD = Util::Strdupnew("env | grep PLUGIN_PATH | cut -d= -f2");
         else
           _libDirCMD = Util::Strdupnew(tmp);
           
@@ -107,18 +108,18 @@
 			_libDir=Util::Strdupnew(tmp);
 			free(line);
 		}else{
-			error("Error while opening pipe! Execute '%s' faild! Could not find plugin path", tmp);
+			dsyslog("Error while opening pipe! Execute '%s' faild! Could not find plugin path", tmp);
 		}
 		debug("We found plugin path: %s", _libDir);
-
+		//_libDir="/media/pc1/vdrRootFS/PLUGINS/lib";
 
         if( (tmp =root->Attribute("sysconfigPre"))==NULL)
-          _sysconfigPre=Util::Strdupnew("/etc/vdr/plugins/setup/local-conf -r -f /tmp/tmp.conf -c /etc");
+          _sysconfigPre=Util::Strdupnew("get-info --eval --config >/tmp/sysconfig.txt");
         else
           _sysconfigPre=Util::Strdupnew(tmp);
 
         if( (tmp =root->Attribute("sysconfigPost"))==NULL)
-          _sysconfigPost=Util::Strdupnew("/etc/vdr/plugins/setup/local-conf -w -f /tmp/tmp.conf -c /etc");
+          _sysconfigPost=Util::Strdupnew("eval `get-info --eval --file`; cat /tmp/sysconfig.txt >>$CONFFILE_CONFIG ; dump-fw-config");
         else
           _sysconfigPost=Util::Strdupnew(tmp);
 		
@@ -127,24 +128,22 @@
   		debug("sysconfigPre: %s",_sysconfigPre);
   		ok = system(_sysconfigPre);
   		if( ok == 0) {//Load sysconfig
-  			if ((ok=_sysconfig.LoadFile(sysconfigFile)) != true){
-  					debug("Could not load sysconfig: [%s]",sysconfigFile);
-  			}
-  	  } else{
-          error("Could not execute sysconfigPre: [%s], errno=%d", _sysconfigPre, ok);	
+  			if ((ok=_sysconfig.LoadFile(sysconfigFile)) != true)
+  				{
+  					DEBUG2("Could not load sysconfig:%s",sysconfigFile);
+  				}
+  	  	} else{
+    		DEBUG4("%s: Could not execute sysconfigPre: %s, errno=%d\n", DBG_PREFIX, _sysconfigPre, ok);	
   		}
 
         root=root->FirstChildElement();
         if( root!= NULL &&
             strcmp(root->Value(), "plugins")==0)
         {
-          _activePlugins.SetProtectedList(root->Attribute("protected"));
           const char* nameSysconfig = root->Attribute("sysconfig");
-          
           if( nameSysconfig != NULL )
           {
            _activePlugins.SetSysconfigName(nameSysconfig);
-           debug("SysPlugin [%s]",nameSysconfig);
            if( ok=loadPlugins(root->FirstChildElement())) //Load Plugins
            {
               ok = _menus.LoadXml(root->NextSibling("menus"));//Load Menus
@@ -163,11 +162,8 @@
   if( ok)
   {	
   	ok=readVdrLib();
-    info("Loaded config file: [%s]",_filename);
   }
 
- 
-
   return(ok);
 }
 
@@ -237,7 +233,7 @@
   		debug("sysconfigPost: %s",_sysconfigPost);
 	  	ok = system(_sysconfigPost);
   	}else{
-    	error("Could not execute sysconfigPost: [%s], errno=%d", _sysconfigPost, ok);	
+    	DEBUG4("%s: Could not execute sysconfigPost: %s, errno=%d\n", DBG_PREFIX, _sysconfigPost, ok);	
   	}	
   }
     
@@ -272,16 +268,10 @@
  * @param elem 
  * @return 
  */
-/*
 bool Config::loadPlugins(TiXmlNode *node)
 {
   bool ok = true;
   TiXmlElement *elem = NULL;
-
-  const char *loaded_plugins =  _sysconfig.GetVariable(_activePlugins.GetSysconfigName());
-  debug("Loaded plugins: [%s]",loaded_plugins);
-  debug("SysPlugin [%s]",_activePlugins.GetSysconfigName());
-
   if(node == NULL || strcmp(node->Value(), "plugin") !=0 )
   {
     dumpXMLError("no <plugin> tag found");
@@ -302,17 +292,9 @@
             if( protect == NULL || Util::isBool(protect, b_protect) == false)
               b_protect = false;
 
-            if( name != NULL && info != NULL ) {
-//ZJ                && active != NULL  && Util::isBool(active, b_active)) {
-              if ( loaded_plugins != NULL && strstr(loaded_plugins,name) == NULL ) {
-                debug("Plugin Desactivado en config [%s]",name);
-                b_active = false;
-              }else {
-                debug("Plugin Activo en config [%s]",name);
-                b_active = true;
-              } 
+            if( name != NULL && info != NULL && active != NULL &&
+                Util::isBool(active, b_active))
               _activePlugins.AddPlugin(name, param,  info, b_active, b_protect);
-            }
             else
             {
               ok = false;
@@ -322,36 +304,6 @@
   }
   return(ok) ;
 }
-*/
-bool Config::loadPlugins(TiXmlNode *node)
-{
-  bool ok = true;
-   
-  const char *loaded_plugins =  _sysconfig.GetVariable(_activePlugins.GetSysconfigName());
-  char *str_plugins = NULL ;
-  char *plugin = NULL;
-  
-      
-  info("Loaded plugins: [%s]",loaded_plugins);
-  
-  if ( loaded_plugins != NULL && strlen(loaded_plugins)>0) {
-    str_plugins =  Util::Strdupnew(loaded_plugins);
-    plugin = strtok(str_plugins," \"-");
-
-    while ( plugin != NULL ) {
-      _activePlugins.AddPlugin(plugin+1, NULL,  plugin+1, true, _activePlugins.checkProtected(plugin+1),false);
-      plugin = strtok(NULL," \"-");
-    }
-    delete []str_plugins;
-  }
-  else {
-    error ("Not found variable [%s]",_activePlugins.GetSysconfigName());
-    ok = false;
-  }
-
-  
-  return(ok) ;
-}
 
 /**
  * Save Plugins
@@ -361,8 +313,10 @@
 {
   TiXmlElement *xml = new TiXmlElement("plugins");
 
-  xml->SetAttribute("sysconfig", _activePlugins.GetSysconfigName());
-  xml->SetAttribute("protected", _activePlugins.GetProtectedList());
+  xml->SetAttribute("sysconfig",   _activePlugins.GetSysconfigName());
+
+  savePlugins(&_activePlugins, xml);
+  savePlugins(&_notInSystemPlugins, xml);
 
   return(xml);
 }
@@ -409,7 +363,7 @@
     errStr = _xmlDoc.ErrorDesc();
     col    = _xmlDoc.ErrorCol();
     row    = _xmlDoc.ErrorRow();
-    error("Error in %s Col=%d Row=%d :%s", _filename, col, row, errStr );
+    DEBUG6("%s: Error in %s Col=%d Row=%d :%s\n",DBG_PREFIX, _filename, col, row, errStr );
   }
 }
 void Config::dumpXMLError(const char* myErrStr)
@@ -422,7 +376,7 @@
     errStr = _xmlDoc.ErrorDesc();
     col    = _xmlDoc.ErrorCol();
     row    = _xmlDoc.ErrorRow();
-    debug("%s: %s: Error in %s Col=%d Row=%d", myErrStr, errStr, _filename, col, row);
+    DEBUG7("%s: %s: %s: Error in %s Col=%d Row=%d\n",DBG_PREFIX, myErrStr, errStr, _filename, col, row);
   }
 }
 
@@ -463,24 +417,23 @@
  * and set the state within the plugin list
  * @return true on success
  */
-bool Config::readVdrLibDir(const char* libDir)
+bool Config::readVdrLib()
 {
   bool ok =true;
-  DIR           *dir = opendir(libDir);
+  DIR           *dir = opendir(_libDir);
   struct dirent *entry = NULL;
   char          *module=NULL;
   char          *tmp=NULL;
   char          *prefix ="libvdr-";
   char          *suffix = NULL;
 
-  debug("config:ReadVdrLib %s",libDir);
+	debug("config:ReadVdrLib");
   //accept only dynamic Link libs with the current vdr version
   asprintf(&suffix, ".so.%s", APIVERSION);
   if( dir !=NULL)
   {
     while( (entry=readdir(dir))!=NULL)
     {
-      debug("Plugin en directorio[%s]\n",entry->d_name +7);
       if( strncmp(entry->d_name, prefix,7)==0  &&
           (tmp=strstr(entry->d_name, suffix))!= NULL)
         {
@@ -489,49 +442,32 @@
           _activePlugins.SetLibDirPlugin(module);
         }
     }
-    
+
+    int nrPlugins = _activePlugins.GetNr();
+    for(int i=0; i<nrPlugins; i++)
+    {
+      Plugin *p=  _activePlugins.Get(i);
+
+      if( p != NULL && p->GetInSystem()==false)
+      {
+        _activePlugins.Del(p, false);
+        _notInSystemPlugins.Add(p);
+        nrPlugins--;
+        i--;
+      }
+    }
+
     closedir(dir);
   }
   else
   {
-    debug("Could not read directory: [%s]", _libDir);
+    DEBUG3("%s: Could not read directory: %s\n", DBG_PREFIX, _libDir);
     ok = false;
   }
   
   if( suffix != NULL) free(suffix);
   return(ok);
 }
- 
-bool Config::readVdrLib()
-{
-  bool result =true;  
-  char *ldir = Util::Strdupnew(_libDir);
-  char *delim = " ";
-  char *dir;
-  
-  dir = strtok(ldir,delim);
-  
-  while ( dir ) {
-	  result = readVdrLibDir(dir)  && result;
-	  dir = strtok(NULL,delim);
-  }
-  int nrPlugins = _activePlugins.GetNr();
-  for(int i=0; i<nrPlugins; i++)
-  {
-    Plugin *p=  _activePlugins.Get(i);
-
-    if( p != NULL && p->GetInSystem()==false)
-    {
-      _activePlugins.Del(p, false);
-      _notInSystemPlugins.Add(p);
-      nrPlugins--;
-      i--;
-    }
-  }
-  
-  if (ldir) delete (ldir);  
-  return(result);
-}
 
 char * Config::GetBootLinux( )
 {
diff -Naur -x svn setup/config.h setup-orig/config.h
--- setup/config.h	2008-09-18 17:44:47.000000000 +0200
+++ setup-orig/config.h	2008-09-18 17:39:34.000000000 +0200
@@ -58,7 +58,6 @@
     void                dumpXMLError(const char* errStr);
     bool                readVdrLib();
     void                prepareSysConfig();
-    bool Config::readVdrLibDir(const char* libDir);
 };
 
 
diff -Naur -x svn setup/debug.h setup-orig/debug.h
--- setup/debug.h	1970-01-01 01:00:00.000000000 +0100
+++ setup-orig/debug.h	2008-09-18 17:39:34.000000000 +0200
@@ -0,0 +1,36 @@
+/*********************************************************
+ * DESCRIPTION: 
+ *             Header File
+ *
+ * $Id$
+ *
+ * Contact:    ranga@vdrtools.de
+ *
+ * Copyright (C) 2004 by Ralf Dotzert 
+ *********************************************************/
+
+#ifndef DEBUG_H
+#define DEBUG_H
+#include "vdr/plugin.h"
+
+#ifdef  DEBUG_STDOUT
+#define DEBUG1(p1)                      printf(p1)
+#define DEBUG2(p1, p2)                  printf(p1, p2)
+#define DEBUG3(p1, p2, p3)              printf(p1, p2, p3)
+#define DEBUG4(p1, p2, p3, p4)          printf(p1, p2, p3, p4)
+#define DEBUG5(p1, p2, p3, p4, p5)      printf(p1, p2, p3, p4, p5)
+#define DEBUG6(p1, p2, p3, p4, p5, p6)  printf(p1, p2, p3, p4, p5, p6)
+#define DEBUG7(p1, p2, p3, p4, p5, p6, p7)  printf(p1, p2, p3, p4, p5, p6, p7)
+#else
+#define DEBUG1(p1)                      esyslog(p1)
+#define DEBUG2(p1, p2)                  esyslog(p1, p2)
+#define DEBUG3(p1, p2, p3)              esyslog(p1, p2, p3)
+#define DEBUG4(p1, p2, p3, p4)          esyslog(p1, p2, p3, p4)
+#define DEBUG5(p1, p2, p3, p4, p5)      esyslog(p1, p2, p3, p4, p5)
+#define DEBUG6(p1, p2, p3, p4, p5, p6)  esyslog(p1, p2, p3, p4, p5, p6)
+#define DEBUG7(p1, p2, p3, p4, p5, p6, p7)  esyslog(p1, p2, p3, p4, p5, p6, p7)
+
+#endif
+#define DBG_PREFIX                      "ERROR: open7x0-Firmware Setup"
+
+#endif /* DEBUG_H */
diff -Naur -x svn setup/HISTORY setup-orig/HISTORY
--- setup/HISTORY	2008-09-18 17:44:47.000000000 +0200
+++ setup-orig/HISTORY	2008-09-18 17:39:34.000000000 +0200
@@ -1,91 +1,83 @@
-VDR Plugin 'setup' Revision History
------------------------------------
-2008-09-18: Version 0.1.0 open7x0 by zjuanma@yahoo.es
-- VDR-NG (M7x0) only.
-- Get active plugins from /etc/rc.local.conf
-- Remove xml plugins configuration.
-2007-12-11: Version 0.0.2 open7x0 by zjuanma@yahoo.es
-- Patched for VDR-NG (M7x0).
-- Supports multiple configurations file (i18n)
-
-2006-03-06: Version 0.0.1 open7x0 by HaPe
-- do not use std::vector
-
-
-2006-03-05
-- handle missing channellists /etc/vdr/channels/*.conf
-- when changing channel lists delete timers with unknown channel
-- correct behaviour if menucommandpatch was applied
-- Show title of current Submenu
-
-
-
-2006-01-15: Version 0.2.3
-- Now Reload Channel-List if Channel-List was changed
-- patched for vdr Version 1.3.38
-
-2005-10-12: Version 0.2.2
-- fixed bug, know do not hide other buttons if help button is shown.
-- know compatible with BIG-PATCH,
-- Supports all VDR System Menu Commands are extracted during buidl process
-  If BIGPATCH was applied, then the new System-Command "LiveBuffer", "BigPatch"
-  are supported
-
-
-2005-10-06: Version 0.2.1
--  fixed Bug: Know changes are saved again in sysConfigFile
-
-2005-10-03: Version 0.2.0
--  Added Setup Menü for plugin. Allowing to edit Menusuffix and Entry-Prefix)
--  Changed minivdr-setup.xml format
-   all <menu>..</menu> tags are now within a <menus>..</menus> tag
-   New system attribute values for the Menu Tags:
-       VDRMenu     ( specifies the position of "Edit Vdr Menu")
-       ActPlugins  ( specifies the position of Activate Plugins-Menu)
--  Added new type "Ip" for Entry holding IP-Addresses Digits and dots.
--  Added new attribut "help1" Entry and Menu in minivdr-setup.xml
--  Added new attribut "help2" Entry and Menu in minivdr-setup.xml
--  The file sysconfig will now be created if it does not exist
--  removed attribute "bootVDR", now using standard vdr Rebot mechanism
--  Configurationfiles (minivdr-setup.xml and vdr-menu.xml) are now stored
-   "PluginConfigDirectory/setup"
-
-2005-06-19: Version 0.1.0
-- changed interaction with user, now the setup plugin behaves as usually known by VDR Users.
-  By pressing the OK Button the user may switch back to the previous menu.
-- fixed bug, if Directory <VDR-Configdir>/channels or <VDR-Configdir>/channels/*.conf are missing
-- do not remove plugins from "minivdr-setup.xml", when plugin is not in system
-- Allows to edit Plugin-Parameters
-- Support executing User Defined commands when walking through MiniVDR Setup Menu
-  specify new command attribute within menu tag
-- Know supports VDR Standard Setup Menu within Setup-Plugin Menu
-- Support paging plugin in Menu Acivate/Deactivbate Plugins
-
-2005-02-26: Version 0.0.5
-- Added new Attributes in minivdr-setup.xml to configure system dependent pathes
-- improved edit menu handling
-- fixed crash, when no network modules were found
-- now add also network modules in subdirectories
-- Show name of plugin in Pluginmenu
-
-2005-01-28: Version 0.0.4
-- Added Creation of new Menus
-- Added Editing of Menu-name
-- Added deltion of menu
-
-2005-01-23: Version 0.0.3
-- Added new entry types:
-    hex     - hexvalues in the form of AB:FF:12
-    numtext - String holding only digits
-- Added new feature ChildLock
-- Added SubMenu Patch for VDR
-- Added SubMenu Edit-Menu
-
-2005-01-05: Version 0.0.2
-- first released version
-- Enable/Disable Plugins
-- Added Menu / SubMenus
-
-2004-12-05: Version 0.0.1
-
-- Initial revision.
+VDR Plugin 'setup' Revision History
+-----------------------------------
+2006-03-06
+- do not use std::vector
+
+
+2006-03-05
+- handle missing channellists /etc/vdr/channels/*.conf
+- when changing channel lists delete timers with unknown channel
+- correct behaviour if menucommandpatch was applied
+- Show title of current Submenu
+
+
+
+2006-01-15: Version 0.2.3
+- Now Reload Channel-List if Channel-List was changed
+- patched for vdr Version 1.3.38
+
+2005-10-12: Version 0.2.2
+- fixed bug, know do not hide other buttons if help button is shown.
+- know compatible with BIG-PATCH,
+- Supports all VDR System Menu Commands are extracted during buidl process
+  If BIGPATCH was applied, then the new System-Command "LiveBuffer", "BigPatch"
+  are supported
+
+
+2005-10-06: Version 0.2.1
+-  fixed Bug: Know changes are saved again in sysConfigFile
+
+2005-10-03: Version 0.2.0
+-  Added Setup Menü for plugin. Allowing to edit Menusuffix and Entry-Prefix)
+-  Changed minivdr-setup.xml format
+   all <menu>..</menu> tags are now within a <menus>..</menus> tag
+   New system attribute values for the Menu Tags:
+       VDRMenu     ( specifies the position of "Edit Vdr Menu")
+       ActPlugins  ( specifies the position of Activate Plugins-Menu)
+-  Added new type "Ip" for Entry holding IP-Addresses Digits and dots.
+-  Added new attribut "help1" Entry and Menu in minivdr-setup.xml
+-  Added new attribut "help2" Entry and Menu in minivdr-setup.xml
+-  The file sysconfig will now be created if it does not exist
+-  removed attribute "bootVDR", now using standard vdr Rebot mechanism
+-  Configurationfiles (minivdr-setup.xml and vdr-menu.xml) are now stored
+   "PluginConfigDirectory/setup"
+
+2005-06-19: Version 0.1.0
+- changed interaction with user, now the setup plugin behaves as usually known by VDR Users.
+  By pressing the OK Button the user may switch back to the previous menu.
+- fixed bug, if Directory <VDR-Configdir>/channels or <VDR-Configdir>/channels/*.conf are missing
+- do not remove plugins from "minivdr-setup.xml", when plugin is not in system
+- Allows to edit Plugin-Parameters
+- Support executing User Defined commands when walking through MiniVDR Setup Menu
+  specify new command attribute within menu tag
+- Know supports VDR Standard Setup Menu within Setup-Plugin Menu
+- Support paging plugin in Menu Acivate/Deactivbate Plugins
+
+2005-02-26: Version 0.0.5
+- Added new Attributes in minivdr-setup.xml to configure system dependent pathes
+- improved edit menu handling
+- fixed crash, when no network modules were found
+- now add also network modules in subdirectories
+- Show name of plugin in Pluginmenu
+
+2005-01-28: Version 0.0.4
+- Added Creation of new Menus
+- Added Editing of Menu-name
+- Added deltion of menu
+
+2005-01-23: Version 0.0.3
+- Added new entry types:
+    hex     - hexvalues in the form of AB:FF:12
+    numtext - String holding only digits
+- Added new feature ChildLock
+- Added SubMenu Patch for VDR
+- Added SubMenu Edit-Menu
+
+2005-01-05: Version 0.0.2
+- first released version
+- Enable/Disable Plugins
+- Added Menu / SubMenus
+
+2004-12-05: Version 0.0.1
+
+- Initial revision.
diff -Naur -x svn setup/i18n.cpp setup-orig/i18n.cpp
--- setup/i18n.cpp	2008-09-18 17:44:47.000000000 +0200
+++ setup-orig/i18n.cpp	2008-09-18 17:39:34.000000000 +0200
@@ -1166,48 +1166,6 @@
     "",
     "",
 },
-{ "YES",
-    "ja",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "sí",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "",
-},
-{ "NO",
-    "nein",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "no",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "",
-    "",
-},
 
 { NULL }
 };
diff -Naur -x svn setup/M7x0 config/local-conf setup-orig/M7x0 config/local-conf
--- setup/M7x0 config/local-conf	2008-09-18 17:44:47.000000000 +0200
+++ setup-orig/M7x0 config/local-conf	1970-01-01 01:00:00.000000000 +0100
@@ -1,51 +0,0 @@
-#!/bin/ash
-
-#Default options
-CONF_DIR=.
-CONF_FILES="rc.local.conf rc.conf" 
-OPTION=
-DATA_FILE=/tmp/conf.tmp
-
-GREP_CMD=grep
-SED_CMD=sed
-
-#### Main loop
-
-set -- $(getopt  wr:f:c "$*")
-if	[ $? -ne 0 ]
-then	print >&2 "Usage: $0 -w|-r [-f output] [-c dir]"
-	exit 1
-fi
-for o
-do	case "$o" in
-	-w | -r) shift; OPTION=$o ;;
-	-f)	shift; DATA_FILE=$1; shift;;
-	-c)	shift; CONF_DIR=$1; shift;;
-	--)	shift; break;;
-	esac
-done
-
-if [ "$OPTION" = "-w" ] ; then
-        for f in `echo $CONF_FILES` ; do
-                #Delete file
-                `cat /dev/null > $CONF_DIR/$f`
-
-                `$GREP_CMD "^$f--" $DATA_FILE | $SED_CMD "s/^$f--//" >> $CONF_DIR/$f`
-        done
-else
-        if [ "$OPTION" = "-r" ] ; then
-                #Delete temp file 
-                `cat /dev/null > $DATA_FILE`
-
-                for f in  `echo $CONF_FILES` ; do
-
-                        `$GREP_CMD "^[^#].*=" $CONF_DIR/$f | $SED_CMD -e "s/^/$f--/" -e "s/#.*$//" >> $DATA_FILE`
-echo
-
-                done 
-        else
-                echo "Error: -r or -w option not found"
-                exit 2
-        fi
-fi
-
diff -Naur -x svn setup/M7x0 config/local-plugins setup-orig/M7x0 config/local-plugins
--- setup/M7x0 config/local-plugins	2008-09-18 17:44:47.000000000 +0200
+++ setup-orig/M7x0 config/local-plugins	1970-01-01 01:00:00.000000000 +0100
@@ -1,3 +0,0 @@
-#!/bin/sh
-
-grep '^vdr_plugindirs' /etc/rc.local.conf | cut -d= -f2 | sed 's/^\"//' | sed 's/\"$//'
diff -Naur -x svn setup/M7x0 config/open7x0-setup_eng.xml setup-orig/M7x0 config/open7x0-setup_eng.xml
--- setup/M7x0 config/open7x0-setup_eng.xml	2008-09-18 18:01:44.000000000 +0200
+++ setup-orig/M7x0 config/open7x0-setup_eng.xml	1970-01-01 01:00:00.000000000 +0100
@@ -1,48 +0,0 @@
-<!--
--   open7x0-VDR Configuration File
--   based on Minisetup by Ralf Dotzert
--->
-<setup sysconfigFile="/tmp/tmp.conf" sysconfigPre="/etc/vdr/plugins/setup/local-conf -r -f /tmp/tmp.conf -c /etc" sysconfigPost="/etc/vdr/plugins/setup/local-conf -w -f /tmp/tmp.conf -c /etc" bootLinux="/etc/vdr/rebootvdr" VDRlibDirCMD="/etc/vdr/plugins/setup/local-plugins">
-    <plugins sysconfig="rc.local.conf--vdr_pluginopts" protected="setup">
-    </plugins>
-    <menus>
-        <menu name="Manager plugins" help1="Enable and disable plugins" system="ActPlugins" />        
-        <menu name="General Configuration">
-            <entry name="Mount USB" sysconfig="rc.local.conf--do_usb_mount" type="bool" value="YES" help1="Mount usb disk" />
-            <entry name="Mount Network" sysconfig="rc.local.conf--do_net_mount" type="bool" value="YES" help1="Mount network shared directories" />
-            <entry name="Start time" sysconfig="rc.local.conf--vdr_startup_time" type="number" value="302" help1="Start time" />
-        </menu>
-        <menu name="Directories">
-            <entry name="Configuration" sysconfig="rc.local.conf--vdr_confdir" type="text" value='&quot;/etc/vdr&quot;' help1="Configuration files. NOT MODIFY" />
-            <entry name="Videos" sysconfig="rc.local.conf--vdr_videodirs" type="text" value='&quot;/media/LACIE/vdr&quot;' help1="Videos directory" />
-            <entry name="Plugins" sysconfig="rc.local.conf--vdr_plugindirs" type="text" value='&quot;/var/media/disk-volume-3/vdr_ng&quot;' help1="Plugins Directories. NOT MODIFY" />
-        </menu>
-        <menu name="EPG Configuration">
-            <entry name="EPG File" sysconfig="rc.local.conf--vdr_epgfile" type="text" value='&quot;/etc/vdr/epg.data&quot;' help1="EPG file" />
-            <entry name="EPG direct" sysconfig="rc.local.conf--vdr_epgdirect" type="bool" value="NO" help1="EPG direct" />
-        </menu>
-        <menu name="Network Configuration">
-            <entry name="Mode" sysconfig="rc.conf--net" type="selection" value="inet" help1="Configuration IP mode. DHCP/INET">
-                <value>dhcp</value>
-                <value>inet</value>
-            </entry>
-            <entry name="IP Address" sysconfig="rc.conf--ip" type="ip" value="192.168.2.100" help1="IP Address M7XX" />
-            <entry name="Network Mask" sysconfig="rc.conf--netmask" type="ip" value="255.255.255.0" help1="Network mask" />
-            <entry name="Gateway" sysconfig="rc.conf--gateway" type="ip" value='&quot;NO&quot;' help1="Gatewary IP address. default NO" />
-            <entry name="MAC Address" sysconfig="rc.conf--mac_cmd" type="text" value='&quot;pic_tool -s get-mac&quot;' help1="NOT MODIFY" />
-            <entry name="Hostname" sysconfig="rc.conf--hostname" type="text" value="`cat /etc/hostname`" help1="M7XX Name (DNS). default /etc/hostname. NOT MODIFY" />  
-        </menu>
-        <menu name="Services">
-            <entry name="Inetd" sysconfig="rc.conf--inetd" type="bool" value="YES" help1="Start inetd server (ftp,telnet)" />
-            <entry name="Web server" sysconfig="rc.local.conf--webif_start" type="bool" value="YES" help1="Start administration web" />
-        </menu>
-        <menu name="Boot menu">
-            <entry name="Enabled" sysconfig="rc.local.conf--bootmenu_start" type="bool" value="YES" help1="Enable boot menu" />
-            <entry name="Default option" sysconfig="rc.local.conf--bootmenu_default" type="number" value="0" help1="Default option" />
-            <entry name="Video mode" sysconfig="rc.local.conf--bootmenu_video_mode" type="selection" value="0" help1="0 = fbase, 1 = svideo">
-                <value>0</value>
-                <value>1</value>
-            </entry>
-        </menu>
-    </menus>
-</setup>
diff -Naur -x svn setup/M7x0 config/open7x0-setup_esl.xml setup-orig/M7x0 config/open7x0-setup_esl.xml
--- setup/M7x0 config/open7x0-setup_esl.xml	2008-09-18 18:01:07.000000000 +0200
+++ setup-orig/M7x0 config/open7x0-setup_esl.xml	1970-01-01 01:00:00.000000000 +0100
@@ -1,48 +0,0 @@
-<!--
--   open7x0-VDR Configuration File
--   based on Minisetup by Ralf Dotzert
--->
-<setup sysconfigFile="/tmp/tmp.conf" sysconfigPre="/etc/vdr/plugins/setup/local-conf -r -f /tmp/tmp.conf -c /etc" sysconfigPost="/etc/vdr/plugins/setup/local-conf -w -f /tmp/tmp.conf -c /etc" bootLinux="/etc/vdr/rebootvdr" VDRlibDirCMD="/etc/vdr/plugins/setup/local-plugins">
-    <plugins sysconfig="rc.local.conf--vdr_pluginopts" protected="setup">
-    </plugins>
-    <menus>
-        <menu name="Gestionar plugins" help1="Activar y desactivar plugins" system="ActPlugins" />
-        <menu name="Configuración general">
-            <entry name="Montar discos USB" sysconfig="rc.local.conf--do_usb_mount" type="bool" value="YES" help1="Montar los discos USB" />
-            <entry name="Montar directorios en Red" sysconfig="rc.local.conf--do_net_mount" type="bool" value="NO" help1="Montar los directorios en Red" />
-            <entry name="Tiempo de arranque" sysconfig="rc.local.conf--vdr_startup_time" type="number" value="298" help1="Valor en segundos del tiempo de arranque del VDR cuando hay una grabacion programada." />
-        </menu>
-        <menu name="Configuración de directorios">
-            <entry name="Directorio ficheros de configuración" sysconfig="rc.local.conf--vdr_confdir" type="text" value='&quot;/etc/vdr&quot;' help1="Directorio donde se encuentra los ficheros de configuración. NO DEBE SER MODIFICADO" />
-            <entry name="Directorio de grabaciones" sysconfig="rc.local.conf--vdr_videodirs" type="text" value="/var/media/disk-volume-0/vdr/videoVDR" help1="Directorio donde se almacenan las grabaciones" />
-            <entry name="Directorio de plugins" sysconfig="rc.local.conf--vdr_plugindirs" type="text" value='&quot;/var/media/disk-volume-3/vdr/plugins&quot;' help1="Directorio donde se encuentran los plugins. NO DEBE SER MODIFICADO" />
-        </menu>
-        <menu name="Configuración EPG">
-            <entry name="Fichero EPG" sysconfig="rc.local.conf--vdr_epgfile" type="text" value='&quot;/etc/vdr/epg.data&quot;' help1="Fichero donde se almacena la información de EPG." />
-            <entry name="EPG directa" sysconfig="rc.local.conf--vdr_epgdirect" type="bool" value="YES" help1="Determina si la EPG se almacena inmediatamente o sólo cuando se apaga el VDR" />
-        </menu>
-        <menu name="Configuración Red">
-            <entry name="Modo configuración" sysconfig="rc.conf--net" type="selection" value="inet" help1="Modo de configuración de la dirección IP.">
-                <value>dhcp</value>
-                <value>inet</value>
-            </entry>
-            <entry name="Dirección IP" sysconfig="rc.conf--ip" type="ip" value="192.168.2.100" help1="Dirección IP del M7XX" />
-            <entry name="Máscara de Red" sysconfig="rc.conf--netmask" type="ip" value="255.255.255.0" help1="Máscara de Red" />
-            <entry name="Puerta de Enlace" sysconfig="rc.conf--gateway" type="ip" value='&quot;192.168.2.1&quot;' help1="Dirección IP de la puerta de Enlace. Si no se usa valor NO" />
-            <entry name="Dirección MAC" sysconfig="rc.conf--mac_cmd" type="text" value='&quot;pic_tool -s get-mac&quot;' help1="Define el comando con el que se obtiene la dirección MAC. NO DEBE SER MODIFICADO" />
-            <entry name="Hostname" sysconfig="rc.conf--hostname" type="text" value="`cat /etc/hostname`" help1="Nombre del M7XX (DNS). Por defecto lo toma del fichero /etc/hostname. NO DEBE SER MODIFICADO" />
-        </menu>
-        <menu name="Servicios">
-            <entry name="Inetd" sysconfig="rc.conf--inetd" type="bool" value="YES" help1="Arrancar servidor inetd (ftp,telnet)" />
-            <entry name="Servidor Web" sysconfig="rc.local.conf--webif_start" type="bool" value="YES" help1="Arrancar servidor web de administración" />
-        </menu>
-        <menu name="Menu de arranque">
-            <entry name="Habilitar" sysconfig="rc.local.conf--bootmenu_start" type="bool" value="NO" help1="Habilitar menu de arranque" />
-            <entry name="Opción por defecto" sysconfig="rc.local.conf--bootmenu_default" type="number" value="0" help1="Opción por defecto del menu de arranque" />
-            <entry name="Modo de video" sysconfig="rc.local.conf--bootmenu_video_mode" type="selection" value="0" help1="0 = fbase, 1 = svideo">
-                <value>0</value>
-                <value>1</value>
-            </entry>
-        </menu>
-    </menus>
-</setup>
diff -Naur -x svn setup/M7x0 config/open7x0-setup V1.0.xml setup-orig/M7x0 config/open7x0-setup V1.0.xml
--- setup/M7x0 config/open7x0-setup V1.0.xml	1970-01-01 01:00:00.000000000 +0100
+++ setup-orig/M7x0 config/open7x0-setup V1.0.xml	2008-09-18 17:39:34.000000000 +0200
@@ -0,0 +1,184 @@
+<!--
+-   Mini-VDR Configuration File
+-   (c) 2005,2006 by Ralf Dotzert
+-
+-
+-   Example:
+-
+-  <setup sysconfigFile="/etc/sysconfig"> 
+-         bootLinux="/sbin/reboot" 
+-         VDRlibDir="/usr/lib/vdr/plugins" 
+-    <plugins sysconfig="PLUGINLIST">
+-         <plugin name="setup" info="Minivdr-Setup" active="yes" />
+-         <plugin name="vdrcd" param="-c /dev/dvd -c /media/dvd -c /media/cdfs" info="VDR CD"/>
+-         ...
+-    </plugin>
+-    <menus>
+-       <menu name="VDR-Einstellungen">
+-         <menu name="Edit VDR-Menü" system="VDRMenu" />
+-         <menu name="OSD" system="OSD" />
+-         <menu name="EPG" system="EPG" />
+-         <menu name="DVB" system="DVB" />
+-         <menu name="LNB" system="LNB" />
+-         <menu name="CICAM" system="CICAM" />
+-         <menu name="Aufnahme" system="Record" />
+-         <menu name="Wiedergabe" system="Replay" />
+-         <menu name="Sonstiges" system="Misc" />
+-         <menu name="Aktiviere Plugins" system="ActPlugins" />
+-         <menu name="Plugins-Setup" system="Plugins" />
+-         <menu name="BigPatch-Setup" system="BigPatch" /> -- Wenn der Bigpatch angwendet wurde --
+-         <menu name="LiveBuffer-Setup" system="LiveBuffer" /> -- Wenn der LiveBuffer-Patch angewendet wurde --
+-       </menu>
+-    <menu name="Netzwerk" help1="Netzwerkeinstellungen" help2="network.hlp" command="/etc/init.d/network restart">
+-       <entry name="Benutze DHCP" sysconfig="USE_DHCP" type="bool" value="true" command="/etc/init.d/network restart/>
+-       <entry name="IP-Adresse" sysconfig="IP" type="ip" value="192.168.0.111" />
+-        ...
+-    </menu>
+-    <menu name="Sonstiges">
+-        <entry name="Ramddisk im MB" sysconfig="RAMDISK" type="number" value="16" />
+-        <menu name="SubMenu">
+-          <entry ... />
+-        </menu>
+-    </menu>
+-   </menus>
+-  </setup>
+-->
+<setup sysconfigFile="/pc1/vdrRootFS/conf-M740AV2-0001E3FA8FD3/plugins/setup/fw-config.txt" bootLinux="/sbin/reboot" VDRlibDir="/pc1/vdrRootFS/PLUGINS\lib">
+    <plugins sysconfig="CONFIG_VDR_PLUGINS">
+        <plugin name="epgsearch" info="EPG zeigt Wiederholungen" active="yes" />
+        <plugin name="solitaire" info="Kartenspiel Solitaire" active="yes" />
+        <plugin name="setup" info="Minivdr-Setup" active="yes" protected="yes" />
+        <plugin name="sleeptimer" info="Shutdown" active="yes" />
+        <plugin name="streamdev-server" info="Streaming Server" active="yes" />
+        <plugin name="osdteletext" info="Teletext Dekoder für das OSD" active="yes" />
+        <plugin name="admin" info="admin" active="no" />
+        <plugin name="fepg" info="fepg" active="no" />
+        <plugin name="filebrowser" info="filebrowser" active="no" />
+        <plugin name="skinsoppalusikka" info="skinsoppalusikka" active="no" />
+        <plugin name="sudoku" info="sudoku" active="no" />
+        <plugin name="surfer" info="surfer" active="no" />
+        <plugin name="alcd" info="Activy300 LCD Display" active="no" />
+        <plugin name="lcdproc" info="Alphanummerische Displays" active="no" />
+        <plugin name="vbox" info="Anrufbeantworter Plugin" active="yes" />
+        <plugin name="rotor" info="Ansteuerung Rotor" active="yes" />
+        <plugin name="femon" info="Anzeige Signalstärke/Qualität" active="yes" />
+        <plugin name="vdrcd" param="-c /dev/dvd -c /media/dvd -c /media/cdfs" info="AutoPlay Funktion" active="yes" />
+        <plugin name="autotimeredit" param="-f /etc/vdradmin/vdradmind.at -p" info="Autotimer-Edit" active="yes" />
+        <plugin name="prefermenu" info="Bevorzugte Kanäle" active="yes" />
+        <plugin name="osdpip" info="Bild in Bild" active="yes" />
+        <plugin name="image" param="-C /usr/bin/imageplugin.sh -m /usr/bin/mount.sh" info="Bildbetrachter Plugin" active="yes" />
+        <plugin name="osdimage" param="--directory /video0/media/photos/vdr" info="Bildern über OSD" active="yes" />
+        <plugin name="burn" param="-D /dev/sr0 -d /video0 -i /video0/convert/dvd/" info="Brennt Filme auf DVD" active="yes" />
+        <plugin name="remote" info="DVB Karten Fernbedienung" active="no" protected="yes" />
+        <plugin name="dvdselect" info="DVD Laufwerke einhängen" active="yes" />
+        <plugin name="dvd" info="DVD Spieler Plugin" active="yes" />
+        <plugin name="freecell" info="Das Kartenspiel Freecell" active="yes" />
+        <plugin name="osddemo" info="Demo On-Screen-Display" active="no" />
+        <plugin name="tvonscreen" param="-v /etc/vdradmin/vdradmind.at" info="EPG Programmzeitschrift" active="yes" />
+        <plugin name="hello" info="Ein &apos;hello&apos; Plugin" active="no" />
+        <plugin name="screenshot" info="Erstellt Screenshots" active="yes" />
+        <plugin name="graphlcd" param="-d ks0108" info="GraphLCD" active="yes" />
+        <plugin name="graphtft" param="-d /dev/fb0" info="GraphTFT" active="no" />
+        <plugin name="radio" info="Hintergrundbild für Radiosender" active="yes" />
+        <plugin name="chanorg" info="Kanäle sortieren" active="yes" />
+        <plugin name="vdrc" info="Linux Commander OSD" active="yes" />
+        <plugin name="sky" info="Mpeg2 Encoder Plugin" active="yes" />
+        <plugin name="newsticker" info="Newsticker" active="yes" />
+        <plugin name="status" info="Osd-Status" active="no" />
+        <plugin name="pim" info="PIM-Kalender Plugin" active="yes" />
+        <plugin name="channelscan" info="Sender Suchlauf" active="yes" />
+        <plugin name="taste" info="Sendungen-Keywords sperren" active="yes" />
+        <plugin name="mlcd" info="Siemens Multitainer LC-Display" active="no" />
+        <plugin name="text2skin" info="Skin-Plugin für VDR" active="yes" />
+        <plugin name="mp3" info="Spielt MP3-Dateien ab" active="yes" />
+        <plugin name="mplayer" info="Spielt diverse Videoformate" active="yes" />
+        <plugin name="streamdev-client" info="Streaming Client" active="no" />
+        <plugin name="sysinfo" info="Systeminformationen" active="yes" />
+        <plugin name="console" info="Textkonsole auf dem Fernseher" active="yes" />
+        <plugin name="tvtv" info="TvTv im OSD" active="yes" />
+        <plugin name="clock" info="Uhr" active="yes" />
+        <plugin name="skincurses" info="VDR-Skin für Konsole" active="yes" />
+        <plugin name="games" info="Verschiedene Spiele" active="yes" />
+        <plugin name="playlist" info="Verw.von Playlisten" active="yes" />
+        <plugin name="vcd" param="--vcd=/dev/dvd" info="VideoCD Plugin" active="yes" />
+        <plugin name="yaepg" info="Weitere EPG-Ansicht" active="yes" />
+        <plugin name="undelete" info="Wiederh. gelöschter Aufnahmen" active="yes" />
+        <plugin name="pilot" info="Zapping Co-Pilot" active="yes" />
+        <plugin name="director" info="Zeigt Multifeed Kanäle" active="yes" />
+        <plugin name="timeline" info="Zeitleiste" active="yes" />
+        <plugin name="wapd" info="wapd" active="yes" />
+    </plugins>
+    <menus>
+        <menu name="VDR-Einstellungen">
+            <menu name="OSD" help1="Einstellungen zum On Screen Display" help2="setup_osd.hlp" system="OSD" />
+            <menu name="EPG" help1="Einstellungen zum Electronic Prgramming Guide" help2="setup_epg.hlp" system="EPG" />
+            <menu name="DVB" help1="Einstellungen zur DVB-Karte" help2="setup_dvb.hlp" system="DVB" />
+            <menu name="LNB" help2="setup_lnb.hlp" system="LNB" />
+            <menu name="CICAM" help2="setup_cicam.hlp" system="CICAM" />
+            <menu name="Aufnahme" help2="setup_record.hlp" system="Record" />
+            <menu name="Wiedergabe" help2="setup_replay.hlp" system="Replay" />
+            <menu name="Sonstiges" help2="setup_misc.hlp" system="Misc" />
+            <menu name="Aktiviere Plugins" help1="Aktiviere und Deaktiviere Plugins" help2="setup_actplugins.hlp" system="ActPlugins" />
+            <menu name="Plugins-Setup" help1="Plugins konfigurieren" system="Plugins" />
+        </menu>
+        <menu name="Informationen zur Konfiguration">
+            <entry name="Konfig name" sysconfig="CONFIG_NAME" type="text" value="Lemmi-1" />
+            <entry name="Erzeugt am" sysconfig="CONFIG_CREATION_TIME" type="text" value="2007-01-09 00:00:00" />
+            <entry name="Zuletzt benutzt am" sysconfig="CONFIG_LAST_USAGE_TIME" type="text" value="2007-01-11 23:57:09" />
+        </menu>
+        <menu name="Datum und Zeit">
+            <entry name="Zeitzone" sysconfig="CONFIG_TIMEZONE" type="text" value="CET-1CDT,M3.5.0,M10.5.0" />
+            <entry name="Zeitserver" sysconfig="CONFIG_NETDATE_PARAM" type="text" value="ntp1.ptb.de" />
+        </menu>
+        <menu name="Video und LED">
+            <entry name="Video mode TV" sysconfig="CONFIG_VIDEO_TV_MODE" type="text" value="svideo" />
+            <entry name="Video mode VCR" sysconfig="CONFIG_VIDEO_VCR_MODE" type="text" value="svideo" />
+        </menu>
+        <menu name="System">
+            <entry name="Logging" sysconfig="CONFIG_LOG_KMSG" type="bool" value="no" />
+        </menu>
+        <menu name="Netzwerk" >
+            <entry name="Netzwerkmodus" sysconfig="CONFIG_NET_MODE" type="selection" value="ip">
+                <value>undef</value>
+                <value>dhcp</value>
+                <value>ip</value>
+                <value>off</value>
+            </entry>
+            <entry name="MAC Adresse" sysconfig="CONFIG_MAC" type="hex" value="E8:B5:5E:65:B9" />
+            <entry name="Hostname" sysconfig="CONFIG_HOSTNAME" type="text" value="TiniVDR" />
+            <entry name="IP-Adresse" sysconfig="CONFIG_IP" type="ip" value="192.168.1.200" />
+            <entry name="Mask" sysconfig="CONFIG_NETMASK" type="ip" value="192.168.1.1" />
+            <entry name="Gateway" sysconfig="CONFIG_GATEWAY" type="ip" value="192.168.1.1" />
+            <entry name="Nameserver" sysconfig="CONFIG_NAMESERVER" type="ip" value="192.168.1.1" />
+            <entry name="CONFIG_SEARCH" sysconfig="CONFIG_SEARCH" type="text" value="MiniVDR" />
+        </menu>
+        <menu name="Mail">
+            <entry name="Mailserver" sysconfig="CONFIG_MAIL_SERVER" type="text" value="MiniVDR" />
+            <entry name="Absender" sysconfig="CONFIG_MAIL_FROM" type="text" value="MiniVDR" />
+            <entry name="Antwort an" sysconfig="CONFIG_MAIL_REPLY" type="text" value="MiniVDR" />
+        </menu>
+        <menu name="VDR">
+            <entry name="USB Automount" sysconfig="CONFIG_USB_AUTO_MOUNT" type="text" value="" />
+            <entry name="Pfad zum VDR" sysconfig="CONFIG_VDR_PATH" type="text" value="MiniVDR" />
+            <entry name="Starter" sysconfig="CONFIG_VDR_STARTER" type="text" value="" />
+            <entry name="Aufnahmeverzeichniss" sysconfig="CONFIG_VDR_VIDEO_PATH" type="text" value="MiniVDR" />
+            <entry name="Pfad für das EPG" sysconfig="CONFIG_VDR_EPG_PATH" type="text" value="" />
+            <entry name="Logging" sysconfig="CONFIG_VDR_LOG" type="selection" value="none">
+                <value>file</value>
+                <value>net</value>
+                <value>both</value>
+                <value>none</value>
+            </entry>
+            <entry name="Größe des Puffers für logging" sysconfig="CONFIG_VDR_LOG_BUFSIZE" type="numtext" value="0" />
+            <entry name="Anzahl der log Dateien" sysconfig="CONFIG_VDR_N_LOG_BACKUP" type="numtext" value="3" />
+        </menu>
+        <menu name="BoxAmp">
+            <entry name="Pfad zu BoxAmp" sysconfig="CONFIG_BOXAMP_PATH" type="text" value="" />
+            <entry name="Pfad zu dem MP3s" sysconfig="CONFIG_BOXAMP_MP3_PATH" type="text" value="" />
+            <entry name="Zusätzliche Optionen" sysconfig="CONFIG_BOXAMP_OPTIONS" type="text" value="" />
+        </menu>
+        <menu name="Kindersicherung">
+            <entry name="Geheimcode (0000=AUS)" sysconfig="CHILDLOCK" type="numtext" setup="childlock" value="0000" />
+        </menu>
+    </menus>
+</setup>
diff -Naur -x svn setup/M7x0 config/open7x0-setup.xml setup-orig/M7x0 config/open7x0-setup.xml
--- setup/M7x0 config/open7x0-setup.xml	1970-01-01 01:00:00.000000000 +0100
+++ setup-orig/M7x0 config/open7x0-setup.xml	2008-09-18 17:39:34.000000000 +0200
@@ -0,0 +1,162 @@
+<!--
+-   (c) 2005,2006 by Ralf Dotzert
+-   Mini-VDR Configuration File
+-->
+<setup sysconfigPre="get-info --eval --config >/tmp/sysconfig.txt" sysconfigPost="eval `get-info --eval --file`; cat /tmp/sysconfig.txt >>$CONFFILE_CONFIG ; dump-fw-config" sysconfigFile="/tmp/sysconfig.txt" bootLinux="/sbin/reboot" VDRlibDirCMD="env | grep PLUGIN_PATH | cut -d= -f2">
+    <plugins sysconfig="CONFIG_VDR_PLUGINS">
+        <plugin name="epgsearch" info="EPG zeigt Wiederholungen" active="yes" />
+        <plugin name="solitaire" info="Kartenspiel Solitaire" active="yes" />
+        <plugin name="setup" info="Gigaset 7x0 Setup" active="yes" protected="yes" />
+        <plugin name="sleeptimer" info="Shutdown" active="yes" />
+        <plugin name="streamdev-server" info="Streaming Server" active="yes" />
+        <plugin name="osdteletext" info="Teletext Dekoder für das OSD" active="yes" />
+        <plugin name="admin" info="admin" active="no" />
+        <plugin name="fepg" info="fepg" active="no" />
+        <plugin name="filebrowser" info="filebrowser" active="no" />
+        <plugin name="skinsoppalusikka" info="skinsoppalusikka" active="no" />
+        <plugin name="sudoku" info="sudoku" active="no" />
+        <plugin name="surfer" info="surfer" active="no" />
+        <plugin name="femon" info="Anzeige Signalstärke/Qualität" active="yes" />
+        <plugin name="vdrcd" param="-c /dev/dvd -c /media/dvd -c /media/cdfs" info="AutoPlay Funktion" active="yes" />
+        <plugin name="autotimeredit" param="-f /etc/vdradmin/vdradmind.at -p" info="Autotimer-Edit" active="yes" />
+        <plugin name="prefermenu" info="Bevorzugte Kanäle" active="yes" />
+        <plugin name="osdpip" info="Bild in Bild" active="yes" />
+        <plugin name="image" param="-C /usr/bin/imageplugin.sh -m /usr/bin/mount.sh" info="Bildbetrachter Plugin" active="yes" />
+        <plugin name="osdimage" param="--directory /video0/media/photos/vdr" info="Bildern über OSD" active="yes" />
+        <plugin name="freecell" info="Das Kartenspiel Freecell" active="yes" />
+        <plugin name="osddemo" info="Demo On-Screen-Display" active="no" />
+        <plugin name="tvonscreen" param="-v /etc/vdradmin/vdradmind.at" info="EPG Programmzeitschrift" active="yes" />
+        <plugin name="hello" info="Ein &apos;hello&apos; Plugin" active="no" />
+        <plugin name="vdrc" info="Linux Commander OSD" active="yes" />
+        <plugin name="newsticker" info="Newsticker" active="yes" />
+        <plugin name="status" info="Osd-Status" active="no" />
+        <plugin name="pim" info="PIM-Kalender Plugin" active="yes" />
+        <plugin name="channelscan" info="Sender Suchlauf" active="yes" />
+        <plugin name="taste" info="Sendungen-Keywords sperren" active="yes" />
+        <plugin name="text2skin" info="Skin-Plugin für VDR" active="yes" />
+        <plugin name="mp3" info="Spielt MP3-Dateien ab" active="yes" />
+        <plugin name="streamdev-client" info="Streaming Client" active="no" />
+        <plugin name="sysinfo" info="Systeminformationen" active="yes" />
+        <plugin name="console" info="Textkonsole auf dem Fernseher" active="yes" />
+        <plugin name="tvtv" info="TvTv im OSD" active="yes" />
+        <plugin name="clock" info="Uhr" active="yes" />
+        <plugin name="skincurses" info="VDR-Skin für Konsole" active="yes" />
+        <plugin name="games" info="Verschiedene Spiele" active="yes" />
+        <plugin name="playlist" info="Verw.von Playlisten" active="yes" />
+        <plugin name="vcd" param="--vcd=/dev/dvd" info="VideoCD Plugin" active="yes" />
+        <plugin name="yaepg" info="Weitere EPG-Ansicht" active="yes" />
+        <plugin name="undelete" info="Wiederh. gelöschter Aufnahmen" active="yes" />
+        <plugin name="pilot" info="Zapping Co-Pilot" active="yes" />
+        <plugin name="director" info="Zeigt Multifeed Kanäle" active="yes" />
+        <plugin name="timeline" info="Zeitleiste" active="yes" />
+        <plugin name="wapd" info="wapd" active="yes" />
+    </plugins>
+    <menus>
+        <menu name="Informationen zur Konfiguration">
+            <entry name="Konfig name" sysconfig="CONFIG_NAME" type="text" value="" help1="Ein freier Text, der diese Konfiguration beschreibt."/>
+            <entry name="Zeitzone" sysconfig="CONFIG_TIMEZONE" type="text" value="CET-1CDT,M3.5.0,M10.5.0" help1="Standard Zeitzone, in der die Anwendungen laufen sollen."/>
+            <entry name="Zeitserver" sysconfig="CONFIG_NETDATE_PARAM" type="text" value="ntp1.ptb.de" help1="Ist dieser Parameter nicht leer, dann wird 'netdate' gestartet."/>
+            <entry name="Video mode TV" sysconfig="CONFIG_VIDEO_TV_MODE" type="selection" value="undef" help1="Video-Modus fuer den TV-Ausgang (SCART)">
+                <value>undef</value>
+                <value>svideo</value>
+                <value>fbas</value>
+            </entry>
+            <entry name="Video mode VCR" sysconfig="CONFIG_VIDEO_VCR_MODE" type="selection" value="undef" help1="Video-Modus fuer den VCR-Ausgang (SCART)">
+                <value>undef</value>
+                <value>svideo</value>
+                <value>fbas</value>
+            </entry>
+            <entry name="LED Helligkeit" sysconfig="CONFIG_LED_BRIGHTNESS" type="number" value="40" help1="Dieser Wert gibt die LED-Helligkeit der aeusseren beiden LEDs an." />
+            <entry name="System Logging" sysconfig="CONFIG_LOG_KMSG" type="bool" value="no" help1="Logging von /proc/kmsg via Port 12002 aktivieren?"/>
+            <entry name="USB Automount" sysconfig="CONFIG_USB_AUTO_MOUNT" type="text" value="" help1="Dieser Parameter legt die USB-Laufwerke fest, die automatisch gemountet werden."/>
+        </menu>
+        <menu name="Netzwerk" >
+            <entry name="Netzwerkmodus" sysconfig="CONFIG_NET_MODE" type="selection" value="undef" help1="Hiermit wird der Netzwerkmodus einegstellt.">
+                <value>undef</value>
+                <value>dhcp</value>
+                <value>ip</value>
+                <value>off</value>
+            </entry>
+            <entry name="MAC Adresse" sysconfig="CONFIG_MAC" type="hex" value="" />
+            <entry name="Hostname" sysconfig="CONFIG_HOSTNAME" type="text" value="" help1="Rechnername des Box (DNS)."/>
+            <entry name="IP-Adresse" sysconfig="CONFIG_IP" type="ip" value="" help1="IP-Adresse der Box."/>
+            <entry name="Mask" sysconfig="CONFIG_NETMASK" type="ip" value="" help1="Netzwerkmaske der Box"/>
+            <entry name="Gateway" sysconfig="CONFIG_GATEWAY" type="ip" value="" help1="Gateway Adresse"/>
+            <entry name="Nameserver" sysconfig="CONFIG_NAMESERVER" type="text" value="" help1="Eine mit Leerzeichen getrennte Liste der DNS-Server."/>
+            <entry name="Such-Postfixe" sysconfig="CONFIG_SEARCH" type="text" value="" help1="Eine mit Leerzeichen getrennte Liste der Such-Postfixe."/>
+        </menu>
+        <menu name="Mail">
+            <entry name="Mailserver" sysconfig="CONFIG_MAIL_SERVER" type="text" value="" help1="Dieser Parameter legt den SMTP-Server fest" />
+            <entry name="Absender" sysconfig="CONFIG_MAIL_FROM" type="text" value="" help1="Legt den Absender fest, wenn keiner übergeben wurde."/>
+            <entry name="Antwort an" sysconfig="CONFIG_MAIL_REPLY" type="text" value="" help1="Dieser Parameter legt die Antwortadresse (Reply-To:) fest."/>
+        </menu>
+        <menu name="Menüs">
+            <entry name="Startmenu" sysconfig="CONFIG_STARTMENU_MODE" type="selection" value="normal" help1="Anzeigeart des Startmenus." >
+                <value>normal</value>
+                <value>fast</value>
+                <value>off</value>	    
+	    </entry>
+            <entry name="Startmenu Timer" sysconfig="CONFIG_STARTMENU_WAIT" type="number" value="30" help1="Wartezeit in Sekunden an, bis die Standardkonfiguration startet" />
+            <entry name="Startmenu Timer Short" sysconfig="CONFIG_STARTMENU_SHORT_WAIT" type="number" value="5" help1="Kurze Wartezeit in Sekunden" />
+            <entry name="Haltmneu" sysconfig="CONFIG_HALTMENU_ENABLED" type="bool" value="yes" help1="Mit dieser Option wird das Shutdown-Menu aktiviert."/>
+            <entry name="Haltmenu Timer" sysconfig="CONFIG_HALTMENU_WAIT" type="number" value="5" help1="Wartezeit bis zum automatischn Shutdown" />
+            <entry name="Haltmenu Screensaver" sysconfig="CONFIG_MENU_HIDE" type="number" value="90" help1="Falls >0, dann werden die Menus durch einen schwarzen Bildschirm" />
+        </menu>
+        <menu name="VDR">
+	    <menu name="Aktiviere Plugins" help1="Aktiviere und Deaktiviere Plugins" help2="setup_actplugins.hlp" system="ActPlugins" />
+            <entry name="Aufnahmeverzeichniss" sysconfig="CONFIG_VDR_VIDEO_PATH" type="text" value="MiniVDR" />
+            <entry name="Pfad für das EPG" sysconfig="CONFIG_VDR_EPG_PATH" type="text" value="" />
+            <entry name="Anzahl der log Dateien" sysconfig="CONFIG_VDR_N_LOG_BACKUP" type="number" value="3" />
+            <entry name="Standart VDR Konfiguration" sysconfig="CONFIG_VDR_STANDARD" type="selection" value="vdr1" help1="Standart VDR Konfiguration" >
+                <value>vdr1</value>	    
+                <value>vdr2</value>	    
+                <value>vdr3</value>	    
+                <value>vdr4</value>	    
+	    </entry>
+            <entry name="Logging für VDR1" sysconfig="CONFIG_VDR1_LOG" type="selection" value="none">
+                <value>file</value>
+                <value>net</value>
+                <value>both</value>
+                <value>none</value>
+            </entry>
+	    <entry name="Name VDR2" sysconfig="CONFIG_VDR2_NAME" type="text" value="" />
+            <entry name="Pfad zum VDR2" sysconfig="CONFIG_VDR2_PATH" type="text" value="*" />
+            <entry name="Starter für VDR2" sysconfig="CONFIG_VDR2_STARTER" type="text" value="" />
+            <entry name="Config Pfad für VDR2" sysconfig="CONFIG_VDR2_CONFIG_PATH" type="text" value="*" />
+            <entry name="Logging für VDR2" sysconfig="CONFIG_VDR2_LOG" type="selection" value="none">
+                <value>file</value>
+                <value>net</value>
+                <value>both</value>
+                <value>none</value>
+            </entry>
+	    <entry name="Name VDR3" sysconfig="CONFIG_VDR3_NAME" type="text" value="" />
+            <entry name="Pfad zum VDR3" sysconfig="CONFIG_VDR3_PATH" type="text" value="*" />
+            <entry name="Starter für VDR3" sysconfig="CONFIG_VDR3_STARTER" type="text" value="" />
+            <entry name="Config Pfad für VDR3" sysconfig="CONFIG_VDR3_CONFIG_PATH" type="text" value="*" />
+            <entry name="Logging für VDR3" sysconfig="CONFIG_VDR3_LOG" type="selection" value="none">
+                <value>file</value>
+                <value>net</value>
+                <value>both</value>
+                <value>none</value>
+            </entry>
+	    <entry name="Name VDR4" sysconfig="CONFIG_VDR4_NAME" type="text" value="" />
+            <entry name="Pfad zum VDR4" sysconfig="CONFIG_VDR4_PATH" type="text" value="*" />
+            <entry name="Starter für VDR4" sysconfig="CONFIG_VDR4_STARTER" type="text" value="" />
+            <entry name="Config Pfad für VDR4" sysconfig="CONFIG_VDR4_CONFIG_PATH" type="text" value="*" />
+            <entry name="Logging für VDR4" sysconfig="CONFIG_VDR4_LOG" type="selection" value="none">
+                <value>file</value>
+                <value>net</value>
+                <value>both</value>
+                <value>none</value>
+            </entry>
+	</menu>
+        <menu name="BoxAmp">
+            <entry name="Pfad zu BoxAmp" sysconfig="CONFIG_BOXAMP_PATH" type="text" value="" />
+            <entry name="Pfad zu dem MP3s" sysconfig="CONFIG_BOXAMP_MP3_PATH" type="text" value="" />
+            <entry name="Zusätzliche Optionen" sysconfig="CONFIG_BOXAMP_OPTIONS" type="text" value="" />
+        </menu>
+        <menu name="Kindersicherung">
+            <entry name="Geheimcode (0000=AUS)" sysconfig="CHILDLOCK" type="numtext" setup="childlock" value="0000" />
+        </menu>
+    </menus>
+</setup>
diff -Naur -x svn setup/Makefile setup-orig/Makefile
--- setup/Makefile	2008-09-18 17:53:34.000000000 +0200
+++ setup-orig/Makefile	2008-09-18 17:39:34.000000000 +0200
@@ -1,12 +1,10 @@
 #
 # Makefile for a Video Disk Recorder plugin
 #
-# $Id: Makefile 43 2007-05-15 06:20:51Z hape $
+# $Id$
 
 # Debugging on/off 
-#OPEN7x0_Setup_DEBUG = 1
-#DEBUG = -DDEBUG 
-
+# OPEN7x0_Setup_DEBUG = 1
 
 # Strip debug symbols?  Set eg. to /bin/true if not
 STRIP = mips-linux-uclibc-strip
@@ -58,7 +56,7 @@
 #just for TESTING VDR SUBmenu
 INCLUDES += -I$(VDRDIR)
 
-DEFINES += $(DEBUG) -D_GNU_SOURCE -DPLUGIN_NAME_I18N='"$(PLUGIN)"' -DVDRVERSION="\"$(VDRVERSION)\"" 
+DEFINES += -D_GNU_SOURCE -DPLUGIN_NAME_I18N='"$(PLUGIN)"' -DVDRVERSION="\"$(VDRVERSION)\"" 
 
 
 ifdef BIGPATCH_INSTALLED
@@ -106,9 +104,8 @@
 	@cp -f $@ $(LIBDIR)/$@.$(APIVERSION)
 
 setupsystemmenu.cpp:  $(VDRDIR)/menu.c
-#	@echo "Extracting class definitions from VDR's menu.c for standard Setup"
-#	@chmod +x ./getVdrSetupMenu.pl
-#	@./getVdrSetupMenu.pl $(VDRDIR)/menu.c
+	@echo "Extracting class definitions from VDR's menu.c for standard Setup"
+	@./getVdrSetupMenu.pl $(VDRDIR)/menu.c
 
 	
 dist: clean
diff -Naur -x svn setup/menus.cpp setup-orig/menus.cpp
--- setup/menus.cpp	2008-09-18 17:44:47.000000000 +0200
+++ setup-orig/menus.cpp	2008-09-18 17:39:34.000000000 +0200
@@ -21,6 +21,7 @@
 #include "menus.h"
 #include "config.h"
 #include "sysconfig.h"
+#include "debug.h"
 #include "common.h"
 
 #define VALUETEXTMAXLEN 250
@@ -862,13 +863,13 @@
         }
         else
         {
-          debug("Can not resolve realpath of %s  errno=%d", channelFile, errno );
+          DEBUG4("%s Can not resolve realpath of %s  errno=%d\n", DBG_PREFIX, channelFile, errno );
           ok = false;
         }
    }
    else
    {
-     debug("Can not read directory:%s  errno=%d", channelDir, errno);
+     DEBUG4("%s Can not read directory:%s  errno=%d\n", DBG_PREFIX, channelDir, errno);
    }
    
    if( dirFP!=NULL ) closedir(dirFP);
@@ -900,7 +901,7 @@
  }
  else
  {
-   debug("UNAME failed, errno=%d", errno);
+   DEBUG3("%s UNAME failed, errno=%d\n", DBG_PREFIX, errno);
    ok=false;
  }
 
@@ -967,7 +968,7 @@
    else
    {
      ok=false;
-     debug("Can not read directory:%s  errno=%d", path, errno);
+     DEBUG4("%s Can not read directory:%s  errno=%d\n", DBG_PREFIX, path, errno);
    }
    if( modDir!=NULL ) closedir(modDir);
    if( path!=NULL) free(path);
@@ -1031,7 +1032,7 @@
       }
       else
       {
-        error("Error in XML File, Column=%d Row%d", node->Column(), node->Row());
+        DEBUG4("%s Error in XML File, Column=%d Row%d\n", DBG_PREFIX, node->Column (), node->Row());
         ok = false;
       }
 
diff -Naur -x svn setup/plugins.cpp setup-orig/plugins.cpp
--- setup/plugins.cpp	2008-09-18 17:44:47.000000000 +0200
+++ setup-orig/plugins.cpp	2008-09-18 17:39:34.000000000 +0200
@@ -14,7 +14,7 @@
 #include "vdr/plugin.h"
 #include "plugins.h"
 #include "util.h"
-#include "common.h"
+#include "debug.h"
 
 
 /**
@@ -25,7 +25,6 @@
 {
     _activePlugins = NULL;
     _sysconfigName = Util::Strdupnew("PLUGINLIST");
-    _protectedList = Util::Strdupnew("");
 }
 
 
@@ -37,7 +36,6 @@
 {
     delete [] _activePlugins;
     delete [] _sysconfigName;
-    delete [] _protectedList;
 }
 
 //---------------------------------------------
@@ -61,7 +59,6 @@
     p->SetActive(flag);
     p->SetProtect(protect);
     p->SetInSystem(inSystem);
-    debug("AddPlugin: [%s] act [%d] inSystem [%d] Protect[%d]\n", name,flag, inSystem,protect);	
     Add(p);
 }
 
@@ -130,8 +127,8 @@
 //----------------------------------------------------------------------------
 //----------------------------------------------------------------------------
 //----------------------------------------------------------------------------
-const char * Plugin::_trueStr =STR_YES;
-const char * Plugin::_falseStr = STR_NO;
+const char * Plugin::_trueStr ="yes";
+const char * Plugin::_falseStr = "no";
 
 /**
  * Constructor 
@@ -225,7 +222,7 @@
     int nr = Count();
     for(int i=0; i<nr; i++)
     {
-        debug("Plugin[%d] ", i);
+        printf("Plugin[%d] ", i);
         Get(i)->Print();
     }
 }
@@ -236,7 +233,7 @@
  */
 void  Plugin::Print()
 {
-    debug("Name=%s  Parameter=%s Info=%s Active=%d inSystem=%d\n",
+    dsyslog("Name=%s  Parameter=%s Info=%s Active=%d inSystem=%d\n",
             _name, _parameter, _info, _active, _inSystem);
 }
 void Plugins::SetSysconfigName( const char * name )
@@ -250,27 +247,9 @@
  return(_sysconfigName);
 }
 
-void Plugins::SetProtectedList( const char * name )
-{
-  delete [] _protectedList;
-  debug("Protected plugins [%s]",name);  
-  if ( name != NULL ) {
-    _protectedList =  Util::Strdupnew(name);
-  }else {
-    _protectedList = Util::Strdupnew("");
-  }
-}
 
-const char* Plugins::GetProtectedList()
-{
-  return _protectedList;  
-}
 
-bool Plugins::checkProtected(const char* pluginName)
-{
-  debug ("Check protected [%s] [%s] [%d]",pluginName,_protectedList,(_protectedList != NULL) && (strstr(_protectedList,pluginName) != NULL ));
-  return (_protectedList != NULL) && (strstr(_protectedList,pluginName) != NULL );
-}
+
 /**
  * retrieves the name of the plugin
  * @return the name
@@ -312,7 +291,7 @@
  */
 void Plugin::Plugin::SetActive( int flag )
 {
- _active=flag; 
+ _active=flag;
 }
 
 /**
@@ -366,13 +345,11 @@
 
 void Plugin::Plugin::SetProtect( bool protect )
 {
-   debug("SetProtect [%s] [%d]",GetName(),protect);
   _protect = protect;
 }
 
 bool Plugin::Plugin::GetProtect( )
 {
-  debug("GetProtect [%s] [%d]",GetName(),_protect);
   return(_protect);
 }
 
@@ -397,9 +374,9 @@
     {
       const char *p = plugin->GetParameter();
       if( p == NULL)
-        sprintf(buf, "%s -P%s ", buf,  plugin->GetName());
+        sprintf(buf, "%s \\\"-P%s \\\"", buf,  plugin->GetName());
       else
-        sprintf(buf, "%s -P%s %s", buf, plugin->GetName(), p);
+        sprintf(buf, "%s \\\"-P%s %s\\\"", buf, plugin->GetName(), p);
     }
   } 
   //HaPe: Add " around the string
@@ -427,7 +404,6 @@
   if( p != NULL)
   { 
     p->SetInSystem(true);
-    debug("In System plugin [%s]\n",pluginName);
   }
   else
   {
@@ -449,9 +425,8 @@
  
   for(int i=0; i<nr && p==NULL; i++)
   { 
-   if( strcmp(Get(i)->GetName(),name)==0) {
+   if( strcmp(Get(i)->GetName(),name)==0)
       p= Get(i);
-   }
   }
   return(p);
 }
diff -Naur -x svn setup/plugins.h setup-orig/plugins.h
--- setup/plugins.h	2008-09-18 17:44:47.000000000 +0200
+++ setup-orig/plugins.h	2008-09-18 17:39:34.000000000 +0200
@@ -54,7 +54,6 @@
 {
 private:
 const char  *_sysconfigName;
-const char  *_protectedList;
 char        *_activePlugins;
 public:
    
@@ -68,15 +67,10 @@
     void SetSysconfigName(const char *name);
     const char *GetSysconfigName();
     Plugin *GetPluginByName(const char*name);
-    
-    void SetProtectedList(const char *name);  
-    const char *GetProtectedList();     
-    bool checkProtected(const char* pluginName);  
-    
+
     void MovePlugin(int index, int toindex, enum Where);
     void Sort( enum Order order);
     const char *GetActivePlugins();
-    
     void SetLibDirPlugin(const char* pluginName);
     void Print();
 
diff -Naur -x svn setup/setup.cpp setup-orig/setup.cpp
--- setup/setup.cpp	2008-09-18 17:44:47.000000000 +0200
+++ setup-orig/setup.cpp	2008-09-18 17:39:34.000000000 +0200
@@ -15,7 +15,7 @@
 //#include "setupsetup.h"
 #include "i18n.h"
 
-static const char *VERSION        = "0.0.2";
+static const char *VERSION        = "0.0.1";
 static const char *DESCRIPTION    = "open7x0-Setup (based on Minivdr-Setup)";
 static const char *MAINMENUENTRY  = "Gigaset 7x0 Firmware Setup";
 
diff -Naur -x svn setup/setupmenu.cpp setup-orig/setupmenu.cpp
--- setup/setupmenu.cpp	2008-09-18 17:44:47.000000000 +0200
+++ setup-orig/setupmenu.cpp	2008-09-18 17:39:34.000000000 +0200
@@ -16,8 +16,9 @@
 #include "setupmenu.h"
 #include "config.h"
 #include "plugins.h"
+#include "debug.h"
 #include "i18n.h"
-#include "common.h"
+//#include "setupsetup.h"
 #include "setupsystemmenu.h"
 
 
@@ -54,7 +55,6 @@
   Clear();
   for(int i=0; i<_plugins->GetNr(); i++)
   {
-debug("===> %s",_plugins->Get(i)->GetInfo());
     if(_plugins->Get(i)->GetProtect())
     {
         char *title = NULL;
@@ -64,7 +64,7 @@
         free(title);
     }
     else
-       Add(new cMenuEditBoolItem(_plugins->Get(i)->GetInfo(), _plugins->Get(i)->GetActiveRef(), tr(STR_NO), tr(STR_YES)));
+       Add(new cMenuEditBoolItem(_plugins->Get(i)->GetInfo(), _plugins->Get(i)->GetActiveRef(), tr("no"), tr("yes")));
   }
   if( current == -1 && _plugins->GetNr() >0)
     current=0;
@@ -136,7 +136,6 @@
 
 void cSetupPluginMenu::setHelp()
 {
-/*JMG
   char *tmp=NULL;
   int  current = Current();
   if(current >= 0 )
@@ -145,7 +144,7 @@
     SetStatus(tmp);
     free(tmp);
   }
-*/
+
 }
 
 
@@ -178,11 +177,10 @@
     _editParameter[sizeof(_editParameter)]='\0';
   }
   Add(new cMenuEditStrItem(tr("Plugin-Parameter"), _editParameter, sizeof(_editParameter), ALLOW_ALL_PARAM_CHARS));
-/*JMG
+
   asprintf(&tmp, "%s: %s", tr("Plugin"), _plugin->GetName());
   SetStatus(tmp);
   free(tmp);
-  */
   Display();
 }
 
@@ -215,7 +213,7 @@
 //################################################################################################
 
 /** Removed as we du not uses the submenupatch
-static char *ALLOW_ALL_CHARS =" aäbcdefghijklmnoöpqrsßtuüvwxyz0123456789-.\"";
+static char *ALLOW_ALL_CHARS =" aäbcdefghijklmnoöpqrsßtuüvwxyz0123456789-.";
 cSetupVdrMenu::cSetupVdrMenu(const char *title): cOsdMenu(title, 25)
 {
   _startIndex           =0;
@@ -524,14 +522,14 @@
       switch( e->GetType() )
       {
         case Util::BOOL:
-          Add(new cMenuEditBoolItem(nohk(e->GetName()), e->GetValueBoolRef(), tr(STR_NO), tr(STR_YES)));
+          Add(new cMenuEditBoolItem(nohk(e->GetName()), e->GetValueBoolRef(), tr("no"), tr("yes")));
              break;
         case Util::NUMBER:
           Add(new cMenuEditIntItem(nohk(e->GetName()), e->GetValueNumberRef(), 0, 999999999));
              break;
         case Util::TEXT:
           Add(new cMenuEditStrItem(nohk((char*)e->GetName()), (char*)e->GetValue(), e->GetValueTextMaxLen(),
-                                      " aäbcdefghijklmnoöpqrsßtuüvwxyz0123456789-+.#~_/:\"\'"));
+                                      " aäbcdefghijklmnoöpqrsßtuüvwxyz0123456789-+.#~_/:"));
              break;
          case Util::NUMBER_TEXT:
            Add(new cMenuEditStrItem(nohk((char*)e->GetName()), (char*)e->GetValue(), e->GetValueTextMaxLen(),
@@ -626,10 +624,8 @@
                         _editItem=false;
                       }
                       break;
-              case kRed:                        
-                        break;
-              case kGreen:
-					{
+              case kRed:
+                        {
                           char *help = NULL;
                           if(n->GetType() == MenuNode::ENTRY)
                             help = GetLongHelp(n->GetMenuEntry()->GetHelp2());
@@ -645,6 +641,8 @@
 
                         }
                         break;
+              case kGreen:
+                        break;
               case kYellow:
                         break;
               case kBlue:
@@ -698,14 +696,12 @@
     SetStatus(tmp);
     free(tmp);
     status=system(cmd);
-    if( status == -1 ) {
-      debug("fork of command %s failed", cmd);
-    }
+    if( status == -1 )
+      DEBUG3("%s: fork of command %s failed\n", DBG_PREFIX, cmd);
     else
     {
-      if( WEXITSTATUS(status) != 0) {
-        debug("executing of command %s returned=%d", cmd, WEXITSTATUS(status));
-      }
+      if( WEXITSTATUS(status) != 0)
+        DEBUG4("%s: executing of command %s returned=%d\n", DBG_PREFIX, cmd, WEXITSTATUS(status));
     }
   }
 }
@@ -722,40 +718,16 @@
 cSetupMenu::cSetupMenu() : cOsdMenu(tr("Gigaset Firmware setup - open7x0.org (based on setup plugin by minivdr.de)"))
 {
    char *configFile=NULL;
-   char *langs=NULL,*lang=NULL;
-   int loaded=0;
    SetCols(20);
 	
-   //Get languages
-   langs  = Util::Strdupnew(I18nLanguageCode(Setup.OSDLanguage));      
-   lang = strtok(langs,",");
-   while (lang != NULL) {
-	  asprintf(&configFile, "%s/setup/open7x0-setup_%s.xml", cPlugin::ConfigDirectory(),lang);
-	  debug("cSetupMenu.constr: Load config file %s", configFile);
-	  _config = new Config( configFile );
-	  free(configFile);
-	  if ( _config!= NULL && _config->LoadFile() == true ) {
-		loaded=1;
-		break;
-	  }else if (_config != NULL) {
-		debug("cSetupMenu.constr: error loading config file");
-		delete _config;
-	  }
-      lang = strtok(NULL,",");
-   }
-   if ( langs ) free(langs);
-   
-   //Default language
-   if ( !loaded ) {
-      asprintf(&configFile, "%s/setup/open7x0-setup.xml", cPlugin::ConfigDirectory());
-	  debug("cSetupMenu.constr: Load config file %s", configFile);
-	  _config = new Config( configFile );
-	  free(configFile);
-	  if ( _config!= NULL && _config->LoadFile() == true ) {
-		loaded=1;		
-	  }
-   }   
-			
+	//Load config file
+   asprintf(&configFile, "%s/setup/open7x0-setup.xml", cPlugin::ConfigDirectory());
+
+   debug("cSetupMenu.constr: Load config file %s", configFile);
+
+  _config    = new Config( configFile );
+   free(configFile);
+
 	//Check for child lock
   _number    = 0;
   _error     = false;
@@ -763,7 +735,7 @@
   _childLockEntered = NULL;
   _childLockEnteredHidden = NULL;
   
-  if( loaded )
+  if( _config!= NULL && _config->LoadFile() == true)
   {
     if(_config->GetChildLock() != NULL && strcmp(_config->GetChildLock(), "0000") != 0)
     {//Ask for childlock code
@@ -881,7 +853,7 @@
                     else
                       if(node->GetType() == MenuNode::MENUSYSTEM )
                       {
-                        cOsdMenu * menu =cSetupSystemMenu::GetSystemMenu (node->GetMenu()->GetSystem(), _config);
+                        cOsdMenu * menu =cSetupSystemMenu::GetSystemMenu (hk(node->GetMenu()->GetSystem()), _config);
                         if(menu != NULL)
                           AddSubMenu(menu);
                       }
@@ -1023,19 +995,17 @@
       long len = ftell(fp);
       fseek(fp, 0L, SEEK_SET);
       int byteRead=0;
-      if( (result=(char*) malloc(len+1)) == NULL){
-        debug("can not allocate buffer for Helpfile: %s", helpfile );
-      }else
-        if( (byteRead=fread(result, 1, len, fp)) == -1){
-          debug("can not read file: Helpfile: %s", helpfile);
-	}
+      if( (result=(char*) malloc(len+1)) == NULL)
+        DEBUG3("%s: can not allocate buffer for Helpfile: %s\n", DBG_PREFIX, helpfile );
+      else
+        if( (byteRead=fread(result, 1, len, fp)) == -1)
+          DEBUG3("%s: can not read file: Helpfile: %s\n", DBG_PREFIX, helpfile);
       else
         result[byteRead] = '\0';
       fclose(fp);
     }
-    else{
-      debug("can not open Helpfile: %s, %s", helpfile, strerror(errno));
-    }
+    else
+      DEBUG4("%s: can not open Helpfile: %s, %s\n", DBG_PREFIX, helpfile, strerror(errno));
     free(helpfile);
   }
   return(result);
diff -Naur -x svn setup/setupsetup.cpp setup-orig/setupsetup.cpp
--- setup/setupsetup.cpp	2008-09-18 17:44:47.000000000 +0200
+++ setup-orig/setupsetup.cpp	2008-09-18 17:39:34.000000000 +0200
@@ -15,7 +15,7 @@
 #include <string.h>
 
 #include "setupsetup.h"
-
+#include "debug.h"
 
 //*****************************************************************************
 // Setup Configuration
diff -Naur -x svn setup/setupsystemmenu.cpp setup-orig/setupsystemmenu.cpp
--- setup/setupsystemmenu.cpp	2008-09-18 17:44:47.000000000 +0200
+++ setup-orig/setupsystemmenu.cpp	2008-09-18 17:39:34.000000000 +0200
@@ -8,24 +8,20 @@
 
 // Headers included by <VDR>/menu.c
 #include "menu.h"
-#include "iconpatch.h"
 #include <ctype.h>
 #include <limits.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <math.h>
 #include "channels.h"
 #include "config.h"
 #include "cutter.h"
 #include "eitscan.h"
-#include "epgmode.h"
 #include "i18n.h"
 #include "interface.h"
 #include "plugin.h"
 #include "recording.h"
 #include "remote.h"
-#include "shutdown.h"
 #include "sources.h"
 #include "status.h"
 #include "themes.h"
@@ -36,7 +32,7 @@
 // local headers needed
 #include "setupsystemmenu.h"
 #include "setupmenu.h"
-#include "common.h"
+#include "debug.h"
 // ************************************
 // SetupClasses defined in <VDR>/menu.c
 // ************************************
@@ -71,7 +67,6 @@
 private:
   int originalNumLanguages;
   int numLanguages;
-  const char *epgModeStr[5];
   void Setup(void);
 public:
   cMenuSetupEPG(void);
@@ -84,7 +79,7 @@
   int originalNumAudioLanguages;
   int numAudioLanguages;
   void Setup(void);
-  const char *videoDisplayFormatTexts[2];
+  const char *videoDisplayFormatTexts[3];
   const char *videoFormatTexts[3];
   const char *updateChannelsTexts[6];
 public:
@@ -113,8 +108,6 @@
 
 
 class cMenuSetupRecord : public cMenuSetupBase {
-private:
-  const char *RecordingsSortModeTexts[MAXSORTMODES];
 public:
   cMenuSetupRecord(void);
   };
@@ -129,8 +122,6 @@
 
 
 class cMenuSetupMisc : public cMenuSetupBase {
-private:
-  const char *channelFilterTexts[3];
 public:
   cMenuSetupMisc(void);
   };
@@ -191,7 +182,7 @@
       if(strcmp(sysCommand, "Plugins") == 0)
          menu = new cMenuSetupPlugins;
    else
-      debug("<menu System=%s> not known", sysCommand);
+      DEBUG3("%s: <menu System=%s> not known\n", DBG_PREFIX, sysCommand);
   }
   return (menu);
 }
diff -Naur -x svn setup/sysconfig.cpp setup-orig/sysconfig.cpp
--- setup/sysconfig.cpp	2008-09-18 17:44:47.000000000 +0200
+++ setup-orig/sysconfig.cpp	2008-09-18 17:39:34.000000000 +0200
@@ -15,6 +15,7 @@
 #include "vdr/plugin.h"
 #include "vdr/tools.h"
 #include "sysconfig.h"
+#include "debug.h"
 #include "util.h"
 #include "common.h"
 
@@ -116,10 +117,10 @@
    destroy();
    _filename = Util::Strdupnew(fname);
    FILE * fp = fopen(fname, "r");
-	debug("Load sysconf:[%s]",fname);
+	debug("Load sysconf:%s",fname);
 	
    if( fp != NULL)
-   {     
+   {
      while( (line = readLine(fp)) != NULL)
      {
        addLine(line);
@@ -130,7 +131,7 @@
    }
    else
    {
-     error("Could not read sysconfigFile: [%s], errno=%d", fname, errno);
+     DEBUG4("%s: Could not read file: %s, errno=%d\n", DBG_PREFIX, _filename, errno);
    }
 
    return(result);
@@ -143,7 +144,7 @@
 bool Sysconfig::SaveFile( )
 {
   bool result=true;
-  debug("Saving sysconf: [%s]",_filename);
+  debug("Save sysconf:%s",_filename);
    FILE * fp = fopen(_filename, "w");
    if(fp != NULL)
    {
@@ -151,12 +152,11 @@
      {
        		fprintf(fp, "%s=%s\n", _variables[i].GetName(), _variables[i].GetValue());
      }
-     isyslog("Saved file [%s]",_filename);
      fclose(fp);
    }
    else
     {
-     debug("Could not write file: [%s], errno=%d", _filename, errno);
+     DEBUG4("%s: Could not write file: %s, errno=%d\n", DBG_PREFIX, _filename, errno);
      result=false;
     }
   return(result);
diff -Naur -x svn setup/util.cpp setup-orig/util.cpp
--- setup/util.cpp	2008-09-18 17:44:47.000000000 +0200
+++ setup-orig/util.cpp	2008-09-18 17:39:34.000000000 +0200
@@ -13,7 +13,7 @@
 #include <string.h>
 #include <ctype.h>
 #include "util.h"
-#include "common.h"
+#include "debug.h"
 
 #define NUL '\0'
 
@@ -38,8 +38,8 @@
 
 static const char *trueStr="true";
 static const char *falseStr="false";
-static const char *yesStr=STR_YES;
-static const char *noStr=STR_NO;
+static const char *yesStr="yes";
+static const char *noStr="no";
 
 
 char * Util::Strdupnew( const char  *str )
@@ -101,7 +101,7 @@
     ok = false;
 
   if(! ok)
-    debug("Illegal Bool value %s found", string);
+    DEBUG3("%s: Illegal Bool value %s found\n", DBG_PREFIX, string);
     
   return(ok);
 }
@@ -164,7 +164,7 @@
     ok = false;
 
   if(! ok)
-    error("Illegal Type value %s found", string);
+    DEBUG3("%s: Illegal Type value %s found\n", DBG_PREFIX, string);
     
   return(ok);
 }
@@ -271,7 +271,7 @@
 
   if( realpath(channelFile, resolvedPath) == NULL)
   {
-    debug("Can not resolve realpath of %s  errno=%d", channelFile, errno );
+    DEBUG4("%s Can not resolve realpath of %s  errno=%d\n", DBG_PREFIX, channelFile, errno );
   }
   else
   {
@@ -281,7 +281,7 @@
       if( unlink(channelFile) == 0 )
       {
         if( symlink(newChannelFile, channelFile)!= 0){
-          debug("Can not link File %s  to %s errno=%d",  tmp, channelFile, errno);
+          DEBUG5("%s Can not link File %s  to %s errno=%d\n",DBG_PREFIX,  tmp, channelFile, errno);
         }
         TimerList timers;
         timers.SaveTimer();
@@ -290,7 +290,7 @@
         timers.RestoreTimer();
       }
       else{
-        debug("Can not unlink File %s  errno=%d", channelFile, errno);
+        DEBUG4("%s Can not unlink File %s  errno=%d\n",DBG_PREFIX, channelFile, errno);
       }
     }
   }
