--- busybox-1.14.3/util-linux/mount.c.orig	2009-08-12 00:11:27.000000000 +0200
+++ busybox-1.14.3/util-linux/mount.c	2009-08-12 00:13:19.000000000 +0200
@@ -1605,7 +1605,6 @@
 	 && (mp->mnt_fsname[0] == '/' || mp->mnt_fsname[0] == '\\')
 	 && mp->mnt_fsname[0] == mp->mnt_fsname[1]
 	) {
-#if 0 /* reported to break things */
 		len_and_sockaddr *lsa;
 		char *ip, *dotted;
 		char *s;
@@ -1626,25 +1625,24 @@
 
 		// Insert ip=... option into string flags.
 		dotted = xmalloc_sockaddr2dotted_noport(&lsa->u.sa);
-		ip = xasprintf("ip=%s", dotted);
+		ip = xasprintf("ip=%s,unc=\\\\%s%s", dotted, dotted, s);
 		parse_mount_options(ip, &filteropts);
 
 		// Compose new unc '\\server-ip\share'
 		// (s => slash after hostname)
 		mp->mnt_fsname = xasprintf("\\\\%s%s", dotted, s);
-#endif
+
 		// Lock is required [why?]
 		vfsflags |= MS_MANDLOCK;
 		mp->mnt_type = (char*)"cifs";
 		rc = mount_it_now(mp, vfsflags, filteropts);
-#if 0
+
 		if (ENABLE_FEATURE_CLEAN_UP) {
 			free(mp->mnt_fsname);
 			free(ip);
 			free(dotted);
 			free(lsa);
 		}
-#endif
 		goto report_error;
 	}
 
