diff -Naur wapd-0.8-orig/i18n.c wapd-0.8/i18n.c
--- wapd-0.8-orig/i18n.c	2005-12-24 14:23:36.000000000 +0100
+++ wapd-0.8/i18n.c	2007-06-05 19:47:41.000000000 +0200
@@ -23,7 +23,7 @@
     "", // TODO
     "VDR:n kauko-ohjaus (WAP)",
     "", // TODO
-    "", // TODO
+    "Control remoto por WAP",
     "", // TODO
     "", // TODO
     "", // TODO
@@ -44,7 +44,7 @@
     "", // TODO
     "Tallenna",
     "", // TODO
-    "", // TODO
+    "Guardar",
     "", // TODO
     "", // TODO
     "", // TODO
@@ -65,7 +65,7 @@
     "", // TODO
     "ei nimeä",
     "", // TODO
-    "", // TODO
+    "sin título",
     "", // TODO
     "", // TODO
     "", // TODO
@@ -86,7 +86,7 @@
     "", // TODO
     "ei alaotsikkoa",
     "", // TODO
-    "", // TODO
+    "sin subtítulo",
     "", // TODO
     "", // TODO
     "", // TODO
@@ -107,7 +107,7 @@
     "", // TODO
     "ei kuvausta",
     "", // TODO
-    "", // TODO
+    "sin descripción",
     "", // TODO
     "", // TODO
     "", // TODO
@@ -128,7 +128,7 @@
     "", // TODO
     "Yhteenvetoa ei saatavilla.",
     "", // TODO
-    "", // TODO
+    "Ninguna descripción disponible.",
     "", // TODO
     "", // TODO
     "", // TODO
@@ -149,7 +149,7 @@
     "", // TODO
     "Virhe tallenteen \"%s\" ID:ssä.",
     "", // TODO
-    "", // TODO
+    "Error grabando ID \"%s\".",
     "", // TODO
     "", // TODO
     "", // TODO
@@ -170,7 +170,7 @@
     "", // TODO
     "Tallenteen ID puuttuu.",
     "", // TODO
-    "", // TODO
+    "Falta id. grabación.",
     "", // TODO
     "", // TODO
     "", // TODO
@@ -191,7 +191,7 @@
     "", // TODO
     "Tapahtumaa \"%s\" ei löydy.",
     "", // TODO
-    "", // TODO
+    "Evento \"%s\" no encontrado.",
     "", // TODO
     "", // TODO
     "", // TODO
@@ -212,7 +212,7 @@
     "", // TODO
     "Virhe tapahtuman \"%s\" ID:ssä.",
     "", // TODO
-    "", // TODO
+    "Error en ID evento \"%s\".",
     "", // TODO
     "", // TODO
     "", // TODO
@@ -233,7 +233,7 @@
     "", // TODO
     "Ohjelmistoa ei saatavilla.",
     "", // TODO
-    "", // TODO
+    "Ningún programa disponible.",
     "", // TODO
     "", // TODO
     "", // TODO
@@ -254,7 +254,7 @@
     "", // TODO
     "Kanavaa \"%s\" ei määritelty.",
     "", // TODO
-    "", // TODO
+    "Canal \"%d\" no definido.",
     "", // TODO
     "", // TODO
     "", // TODO
@@ -275,7 +275,7 @@
     "", // TODO
     "EPG-tieto ei saatavilla.",
     "", // TODO
-    "", // TODO
+    "No puedo obtener datos EPG.",
     "", // TODO
     "", // TODO
     "", // TODO
@@ -296,7 +296,7 @@
     "", // TODO
     "Kanavaa \"%s\" ei löydy.",
     "", // TODO
-    "", // TODO
+    "Canal \"%d\" no encontrado.",
     "", // TODO
     "", // TODO
     "", // TODO
@@ -317,7 +317,7 @@
     "", // TODO
     "Kanavia ei määritelty.",
     "", // TODO
-    "", // TODO
+    "Ningún canal definido.",
     "", // TODO
     "", // TODO
     "", // TODO
@@ -338,7 +338,7 @@
     "", // TODO
     "Ajastinta \"%s\" ei löydy.",
     "", // TODO
-    "", // TODO
+    "Programador \"%d\" no encontrado.",
     "", // TODO
     "", // TODO
     "", // TODO
@@ -359,7 +359,7 @@
     "", // TODO
     "Ajastimia ei määritelty.",
     "", // TODO
-    "", // TODO
+    "Ninguna programación definida.",
     "", // TODO
     "", // TODO
     "", // TODO
@@ -380,7 +380,7 @@
     "", // TODO
     "Tallenteita ei ole saatavilla.",
     "", // TODO
-    "", // TODO
+    "Ninguna grabación disponible.",
     "", // TODO
     "", // TODO
     "", // TODO
@@ -401,7 +401,7 @@
     "", // TODO
     "Parametrit",
     "", // TODO
-    "", // TODO
+    "Parámetros",
     "", // TODO
     "", // TODO
     "", // TODO
diff -Naur wapd-0.8-orig/Makefile wapd-0.8/Makefile
--- wapd-0.8-orig/Makefile	2005-01-25 21:33:05.000000000 +0100
+++ wapd-0.8/Makefile	2007-07-30 11:07:28.000000000 +0200
@@ -16,13 +16,13 @@
 ### The C++ compiler and options:
 
 CXX      ?= g++
-CXXFLAGS ?= -O2 -Wall -Woverloaded-virtual -ggdb
+CXXFLAGS ?= -fPIC -O2 -Wall -Woverloaded-virtual -ggdb
 
 ### The directory environment:
 
-DVBDIR = ../../../../DVB
-VDRDIR = ../../..
-LIBDIR = ../../lib
+DVBDIR ?= ../../..
+VDRDIR ?= ../../..
+LIBDIR ?= ../../lib
 TMPDIR = /tmp
 
 ### Allow user defined options to overwrite defaults:
@@ -31,7 +31,7 @@
 
 ### The version number of VDR (taken from VDR's "config.h"):
 
-VDRVERSION = $(shell grep 'define VDRVERSION ' $(VDRDIR)/config.h | awk '{ print $$3 }' | sed -e 's/"//g')
+APIVERSION = $(shell grep 'define APIVERSION ' $(VDRDIR)/config.h | awk '{ print $$3 }' | sed -e 's/"//g')
 
 ### The name of the distribution archive:
 
@@ -40,7 +40,7 @@
 
 ### Includes and Defines (add further entries here):
 
-INCLUDES += -I$(VDRDIR)/include -I$(DVBDIR)/include
+INCLUDES += -I$(VDRDIR)/include -I$(DVBDIR)
 
 DEFINES += -D_GNU_SOURCE -DPLUGIN_NAME_I18N='"$(PLUGIN)"'
 
@@ -57,7 +57,7 @@
 
 # Dependencies:
 
-MAKEDEP = g++ -MM -MG
+MAKEDEP = $(CXX) -MM -MG
 DEPFILE = .dependencies
 $(DEPFILE): Makefile
 	@$(MAKEDEP) $(DEFINES) $(INCLUDES) $(OBJS:%.o=%.c) > $@
@@ -70,7 +70,7 @@
 
 libvdr-$(PLUGIN).so: $(OBJS)
 	$(CXX) $(CXXFLAGS) -shared $(OBJS) $(LIBS) -o $@
-	@cp $@ $(LIBDIR)/$@.$(VDRVERSION)
+	@cp -f $@ $(LIBDIR)/$@.$(APIVERSION)
 
 wappasswd: wappasswd.c
 	$(CXX) $(CXXFLAGS) $(LIBS) -o wappasswd wappasswd.c
diff -Naur wapd-0.8-orig/README wapd-0.8/README
--- wapd-0.8-orig/README	2006-01-08 20:45:34.000000000 +0100
+++ wapd-0.8/README	2006-02-26 17:10:49.000000000 +0100
@@ -46,7 +46,7 @@
 Files:
 ------------------------------------------------------------------------------
 
-The plugin needs 2 files under /video/plugins (default):
+The plugin needs 2 files under /video/plugins/wapd (default):
 
 'waphosts' controls host based access and is similar to svdrphosts.conf
 
diff -Naur wapd-0.8-orig/server.c wapd-0.8/server.c
--- wapd-0.8-orig/server.c	2005-12-24 15:19:20.000000000 +0100
+++ wapd-0.8/server.c	2007-06-05 19:47:41.000000000 +0200
@@ -694,7 +694,7 @@
            int offset = strtol(getopt, NULL, 10);
            int i = 0;
            const cEvent *PresentEvent =
-              Schedule->GetPresentEvent(channel->Number() == cDevice::CurrentChannel());
+              Schedule->GetPresentEvent();
            time_t now = time(NULL) - Setup.EPGLinger * 60;
            for (const cEvent *Event = Schedule->Events()->First();
                 Event;
diff -Naur wapd-0.8-orig/wapd.c wapd-0.8/wapd.c
--- wapd-0.8-orig/wapd.c	2005-12-24 13:09:08.000000000 +0100
+++ wapd-0.8/wapd.c	2006-02-26 17:10:50.000000000 +0100
@@ -80,8 +80,8 @@
 {
   // Start any background activities the plugin shall perform.
   RegisterI18n(Phrases);
-  if (WapHosts.Load(AddDirectory(ConfigDirectory(), "waphosts"), true)
-     && WapAccess.Load(AddDirectory(ConfigDirectory(), "wapaccess"), true)
+  if (WapHosts.Load(AddDirectory(ConfigDirectory("wapd"), "waphosts"), true)
+     && WapAccess.Load(AddDirectory(ConfigDirectory("wapd"), "wapaccess"), true)
      && (server = new cWapServer(tcpport)))
      return true;
   return false;
