diff -Naur vdr-m7x0-orig/channels.c vdr-m7x0/channels.c
--- vdr-m7x0-orig/channels.c	2007-06-30 10:22:11.000000000 +0200
+++ vdr-m7x0/channels.c	2007-07-01 20:56:18.000000000 +0200
@@ -861,6 +861,15 @@
     }
   };
 
+bool cChannel::Filtered(void)
+{
+  switch(Setup.ChannelFilter) {
+    case 1:    return (groupSep || vpid != 0);
+    case 2:    return (groupSep || vpid == 0);
+    default:   return true;
+  } 
+}
+
 // -- cChannels --------------------------------------------------------------
 
 cChannels Channels;
diff -Naur vdr-m7x0-orig/channels.h vdr-m7x0/channels.h
--- vdr-m7x0-orig/channels.h	2007-06-30 10:22:11.000000000 +0200
+++ vdr-m7x0/channels.h	2007-07-01 20:56:18.000000000 +0200
@@ -216,6 +216,7 @@
   void SetCaDescriptors(int Level);
   void SetLinkChannels(cLinkChannels *LinkChannels);
   void SetRefChannel(cChannel *RefChannel);
+  bool Filtered(void);
   };
 
 class cChannels : public cRwLock, public cConfig<cChannel> {
diff -Naur vdr-m7x0-orig/config.c vdr-m7x0/config.c
--- vdr-m7x0-orig/config.c	2007-07-01 20:33:24.000000000 +0200
+++ vdr-m7x0/config.c	2007-07-01 20:56:18.000000000 +0200
@@ -279,6 +279,7 @@
   CurrentChannel = -1;
   CurrentVolume = MAXVOLUME;
   CurrentDolby = 0;
+  ChannelFilter = 0;
   InitialChannel = 0;
   InitialVolume = -1;
   ShowRecDate = 1;
@@ -455,6 +456,7 @@
   else if (!strcasecmp(Name, "CurrentChannel"))      CurrentChannel     = atoi(Value);
   else if (!strcasecmp(Name, "CurrentVolume"))       CurrentVolume      = atoi(Value);
   else if (!strcasecmp(Name, "CurrentDolby"))        CurrentDolby       = atoi(Value);
+  else if (!strcasecmp(Name, "ChannelFilter"))       ChannelFilter      = atoi(Value);
   else if (!strcasecmp(Name, "InitialChannel"))      InitialChannel     = atoi(Value);
   else if (!strcasecmp(Name, "InitialVolume"))       InitialVolume      = atoi(Value);
   else if (!strcasecmp(Name, "ShowRecDate"))         ShowRecDate        = atoi(Value);
@@ -538,6 +540,7 @@
   Store("CurrentChannel",     CurrentChannel);
   Store("CurrentVolume",      CurrentVolume);
   Store("CurrentDolby",       CurrentDolby);
+  Store("ChannelFilter",      ChannelFilter);
   Store("InitialChannel",     InitialChannel);
   Store("InitialVolume",      InitialVolume);
   Store("ShowRecDate",        ShowRecDate);
diff -Naur vdr-m7x0-orig/config.h vdr-m7x0/config.h
--- vdr-m7x0-orig/config.h	2007-07-01 20:33:24.000000000 +0200
+++ vdr-m7x0/config.h	2007-07-01 20:56:18.000000000 +0200
@@ -259,6 +259,7 @@
   int CurrentChannel;
   int CurrentVolume;
   int CurrentDolby;
+  int ChannelFilter;
   int InitialChannel;
   int InitialVolume;
   int ShowRecDate, ShowRecTime, ShowRecLength, ShowProgressBar, MenuCmdPosition;
diff -Naur vdr-m7x0-orig/device.c vdr-m7x0/device.c
--- vdr-m7x0-orig/device.c	2007-06-30 10:22:11.000000000 +0200
+++ vdr-m7x0/device.c	2007-07-01 20:56:18.000000000 +0200
@@ -815,7 +815,7 @@
      cChannel *channel;
      while ((channel = Channels.GetByNumber(n, Direction)) != NULL) {
            // try only channels which are currently available
-           if (PrimaryDevice()->ProvidesChannel(channel, Setup.PrimaryLimit) || /* PrimaryDevice()->CanReplay() && */ GetDevice(channel, 0))
+           if (channel->Filtered() && (PrimaryDevice()->ProvidesChannel(channel, Setup.PrimaryLimit) || /* PrimaryDevice()->CanReplay() && */ GetDevice(channel, 0)))
               break;
            n = channel->Number() + Direction;
            }
diff -Naur vdr-m7x0-orig/i18n.c vdr-m7x0/i18n.c
--- vdr-m7x0-orig/i18n.c	2007-07-01 20:33:24.000000000 +0200
+++ vdr-m7x0/i18n.c	2007-07-01 20:56:18.000000000 +0200
@@ -4586,6 +4586,63 @@
     "",// TODO
     ""// TODO
   },
+  { "Setup.Miscellaneous$Filter channels",
+    "",// TODO
+    "",// TODO
+    "Filtra canali",
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "Filtrar canales",
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "Filtra canals",
+    "",// TODO
+    "",// hrv TODO
+  },
+  { "Tv only",
+    "",// TODO
+    "",// TODO
+    "Solo Tv",
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "Sólo Tv",
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "Només Tv",
+    "",// TODO
+    "",// hrv TODO
+  },
+  { "Radio only",
+    "",// TODO
+    "",// TODO
+    "Solo Radio",
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "Sólo Radio",
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "",// TODO
+    "Només Radio",
+    "",// TODO
+    "",// hrv TODO
+  },
   // The days of the week:
   { "MTWTFSS",
     "MDMDFSS",
diff -Naur vdr-m7x0-orig/menu.c vdr-m7x0/menu.c
--- vdr-m7x0-orig/menu.c	2007-07-01 20:33:24.000000000 +0200
+++ vdr-m7x0/menu.c	2007-07-01 20:57:12.000000000 +0200
@@ -428,7 +428,7 @@
   cMenuChannelItem *currentItem = NULL;
   Clear();
   for (cChannel *channel = Channels.First(); channel; channel = Channels.Next(channel)) {
-      if (!channel->GroupSep() || cMenuChannelItem::SortMode() == cMenuChannelItem::csmNumber && *channel->Name()) {
+      if ((!channel->GroupSep() || cMenuChannelItem::SortMode() == cMenuChannelItem::csmNumber && *channel->Name()) && ( channel == currentChannel || channel->Filtered() )) {
          cMenuChannelItem *item = new cMenuChannelItem(channel);
          Add(item);
          if (channel == currentChannel)
@@ -3047,6 +3047,8 @@
 // --- cMenuSetupMisc --------------------------------------------------------
 
 class cMenuSetupMisc : public cMenuSetupBase {
+private:
+  const char *channelFilterTexts[3];
 public:
   cMenuSetupMisc(void);
   };
@@ -3054,10 +3056,14 @@
 cMenuSetupMisc::cMenuSetupMisc(void)
 {
   SetSection(tr("Miscellaneous"));
+  channelFilterTexts[0] = tr("no");
+  channelFilterTexts[1] = tr("Tv only");
+  channelFilterTexts[2] = tr("Radio only");
   Add(new cMenuEditIntItem( tr("Setup.Miscellaneous$Min. event timeout (min)"),   &data.MinEventTimeout));
   Add(new cMenuEditIntItem( tr("Setup.Miscellaneous$Min. user inactivity (min)"), &data.MinUserInactivity));
   Add(new cMenuEditIntItem( tr("Setup.Miscellaneous$SVDRP timeout (s)"),          &data.SVDRPTimeout));
   Add(new cMenuEditIntItem( tr("Setup.Miscellaneous$Zap timeout (s)"),            &data.ZapTimeout));
+  Add(new cMenuEditStraItem(tr("Setup.Miscellaneous$Filter channels"),            &data.ChannelFilter, 3, channelFilterTexts));;    
   Add(new cMenuEditChanItem(tr("Setup.Miscellaneous$Initial channel"),            &data.InitialChannel, tr("Setup.Miscellaneous$as before")));
   Add(new cMenuEditIntItem( tr("Setup.Miscellaneous$Initial volume"),             &data.InitialVolume, -1, 255, tr("Setup.Miscellaneous$as before")));
   Add(new cMenuEditBoolItem( tr("Setup.Miscellaneous$Abort when Plugin fails to load"),    &data.AbortWhenPluginFails));
@@ -3612,7 +3618,7 @@
   if (Direction) {
      while (Channel) {
            Channel = Direction > 0 ? Channels.Next(Channel) : Channels.Prev(Channel);
-           if (Channel && !Channel->GroupSep() && (cDevice::PrimaryDevice()->ProvidesChannel(Channel, Setup.PrimaryLimit) || cDevice::GetDevice(Channel, 0)))
+           if (Channel && Channel->Filtered() && !Channel->GroupSep() && (cDevice::PrimaryDevice()->ProvidesChannel(Channel, Setup.PrimaryLimit) || cDevice::GetDevice(Channel, 0)))
               return Channel;
            }
      }
