diff -Naur vdr-m7x0-orig/channels.c vdr-m7x0/channels.c
--- vdr-m7x0-orig/channels.c	2007-05-18 21:36:33.000000000 +0200
+++ vdr-m7x0/channels.c	2007-05-18 22:06:32.000000000 +0200
@@ -7,6 +7,7 @@
  * $Id: channels.c 1.53 2006/05/28 15:03:40 kls Exp $
  */
 
+#include "config.h"
 #include "channels.h"
 //M7X0 BEGIN AK
 #include "m7x0_dvb/frontend.h"
@@ -1069,6 +1070,17 @@
   return NULL;
 }
 
+//TK
+bool cChannels::keinEPGScan(tChannelID kanalID)
+{
+  bool rc=false;
+
+  if (strstr(::Setup.noEPG,kanalID.ToString())!=NULL){
+    rc=true;
+  }
+  return rc;
+}
+
 cString ChannelString(const cChannel *Channel, int Number)
 {
   char buffer[256];
diff -Naur vdr-m7x0-orig/channels.h vdr-m7x0/channels.h
--- vdr-m7x0-orig/channels.h	2007-05-18 21:40:33.000000000 +0200
+++ vdr-m7x0/channels.h	2007-05-18 22:06:32.000000000 +0200
@@ -253,6 +253,7 @@
       ///< modification has been made, and 2 if the user has made a modification.
       ///< Calling this function resets the 'modified' flag to 0.
   cChannel *NewChannel(const cChannel *Transponder, const char *Name, const char *ShortName, const char *Provider, int Nid, int Tid, int Sid, int Rid = 0);
+  bool keinEPGScan(tChannelID kanalID);//TK
   };
 
 extern cChannels Channels;
diff -Naur vdr-m7x0-orig/config.c vdr-m7x0/config.c
--- vdr-m7x0-orig/config.c	2007-05-18 21:57:39.000000000 +0200
+++ vdr-m7x0/config.c	2007-05-18 22:07:08.000000000 +0200
@@ -286,6 +286,7 @@
   ChannelFilter = 0;
   InitialChannel = 0;
   InitialVolume = -1;
+  strcpy(noEPG, "");
   ShowRecDate = 1;
   ShowRecTime = 1;
   ShowRecLength = 1;
@@ -455,6 +456,7 @@
   else if (!strcasecmp(Name, "NextWakeupTime"))      NextWakeupTime     = atoi(Value);
   else if (!strcasecmp(Name, "MultiSpeedMode"))      MultiSpeedMode     = atoi(Value);
   else if (!strcasecmp(Name, "ShowReplayMode"))      ShowReplayMode     = atoi(Value);
+  else if (!strcasecmp(Name, "noEPG"))               strn0cpy(noEPG, Value, 999);
   else if (!strcasecmp(Name, "DoubleEpgTimeDelta"))  DoubleEpgTimeDelta = atoi(Value);
   else if (!strcasecmp(Name, "DoubleEpgAction"))     DoubleEpgAction    = atoi(Value);
   else if (!strcasecmp(Name, "MixEpgAction"))        MixEpgAction    	= atoi(Value);
@@ -553,6 +555,7 @@
   Store("ChannelFilter",      ChannelFilter);
   Store("InitialChannel",     InitialChannel);
   Store("InitialVolume",      InitialVolume);
+  Store("noEPG",              noEPG);
   Store("ShowRecDate",        ShowRecDate);
   Store("ShowRecTime",        ShowRecTime);
   Store("ShowRecLength",      ShowRecLength);
diff -Naur vdr-m7x0-orig/config.h vdr-m7x0/config.h
--- vdr-m7x0-orig/config.h	2007-05-18 21:57:39.000000000 +0200
+++ vdr-m7x0/config.h	2007-05-18 22:06:32.000000000 +0200
@@ -261,6 +261,7 @@
   int DisableVPS;
   int ResumeID;
   int CurrentChannel;
+  char noEPG[999];
   int CurrentVolume;
   int CurrentDolby;
   int ChannelFilter;
diff -Naur vdr-m7x0-orig/eit.c vdr-m7x0/eit.c
--- vdr-m7x0-orig/eit.c	2007-05-18 21:52:48.000000000 +0200
+++ vdr-m7x0/eit.c	2007-05-18 22:06:32.000000000 +0200
@@ -35,6 +35,13 @@
   if (!channel)
      return; // only collect data for known channels
 
+ //TK
+   tChannelID kanalID;
+   kanalID=channel->GetChannelID();
+   if (Channels.keinEPGScan(kanalID)){
+     return;
+   }
+
   cSchedule *pSchedule = (cSchedule *)Schedules->GetSchedule(channel, true);
 
   bool Empty = true;
