diff -Naur 178-gambler-unstable/dvbdevice.c 184-andreas-unstable/dvbdevice.c
--- 178-gambler-unstable/dvbdevice.c	2007-07-27 21:53:10.000000000 +0200
+++ 184-andreas-unstable/dvbdevice.c	2007-07-27 21:53:04.000000000 +0200
@@ -4304,8 +4258,8 @@
 
   // Wait for tuner lock:
   // This seems to be need sometimes on M7x0
-//  if (!HasLock(-1))
-//      esyslog("ERROR: Cannot get Tuner-Lock!");
+  if (!HasLock(-1))
+      esyslog("ERROR: Cannot get Tuner-Lock!");
   // PID settings:
 
   if (TurnOnLivePIDs) {
