diff -Naur osdserver-0.1.1-orig/Makefile osdserver-0.1.1/Makefile
--- osdserver-0.1.1-orig/Makefile	2007-06-07 20:31:11.000000000 +0200
+++ osdserver-0.1.1/Makefile	2007-07-30 17:54:26.000000000 +0200
@@ -22,8 +22,8 @@
 
 ### The directory environment:
 
-VDRDIR = ../../..
-LIBDIR = ../../lib
+VDRDIR ?= ../../..
+LIBDIR ?= ../../lib
 TMPDIR = /tmp
 
 ### Allow user defined options to overwrite defaults:
