diff -Naur slin_m740_orig/drivers/usb/usb.c slin_m740/drivers/usb/usb.c
--- slin_m740_orig/drivers/usb/usb.c	Mon May 23 15:42:28 2005
+++ slin_m740/drivers/usb/usb.c	Wed May 20 12:28:05 2009
@@ -814,6 +814,7 @@
 {
 	char *argv [3], **envp, *buf, *scratch;
 	int i = 0, value;
+	int InterfaceClass = -1;
 
 	if (!hotplug_path [0])
 		return;
@@ -840,11 +841,6 @@
 		return;
 	}
 
-	/* only one standardized param to hotplug command: type */
-	argv [0] = hotplug_path;
-	argv [1] = "usb";
-	argv [2] = 0;
-
 	/* minimal command environment */
 	envp [i++] = "HOME=/";
 	envp [i++] = "PATH=/sbin:/bin:/usr/sbin:/usr/bin";
@@ -890,10 +886,11 @@
 			    dev->descriptor.bDeviceProtocol) + 1;
 	if (dev->descriptor.bDeviceClass == 0) {
 		int alt = dev->actconfig->interface [interface].act_altsetting;
+		InterfaceClass = dev->actconfig->interface [interface].altsetting [alt].bInterfaceClass;
 
 		envp [i++] = scratch;
 		scratch += sprintf (scratch, "INTERFACE=%d/%d/%d",
-			dev->actconfig->interface [interface].altsetting [alt].bInterfaceClass,
+			InterfaceClass,
 			dev->actconfig->interface [interface].altsetting [alt].bInterfaceSubClass,
 			dev->actconfig->interface [interface].altsetting [alt].bInterfaceProtocol)
 			+ 1;
@@ -903,12 +900,20 @@
 
 	/* NOTE: user mode daemons can call the agents too */
 
-	dbg ("kusbd: %s %s %d", argv [0], verb, dev->devnum);
-	value = call_usermodehelper (argv [0], argv, envp);
+	/* Only usbautomounter is used for USB Mass Storage (Interface 8) */
+	if( InterfaceClass == 8 ) {
+		argv [0] = "/sbin/usbautomounter";
+		argv [1] = "usb";
+		argv [2] = 0;
+
+		dbg ("kusbd: %s %s %d", argv [0], verb, dev->devnum);
+		value = call_usermodehelper (argv [0], argv, envp);
+		if (value != 0)
+			dbg ("kusbd policy returned 0x%x", value);
+	}
+
 	kfree (buf);
 	kfree (envp);
-	if (value != 0)
-		dbg ("kusbd policy returned 0x%x", value);
 }
 
 static void call_policy (char *verb, struct usb_device *dev)
