diff -Naur slin_m740_orig/fs/jffs2/compr.c slin_m740/fs/jffs2/compr.c
--- slin_m740_orig/fs/jffs2/compr.c	Sat May 29 15:45:42 2004
+++ slin_m740/fs/jffs2/compr.c	Mon Sep 28 11:03:25 2009
@@ -136,7 +136,7 @@
 #endif
 		break;
 	case JFFS2_COMPR_DYNRUBIN:
-#if 1 /* Phase this one out */
+#if 0 /* Phase this one out */
 		dynrubin_decompress(cdata_in, data_out, cdatalen, datalen);
 #else
 		printk(KERN_WARNING "JFFS2: Dynrubin compression encountered but support not compiled in!\n");
diff -Naur slin_m740_orig/fs/jffs2/Makefile slin_m740/fs/jffs2/Makefile
--- slin_m740_orig/fs/jffs2/Makefile	Sat May 29 15:45:42 2004
+++ slin_m740/fs/jffs2/Makefile	Mon Sep 28 11:03:57 2009
@@ -10,8 +10,7 @@
 # Note 2! The CFLAGS definitions are now in the main makefile...
 
 
-COMPR_OBJS	:= compr.o compr_rubin.o compr_rtime.o pushpull.o \
-			compr_zlib.o
+COMPR_OBJS	:= compr.o compr_rtime.o compr_zlib.o
 JFFS2_OBJS	:= dir.o file.o ioctl.o nodelist.o malloc.o \
 	read.o nodemgmt.o readinode.o super.o write.o scan.o gc.o \
 	symlink.o build.o erase.o background.o
