diff -Naur epgsearch-0.9.24.beta22-orig/i18n-generated.h epgsearch-0.9.24.beta22/i18n-generated.h
--- epgsearch-0.9.24.beta22-orig/i18n-generated.h	2008-03-05 20:07:40.000000000 +0100
+++ epgsearch-0.9.24.beta22/i18n-generated.h	2008-03-08 12:49:19.000000000 +0100
@@ -3718,7 +3718,7 @@
     "",
     "Luo ajastimet uudelleen",
     "",
-    "Recrear programaciones tras borrado",
+    "Recrear programaciones tras borrar",
     "",
     "",
     "",
@@ -5468,7 +5468,7 @@
     "",
     "",
     "",
-    "",
+    "Ha cambiado el tiempo inicial/final",
     "",
     "",
     "",
@@ -5503,7 +5503,7 @@
     "",
     "",
     "",
-    "",
+    "Ha cambiado el título/episodio",
     "",
     "",
     "",
diff -Naur epgsearch-0.9.24.beta22-orig/Makefile epgsearch-0.9.24.beta22/Makefile
--- epgsearch-0.9.24.beta22-orig/Makefile	2008-02-20 19:45:11.000000000 +0100
+++ epgsearch-0.9.24.beta22/Makefile	2008-03-08 12:41:25.000000000 +0100
@@ -34,9 +34,9 @@
 
 ### The directory environment:
 
-DVBDIR = ../../../../DVB
-VDRDIR = ../../..
-LIBDIR = ../../lib
+DVBDIR ?= ../../../../DVB
+VDRDIR ?= ../../..
+LIBDIR ?= ../../lib
 TMPDIR = /tmp
 
 ### Allow user defined options to overwrite defaults:
@@ -69,7 +69,7 @@
 
 ### Includes and Defines (add further entries here):
 
-INCLUDES += -I$(VDRDIR)/include -I$(DVBDIR)/include
+INCLUDES += -I$(VDRDIR)/include -I$(DVBDIR)
 #INCLUDES += -I$(VDRDIR)/include
 
 DEFINES += -D_GNU_SOURCE -DPLUGIN_NAME_I18N='"$(PLUGIN)"'
@@ -128,7 +128,7 @@
 
 # Dependencies:
 
-MAKEDEP = g++ -MM -MG
+MAKEDEP = $(CXX) -MM -MG
 DEPFILE = .dependencies
 $(DEPFILE): Makefile
 	@$(MAKEDEP) $(CXXFLAGS) $(DEFINES) $(INCLUDES) $(OBJS:%.o=%.c) $(OBJS2:%.o=%.c) $(OBJS3:%.o=%.c) $(OBJS4:%.o=%.c)> $@
diff -Naur epgsearch-0.9.24.beta22-orig/po/es_ES.po epgsearch-0.9.24.beta22/po/es_ES.po
--- epgsearch-0.9.24.beta22-orig/po/es_ES.po	2008-03-05 20:06:15.000000000 +0100
+++ epgsearch-0.9.24.beta22/po/es_ES.po	2008-03-08 12:45:48.000000000 +0100
@@ -394,7 +394,7 @@
 msgstr "Establecer a 'sí' cuando no desee recibir ningún aviso de emisiones durante una reproducción."
 
 msgid "Recreate timers after deletion"
-msgstr "Recrear programaciones tras borrado"
+msgstr "Recrear programaciones tras borrar"
 
 msgid "Help$Set this to 'yes' if you want timers to be recreated with the next search timer update after deleting them."
 msgstr "Establecer a 'sí' cuando desee que las programaciones se creen de nuevo con la siguiente actualización de programación por búsqueda después de borrarlas."
@@ -556,10 +556,10 @@
 msgstr " aábcdeéfghiíjklmnñoópqrstuúvwxyz0123456789-.,#~\\^$[]|()*+?{}/:%@&"
 
 msgid "Start/Stop time has changed"
-msgstr ""
+msgstr "Ha cambiado el tiempo inicial/final"
 
 msgid "Title/episode has changed"
-msgstr ""
+msgstr "Ha cambiado el título/episodio"
 
 msgid "No new timers were added."
 msgstr "No se añadieron nuevas programaciones."
