diff -Naur channellists-0.0.4-orig/i18n.c channellists-0.0.4/i18n.c
--- channellists-0.0.4-orig/i18n.c	Mon Jan 21 14:36:24 2008
+++ channellists-0.0.4/i18n.c	Fri Jan 25 12:52:41 2008
@@ -18,7 +18,7 @@
     "", // Norsk
     "", // suomi
     "", // Polski
-    "", // Español
+    "Canales presintonizados", // Español
     "", // ÅëëçíéêÜ (Greek)
     "", // Svenska
     "", // Romaneste
@@ -40,7 +40,7 @@
     "", // Norsk
     "", // suomi
     "", // Polski
-    "", // Español
+    "Administra tus listas de canales", // Español
     "", // ÅëëçíéêÜ (Greek)
     "", // Svenska
     "", // Romaneste
@@ -62,7 +62,7 @@
     "", // Norsk
     "", // suomi
     "", // Polski
-    "", // Español
+    "Ocultar entrada del menú principal", // Español
     "", // ÅëëçíéêÜ (Greek)
     "", // Svenska
     "", // Romaneste
@@ -84,7 +84,7 @@
     "", // Norsk
     "", // suomi
     "", // Polski
-    "", // Español
+    "Acción después de cambiar", // Español
     "", // ÅëëçíéêÜ (Greek)
     "", // Svenska
     "", // Romaneste
@@ -128,7 +128,7 @@
     "", // Norsk
     "", // suomi
     "", // Polski
-    "", // Español
+    "Cerrar", // Español
     "", // ÅëëçíéêÜ (Greek)
     "", // Svenska
     "", // Romaneste
@@ -150,7 +150,7 @@
     "", // Norsk
     "", // suomi
     "", // Polski
-    "", // Español
+    "Nada", // Español
     "", // ÅëëçíéêÜ (Greek)
     "", // Svenska
     "", // Romaneste
@@ -172,7 +172,7 @@
     "", // Norsk
     "", // suomi
     "", // Polski
-    "", // Español
+    "Botón rojo", // Español
     "", // ÅëëçíéêÜ (Greek)
     "", // Svenska
     "", // Romaneste
@@ -194,7 +194,7 @@
     "", // Norsk
     "", // suomi
     "", // Polski
-    "", // Español
+    "Botón verde", // Español
     "", // ÅëëçíéêÜ (Greek)
     "", // Svenska
     "", // Romaneste
@@ -216,7 +216,7 @@
     "", // Norsk
     "", // suomi
     "", // Polski
-    "", // Español
+    "¿Cambiar la lista de canales actual?", // Español
     "", // ÅëëçíéêÜ (Greek)
     "", // Svenska
     "", // Romaneste
@@ -238,7 +238,7 @@
     "", // Norsk
     "", // suomi
     "", // Polski
-    "", // Español
+    "Script", // Español
     "", // ÅëëçíéêÜ (Greek)
     "", // Svenska
     "", // Romaneste
@@ -260,7 +260,7 @@
     "", // Norsk
     "", // suomi
     "", // Polski
-    "", // Español
+    "Renombrar", // Español
     "", // ÅëëçíéêÜ (Greek)
     "", // Svenska
     "", // Romaneste
@@ -282,7 +282,7 @@
     "", // Norsk
     "", // suomi
     "", // Polski
-    "", // Español
+    "Copiar", // Español
     "", // ÅëëçíéêÜ (Greek)
     "", // Svenska
     "", // Romaneste
@@ -304,7 +304,7 @@
     "", // Norsk
     "", // suomi
     "", // Polski
-    "", // Español
+    "Guardar", // Español
     "", // ÅëëçíéêÜ (Greek)
     "", // Svenska
     "", // Romaneste
@@ -348,7 +348,7 @@
     "", // Norsk
     "", // suomi
     "", // Polski
-    "", // Español
+    "¿Borrar la lista de canales?", // Español
     "", // ÅëëçíéêÜ (Greek)
     "", // Svenska
     "", // Romaneste
@@ -370,7 +370,7 @@
     "", // Norsk
     "", // suomi
     "", // Polski
-    "", // Español
+    "¿Ejecutar script de actualización?", // Español
     "", // ÅëëçíéêÜ (Greek)
     "", // Svenska
     "", // Romaneste
@@ -392,7 +392,7 @@
     "", // Norsk
     "", // suomi
     "", // Polski
-    "", // Español
+    "Espere...", // Español
     "", // ÅëëçíéêÜ (Greek)
     "", // Svenska
     "", // Romaneste
@@ -414,7 +414,7 @@
     "", // Norsk
     "", // suomi
     "", // Polski
-    "", // Español
+    "Se produjo un error, ver syslog...", // Español
     "", // ÅëëçíéêÜ (Greek)
     "", // Svenska
     "", // Romaneste
@@ -436,7 +436,7 @@
     "", // Norsk
     "", // suomi
     "", // Polski
-    "", // Español
+    "Renombrar", // Español
     "", // ÅëëçíéêÜ (Greek)
     "", // Svenska
     "", // Romaneste
@@ -458,7 +458,7 @@
     "", // Norsk
     "", // suomi
     "", // Polski
-    "", // Español
+    "Script", // Español
     "", // ÅëëçíéêÜ (Greek)
     "", // Svenska
     "", // Romaneste
@@ -480,7 +480,7 @@
     "", // Norsk
     "", // suomi
     "", // Polski
-    "", // Español
+    "Copiar", // Español
     "", // ÅëëçíéêÜ (Greek)
     "", // Svenska
     "", // Romaneste
@@ -524,7 +524,7 @@
     "", // Norsk
     "", // suomi
     "", // Polski
-    "", // Español
+    "Guardar", // Español
     "", // ÅëëçíéêÜ (Greek)
     "", // Svenska
     "", // Romaneste
diff -Naur channellists-0.0.4-orig/Makefile channellists-0.0.4/Makefile
--- channellists-0.0.4-orig/Makefile	Mon Jan 21 18:12:36 2008
+++ channellists-0.0.4/Makefile	Fri Jan 25 12:36:55 2008
@@ -22,8 +22,8 @@
 
 ### The directory environment:
 
-VDRDIR = ../../..
-LIBDIR = ../../lib
+VDRDIR ?= ../../..
+LIBDIR ?= ../../lib
 TMPDIR = /tmp
 
 ### Allow user defined options to overwrite defaults:
diff -Naur channellists-0.0.4-orig/po/es_ES.po channellists-0.0.4/po/es_ES.po
--- channellists-0.0.4-orig/po/es_ES.po	Mon Jan 21 22:09:26 2008
+++ channellists-0.0.4/po/es_ES.po	Fri Jan 25 12:54:14 2008
@@ -1,88 +1,88 @@
 # VDR plugin language source file.
 # Copyright (C) 2007 Klaus Schmidinger <kls@cadsoft.de>
 # This file is distributed under the same license as the VDR package.
-# Ruben Nunez Francisco <ruben.nunez@tang-it.com>, 2002
+# bittor from open7x0.org <bittor7x0 _at_ gmail.com>, 2008
 #
 msgid ""
 msgstr ""
 "Project-Id-Version: VDR 1.5.7\n"
 "Report-Msgid-Bugs-To: <see README>\n"
 "POT-Creation-Date: 2008-01-21 18:23+0100\n"
-"PO-Revision-Date: 2008-01-21 18:12+0100\n"
-"Last-Translator: Ruben Nunez Francisco <ruben.nunez@tang-it.com>\n"
+"PO-Revision-Date: 2008-01-25 14:21+0100\n"
+"Last-Translator: bittor from open7x0.org <bittor7x0 _at_ gmail.com>\n"
 "Language-Team: <vdr@linuxtv.org>\n"
 "MIME-Version: 1.0\n"
 "Content-Type: text/plain; charset=ISO-8859-15\n"
 "Content-Transfer-Encoding: 8bit\n"
 
 msgid "Manage your channellists"
-msgstr ""
+msgstr "Administra tus listas de canales"
 
 msgid "Channellists"
-msgstr ""
+msgstr "Canales presintonizados"
 
 msgid "Menu"
 msgstr "Menú"
 
 msgid "close"
-msgstr ""
+msgstr "Cerrar"
 
 msgid "none"
-msgstr ""
+msgstr "Nada"
 
 msgid "Hide main menu entry"
-msgstr ""
+msgstr "Ocultar entrada del menú principal"
 
 msgid "Action after switch"
-msgstr ""
+msgstr "Acción después de cambiar"
 
 msgid "Button Red"
-msgstr ""
+msgstr "Botón Rojo"
 
 msgid "Rename"
-msgstr ""
+msgstr "Renombrar"
 
 msgid "Script"
-msgstr ""
+msgstr "Script"
 
 msgid "Button Green"
-msgstr ""
+msgstr "Botón Verde"
 
 msgid "Copy"
-msgstr ""
+msgstr "Copiar"
 
 msgid "Name"
 msgstr "Nombre"
 
 msgid "An error occurs, see syslog..."
-msgstr ""
+msgstr "Se produjo un error, ver syslog..."
 
 msgid "Switch current channellist?"
-msgstr ""
+msgstr "¿Cambiar la lista de canales actual?"
 
 msgid "Delete channellist?"
-msgstr ""
+msgstr "¿Borrar la lista de canales?"
 
 msgid "Save"
-msgstr ""
+msgstr "Guardar"
 
 msgid "Execute update script?"
-msgstr ""
+msgstr "¿Ejecutar script de actualización?"
 
 msgid "Waiting..."
-msgstr ""
+msgstr "Espere..."
 
 msgid "Button$Script"
-msgstr ""
+msgstr "Script"
 
 msgid "Button$Rename"
-msgstr ""
+msgstr "Renombrar"
 
 msgid "Button$Copy"
-msgstr ""
+msgstr "Copiar"
 
 msgid "Button$Delete"
 msgstr "Borrar"
 
 msgid "Button$Save"
-msgstr ""
+msgstr "Guardar"
