diff -Naur remotetimers-0.0.1-orig/i18n.c remotetimers-0.0.1/i18n.c
--- remotetimers-0.0.1-orig/i18n.c	Wed Dec 05 17:12:17 2007
+++ remotetimers-0.0.1/i18n.c	Mon Dec 10 16:22:56 2007
@@ -11,7 +11,7 @@
 	"",// TODO Norsk
 	"",// TODO Finnish
 	"",// TODO Polski
-	"",// TODO Español
+	"Programador remoto",// Español
 	"",// TODO ÅëëçíéêÜ (Greek)
 	"",// TODO Svenska
 	"",// TODO Romaneste
@@ -33,7 +33,7 @@
 	"",// TODO Norsk
 	"",// TODO Finnish
 	"",// TODO Polski
-	"",// TODO Español
+	"Editar programaciones en VDR remoto",// Español
 	"",// TODO ÅëëçíéêÜ (Greek)
 	"",// TODO Svenska
 	"",// TODO Romaneste
@@ -55,7 +55,7 @@
 	"",// TODO Norsk
 	"",// TODO Finnish
 	"",// TODO Polski
-	"",// TODO Español
+	"Ocultar en el menú principal",// Español
 	"",// TODO ÅëëçíéêÜ (Greek)
 	"",// TODO Svenska
 	"",// TODO Romaneste
@@ -77,7 +77,7 @@
 	"",// TODO Norsk
 	"",// TODO Finnish
 	"",// TODO Polski
-	"",// TODO Español
+	"Reemplazar en el menú principal \"Programación\"",// Español
 	"",// TODO ÅëëçíéêÜ (Greek)
 	"",// TODO Svenska
 	"",// TODO Romaneste
@@ -99,7 +99,7 @@
 	"",// TODO Norsk
 	"",// TODO Finnish
 	"",// TODO Polski
-	"",// TODO Español
+	"Reemplazar en el menu principal \"Programaciones\"",// Español
 	"",// TODO ÅëëçíéêÜ (Greek)
 	"",// TODO Svenska
 	"",// TODO Romaneste
@@ -121,7 +121,7 @@
 	"",// TODO Norsk
 	"",// TODO Finnish
 	"",// TODO Polski
-	"",// TODO Español
+	"Servidor IP",// Español
 	"",// TODO ÅëëçíéêÜ (Greek)
 	"",// TODO Svenska
 	"",// TODO Romaneste
@@ -143,7 +143,7 @@
 	"",// TODO Norsk
 	"",// TODO Finnish
 	"",// TODO Polski
-	"",// TODO Español
+	"Puerto del servidor",// Español
 	"",// TODO ÅëëçíéêÜ (Greek)
 	"",// TODO Svenska
 	"",// TODO Romaneste
@@ -165,7 +165,7 @@
 	"",// TODO Norsk
 	"",// TODO Finnish
 	"",// TODO Polski
-	"",// TODO Español
+	"Valores predefinidos de nuevas programaciones",// Español
 	"",// TODO ÅëëçíéêÜ (Greek)
 	"",// TODO Svenska
 	"",// TODO Romaneste
@@ -187,7 +187,7 @@
 	"",// TODO Norsk
 	"",// TODO Finnish
 	"",// TODO Polski
-	"",// TODO Español
+	"Localización",// Español
 	"",// TODO ÅëëçíéêÜ (Greek)
 	"",// TODO Svenska
 	"",// TODO Romaneste
@@ -209,7 +209,7 @@
 	"",// TODO Norsk
 	"",// TODO Finnish
 	"",// TODO Polski
-	"",// TODO Español
+	"Local",// Español
 	"",// TODO ÅëëçíéêÜ (Greek)
 	"",// TODO Svenska
 	"",// TODO Romaneste
@@ -231,7 +231,7 @@
 	"",// TODO Norsk
 	"",// TODO Finnish
 	"",// TODO Polski
-	"",// TODO Español
+	"Servidor",// Español
 	"",// TODO ÅëëçíéêÜ (Greek)
 	"",// TODO Svenska
 	"",// TODO Romaneste
@@ -253,7 +253,7 @@
 	"",// TODO Norsk
 	"",// TODO Finnish
 	"",// TODO Polski
-	"",// TODO Español
+	"Id de Usuario",// Español
 	"",// TODO ÅëëçíéêÜ (Greek)
 	"",// TODO Svenska
 	"",// TODO Romaneste
@@ -275,7 +275,7 @@
 	"",// TODO Norsk
 	"",// TODO Finnish
 	"",// TODO Polski
-	"",// TODO Español
+	"Ajustes del menu \"Timers\"",// Español
 	"",// TODO ÅëëçíéêÜ (Greek)
 	"",// TODO Svenska
 	"",// TODO Romaneste
@@ -297,7 +297,7 @@
 	"",// TODO Norsk
 	"",// TODO Finnish
 	"",// TODO Polski
-	"",// TODO Español
+	"Filtros para el usuario",// Español
 	"",// TODO ÅëëçíéêÜ (Greek)
 	"",// TODO Svenska
 	"",// TODO Romaneste
@@ -319,7 +319,7 @@
 	"",// TODO Norsk
 	"",// TODO Finnish
 	"",// TODO Polski
-	"",// TODO Español
+	"Programaciones - Usuario %d",// Español
 	"",// TODO ÅëëçíéêÜ (Greek)
 	"",// TODO Svenska
 	"",// TODO Romaneste
@@ -341,7 +341,7 @@
 	"",// TODO Norsk
 	"",// TODO Finnish
 	"",// TODO Polski
-	"",// TODO Español
+	"RL",// Español
 	"",// TODO ÅëëçíéêÜ (Greek)
 	"",// TODO Svenska
 	"",// TODO Romaneste
@@ -363,7 +363,7 @@
 	"",// TODO Norsk
 	"",// TODO Finnish
 	"",// TODO Polski
-	"",// TODO Español
+	"Programaciones remotas no disponibles",// Español
 	"",// TODO ÅëëçíéêÜ (Greek)
 	"",// TODO Svenska
 	"",// TODO Romaneste
@@ -385,7 +385,7 @@
 	"",// TODO Norsk
 	"",// TODO Finnish
 	"",// TODO Polski
-	"",// TODO Español
+	"Lista de programaciones remotas actualizada",// Español
 	"",// TODO ÅëëçíéêÜ (Greek)
 	"",// TODO Svenska
 	"",// TODO Romaneste
@@ -407,7 +407,7 @@
 	"",// TODO Norsk
 	"",// TODO Finnish
 	"",// TODO Polski
-	"",// TODO Español
+	"Programaciones actualizandose remotamente - inténtelo de nuevo",// Español
 	"",// TODO ÅëëçíéêÜ (Greek)
 	"",// TODO Svenska
 	"",// TODO Romaneste
@@ -429,7 +429,7 @@
 	"",// TODO Norsk
 	"",// TODO Finnish
 	"",// TODO Polski
-	"",// TODO Español
+	"Programación grabandose - desactivarla primero",// Español
 	"",// TODO ÅëëçíéêÜ (Greek)
 	"",// TODO Svenska
 	"",// TODO Romaneste
@@ -451,7 +451,7 @@
 	"",// TODO Norsk
 	"",// TODO Finnish
 	"",// TODO Polski
-	"",// TODO Español
+	"Programación ya definida",// Español
 	"",// TODO ÅëëçíéêÜ (Greek)
 	"",// TODO Svenska
 	"",// TODO Romaneste
@@ -473,7 +473,7 @@
 	"",// TODO Norsk
 	"",// TODO Finnish
 	"",// TODO Polski
-	"",// TODO Español
+	"Programaciín modificada remotamente - Comprobar programaciones remotas",// Español
 	"",// TODO ÅëëçíéêÜ (Greek)
 	"",// TODO Svenska
 	"",// TODO Romaneste
@@ -495,7 +495,7 @@
 	"",// TODO Norsk
 	"",// TODO Finnish
 	"",// TODO Polski
-	"",// TODO Español
+	"Error inexperado - Comprobar programaciones remotas",// Español
 	"",// TODO ÅëëçíéêÜ (Greek)
 	"",// TODO Svenska
 	"",// TODO Romaneste
@@ -517,7 +517,7 @@
 	"",// TODO Norsk
 	"",// TODO Finnish
 	"",// TODO Polski
-	"",// TODO Español
+	"Conexión SVDRP perdida - Comprobar programaciones remotas",// Español
 	"",// TODO ÅëëçíéêÜ (Greek)
 	"",// TODO Svenska
 	"",// TODO Romaneste
@@ -539,7 +539,7 @@
 	"",// TODO Norsk
 	"",// TODO Finnish
 	"",// TODO Polski
-	"",// TODO Español
+	"Programación grabandose - no se puede cambiar al servidor",// Español
 	"",// TODO ÅëëçíéêÜ (Greek)
 	"",// TODO Svenska
 	"",// TODO Romaneste
diff -Naur remotetimers-0.0.1-orig/Makefile remotetimers-0.0.1/Makefile
--- remotetimers-0.0.1-orig/Makefile	Wed Dec 05 17:12:17 2007
+++ remotetimers-0.0.1/Makefile	Mon Dec 10 11:37:27 2007
@@ -20,8 +20,8 @@
 
 ### The directory environment:
 
-VDRDIR = ../../..
-LIBDIR = ../../lib
+VDRDIR ?= ../../..
+LIBDIR ?= ../../lib
 TMPDIR = /tmp
 
 ### Allow user defined options to overwrite defaults:
@@ -67,7 +67,7 @@
 
 libvdr-$(PLUGIN).so: $(OBJS)
 	$(CXX) $(CXXFLAGS) -shared $(OBJS) -o $@
-	@cp $@ $(LIBDIR)/$@.$(APIVERSION)
+	@cp -f $@ $(LIBDIR)/$@.$(APIVERSION)
 
 dist: clean
 	@-rm -rf $(TMPDIR)/$(ARCHIVE)
diff -Naur remotetimers-0.0.1-orig/timers.c remotetimers-0.0.1/timers.c
--- remotetimers-0.0.1-orig/timers.c	Wed Dec 05 17:12:17 2007
+++ remotetimers-0.0.1/timers.c	Mon Dec 10 11:37:27 2007
@@ -48,7 +48,7 @@
 // cRemoteTimers -------------------------------------------------
 
 cRemoteTimers::cRemoteTimers(): refcount(0) {
-	plugin = cPluginManager::GetPlugin("svdrpservice");
+	plugin = NULL;
 	svdrp.handle = -1;
 }
 
@@ -59,6 +59,8 @@
 
 bool cRemoteTimers::Connect() {
 	refcount++;
+	if (!plugin)
+		plugin = cPluginManager::GetPlugin("svdrpservice");
 	if (plugin && svdrp.handle < 0) {
 		svdrp.serverIp = RemoteTimersSetup.serverIp;
 		svdrp.serverPort = RemoteTimersSetup.serverPort;
