From: Dmitry Bogatov <KAction@gnu.org>
Date: Mon, 28 May 2018 21:46:39 +0300
Subject: [PATCH] Fix spin lock on systems with poor clock

---
 runit-2.1.2/src/svlogd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/svlogd.c b/src/svlogd.c
index 90a3321..fab8441 100644
--- a/src/svlogd.c
+++ b/src/svlogd.c
@@ -234,7 +234,7 @@ void rmoldest(struct logdir *ld) {
         if (unlink(f->d_name) == -1)
           warn2("unable to unlink processor leftover", f->d_name);
       }
-      else {
+      else if (f->d_name[26] != 'u') {
         ++n;
         if (str_diff(f->d_name, oldest) < 0) byte_copy(oldest, 27, f->d_name);
       }
