diff -Naur a/src/sv.c b/src/sv.c
--- a/src/sv.c	2020-10-28 16:05:58.000000000 +0100
+++ b/src/sv.c	2020-10-28 16:07:33.933099649 +0100
@@ -33,14 +33,14 @@
 char *progname;
 char *action;
 char *acts;
-char *varservice ="/etc/service/";
+char *varservice ="/var/service/";
 char **service;
 char **servicex;
 unsigned int services;
 unsigned int rc =0;
 unsigned int lsb;
 unsigned int verbose =0;
-unsigned long wait =7;
+unsigned long wait =30;
 unsigned int kll =0;
 unsigned int islog =0;
 struct taia tstart, tnow, tdiff;
