diff -Naur remotetimers-0.0.1-orig/Makefile remotetimers-0.0.1/Makefile
--- remotetimers-0.0.1-orig/Makefile	2007-01-21 00:15:03.000000000 +0100
+++ remotetimers-0.0.1/Makefile	2007-06-16 20:11:12.000000000 +0200
@@ -20,8 +20,8 @@
 
 ### The directory environment:
 
-VDRDIR = ../../..
-LIBDIR = ../../lib
+VDRDIR ?= ../../..
+LIBDIR ?= ../../lib
 TMPDIR = /tmp
 
 ### Allow user defined options to overwrite defaults:
@@ -67,7 +67,7 @@
 
 libvdr-$(PLUGIN).so: $(OBJS)
 	$(CXX) $(CXXFLAGS) -shared $(OBJS) -o $@
-	@cp $@ $(LIBDIR)/$@.$(APIVERSION)
+	@cp -f $@ $(LIBDIR)/$@.$(APIVERSION)
 
 dist: clean
 	@-rm -rf $(TMPDIR)/$(ARCHIVE)
diff -Naur remotetimers-0.0.1-orig/timers.c remotetimers-0.0.1/timers.c
--- remotetimers-0.0.1-orig/timers.c	2007-03-04 16:46:27.000000000 +0100
+++ remotetimers-0.0.1/timers.c	2007-05-09 20:36:06.000000000 +0200
@@ -48,7 +48,7 @@
 // cRemoteTimers -------------------------------------------------
 
 cRemoteTimers::cRemoteTimers(): refcount(0) {
-	plugin = cPluginManager::GetPlugin("svdrpservice");
+	plugin = NULL;
 	svdrp.handle = -1;
 }
 
@@ -59,6 +59,8 @@
 
 bool cRemoteTimers::Connect() {
 	refcount++;
+	if (!plugin)
+		plugin = cPluginManager::GetPlugin("svdrpservice");
 	if (plugin && svdrp.handle < 0) {
 		svdrp.serverIp = RemoteTimersSetup.serverIp;
 		svdrp.serverPort = RemoteTimersSetup.serverPort;
