diff -Naurx .svn mpgplayer-0.0.1-orig/mpgplayer_i18n.c mpgplayer-0.0.1/mpgplayer_i18n.c
--- mpgplayer-0.0.1-orig/mpgplayer_i18n.c	2009-08-21 15:15:53.000000000 +0200
+++ mpgplayer-0.0.1/mpgplayer_i18n.c	2009-08-22 17:56:28.000000000 +0200
@@ -33,7 +33,7 @@
   },
   {
      "A player for Mpg-PS-Recordings",
-     "Ein Player fï¿½r Mpg PS Aufnahmen",
+     "Ein Player für Mpg PS Aufnahmen",
      "",
      "",
      "",
@@ -56,7 +56,7 @@
   },
   {
      "Cannot open directory",
-     "Kann Verzeichnis nicht ï¿½ffnen",
+     "Kann Verzeichnis nicht öffnen",
      "",
      "",
      "",
@@ -79,7 +79,7 @@
   },
   {
      "Cannot delete running recording",
-     "Kann laufende Aufnahme nicht lï¿½schen",
+     "Kann laufende Aufnahme nicht löschen",
      "",
      "",
      "",
@@ -88,7 +88,7 @@
      "",
      "",
      "",
-     "No puedo borrar grabaciï¿½n actual",
+     "No puedo borrar grabación actual",
      "",
      "",
      "",
@@ -102,7 +102,7 @@
   },
   {
      "Cannot delete recording",
-     "Kann Aufnahme nicht lï¿½schen",
+     "Kann Aufnahme nicht löschen",
      "",
      "",
      "",
@@ -111,7 +111,7 @@
      "",
      "",
      "",
-     "No puedo borrar grabaciï¿½n",
+     "No puedo borrar grabación",
      "",
      "",
      "",
diff -Naurx .svn mpgplayer-0.0.1-orig/mpgplayer_menu.c mpgplayer-0.0.1/mpgplayer_menu.c
--- mpgplayer-0.0.1-orig/mpgplayer_menu.c	2009-08-21 15:15:53.000000000 +0200
+++ mpgplayer-0.0.1/mpgplayer_menu.c	2009-08-22 17:56:28.000000000 +0200
@@ -735,7 +735,7 @@
 {
   return new cMpgMenuRecording(Recording(), false);
 }
-
+//
 eOSState cMpgReplayControl::ProcessKey(eKeys Key)
 {
   if (!Active())
@@ -764,6 +764,20 @@
      return osContinue;
      }
   bool DoShowMode = true;
+  if (!Setup.LRForwardRewind || (Setup.LRForwardRewind == 1 && !visible)) {
+    switch (Key) {
+      // Left/Right volume control
+      case kLeft|k_Repeat:
+      case kLeft:
+      case kRight|k_Repeat:
+      case kRight:
+	  cRemote::Put(NORMALKEY(Key) == kLeft ? kVolDn : kVolUp, true);
+        return osContinue;
+        break;
+      default:
+        break;
+    }
+  }
   switch (Key) {
     // Positioning:
     case kPlay:
@@ -783,10 +797,11 @@
 //    case kRed:     TimeSearch(); break;
     case kRed|k_Repeat:
     case kRed:   SkipSeconds(-100); break;
-    case kGreen|k_Repeat:
-    case kGreen:   SkipSeconds(-10); break;
-    case kYellow|k_Repeat:
-    case kYellow:  SkipSeconds( 10); break;
+    case kGreen|k_Repeat:   SkipSeconds(-(Setup.JumpSecondsRepeat)); break;
+    case kGreen:   SkipSeconds(-(Setup.JumpSeconds)); break;
+    case kYellow|k_Repeat: 
+                   SkipSeconds(Setup.JumpSecondsRepeat); break;
+    case kYellow:  SkipSeconds(Setup.JumpSeconds); break;
     case kBlue|k_Repeat:
     case kBlue:    SkipSeconds( 100); break;
     case kStop:
@@ -835,7 +850,7 @@
      ShowMode();
   return osContinue;
 }
-
+//
 // --- cMpgSetupMenu --------------------------------------------------------
 
 cMpgSetupMenu::cMpgSetupMenu(cMpgController *Controller)
